<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-1.8.25">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<meta name="author" content="Guangchuang Yu">
<title>8&nbsp; Disease enrichment analysis – Biomedical Knowledge Mining using GOSemSim and clusterProfiler</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>

<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./026-meshes-enrichment.html" rel="next">
<link href="./024-reactome.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js" type="module"></script>
<script src="site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="site_libs/quarto-html/axe/axe-check.js" type="module"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-7b89279ff1a6dce999919e0e67d4d9ec.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-70a47bd5681a7291082a5b9f83d58762.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light"><script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
</head>
<body class="nav-sidebar floating quarto-light">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top"><nav class="quarto-secondary-nav"><div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./02-Enrichment.html">Enrichment Analysis</a></li><li class="breadcrumb-item"><a href="./025-do-enrichment.html"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Disease enrichment analysis</span></a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav></header><!-- content --><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto"><div class="pt-lg-2 mt-2 text-left sidebar-header sidebar-header-stacked">
      <a href="./index.html" class="sidebar-logo-link">
      </a>
    <div class="sidebar-title mb-0 py-0">
      <a href="./">Biomedical Knowledge Mining using GOSemSim and clusterProfiler</a> 
        <div class="sidebar-tools-main">
    <a href="https://github.com/YuLab-SMU/biomedical-knowledge-mining-book/" title="Source Code" class="quarto-navigation-tool px-1" aria-label="Source Code"><i class="bi bi-github"></i></a>
  <a href="" class="quarto-reader-toggle quarto-navigation-tool px-1" onclick="window.quartoToggleReader(); return false;" title="Toggle reader mode">
  <div class="quarto-reader-toggle-btn">
  <i class="bi"></i>
  </div>
</a>
</div>
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Welcome</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="./01-semantic-similarity.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Semantic Similarity Analysis</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./011-GOSemSim.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">GO semantic similarity analysis</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./012-DOSE-semantic-similarity.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">DO semantic similarity analysis</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./013-meshes-semantic-similarity.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">MeSH semantic similarity analysis</span></span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="./02-Enrichment.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Enrichment Analysis</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./021-go.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">GO enrichment analysis</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./022-kegg.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">KEGG enrichment analysis</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./023-wikipathways.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">WikiPathways analysis</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./024-reactome.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Reactome enrichment analysis</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./025-do-enrichment.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Disease enrichment analysis</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./026-meshes-enrichment.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">9</span>&nbsp; <span class="chapter-title">MeSH enrichment analysis</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./027-universal-enrichment.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">10</span>&nbsp; <span class="chapter-title">Universal enrichment analysis</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./028-chipseeker.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">11</span>&nbsp; <span class="chapter-title">Functional enrichment analysis of genomic coordinations</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./029-compareCluster.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">12</span>&nbsp; <span class="chapter-title">Biological theme comparison</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./enrichplot.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">13</span>&nbsp; <span class="chapter-title">Visualization of functional enrichment result</span></span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="./misc.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Miscellaneous topics</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./dplyr.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">14</span>&nbsp; <span class="chapter-title">dplyr verbs for manipulating enrichment result</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./utilities.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">15</span>&nbsp; <span class="chapter-title">Useful utilities</span></span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="./appendix.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Appendix</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./FAQ.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">16</span>&nbsp; <span class="chapter-title">Frequently asked questions</span></span></a>
  </div>
</li>
      </ul>
</li>
    </ul>
</div>
</nav><div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active"><h2 id="toc-title">Table of contents</h2>
   
  <ul>
<li>
<a href="#dose-ora" id="toc-dose-ora" class="nav-link active" data-scroll-target="#dose-ora"><span class="header-section-number">8.1</span> Disease over-representation analysis</a>
  <ul class="collapse">
<li><a href="#dose-do-ora" id="toc-dose-do-ora" class="nav-link" data-scroll-target="#dose-do-ora"><span class="header-section-number">8.1.1</span> Over-representation analysis for disease ontology</a></li>
  <li><a href="#dose-ncg-ora" id="toc-dose-ncg-ora" class="nav-link" data-scroll-target="#dose-ncg-ora"><span class="header-section-number">8.1.2</span> Over-representation analysis for the network of cancer gene</a></li>
  <li><a href="#dose-dgn-ora" id="toc-dose-dgn-ora" class="nav-link" data-scroll-target="#dose-dgn-ora"><span class="header-section-number">8.1.3</span> Over-representation analysis for the disease gene network</a></li>
  </ul>
</li>
  <li>
<a href="#dose-gsea" id="toc-dose-gsea" class="nav-link" data-scroll-target="#dose-gsea"><span class="header-section-number">8.2</span> Disease gene set enrichment analysis</a>
  <ul class="collapse">
<li><a href="#gsedo-fuction" id="toc-gsedo-fuction" class="nav-link" data-scroll-target="#gsedo-fuction"><span class="header-section-number">8.2.1</span> <code>gseDO</code> fuction</a></li>
  <li><a href="#gsencg-fuction" id="toc-gsencg-fuction" class="nav-link" data-scroll-target="#gsencg-fuction"><span class="header-section-number">8.2.2</span> <code>gseNCG</code> fuction</a></li>
  <li><a href="#gsedgn-fuction" id="toc-gsedgn-fuction" class="nav-link" data-scroll-target="#gsedgn-fuction"><span class="header-section-number">8.2.3</span> <code>gseDGN</code> fuction</a></li>
  </ul>
</li>
  <li><a href="#references" id="toc-references" class="nav-link" data-scroll-target="#references">References</a></li>
  </ul><div class="toc-actions"><ul><li><a href="https://github.com/YuLab-SMU/biomedical-knowledge-mining-book/edit/main/025-do-enrichment.qmd" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/YuLab-SMU/biomedical-knowledge-mining-book/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content"><header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./02-Enrichment.html">Enrichment Analysis</a></li><li class="breadcrumb-item"><a href="./025-do-enrichment.html"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Disease enrichment analysis</span></a></li></ol></nav><div class="quarto-title">
<h1 class="title"><span id="dose-enrichment" class="quarto-section-identifier"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Disease enrichment analysis</span></span></h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header><p>We developed <a href="http://bioconductor.org/packages/DOSE">DOSE</a> <span class="citation" data-cites="yu_dose_2015">(<a href="#ref-yu_dose_2015" role="doc-biblioref">Yu et al. 2015</a>)</span> package to promote the investigation of diseases. <a href="http://bioconductor.org/packages/DOSE">DOSE</a> provides five methods for <a href="012-DOSE-semantic-similarity.html">measuring semantic similarities among DO terms and gene products</a>, hypergeometric model and gene set enrichment analysis (GSEA) for associating disease with gene list and extracting disease association insight from genome wide expression profiles.</p>
<section id="dose-ora" class="level2" data-number="8.1"><h2 data-number="8.1" class="anchored" data-anchor-id="dose-ora">
<span class="header-section-number">8.1</span> Disease over-representation analysis</h2>
<p><a href="http://bioconductor.org/packages/DOSE">DOSE</a> supports enrichment analysis of Disease Ontology (DO) <span class="citation" data-cites="schriml_disease_2011">(<a href="#ref-schriml_disease_2011" role="doc-biblioref">Schriml et al. 2011</a>)</span>, <a href="http://ncg.kcl.ac.uk/">Network of Cancer Gene</a> <span class="citation" data-cites="omer_ncg">(<a href="#ref-omer_ncg" role="doc-biblioref">A. et al. 2016</a>)</span> and <a href="http://disgenet.org/">Disease Gene Network</a> (DisGeNET) <span class="citation" data-cites="janet_disgenet">(<a href="#ref-janet_disgenet" role="doc-biblioref">Janet et al. 2015</a>)</span>. In addition, several visualization methods were provided by <a href="http://bioconductor.org/packages/enrichplot">enrichplot</a> to help interpreting semantic and enrichment results.</p>
<section id="dose-do-ora" class="level3" data-number="8.1.1"><h3 data-number="8.1.1" class="anchored" data-anchor-id="dose-do-ora">
<span class="header-section-number">8.1.1</span> Over-representation analysis for disease ontology</h3>
<p>In the following example, we selected fold change above 1.5 as the differential genes and analyzing their disease association.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb1"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://yulab-smu.top/contribution-knowledge-mining/">DOSE</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">geneList</span><span class="op">)</span></span>
<span><span class="va">gene</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">geneList</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">abs</a></span><span class="op">(</span><span class="va">geneList</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">1.5</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">gene</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "4312"  "8318"  "10874" "55143" "55388" "991"  </code></pre>
</div>
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb3"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DOSE/man/enrichDO.html">enrichDO</a></span><span class="op">(</span>gene          <span class="op">=</span> <span class="va">gene</span>,</span>
<span>              ont           <span class="op">=</span> <span class="st">"HDO"</span>,</span>
<span>              pvalueCutoff  <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>              pAdjustMethod <span class="op">=</span> <span class="st">"BH"</span>,</span>
<span>              universe      <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">geneList</span><span class="op">)</span>,</span>
<span>              minGSSize     <span class="op">=</span> <span class="fl">5</span>,</span>
<span>              maxGSSize     <span class="op">=</span> <span class="fl">500</span>,</span>
<span>              qvalueCutoff  <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>              readable      <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>                       ID           Description GeneRatio  BgRatio RichFactor
DOID:0060306 DOID:0060306 Meier-Gorlin syndrome     6/273  10/6078  0.6000000
DOID:10534     DOID:10534        stomach cancer    25/273 205/6078  0.1219512
DOID:5041       DOID:5041     esophageal cancer    18/273 132/6078  0.1363636
DOID:820         DOID:820           myocarditis     8/273  31/6078  0.2580645
DOID:1107       DOID:1107  esophageal carcinoma    16/273 117/6078  0.1367521
DOID:1612       DOID:1612         breast cancer    35/273 398/6078  0.0879397
             FoldEnrichment   zScore       pvalue    p.adjust      qvalue
DOID:0060306      13.358242 8.481228 1.402836e-06 0.001273775 0.001058772
DOID:10534         2.715090 5.416997 4.109986e-06 0.001865933 0.001550979
DOID:5041          3.035964 5.128253 2.117499e-05 0.006408964 0.005327182
DOID:820           5.745480 5.744019 4.782074e-05 0.009274126 0.007708728
DOID:1107          3.044614 4.842498 5.908904e-05 0.009274126 0.007708728
DOID:1612          1.957866 4.286443 8.098096e-05 0.009274126 0.007708728
                                                                                                                                                                                 geneID
DOID:0060306                                                                                                                                             8318/81620/4174/990/23594/4998
DOID:10534                                             4312/2305/10403/259266/8140/81930/332/2146/8061/4318/3576/8792/6352/4288/3131/2952/347902/3572/563/7031/6926/3117/2018/2066/3169
DOID:5041                                                                                      4312/3868/8140/7850/2146/4321/2921/4318/3576/6890/2952/563/7373/771/2018/2066/3169/11122
DOID:820                                                                                                                                       6280/6279/3627/29851/8792/1493/2697/4982
DOID:1107                                                                                                4312/3868/8140/2146/4321/4318/3576/6890/2952/563/7373/771/2018/2066/3169/11122
DOID:1612    4312/6241/983/2146/3887/6790/891/23532/3161/993/5347/55215/55723/875/8438/9700/5888/7083/898/1493/6352/4288/3551/185/2952/367/1634/4582/3479/9370/3117/652/4137/3169/10551
             Count
DOID:0060306     6
DOID:10534      25
DOID:5041       18
DOID:820         8
DOID:1107       16
DOID:1612       35</code></pre>
</div>
</div>
<p>The <code><a href="https://rdrr.io/pkg/DOSE/man/enrichDO.html">enrichDO()</a></code> function requires an entrezgene ID vector as input, mostly is the differential gene list of gene expression profile studies. Please refer to <a href="utilities.html#id-convert">session 16.1</a> if you need to conver other gene ID types to entrezgene ID.</p>
<p>The <code>ont</code> parameter can be “HDO” (Human Disease Ontology), “HPO” (Human Phenotype Ontology) or “MPO” (Mouse Phenotype Ontology). <code>pvalueCutoff</code> setting the cutoff value of <em>p</em> value and adjusted <em>p</em> value; <code>pAdjustMethod</code> setting the <em>p</em> value correction methods, include the Bonferroni correction (“bonferroni”), Holm (“holm”), Hochberg (“hochberg”), Hommel (“hommel”), Benjamini &amp; Hochberg (“BH”) and Benjamini &amp; Yekutieli (“BY”) while <code>qvalueCutoff</code> is used to control <em>q</em>-values.</p>
<p>The <code>universe</code> setting the background gene universe for testing. If user do not explicitly setting this parameter, <code><a href="https://rdrr.io/pkg/DOSE/man/enrichDO.html">enrichDO()</a></code> will set the universe to all human genes that have DO annotation.</p>
<p>The <code>minGSSize</code> (and <code>maxGSSize</code>) indicates that only those DO terms that have more than <code>minGSSize</code> (and less than <code>maxGSSize</code>) genes annotated will be tested.</p>
<p>The <code>readable</code> is a logical parameter, indicates whether the entrezgene IDs will mapping to gene symbols or not (see also <a href="utilities.html#setReadable">setReadable</a>).</p>
</section><section id="dose-ncg-ora" class="level3" data-number="8.1.2"><h3 data-number="8.1.2" class="anchored" data-anchor-id="dose-ncg-ora">
<span class="header-section-number">8.1.2</span> Over-representation analysis for the network of cancer gene</h3>
<p><a href="http://ncg.kcl.ac.uk/">Network of Cancer Gene</a> (NCG) <span class="citation" data-cites="omer_ncg">(<a href="#ref-omer_ncg" role="doc-biblioref">A. et al. 2016</a>)</span> is a manually curated repository of cancer genes. NCG release 5.0 (Aug.&nbsp;2015) collects 1,571 cancer genes from 175 published studies. <a href="http://bioconductor.org/packages/DOSE">DOSE</a> supports analyzing gene list and determine whether they are enriched in genes known to be mutated in a given cancer type.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb5"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">gene2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">geneList</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">abs</a></span><span class="op">(</span><span class="va">geneList</span><span class="op">)</span> <span class="op">&lt;</span> <span class="fl">3</span><span class="op">]</span></span>
<span><span class="va">ncg</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DOSE/man/enrichNCG.html">enrichNCG</a></span><span class="op">(</span><span class="va">gene2</span><span class="op">)</span> </span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">ncg</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>                                                                     ID
pan-cancer_paediatric                             pan-cancer_paediatric
triple_negative_breast_cancer             triple_negative_breast_cancer
bladder_cancer                                           bladder_cancer
pancreatic_cancer_(all_histologies) pancreatic_cancer_(all_histologies)
soft_tissue_sarcoma                                 soft_tissue_sarcoma
paediatric_high-grade_glioma               paediatric_high-grade_glioma
                                                            Description
pan-cancer_paediatric                             pan-cancer_paediatric
triple_negative_breast_cancer             triple_negative_breast_cancer
bladder_cancer                                           bladder_cancer
pancreatic_cancer_(all_histologies) pancreatic_cancer_(all_histologies)
soft_tissue_sarcoma                                 soft_tissue_sarcoma
paediatric_high-grade_glioma               paediatric_high-grade_glioma
                                    GeneRatio  BgRatio RichFactor
pan-cancer_paediatric                162/2281 183/3177  0.8852459
triple_negative_breast_cancer         71/2281  75/3177  0.9466667
bladder_cancer                        97/2281 112/3177  0.8660714
pancreatic_cancer_(all_histologies)   40/2281  42/3177  0.9523810
soft_tissue_sarcoma                   26/2281  26/3177  1.0000000
paediatric_high-grade_glioma          25/2281  25/3177  1.0000000
                                    FoldEnrichment   zScore       pvalue
pan-cancer_paediatric                     1.232979 5.179242 1.833773e-08
triple_negative_breast_cancer             1.318527 4.453534 4.290660e-07
bladder_cancer                            1.206273 3.545563 1.253690e-04
pancreatic_cancer_(all_histologies)       1.326486 3.397967 1.262162e-04
soft_tissue_sarcoma                       1.392810 3.208441 1.742793e-04
paediatric_high-grade_glioma              1.392810 3.145636 2.434966e-04
                                        p.adjust       qvalue
pan-cancer_paediatric               1.613721e-06 7.721152e-07
triple_negative_breast_cancer       1.887890e-05 9.032967e-06
bladder_cancer                      2.776757e-03 1.328592e-03
pancreatic_cancer_(all_histologies) 2.776757e-03 1.328592e-03
soft_tissue_sarcoma                 3.067315e-03 1.467615e-03
paediatric_high-grade_glioma        3.073768e-03 1.470702e-03
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                geneID
pan-cancer_paediatric               2146/55353/4609/1029/3575/22806/3418/3066/2120/30012/867/7468/7545/3195/865/64109/4613/613/11177/7490/238/10736/10054/5771/4893/140885/1785/9760/3417/6597/6476/9126/4869/10320/7307/80204/1050/10992/8028/2312/6608/896/894/2196/4849/7023/5093/5079/5293/5727/55181/171017/51322/5781/3718/55294/60/673/8085/5897/4851/1665/51176/1108/7764/10664/6098/2332/2201/6495/3845/7015/1441/2782/64919/4298/23512/8239/29102/6929/8021/6134/6598/4209/5290/22941/8726/207/3717/2033/10716/4928/6932/694/5156/10019/6886/9968/7080/2623/7874/1654/4149/3020/23219/55252/55729/10735/5728/4853/23451/51341/387/3206/6146/79718/2624/63035/3815/171023/23269/25/23592/5896/7403/2260/54880/3716/9203/57178/6777/5789/4297/29072/90/546/120/25836/8289/4345/9611/5925/4763/1997/1499/7157/3399/5295/1387/4602/51564/1027/4005/2322/2078/678/6403/55709/1277/7494/64061/2625
triple_negative_breast_cancer                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       6790/898/4609/1029/1789/4436/2120/867/7128/1788/1030/7490/2271/238/675/2047/4914/1316/5291/5293/5781/55294/8085/4851/4170/3845/355/1616/4854/5290/207/2033/4233/29110/2903/5979/5728/4853/2624/3815/10000/7403/2260/55193/472/5789/4297/2065/4286/8626/8405/8289/10499/55164/5925/4763/23405/1499/4921/7157/5295/1387/2078/324/7248/7048/22894/3480/2045/2066/2625
bladder_cancer                                                                                                                                                                                                                                                                                                                                                            9700/57211/2175/9603/1029/11168/2072/8997/79949/54663/688/6882/4893/8454/6693/56288/2195/10992/1026/64783/896/677/26038/6256/55294/60/8085/4851/841/3265/7175/1999/730051/3845/23484/7015/8243/10605/8295/4854/5290/51043/2033/4780/23224/23217/2064/23385/55252/10735/8241/10672/5728/4853/23451/374291/387/7799/171023/288/30849/4152/9794/7403/287/57634/463/472/4297/2065/2262/3280/23232/8289/9611/5925/2068/56339/4763/7157/2186/1387/3910/7536/2261/7248/23037/6709/54961/23345/57125/7832/79633/10628/22906/388/3169
pancreatic_cancer_(all_histologies)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      1029/4771/8997/7159/2011/6597/7307/10992/3710/6710/55294/7091/3845/23654/7046/3096/4089/91/8241/54549/92/23451/63035/7403/55193/23309/472/800/29072/23077/23499/8289/54894/6416/7157/4088/182/7048/2199/26960
soft_tissue_sarcoma                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   999/6850/4914/4342/2185/55294/2041/4851/2044/4058/5290/4486/5297/5728/3815/2324/7403/546/5925/4763/1499/7157/5159/2045/3667/2066
paediatric_high-grade_glioma                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   4609/1029/1019/4613/1030/1956/4914/896/894/673/8493/5290/4233/5156/1021/63035/54880/4916/90/546/4763/7157/5295/595/4915
                                    Count
pan-cancer_paediatric                 162
triple_negative_breast_cancer          71
bladder_cancer                         97
pancreatic_cancer_(all_histologies)    40
soft_tissue_sarcoma                    26
paediatric_high-grade_glioma           25</code></pre>
</div>
</div>
</section><section id="dose-dgn-ora" class="level3" data-number="8.1.3"><h3 data-number="8.1.3" class="anchored" data-anchor-id="dose-dgn-ora">
<span class="header-section-number">8.1.3</span> Over-representation analysis for the disease gene network</h3>
<p><a href="http://disgenet.org/">DisGeNET</a><span class="citation" data-cites="janet_disgenet">(<a href="#ref-janet_disgenet" role="doc-biblioref">Janet et al. 2015</a>)</span> is an integrative and comprehensive resources of gene-disease associations from several public data sources and the literature. It contains gene-disease associations and snp-gene-disease associations.</p>
<p>The enrichment analysis of disease-gene associations is supported by the <code>enrichDGN</code> function and analysis of snp-gene-disease associations is supported by the <code>enrichDGNv</code> function.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb7"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">dgn</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DOSE/man/enrichDGN.html">enrichDGN</a></span><span class="op">(</span><span class="va">gene</span><span class="op">)</span> </span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">dgn</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>               ID                              Description GeneRatio   BgRatio
C0010278 C0010278                         Craniosynostosis    43/497 488/21671
C0853879 C0853879             Invasive carcinoma of breast    42/497 473/21671
C4733092 C4733092 estrogen receptor-negative breast cancer    34/497 356/21671
C3642347 C3642347              Basal-Like Breast Carcinoma    28/497 245/21671
C3642345 C3642345               Luminal A Breast Carcinoma    22/497 153/21671
C0036202 C0036202                              Sarcoidosis    36/497 413/21671
         RichFactor FoldEnrichment    zScore       pvalue     p.adjust
C0010278 0.08811475       3.842122  9.728932 4.609534e-14 2.267976e-10
C0853879 0.08879493       3.871780  9.674768 7.105190e-14 2.267976e-10
C4733092 0.09550562       4.164391  9.223137 2.446675e-12 4.864593e-09
C3642347 0.11428571       4.983271  9.606353 3.047991e-12 4.864593e-09
C3642345 0.14379085       6.269802 10.021789 7.034749e-12 8.438458e-09
C0036202 0.08716707       3.800800  8.804448 7.930882e-12 8.438458e-09
               qvalue
C0010278 1.636811e-10
C0853879 1.636811e-10
C4733092 3.510804e-09
C3642347 3.510804e-09
C3642345 6.090082e-09
C0036202 6.090082e-09
                                                                                                                                                                                                                          geneID
C0010278 4312/8318/6280/1062/6279/6278/3627/820/27299/6362/81620/2146/3002/29968/990/4318/4069/3576/6890/23594/26279/1493/6352/4998/2152/2697/185/4330/5327/4982/1300/3667/2200/9607/3572/563/7031/3479/6424/1846/3117/1308/2625
C0853879        4312/7153/6278/9787/9582/51203/890/983/5080/2146/1111/9232/10855/4171/6664/4102/2173/4318/701/3576/1978/8836/53335/1894/7980/8792/8842/2151/185/2952/367/4982/4582/6926/3479/1602/23158/2066/3169/5304/2625/5241
C4733092                                         2305/6278/79733/6241/81930/81620/2146/3620/29968/11004/8061/3576/1894/2491/7083/8792/214/5327/367/4982/3667/4582/27324/3479/1846/80129/4137/8839/3169/1408/5304/2625/5241/10551
C3642347                                                                          2305/1062/4605/9833/7368/11065/10232/55765/5163/2146/2568/3620/6790/6664/29127/2173/4318/3576/3159/8792/6663/27324/3479/1846/18/3169/2625/5241
C3642345                                                                                                         2305/9833/7153/55355/1111/3161/4318/3576/2001/6663/4288/2152/185/4128/4582/27324/80129/3169/5304/8614/2625/5241
C0036202                               4312/6280/6279/10403/3627/6373/4283/27299/6362/3002/4321/6355/6364/29851/4318/5004/4069/3576/26227/6890/6352/4485/23541/185/7043/6863/2952/4982/25802/4582/2053/3479/3117/2167/80736/1524
         Count
C0010278    43
C0853879    42
C4733092    34
C3642347    28
C3642345    22
C0036202    36</code></pre>
</div>
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb9"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">snp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"rs1401296"</span>, <span class="st">"rs9315050"</span>, <span class="st">"rs5498"</span>, <span class="st">"rs1524668"</span>, <span class="st">"rs147377392"</span>,</span>
<span>         <span class="st">"rs841"</span>, <span class="st">"rs909253"</span>, <span class="st">"rs7193343"</span>, <span class="st">"rs3918232"</span>, <span class="st">"rs3760396"</span>,</span>
<span>         <span class="st">"rs2231137"</span>, <span class="st">"rs10947803"</span>, <span class="st">"rs17222919"</span>, <span class="st">"rs386602276"</span>, <span class="st">"rs11053646"</span>,</span>
<span>         <span class="st">"rs1805192"</span>, <span class="st">"rs139564723"</span>, <span class="st">"rs2230806"</span>, <span class="st">"rs20417"</span>, <span class="st">"rs966221"</span><span class="op">)</span></span>
<span><span class="va">dgnv</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DOSE/man/enrichDGNv.html">enrichDGNv</a></span><span class="op">(</span><span class="va">snp</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">dgnv</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>               ID               Description GeneRatio    BgRatio RichFactor
C0010054 C0010054 Coronary Arteriosclerosis      6/17 440/194515 0.01363636
C0151744 C0151744       Myocardial Ischemia      4/17 103/194515 0.03883495
C0031099 C0031099             Periodontitis      4/17 116/194515 0.03448276
C0007785 C0007785       Cerebral Infarction      4/17 123/194515 0.03252033
C0003850 C0003850          Arteriosclerosis      4/17 267/194515 0.01498127
C0004153 C0004153           Atherosclerosis      4/17 281/194515 0.01423488
         FoldEnrichment   zScore       pvalue     p.adjust qvalue
C0010054       156.0281 30.43647 1.568917e-12 2.761295e-10     NA
C0151744       444.3518 42.07730 1.754840e-10 1.544259e-08     NA
C0031099       394.5538 39.63952 2.839985e-10 1.583793e-08     NA
C0007785       372.0995 38.48983 3.599531e-10 1.583793e-08     NA
C0003850       171.4166 26.05143 8.145389e-09 2.867177e-07     NA
C0004153       162.8763 25.38727 9.996713e-09 2.932369e-07     NA
                                                            geneID Count
C0010054 rs5498/rs147377392/rs11053646/rs1805192/rs2230806/rs20417     6
C0151744                   rs5498/rs147377392/rs11053646/rs1805192     4
C0031099                         rs5498/rs909253/rs1805192/rs20417     4
C0007785                rs147377392/rs11053646/rs1805192/rs2230806     4
C0003850                        rs5498/rs1805192/rs2230806/rs20417     4
C0004153                        rs5498/rs1805192/rs2230806/rs20417     4</code></pre>
</div>
</div>
</section></section><section id="dose-gsea" class="level2" data-number="8.2"><h2 data-number="8.2" class="anchored" data-anchor-id="dose-gsea">
<span class="header-section-number">8.2</span> Disease gene set enrichment analysis</h2>
<section id="gsedo-fuction" class="level3" data-number="8.2.1"><h3 data-number="8.2.1" class="anchored" data-anchor-id="gsedo-fuction">
<span class="header-section-number">8.2.1</span> <code>gseDO</code> fuction</h3>
<p>In the following example, in order to speedup the compilation of this document, only gene sets with size above 120 were tested and only 100 permutations were performed.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb11"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://yulab-smu.top/contribution-knowledge-mining/">DOSE</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">geneList</span><span class="op">)</span></span>
<span><span class="va">y</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DOSE/man/gseDO.html">gseDO</a></span><span class="op">(</span><span class="va">geneList</span>,</span>
<span>           minGSSize     <span class="op">=</span> <span class="fl">120</span>,</span>
<span>           pvalueCutoff  <span class="op">=</span> <span class="fl">0.2</span>,</span>
<span>           pAdjustMethod <span class="op">=</span> <span class="st">"BH"</span>,</span>
<span>           verbose       <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">y</span>, <span class="fl">3</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>                       ID                      Description setSize
DOID:612         DOID:612 primary immunodeficiency disease     194
DOID:0050117 DOID:0050117      disease by infectious agent     466
DOID:3082       DOID:3082        interstitial lung disease     149
             enrichmentScore      NES       pvalue     p.adjust       qvalue
DOID:612           0.4861389 2.173118 1.000000e-10 1.540000e-08 9.578947e-09
DOID:0050117       0.3382869 1.655880 9.518882e-08 7.329539e-06 4.559044e-06
DOID:3082          0.4182272 1.797816 6.441891e-06 3.306837e-04 2.056885e-04
             rank                   leading_edge
DOID:612     2521 tags=45%, list=20%, signal=36%
DOID:0050117 2199 tags=31%, list=18%, signal=27%
DOID:3082    1864 tags=28%, list=15%, signal=24%
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      core_enrichment
DOID:612                                                                                                                                                                                                                                                                                                     55388/7153/9837/29851/6890/9636/1503/1493/7037/4173/3932/3559/6772/51311/3507/3561/917/3574/3575/919/4860/915/22806/4938/1535/3458/959/5336/11151/3702/925/4688/64135/28755/50615/974/1794/3689/5788/916/4068/3937/30009/3394/10525/100/7374/3659/939/4689/5880/7128/6891/6789/930/6573/11322/204/6850/10095/7852/8772/64170/3119/28985/1053/5971/1536/10125/8456/8625/3071/7293/4478/1380/958/5591/9437/10379/54440/3570/3978/3593/10625/29927/3558/735
DOID:0050117 4312/6279/8685/3627/4283/6362/3002/3620/6355/2921/6364/4318/3576/6890/875/1493/6352/4288/3934/59272/4599/54210/3932/80380/713/5551/133/3559/768/1230/6772/51311/6347/6402/5320/64581/4609/81611/3561/91543/6351/6590/9332/1029/1051/3574/6354/3806/7412/1535/3458/81873/959/4783/2950/3162/925/1594/50615/942/2529/51224/1991/1557/10576/9235/25939/6361/3689/1234/916/467/7096/259197/3383/6367/30009/2539/5806/6374/100/2525/1088/3600/3659/2919/6696/3549/940/58528/10541/6646/6285/6891/6396/671/7422/6059/6573/929/142/26191/4282/468/1312/7124/29949/912/1030/5329/3569/4049/7097/10096/56244/50506/2048/345/1378/5133/5743/348/4151/3119/3053/1956/2213/5971/3135/6376/9021/3596/2833/1080/356/1380/286/7133/887/958/4504/728/9402/5610/5054/4880
DOID:3082                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          4312/8685/3627/6373/4283/3002/4318/3576/6890/1493/6352/3559/6772/6347/2643/1482/1237/727897/6354/2950/3162/942/3329/1116/1991/6361/1234/3383/6367/6374/3600/940/10059/7422/999/142/27087/7124/56244/5743/3119/1956</code></pre>
</div>
</div>
</section><section id="gsencg-fuction" class="level3" data-number="8.2.2"><h3 data-number="8.2.2" class="anchored" data-anchor-id="gsencg-fuction">
<span class="header-section-number">8.2.2</span> <code>gseNCG</code> fuction</h3>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb13"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">ncg</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DOSE/man/gseNCG.html">gseNCG</a></span><span class="op">(</span><span class="va">geneList</span>,</span>
<span>              pvalueCutoff  <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>              pAdjustMethod <span class="op">=</span> <span class="st">"BH"</span>,</span>
<span>              verbose       <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">ncg</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DOSE/man/setReadable.html">setReadable</a></span><span class="op">(</span><span class="va">ncg</span>, <span class="st">'org.Hs.eg.db'</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">ncg</span>, <span class="fl">3</span><span class="op">)</span> </span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>                                                       ID
pan-gynecological and breast pan-gynecological and breast
pan-gastric                                   pan-gastric
esophageal_adenocarcinoma       esophageal_adenocarcinoma
                                              Description setSize
pan-gynecological and breast pan-gynecological and breast      43
pan-gastric                                   pan-gastric      49
esophageal_adenocarcinoma       esophageal_adenocarcinoma      92
                             enrichmentScore       NES      pvalue p.adjust
pan-gynecological and breast      -0.5263429 -1.686853 0.004849844 0.138968
pan-gastric                       -0.4993803 -1.651234 0.005146964 0.138968
esophageal_adenocarcinoma         -0.4235434 -1.570026 0.004691119 0.138968
                                qvalue rank                   leading_edge
pan-gynecological and breast 0.1282226 2464 tags=44%, list=20%, signal=36%
pan-gastric                  0.1282226 3280 tags=49%, list=26%, signal=36%
esophageal_adenocarcinoma    0.1282226 2464 tags=34%, list=20%, signal=27%
                                                                                                                                                                                                       core_enrichment
pan-gynecological and breast                                                                          ATM/ZC3H13/NIPBL/SPOP/ARID1A/RASA1/RB1/RNF43/MAP2K4/NF1/CTNNB1/TP53/PIK3R1/CDKN1B/CCND1/ARID5B/MAP3K1/TBX3/GATA3
pan-gastric                                                             BCOR/SOX9/TCF7L2/ATM/CALD1/SEMG2/HTR7/ARID1A/RASA1/RB1/TTBK2/RNF43/CTNNB1/TP53/BCL9/SMAD3/APC/ZFP36L2/TGFBR2/MUC6/MAP3K1/CACNA1C/ATP8B1/CYP4B1
esophageal_adenocarcinoma    NIPBL/ARID1A/RASA1/MSH3/RNF43/CTNNB1/DNAH7/TP53/GATAD1/PIK3R1/STK11/CDKN1B/SPART/APC/SLIT3/LIN7A/SLIT2/TGFBR2/EPHA3/MUC6/ABCB1/NAV3/SYNE1/MAP3K1/SEMA5A/FTO/NUAK1/PCDH9/CHL1/ABCG2/TRIM58</code></pre>
</div>
</div>
</section><section id="gsedgn-fuction" class="level3" data-number="8.2.3"><h3 data-number="8.2.3" class="anchored" data-anchor-id="gsedgn-fuction">
<span class="header-section-number">8.2.3</span> <code>gseDGN</code> fuction</h3>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb15"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">dgn</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DOSE/man/gseDGN.html">gseDGN</a></span><span class="op">(</span><span class="va">geneList</span>,</span>
<span>              pvalueCutoff  <span class="op">=</span> <span class="fl">0.2</span>,</span>
<span>              pAdjustMethod <span class="op">=</span> <span class="st">"BH"</span>,</span>
<span>              verbose       <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">dgn</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DOSE/man/setReadable.html">setReadable</a></span><span class="op">(</span><span class="va">dgn</span>, <span class="st">'org.Hs.eg.db'</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">dgn</span>, <span class="fl">3</span><span class="op">)</span> </span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>               ID                  Description setSize enrichmentScore      NES
C0024266 C0024266 Lymphocytic Choriomeningitis     120       0.5712593 2.389853
C4721414 C4721414         Mantle cell lymphoma     368       0.4107437 1.989531
C0151779 C0151779           Cutaneous Melanoma     415       0.3963772 1.943568
               pvalue     p.adjust       qvalue rank
C0024266 1.152217e-10 2.365214e-07 2.024264e-07 2579
C4721414 1.000000e-10 2.365214e-07 2.024264e-07 1745
C0151779 1.000000e-10 2.365214e-07 2.024264e-07 1864
                           leading_edge
C0024266 tags=48%, list=21%, signal=38%
C4721414 tags=26%, list=14%, signal=23%
C0151779 tags=27%, list=15%, signal=24%
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       core_enrichment
C0024266                                                                                                                                                                                                                                                                                                                       S100A9/CXCL10/CXCL9/EZH2/GZMB/ICOS/USP18/CXCL8/CTLA4/TREM1/PRF1/ADM/CA9/STAT1/CCL2/SELL/CDKN2A/IL7/IL7R/IFNG/CCR5/IL27RA/SH2D1A/FCER1G/CDK2AP2/CPVL/CD27/PSMB10/PTPN22/SLAMF1/KDM1A/TNF/IL6/FGL2/TLR2/RPAIN/NELFCD/PDCD1/WAS/HIF1A/ATP5F1B/FCGR2B/EGR2/STX11/CXCR3/TYROBP/YME1L1/SOSTDC1/PTPN2/TRAF1/HNF1A/IRF9/PML/NR0B2/IL2/TOX/AGFG1
C4721414                                                                                                            CDC20/MELK/E2F8/APOBEC3B/PBK/TPX2/RAD51AP1/DUSP2/CDT1/EZH2/AURKB/CHEK1/AURKA/CCNB1/PSAT1/SOX11/PRAME/CDC6/PLK1/MMP9/EIF4EBP1/SPIB/RAD51/CD38/MMP7/MCM6/CTSC/LCK/MNX1/SKP2/STAT1/PRDM1/MS4A4A/IGK/MYC/PCNA/IFI27/PSMG1/CCR7/GMNN/E2F1/CDKN2A/PSMB9/NME1/LTB/IGHD/CD40LG/LAIR1/IGF2BP3/LBR/COL11A2/MSH2/CD79B/APRT/NSD2/CDK4/PTPRC/PLSCR1/CCR5/G6PD/CHEK2/HILPDA/DCK/PIM2/WNT3/CD6/CD28/MTAP/PRDX1/MRC1/TUBB3/VEGFA/CD19/HACD1/SOX4/PMCH/ST14/PARP1/TCL1A/DNMT1/IGLL5/SYK/TNF/MYCN/CD1D/NXT1/CDKN2B/RANGAP1/IL6/LTA/PSMD2/CXCR4/BCR/FCER2/FADD/PTGS2
C0151779 MMP1/CDCA8/S100A8/TOP2A/PBK/ISG20/NEK2/CCL18/PDK1/EZH2/AURKB/CHAF1B/KIF18B/MMP12/LAG3/DTL/PTTG1/SOX11/CDKN3/PRAME/PLK1/FABP7/MMP9/GBP1/CXCL8/CTLA4/SOX10/PROM1/MKI67/FANCA/MCM6/MCM4/PITX1/DBF4/IL2RA/TYMS/PLA2G2A/MYC/PCNA/NDRG1/CDKAL1/BLM/E2F1/OCA2/CD163/CDKN2A/NME1/TCF21/KMO/VCAM1/RUNX3/IFNG/MCM7/KYNU/STIP1/CCDC88C/GSTP1/CHAF1A/MSH2/CYP27B1/MAGEA1/CDKN2C/PRDM16/MX2/CDK4/GAPDH/MCM3/ICAM1/CCL22/CHEK2/CXCL5/ERCC4/SPP1/TEAD4/MTAP/HCP5/MRC1/TUBB3/CD27/S100B/CSPG4/PLP2/NBN/VEGFA/IFNW1/CDK16/CDH1/CCNF/CBLIF/SOX4/PARP1/SLC16A1/DNMT1/MAP2K1/ATG5/TNF/NXT1/CDKN2B/RBX1/IL6/TLR2/CXCR4/PDCD1/PTGS2/HIF1A/JMJD6/XRCC3/HLA-DQB1/PVR/CASP10/TGFA/EGFR</code></pre>
</div>
</div>
</section></section><section id="references" class="level2 unnumbered"><h2 class="unnumbered anchored" data-anchor-id="references">References</h2>


<div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list">
<div id="ref-omer_ncg" class="csl-entry" role="listitem">
A., Omer, Giovanni M. D., Thanos P. M., and Francesca D. C. 2016. <span>“NCG 5.0: Updates of a Manually Curated Repository of Cancer Genes and Associated Properties from Cancer Mutational Screenings.”</span> <em>Nucleic Acids Research</em> 44 (D1): D992–99. <a href="https://doi.org/10.1093/nar/gkv1123">https://doi.org/10.1093/nar/gkv1123</a>.
</div>
<div id="ref-janet_disgenet" class="csl-entry" role="listitem">
Janet, P., Núria Q. R., Àlex B., Jordi D. P., Anna B. M., Martin B., Ferran S., and Laura I. F. 2015. <span>“DisGeNET: A Discovery Platform for the Dynamical Exploration of Human Diseases and Their Genes.”</span> <em>Database</em> 2015 (March): bav028. <a href="https://doi.org/10.1093/database/bav028">https://doi.org/10.1093/database/bav028</a>.
</div>
<div id="ref-schriml_disease_2011" class="csl-entry" role="listitem">
Schriml, L. M., C. Arze, S. Nadendla, Y.-W. W. Chang, M. Mazaitis, V. Felix, G. Feng, and W. A. Kibbe. 2011. <span>“Disease Ontology: A Backbone for Disease Semantic Integration.”</span> <em>Nucleic Acids Research</em> 40 (D1): D940–46. <a href="https://doi.org/10.1093/nar/gkr972">https://doi.org/10.1093/nar/gkr972</a>.
</div>
<div id="ref-yu_dose_2015" class="csl-entry" role="listitem">
Yu, Guangchuang, Li-Gen Wang, Guang-Rong Yan, and Qing-Yu He. 2015. <span>“<span>DOSE</span>: An r/Bioconductor Package for Disease Ontology Semantic and Enrichment Analysis.”</span> <em>Bioinformatics</em> 31 (4): 608–9. <a href="https://doi.org/10.1093/bioinformatics/btu684">https://doi.org/10.1093/bioinformatics/btu684</a>.
</div>
</div>
</section></main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
      const outerScaffold = trigger.parentElement.cloneNode(true);
      const codeEl = outerScaffold.querySelector('code');
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp("https:\/\/yulab-smu\.top\/biomedical-knowledge-mining-book\/");
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script><nav class="page-navigation"><div class="nav-page nav-page-previous">
      <a href="./024-reactome.html" class="pagination-link" aria-label="Reactome enrichment analysis">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Reactome enrichment analysis</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./026-meshes-enrichment.html" class="pagination-link" aria-label="MeSH enrichment analysis">
        <span class="nav-page-text"><span class="chapter-number">9</span>&nbsp; <span class="chapter-title">MeSH enrichment analysis</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->
<footer class="footer"><div class="nav-footer">
    <div class="nav-footer-left">
<p>Biomedical Knowledge Mining using GOSemSim and clusterProfiler was written by Guangchuang Yu.</p>
</div>   
    <div class="nav-footer-center">
      &nbsp;
    <div class="toc-actions d-sm-block d-md-none"><ul><li><a href="https://github.com/YuLab-SMU/biomedical-knowledge-mining-book/edit/main/025-do-enrichment.qmd" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/YuLab-SMU/biomedical-knowledge-mining-book/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div></div>
    <div class="nav-footer-right">
<p>This book was built with <a href="https://quarto.org/">Quarto</a>.</p>
</div>
  </div>
</footer>


</body></html>