<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-1.8.25">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<meta name="author" content="Guangchuang Yu">
<title>12&nbsp; Biological theme comparison – Biomedical Knowledge Mining using GOSemSim and clusterProfiler</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>

<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./enrichplot.html" rel="next">
<link href="./028-chipseeker.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js" type="module"></script>
<script src="site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="site_libs/quarto-html/axe/axe-check.js" type="module"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-7b89279ff1a6dce999919e0e67d4d9ec.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-70a47bd5681a7291082a5b9f83d58762.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light"><script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script><script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script><script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script><script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN" && texText && texText.data) {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>
</head>
<body class="nav-sidebar floating quarto-light">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top"><nav class="quarto-secondary-nav"><div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./02-Enrichment.html">Enrichment Analysis</a></li><li class="breadcrumb-item"><a href="./029-compareCluster.html"><span class="chapter-number">12</span>&nbsp; <span class="chapter-title">Biological theme comparison</span></a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav></header><!-- content --><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto"><div class="pt-lg-2 mt-2 text-left sidebar-header sidebar-header-stacked">
      <a href="./index.html" class="sidebar-logo-link">
      </a>
    <div class="sidebar-title mb-0 py-0">
      <a href="./">Biomedical Knowledge Mining using GOSemSim and clusterProfiler</a> 
        <div class="sidebar-tools-main">
    <a href="https://github.com/YuLab-SMU/biomedical-knowledge-mining-book/" title="Source Code" class="quarto-navigation-tool px-1" aria-label="Source Code"><i class="bi bi-github"></i></a>
  <a href="" class="quarto-reader-toggle quarto-navigation-tool px-1" onclick="window.quartoToggleReader(); return false;" title="Toggle reader mode">
  <div class="quarto-reader-toggle-btn">
  <i class="bi"></i>
  </div>
</a>
</div>
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Welcome</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="./01-semantic-similarity.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Semantic Similarity Analysis</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./011-GOSemSim.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">GO semantic similarity analysis</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./012-DOSE-semantic-similarity.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">DO semantic similarity analysis</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./013-meshes-semantic-similarity.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">MeSH semantic similarity analysis</span></span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="./02-Enrichment.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Enrichment Analysis</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./021-go.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">GO enrichment analysis</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./022-kegg.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">KEGG enrichment analysis</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./023-wikipathways.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">WikiPathways analysis</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./024-reactome.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Reactome enrichment analysis</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./025-do-enrichment.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Disease enrichment analysis</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./026-meshes-enrichment.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">9</span>&nbsp; <span class="chapter-title">MeSH enrichment analysis</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./027-universal-enrichment.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">10</span>&nbsp; <span class="chapter-title">Universal enrichment analysis</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./028-chipseeker.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">11</span>&nbsp; <span class="chapter-title">Functional enrichment analysis of genomic coordinations</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./029-compareCluster.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">12</span>&nbsp; <span class="chapter-title">Biological theme comparison</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./enrichplot.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">13</span>&nbsp; <span class="chapter-title">Visualization of functional enrichment result</span></span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="./misc.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Miscellaneous topics</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./dplyr.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">14</span>&nbsp; <span class="chapter-title">dplyr verbs for manipulating enrichment result</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./utilities.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">15</span>&nbsp; <span class="chapter-title">Useful utilities</span></span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="./appendix.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Appendix</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./FAQ.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">16</span>&nbsp; <span class="chapter-title">Frequently asked questions</span></span></a>
  </div>
</li>
      </ul>
</li>
    </ul>
</div>
</nav><div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active"><h2 id="toc-title">Table of contents</h2>
   
  <ul>
<li><a href="#comparing-multiple-gene-lists" id="toc-comparing-multiple-gene-lists" class="nav-link active" data-scroll-target="#comparing-multiple-gene-lists"><span class="header-section-number">12.1</span> Comparing multiple gene lists</a></li>
  <li><a href="#formula-interface-of-comparecluster" id="toc-formula-interface-of-comparecluster" class="nav-link" data-scroll-target="#formula-interface-of-comparecluster"><span class="header-section-number">12.2</span> Formula interface of compareCluster</a></li>
  <li>
<a href="#vis-compare-profile" id="toc-vis-compare-profile" class="nav-link" data-scroll-target="#vis-compare-profile"><span class="header-section-number">12.3</span> Visualization of functional profile comparison</a>
  <ul class="collapse">
<li><a href="#compare-dotplot" id="toc-compare-dotplot" class="nav-link" data-scroll-target="#compare-dotplot"><span class="header-section-number">12.3.1</span> Dot plot</a></li>
  <li><a href="#compare-cnetplot" id="toc-compare-cnetplot" class="nav-link" data-scroll-target="#compare-cnetplot"><span class="header-section-number">12.3.2</span> Gene-Concept Network</a></li>
  </ul>
</li>
  <li><a href="#summary" id="toc-summary" class="nav-link" data-scroll-target="#summary"><span class="header-section-number">12.4</span> Summary</a></li>
  <li><a href="#references" id="toc-references" class="nav-link" data-scroll-target="#references">References</a></li>
  </ul><div class="toc-actions"><ul><li><a href="https://github.com/YuLab-SMU/biomedical-knowledge-mining-book/edit/main/029-compareCluster.qmd" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/YuLab-SMU/biomedical-knowledge-mining-book/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content"><header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./02-Enrichment.html">Enrichment Analysis</a></li><li class="breadcrumb-item"><a href="./029-compareCluster.html"><span class="chapter-number">12</span>&nbsp; <span class="chapter-title">Biological theme comparison</span></a></li></ol></nav><div class="quarto-title">
<h1 class="title"><span id="clusterprofiler-comparecluster" class="quarto-section-identifier"><span class="chapter-number">12</span>&nbsp; <span class="chapter-title">Biological theme comparison</span></span></h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header><p>The <a href="http://bioconductor.org/packages/clusterProfiler">clusterProfiler</a> package was developed for biological theme comparison <span class="citation" data-cites="yu2012 wu_clusterprofiler_2021">(<a href="#ref-yu2012" role="doc-biblioref">Yu et al. 2012</a>; <a href="#ref-wu_clusterprofiler_2021" role="doc-biblioref">Wu et al. 2021</a>)</span>, and it provides a function, <code>compareCluster</code>, to automatically calculate enriched functional profiles of each gene clusters and aggregate the results into a single object. Comparing functional profiles can reveal functional consensus and differences among different experiments and helps in identifying differential functional modules in omics datasets.</p>
<section id="comparing-multiple-gene-lists" class="level2" data-number="12.1"><h2 data-number="12.1" class="anchored" data-anchor-id="comparing-multiple-gene-lists">
<span class="header-section-number">12.1</span> Comparing multiple gene lists</h2>
<p>The <code>compareCluster()</code> function applies selected function (via the <code>fun</code> parameter) to perform enrichment analysis for each gene list.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb1"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">gcSample</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/str.html">str</a></span><span class="op">(</span><span class="va">gcSample</span><span class="op">)</span> </span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>List of 8
 $ X1: chr [1:216] "4597" "7111" "5266" "2175" ...
 $ X2: chr [1:805] "23450" "5160" "7126" "26118" ...
 $ X3: chr [1:392] "894" "7057" "22906" "3339" ...
 $ X4: chr [1:838] "5573" "7453" "5245" "23450" ...
 $ X5: chr [1:929] "5982" "7318" "6352" "2101" ...
 $ X6: chr [1:585] "5337" "9295" "4035" "811" ...
 $ X7: chr [1:582] "2621" "2665" "5690" "3608" ...
 $ X8: chr [1:237] "2665" "4735" "1327" "3192" ...</code></pre>
</div>
</div>
<p>Users can use a named list of gene IDs as the input that passed to the <code>geneCluster</code> parameter.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb3"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">ck</span> <span class="op">&lt;-</span> <span class="fu">compareCluster</span><span class="op">(</span>geneCluster <span class="op">=</span> <span class="va">gcSample</span>, fun <span class="op">=</span> <span class="va">enrichGO</span>, </span>
<span>                    OrgDb <span class="op">=</span> <span class="va">org.Hs.eg.db</span>, ont<span class="op">=</span><span class="st">"BP"</span><span class="op">)</span></span>
<span><span class="va">ck</span> <span class="op">&lt;-</span> <span class="fu">setReadable</span><span class="op">(</span><span class="va">ck</span>, OrgDb <span class="op">=</span> <span class="va">org.Hs.eg.db</span>, keyType<span class="op">=</span><span class="st">"ENTREZID"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">ck</span><span class="op">)</span> </span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>  Cluster         ID
1      X1 GO:0021978
2      X1 GO:0060707
3      X1 GO:0061844
4      X1 GO:0019730
5      X1 GO:0086005
6      X1 GO:0061351
                                                              Description
1                                           telencephalon regionalization
2                                  trophoblast giant cell differentiation
3 antimicrobial humoral immune response mediated by antimicrobial peptide
4                                          antimicrobial humoral response
5                        ventricular cardiac muscle cell action potential
6                                     neural precursor cell proliferation
  GeneRatio   BgRatio RichFactor FoldEnrichment    zScore       pvalue
1     4/199  13/18860 0.30769231      29.161191 10.488659 7.981530e-06
2     4/199  14/18860 0.28571429      27.078248 10.079785 1.108195e-05
3     9/199 135/18860 0.06666667       6.318258  6.403889 1.345178e-05
4    10/199 184/18860 0.05434783       5.150754  5.842684 2.659639e-05
5     5/199  38/18860 0.13157895      12.470246  7.308867 4.703787e-05
6     9/199 168/18860 0.05357143       5.077172  5.481560 7.544726e-05
    p.adjust     qvalue
1 0.01194966 0.01103518
2 0.01194966 0.01103518
3 0.01194966 0.01103518
4 0.01771985 0.01636378
5 0.02507118 0.02315253
6 0.02879290 0.02658943
                                                    geneID Count
1                                      PAX6/LHX2/EMX1/EMX2     4
2                                     SOX15/GJB5/PLK4/ELF5     4
3     CCL20/BPI/KNG1/PLA2G1B/CXCL3/CXCL8/H2BC11/CXCL5/MUC7     9
4 PI3/CCL20/BPI/KNG1/PLA2G1B/CXCL3/CXCL8/H2BC11/CXCL5/MUC7    10
5                           SCN3B/KCNH2/KCND3/KCNE5/CTNNA3     5
6             NF2/PAX6/LHX2/FZD9/LHX5/EPHB1/EMX1/LEF1/EMX2     9</code></pre>
</div>
</div>
</section><section id="formula-interface-of-comparecluster" class="level2" data-number="12.2"><h2 data-number="12.2" class="anchored" data-anchor-id="formula-interface-of-comparecluster">
<span class="header-section-number">12.2</span> Formula interface of compareCluster</h2>
<p>As an alternaitve to using named list, the <code>compareCluster()</code> function also supports passing a formula to describe more complicated experimental designs (<em>e.g.</em>, <span class="math inline">\(Gene \sim time + treatment\)</span>).</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb5"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">geneList</span>, package<span class="op">=</span><span class="st">'DOSE'</span><span class="op">)</span></span>
<span><span class="va">mydf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>Entrez<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">geneList</span><span class="op">)</span>, FC<span class="op">=</span><span class="va">geneList</span><span class="op">)</span></span>
<span><span class="va">mydf</span> <span class="op">&lt;-</span> <span class="va">mydf</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">abs</a></span><span class="op">(</span><span class="va">mydf</span><span class="op">$</span><span class="va">FC</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">1</span>,<span class="op">]</span></span>
<span><span class="va">mydf</span><span class="op">$</span><span class="va">group</span> <span class="op">&lt;-</span> <span class="st">"upregulated"</span></span>
<span><span class="va">mydf</span><span class="op">$</span><span class="va">group</span><span class="op">[</span><span class="va">mydf</span><span class="op">$</span><span class="va">FC</span> <span class="op">&lt;</span> <span class="fl">0</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"downregulated"</span></span>
<span><span class="va">mydf</span><span class="op">$</span><span class="va">othergroup</span> <span class="op">&lt;-</span> <span class="st">"A"</span></span>
<span><span class="va">mydf</span><span class="op">$</span><span class="va">othergroup</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">abs</a></span><span class="op">(</span><span class="va">mydf</span><span class="op">$</span><span class="va">FC</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">2</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"B"</span></span>
<span></span>
<span><span class="va">formula_res</span> <span class="op">&lt;-</span> <span class="fu">compareCluster</span><span class="op">(</span><span class="va">Entrez</span><span class="op">~</span><span class="va">group</span><span class="op">+</span><span class="va">othergroup</span>, data<span class="op">=</span><span class="va">mydf</span>, </span>
<span>                            fun<span class="op">=</span><span class="va">enrichGO</span>, OrgDb <span class="op">=</span> <span class="va">org.Hs.eg.db</span>, ont<span class="op">=</span><span class="st">"BP"</span><span class="op">)</span></span>
<span>  </span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">formula_res</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>          Cluster         group othergroup         ID
1 downregulated.A downregulated          A GO:0061448
2 downregulated.A downregulated          A GO:0030198
3 downregulated.A downregulated          A GO:0043062
4 downregulated.A downregulated          A GO:0045229
5 downregulated.A downregulated          A GO:0051216
6 downregulated.A downregulated          A GO:0007178
                                                              Description
1                                           connective tissue development
2                                       extracellular matrix organization
3                                    extracellular structure organization
4                           external encapsulating structure organization
5                                                   cartilage development
6 cell surface receptor protein serine/threonine kinase signaling pathway
  GeneRatio   BgRatio RichFactor FoldEnrichment   zScore       pvalue
1    37/632 307/18860 0.12052117       3.596565 8.541020 1.602803e-11
2    35/632 331/18860 0.10574018       3.155474 7.366806 2.040034e-09
3    35/632 332/18860 0.10542169       3.145970 7.345591 2.209743e-09
4    35/632 333/18860 0.10510511       3.136523 7.324457 2.392722e-09
5    25/632 216/18860 0.11574074       3.453909 6.754047 7.258181e-08
6    37/632 446/18860 0.08295964       2.475663 5.872578 4.170243e-07
      p.adjust       qvalue
1 7.255888e-08 6.137891e-08
2 2.707963e-06 2.290716e-06
3 2.707963e-06 2.290716e-06
4 2.707963e-06 2.290716e-06
5 6.571557e-05 5.559002e-05
6 3.010653e-04 2.546768e-04
                                                                                                                                                                                           geneID
1     1281/4023/54360/582/7227/81029/3952/585/1513/79068/4488/6653/2202/85477/4856/7494/6097/7481/5159/1396/1289/116039/2690/8817/9658/7704/633/4487/658/2737/5744/2121/2487/1746/6935/3479/80310
2            2192/1281/10218/50509/4319/1290/25903/1513/7075/11096/4313/2621/2191/1471/11117/857/1289/3908/4016/83716/1296/4035/3249/1842/165/10631/1805/2331/3913/4982/7177/7049/2006/7373/10266
3            2192/1281/10218/50509/4319/1290/25903/1513/7075/11096/4313/2621/2191/1471/11117/857/1289/3908/4016/83716/1296/4035/3249/1842/165/10631/1805/2331/3913/4982/7177/7049/2006/7373/10266
4            2192/1281/10218/50509/4319/1290/25903/1513/7075/11096/4313/2621/2191/1471/11117/857/1289/3908/4016/83716/1296/4035/3249/1842/165/10631/1805/2331/3913/4982/7177/7049/2006/7373/10266
5                                                                  1281/54360/582/7227/81029/3952/1513/4488/2202/85477/4856/7481/116039/2690/8817/7704/633/4487/658/2737/5744/2121/2487/1746/6935
6 55198/6469/4681/4926/1281/2530/65997/11167/58495/1149/55273/51232/4488/6653/3625/91851/10468/3693/4856/7494/11117/857/4053/7041/4035/4487/658/2615/5654/2331/7043/6935/9863/2200/7049/7122/6424
  Count
1    37
2    35
3    35
4    35
5    25
6    37</code></pre>
</div>
</div>
</section><section id="vis-compare-profile" class="level2" data-number="12.3"><h2 data-number="12.3" class="anchored" data-anchor-id="vis-compare-profile">
<span class="header-section-number">12.3</span> Visualization of functional profile comparison</h2>
<section id="compare-dotplot" class="level3" data-number="12.3.1"><h3 data-number="12.3.1" class="anchored" data-anchor-id="compare-dotplot">
<span class="header-section-number">12.3.1</span> Dot plot</h3>
<p>We can visualize the result using the <code>dotplot()</code> method.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb7"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu">dotplot</span><span class="op">(</span><span class="va">ck</span><span class="op">)</span></span>
<span><span class="fu">dotplot</span><span class="op">(</span><span class="va">formula_res</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="cell-output-display">
<div id="fig-ccn" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-scap="Comparing enrichment results of multiple gene lists. ">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-ccn-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="029-compareCluster_files/figure-html/fig-ccn-1.png" class="img-fluid figure-img" style="width:100.0%" data-fig-scap="Comparing enrichment results of multiple gene lists. ">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-ccn-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;12.1: <strong>Comparing enrichment results of multiple gene lists.</strong> (A) Using a named list of gene clusters, the results were displayed as multiple columns with each one represents an enrichment result of a gene cluster. (B) Using formula interface, the columns represent gene clusters defined by the formula.
</figcaption></figure>
</div>
</div>
</div>
<p>The fomula interface allows more complicated gene cluster definition. In <a href="#fig-ccn" class="quarto-xref">Figure&nbsp;<span>12.1</span></a>(B), the gene clusters were defined by two variables (i.e.&nbsp;<code>group</code> that divides genes into <code>upregulated</code> and <code>downregulated</code> and <code>othergroup</code> that divides the genes into two categories of <code>A</code> and <code>B</code>.). The <code>dotplot()</code> function allows us to use one variable to divide the result into different facet and plot the result with other variables in each facet panel (<a href="#fig-ccf" class="quarto-xref">Figure&nbsp;<span>12.2</span></a>).</p>
<div class="cell" data-crop="true">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb8"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu">dotplot</span><span class="op">(</span><span class="va">formula_res</span>, x<span class="op">=</span><span class="st">"group"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">facet_grid</span><span class="op">(</span><span class="op">~</span><span class="va">othergroup</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output-display">
<div id="fig-ccf" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-scap="Comparing enrichment results of multiple gene lists defined by multiple variable. ">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-ccf-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="029-compareCluster_files/figure-html/fig-ccf-1.png" class="img-fluid figure-img" style="width:90.0%" data-fig-scap="Comparing enrichment results of multiple gene lists defined by multiple variable. ">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-ccf-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;12.2: <strong>Comparing enrichment results of multiple gene lists defined by multiple variable.</strong> The results were visualized as a dot plot with an x-axis representing one level of conditions (the <em>group</em> variable) and a facet panel indicating another level of conditions (the <em>othergroup</em> variable).
</figcaption></figure>
</div>
</div>
</div>
<p>By default, only top 5 (most significant) categories of each cluster was plotted. User can changes the parameter <code>showCategory</code> to specify how many categories of each cluster to be plotted, and if <code>showCategory</code> was set to <code>NULL</code>, the whole result will be plotted. The <code>showCategory</code> parameter also allows passing a vector of selected categories to <a href="FAQ.html#showing-specific-pathways">plot pathway of interests</a></p>
<p>The <code>dotplot()</code> function tries to make the comparison among different clusters more informative and reasonable. After extracting <em>e.g.</em> 10 categories for each clusters, <a href="http://bioconductor.org/packages/clusterProfiler">clusterProfiler</a> tries to collect overlap of these categories among clusters. For example, <code>term A</code> is enriched in all the gene clusters (<em>e.g.</em>, <code>g1</code> and <code>g2</code>) and is in the 10 most significant categories of <code>g1</code> but not <code>g2</code>. <a href="http://bioconductor.org/packages/clusterProfiler">clusterProfiler</a> will capture this information and include <code>term A</code> in <code>g2</code> cluster to make the comparison in <code>dotplot</code> more reasonable. If users want to ignore these information, they can set <code>includeAll = FALSE</code> in <code>dotplot()</code>, which is not recommended.</p>
<p>The <code>dotplot()</code> function accepts a parameter <code>size</code> for setting the scale of dot sizes. The default parameter <code>size</code> is setting to <code>geneRatio</code>, which corresponding to the <code>GeneRatio</code> column of the output. If it was setting to <code>count</code>, the comparison will be based on gene counts, while if setting to <code>rowPercentage</code>, the dot sizes will be normalized by <code>count/(sum of each row)</code>. Users can also map the dot size to other variables or derived variables (see <a href="dplyr.html">Chapter 16</a>).</p>
<p>To provide the full information, we also provide number of identified genes in each category (numbers in parentheses) when <code>by</code> is setting to <code>rowPercentage</code> and number of gene clusters in each cluster label (numbers in parentheses) when <code>by</code> is setting to <code>geneRatio</code>, as shown in Figure <a href="#fig-ccn" class="quarto-xref">Figure&nbsp;<span>12.1</span></a>.</p>
<p>The p-values indicate that which categories are more likely to have biological meanings. The dots in the plot are color-coded based on their corresponding adjusted p-values. Color gradient ranging from red to blue correspond to the order of increasing adjusted p-values. That is, red indicate low p-values (high enrichment), and blue indicate high p-values (low enrichment). Adjusted p-values were filtered out by the threshold giving by the parameter <code>pvalueCutoff</code>, and FDR can be estimated by <code>qvalue</code>.</p>
</section><section id="compare-cnetplot" class="level3" data-number="12.3.2"><h3 data-number="12.3.2" class="anchored" data-anchor-id="compare-cnetplot">
<span class="header-section-number">12.3.2</span> Gene-Concept Network</h3>
<p>The <a href="enrichplot.html#cnetplot">cnetplot</a> also works with <code>compareCluster()</code> result.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb9"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu">cnetplot</span><span class="op">(</span><span class="va">ck</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output-display">
<div id="fig-mcnetplot" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-scap="`cnetplot()` for comparing functional profiles of multiple gene clusters. ">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-mcnetplot-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="029-compareCluster_files/figure-html/fig-mcnetplot-1.svg" class="img-fluid figure-img" style="width:98.0%" data-fig-scap="`cnetplot()` for comparing functional profiles of multiple gene clusters. ">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-mcnetplot-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;12.3: <strong><code>cnetplot()</code> for comparing functional profiles of multiple gene clusters.</strong> Genes and functional categories (<em>i.e.</em>, pathways) are encoded as pies to distinguish different gene clusters.
</figcaption></figure>
</div>
</div>
</div>
</section></section><section id="summary" class="level2" data-number="12.4"><h2 data-number="12.4" class="anchored" data-anchor-id="summary">
<span class="header-section-number">12.4</span> Summary</h2>
<p>The comparison function was designed as a framework for comparing gene clusters of any kind of ontology associations, not only <a href="021-go.html#go-classification">groupGO</a>, <a href="021-go.html#clusterprofiler-go-ora">enrichGO</a>, <a href="022-kegg.html#clusterprofiler-kegg-pathway-ora">enrichKEGG</a>, <a href="022-kegg.html#clusterprofiler-kegg-module-ora">enrichMKEGG</a>, <a href="023-wikipathways.html#clusterprofiler-wikipathway-ora">enrichWP</a> and <a href="027-universal-enrichment.html">enricher</a> that were provided in this package, but also other biological and biomedical ontologies, including but not limited to <a href="024-reactome.html#reactomepa-ora">enrichPathway</a>, <a href="025-do-enrichment.html#dose-do-ora">enrichDO</a>, <a href="025-do-enrichment.html#dose-ncg-ora">enrichNCG</a>, <a href="025-do-enrichment.html#dose-dgn-ora">enrichDGN</a> and <a href="026-meshes-enrichment.html#meshes-ora">enrichMeSH</a>.</p>
<p>In <span class="citation" data-cites="yu2012">(<a href="#ref-yu2012" role="doc-biblioref">Yu et al. 2012</a>)</span>, we analyzed the publicly available expression dataset of breast tumour tissues from 200 patients (GSE11121, Gene Expression Omnibus) <span class="citation" data-cites="schmidt2008">(<a href="#ref-schmidt2008" role="doc-biblioref">Schmidt et al. 2008</a>)</span>. We identified 8 gene clusters from differentially expressed genes, and using the <code>compareCluster()</code> function to compare these gene clusters by their enriched biological process. In <span class="citation" data-cites="wu_clusterprofiler_2021">(<a href="#ref-wu_clusterprofiler_2021" role="doc-biblioref">Wu et al. 2021</a>)</span>, we analyzed the GSE8057 dataset which contains expression data from ovarian cancer cells at multiple time points and under two treatment conditions. Eight groups of DEG lists were analyzed simultaneously using <code>compareCluster()</code> with WikiPathways. The result indicates that the two drgus have distinct effects at the begining but consistent effects in the later stages (Fig. 4 of <span class="citation" data-cites="wu_clusterprofiler_2021">(<a href="#ref-wu_clusterprofiler_2021" role="doc-biblioref">Wu et al. 2021</a>)</span>).</p>
</section><section id="references" class="level2 unnumbered"><h2 class="unnumbered anchored" data-anchor-id="references">References</h2>


<div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list">
<div id="ref-schmidt2008" class="csl-entry" role="listitem">
Schmidt, Marcus, Daniel B?hm, Christian von T?rne, Eric Steiner, Alexander Puhl, Henryk Pilch, Hans-Anton Lehr, Jan G. Hengstler, Heinz K?lbl, and Mathias Gehrmann. 2008. <span>“The Humoral Immune System Has a Key Prognostic Impact in Node-Negative Breast Cancer.”</span> <em>Cancer Research</em> 68 (13): 5405–13. <a href="https://doi.org/10.1158/0008-5472.CAN-07-5206">https://doi.org/10.1158/0008-5472.CAN-07-5206</a>.
</div>
<div id="ref-wu_clusterprofiler_2021" class="csl-entry" role="listitem">
Wu, Tianzhi, Erqiang Hu, Shuangbin Xu, Meijun Chen, Pingfan Guo, Zehan Dai, Tingze Feng, et al. 2021. <span>“<span class="nocase">clusterProfiler</span> 4.0: A Universal Enrichment Tool for Interpreting Omics Data.”</span> <em>The Innovation</em> 2 (3): 100141. <a href="https://doi.org/10.1016/j.xinn.2021.100141">https://doi.org/10.1016/j.xinn.2021.100141</a>.
</div>
<div id="ref-yu2012" class="csl-entry" role="listitem">
Yu, Guangchuang, Le-Gen Wang, Yanyan Han, and Qing-Yu He. 2012. <span>“clusterProfiler: An r Package for Comparing Biological Themes Among Gene Clusters.”</span> <em>OMICS: A Journal of Integrative Biology</em> 16 (5): 284–87. <a href="https://doi.org/10.1089/omi.2011.0118">https://doi.org/10.1089/omi.2011.0118</a>.
</div>
</div>
</section></main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
      const outerScaffold = trigger.parentElement.cloneNode(true);
      const codeEl = outerScaffold.querySelector('code');
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp("https:\/\/yulab-smu\.top\/biomedical-knowledge-mining-book\/");
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script><nav class="page-navigation"><div class="nav-page nav-page-previous">
      <a href="./028-chipseeker.html" class="pagination-link" aria-label="Functional enrichment analysis of genomic coordinations">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">11</span>&nbsp; <span class="chapter-title">Functional enrichment analysis of genomic coordinations</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./enrichplot.html" class="pagination-link" aria-label="Visualization of functional enrichment result">
        <span class="nav-page-text"><span class="chapter-number">13</span>&nbsp; <span class="chapter-title">Visualization of functional enrichment result</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->
<footer class="footer"><div class="nav-footer">
    <div class="nav-footer-left">
<p>Biomedical Knowledge Mining using GOSemSim and clusterProfiler was written by Guangchuang Yu.</p>
</div>   
    <div class="nav-footer-center">
      &nbsp;
    <div class="toc-actions d-sm-block d-md-none"><ul><li><a href="https://github.com/YuLab-SMU/biomedical-knowledge-mining-book/edit/main/029-compareCluster.qmd" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/YuLab-SMU/biomedical-knowledge-mining-book/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div></div>
    <div class="nav-footer-right">
<p>This book was built with <a href="https://quarto.org/">Quarto</a>.</p>
</div>
  </div>
</footer>


</body></html>