<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Biomedical Knowledge Mining using GOSemSim and clusterProfiler</title>
  <meta name="description" content="Biomedical knowledge mining using GOSemSim and clusterProfiler." />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="Biomedical Knowledge Mining using GOSemSim and clusterProfiler" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Biomedical knowledge mining using GOSemSim and clusterProfiler." />
  <meta name="github-repo" content="YuLab-SMU/biomedical-knowledge-mining-book" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Biomedical Knowledge Mining using GOSemSim and clusterProfiler" />
  
  <meta name="twitter:description" content="Biomedical knowledge mining using GOSemSim and clusterProfiler." />
  

<meta name="author" content="Guangchuang Yu" />


<meta name="date" content="2021-08-16" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  


<script src="libs/header-attrs-2.10/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path=""><a href="#introduction"><i class="fa fa-check"></i>üìñ Introduction</a>
<ul>
<li class="chapter" data-level="" data-path=""><a href="#motivation"><i class="fa fa-check"></i>üéØ Motivation</a></li>
<li class="chapter" data-level="" data-path=""><a href="#citation"><i class="fa fa-check"></i>üìù Citation</a></li>
<li class="chapter" data-level="" data-path=""><a href="#book-structure"><i class="fa fa-check"></i>üìö Book structure</a></li>
<li class="chapter" data-level="" data-path=""><a href="#want-to-help"><i class="fa fa-check"></i>üíñ Want to help?</a></li>
</ul></li>
<li class="part"><span><b>Part I: Semantic Similarity Analysis</b></span></li>
<li class="chapter" data-level="1" data-path=""><a href="#semantic-similarity-overview"><i class="fa fa-check"></i><b>1</b> Overview of semantic similarity analysis</a>
<ul>
<li class="chapter" data-level="1.1" data-path=""><a href="#information-content-based-methods"><i class="fa fa-check"></i><b>1.1</b> Information content-based methods</a>
<ul>
<li class="chapter" data-level="1.1.1" data-path=""><a href="#resnik-method"><i class="fa fa-check"></i><b>1.1.1</b> Resnik method</a></li>
<li class="chapter" data-level="1.1.2" data-path=""><a href="#lin-method"><i class="fa fa-check"></i><b>1.1.2</b> Lin method</a></li>
<li class="chapter" data-level="1.1.3" data-path=""><a href="#rel-method"><i class="fa fa-check"></i><b>1.1.3</b> Rel method</a></li>
<li class="chapter" data-level="1.1.4" data-path=""><a href="#jiang-method"><i class="fa fa-check"></i><b>1.1.4</b> Jiang method</a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path=""><a href="#graph-based-method"><i class="fa fa-check"></i><b>1.2</b> Graph-based method</a>
<ul>
<li class="chapter" data-level="1.2.1" data-path=""><a href="#wang-method"><i class="fa fa-check"></i><b>1.2.1</b> Wang method</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path=""><a href="#combine-methods"><i class="fa fa-check"></i><b>1.3</b> Combine methods</a>
<ul>
<li class="chapter" data-level="1.3.1" data-path=""><a href="#max"><i class="fa fa-check"></i><b>1.3.1</b> max</a></li>
<li class="chapter" data-level="1.3.2" data-path=""><a href="#avg"><i class="fa fa-check"></i><b>1.3.2</b> avg</a></li>
<li class="chapter" data-level="1.3.3" data-path=""><a href="#rcmax"><i class="fa fa-check"></i><b>1.3.3</b> rcmax</a></li>
<li class="chapter" data-level="1.3.4" data-path=""><a href="#bma"><i class="fa fa-check"></i><b>1.3.4</b> BMA</a></li>
</ul></li>
<li class="chapter" data-level="1.4" data-path=""><a href="#summary"><i class="fa fa-check"></i><b>1.4</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path=""><a href="#GOSemSim"><i class="fa fa-check"></i><b>2</b> GO semantic similarity analysis</a>
<ul>
<li class="chapter" data-level="2.1" data-path=""><a href="#semantic-data"><i class="fa fa-check"></i><b>2.1</b> Semantic data</a></li>
<li class="chapter" data-level="2.2" data-path=""><a href="#gosemsim-supported-organisms"><i class="fa fa-check"></i><b>2.2</b> Supported organisms</a></li>
<li class="chapter" data-level="2.3" data-path=""><a href="#go-semantic-simiarlity"><i class="fa fa-check"></i><b>2.3</b> GO semantic similarity measurement</a></li>
<li class="chapter" data-level="2.4" data-path=""><a href="#gene-go-semantic-similarity"><i class="fa fa-check"></i><b>2.4</b> Gene semantic similarity measurement</a></li>
<li class="chapter" data-level="2.5" data-path=""><a href="#gene-cluster-go-semantic-similarity"><i class="fa fa-check"></i><b>2.5</b> Gene cluster semantic similarity measurement</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path=""><a href="#DOSE-semantic-similarity"><i class="fa fa-check"></i><b>3</b> DO semantic similarity analysis</a>
<ul>
<li class="chapter" data-level="3.1" data-path=""><a href="#do-semantic-similarity"><i class="fa fa-check"></i><b>3.1</b> DO term semantic similarity measurement</a></li>
<li class="chapter" data-level="3.2" data-path=""><a href="#gene-do-semantic-similarity"><i class="fa fa-check"></i><b>3.2</b> Gene semantic similarity measurement</a></li>
<li class="chapter" data-level="3.3" data-path=""><a href="#gene-cluster-do-semantic-similarity"><i class="fa fa-check"></i><b>3.3</b> Gene cluster semantic similarity measurement</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path=""><a href="#meshes-semantic-similarity"><i class="fa fa-check"></i><b>4</b> MeSH semantic similarity analysis</a>
<ul>
<li class="chapter" data-level="4.1" data-path=""><a href="#meshes-supported-organisms"><i class="fa fa-check"></i><b>4.1</b> Supported organisms</a></li>
<li class="chapter" data-level="4.2" data-path=""><a href="#meshes-semantic-simiarlity"><i class="fa fa-check"></i><b>4.2</b> MeSH semantic similarity measurement</a></li>
<li class="chapter" data-level="4.3" data-path=""><a href="#gene-meshes-semantic-similarity"><i class="fa fa-check"></i><b>4.3</b> Gene semantic similarity measurement</a></li>
</ul></li>
<li class="part"><span><b>Part II: Enrichment analysis</b></span></li>
<li class="chapter" data-level="5" data-path=""><a href="#enrichment-overview"><i class="fa fa-check"></i><b>5</b> Overview of enrichment analysis</a>
<ul>
<li class="chapter" data-level="5.1" data-path=""><a href="#terminology"><i class="fa fa-check"></i><b>5.1</b> Terminology</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path=""><a href="#gene-sets-and-pathway"><i class="fa fa-check"></i><b>5.1.1</b> Gene sets and pathway</a></li>
<li class="chapter" data-level="5.1.2" data-path=""><a href="#gene-ontology-go"><i class="fa fa-check"></i><b>5.1.2</b> Gene Ontology (GO)</a></li>
<li class="chapter" data-level="5.1.3" data-path=""><a href="#kyoto-encyclopedia-of-genes-and-genomes-kegg"><i class="fa fa-check"></i><b>5.1.3</b> Kyoto Encyclopedia of Genes and Genomes (KEGG)</a></li>
<li class="chapter" data-level="5.1.4" data-path=""><a href="#other-gene-sets"><i class="fa fa-check"></i><b>5.1.4</b> Other gene sets</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path=""><a href="#ora-algorithm"><i class="fa fa-check"></i><b>5.2</b> Over Representation Analysis</a></li>
<li class="chapter" data-level="5.3" data-path=""><a href="#gsea-algorithm"><i class="fa fa-check"></i><b>5.3</b> Gene Set Enrichment Analysis</a></li>
<li class="chapter" data-level="5.4" data-path=""><a href="#leading-edge-analysis-and-core-enriched-genes"><i class="fa fa-check"></i><b>5.4</b> Leading edge analysis and core enriched genes</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path=""><a href="#clusterprofiler-go"><i class="fa fa-check"></i><b>6</b> GO enrichment analysis</a>
<ul>
<li class="chapter" data-level="6.1" data-path=""><a href="#clusterProfiler-go-supported-organisms"><i class="fa fa-check"></i><b>6.1</b> Supported organisms</a></li>
<li class="chapter" data-level="6.2" data-path=""><a href="#go-classification"><i class="fa fa-check"></i><b>6.2</b> GO classification</a></li>
<li class="chapter" data-level="6.3" data-path=""><a href="#clusterprofiler-go-ora"><i class="fa fa-check"></i><b>6.3</b> GO over-representation analysis</a></li>
<li class="chapter" data-level="6.4" data-path=""><a href="#go-gene-set-enrichment-analysis"><i class="fa fa-check"></i><b>6.4</b> GO Gene Set Enrichment Analysis</a></li>
<li class="chapter" data-level="6.5" data-path=""><a href="#clusterprofiler-go-non-model"><i class="fa fa-check"></i><b>6.5</b> GO analysis for non-model organisms</a></li>
<li class="chapter" data-level="6.6" data-path=""><a href="#visualize-enriched-go-terms-as-a-directed-acyclic-graph"><i class="fa fa-check"></i><b>6.6</b> Visualize enriched GO terms as a directed acyclic graph</a></li>
<li class="chapter" data-level="6.7" data-path=""><a href="#clusterprofiler-go-summary"><i class="fa fa-check"></i><b>6.7</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path=""><a href="#clusterprofiler-kegg"><i class="fa fa-check"></i><b>7</b> KEGG enrichment analysis</a>
<ul>
<li class="chapter" data-level="7.1" data-path=""><a href="#clusterProfiler-kegg-supported-organisms"><i class="fa fa-check"></i><b>7.1</b> Supported organisms</a></li>
<li class="chapter" data-level="7.2" data-path=""><a href="#clusterprofiler-kegg-pathway-ora"><i class="fa fa-check"></i><b>7.2</b> KEGG pathway over-representation analysis</a></li>
<li class="chapter" data-level="7.3" data-path=""><a href="#clusterprofiler-kegg-pathway-gsea"><i class="fa fa-check"></i><b>7.3</b> KEGG pathway gene set enrichment analysis</a></li>
<li class="chapter" data-level="7.4" data-path=""><a href="#clusterprofiler-kegg-module-ora"><i class="fa fa-check"></i><b>7.4</b> KEGG module over-representation analysis</a></li>
<li class="chapter" data-level="7.5" data-path=""><a href="#clusterprofiler-kegg-module-gsea"><i class="fa fa-check"></i><b>7.5</b> KEGG module gene set enrichment analysis</a></li>
<li class="chapter" data-level="7.6" data-path=""><a href="#visualize-enriched-kegg-pathways"><i class="fa fa-check"></i><b>7.6</b> Visualize enriched KEGG pathways</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path=""><a href="#wikipathways-analysis"><i class="fa fa-check"></i><b>8</b> WikiPathways analysis</a>
<ul>
<li class="chapter" data-level="8.1" data-path=""><a href="#clusterprofiler-wikipathway-ora"><i class="fa fa-check"></i><b>8.1</b> WikiPathways over-representation analysis</a></li>
<li class="chapter" data-level="8.2" data-path=""><a href="#clusterprofiler-wikipathway-gsea"><i class="fa fa-check"></i><b>8.2</b> WikiPathways gene set enrichment analysis</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path=""><a href="#reactomepa"><i class="fa fa-check"></i><b>9</b> Reactome enrichment analysis</a>
<ul>
<li class="chapter" data-level="9.1" data-path=""><a href="#reactomepa-supported-organisms"><i class="fa fa-check"></i><b>9.1</b> Supported organisms</a></li>
<li class="chapter" data-level="9.2" data-path=""><a href="#reactomepa-ora"><i class="fa fa-check"></i><b>9.2</b> Reactome pathway over-representation analysis</a></li>
<li class="chapter" data-level="9.3" data-path=""><a href="#reactomepa-gsea"><i class="fa fa-check"></i><b>9.3</b> Reactome pathway gene set enrichment analysis</a></li>
<li class="chapter" data-level="9.4" data-path=""><a href="#pathway-visualization"><i class="fa fa-check"></i><b>9.4</b> Pathway Visualization</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path=""><a href="#dose-enrichment"><i class="fa fa-check"></i><b>10</b> Disease enrichment analysis</a>
<ul>
<li class="chapter" data-level="10.1" data-path=""><a href="#dose-ora"><i class="fa fa-check"></i><b>10.1</b> Disease over-representation analysis</a>
<ul>
<li class="chapter" data-level="10.1.1" data-path=""><a href="#dose-do-ora"><i class="fa fa-check"></i><b>10.1.1</b> Over-representation analysis for disease ontology</a></li>
<li class="chapter" data-level="10.1.2" data-path=""><a href="#dose-ncg-ora"><i class="fa fa-check"></i><b>10.1.2</b> Over-representation analysis for the network of cancer gene</a></li>
<li class="chapter" data-level="10.1.3" data-path=""><a href="#dose-dgn-ora"><i class="fa fa-check"></i><b>10.1.3</b> Over-representation analysis for the disease gene network</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path=""><a href="#dose-gsea"><i class="fa fa-check"></i><b>10.2</b> Disease gene set enrichment analysis</a>
<ul>
<li class="chapter" data-level="10.2.1" data-path=""><a href="#gsedo-fuction"><i class="fa fa-check"></i><b>10.2.1</b> <code>gseDO</code> fuction</a></li>
<li class="chapter" data-level="10.2.2" data-path=""><a href="#gsencg-fuction"><i class="fa fa-check"></i><b>10.2.2</b> <code>gseNCG</code> fuction</a></li>
<li class="chapter" data-level="10.2.3" data-path=""><a href="#gsedgn-fuction"><i class="fa fa-check"></i><b>10.2.3</b> <code>gseDGN</code> fuction</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path=""><a href="#meshes-enrichment"><i class="fa fa-check"></i><b>11</b> MeSH enrichment analysis</a>
<ul>
<li class="chapter" data-level="11.1" data-path=""><a href="#meshes-ora"><i class="fa fa-check"></i><b>11.1</b> MeSH over-representation analysis</a></li>
<li class="chapter" data-level="11.2" data-path=""><a href="#meshes-gsea"><i class="fa fa-check"></i><b>11.2</b> MeSH gene set enrichment analysis</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path=""><a href="#universal-api"><i class="fa fa-check"></i><b>12</b> Universal enrichment analysis</a>
<ul>
<li class="chapter" data-level="12.1" data-path=""><a href="#input-data"><i class="fa fa-check"></i><b>12.1</b> Input data</a></li>
<li class="chapter" data-level="12.2" data-path=""><a href="#cell-marker"><i class="fa fa-check"></i><b>12.2</b> Cell Marker</a>
<ul>
<li class="chapter" data-level="12.2.1" data-path=""><a href="#cell-marker-ora"><i class="fa fa-check"></i><b>12.2.1</b> Cell Marker over-presentaton analysis</a></li>
<li class="chapter" data-level="12.2.2" data-path=""><a href="#cell-marker-gsea"><i class="fa fa-check"></i><b>12.2.2</b> Cell Marker gene set enrichment analysis</a></li>
</ul></li>
<li class="chapter" data-level="12.3" data-path=""><a href="#msigdb-analysis"><i class="fa fa-check"></i><b>12.3</b> MSigDb analysis</a>
<ul>
<li class="chapter" data-level="12.3.1" data-path=""><a href="#msigdb-ora"><i class="fa fa-check"></i><b>12.3.1</b> MSigDb over-presentaton analysis</a></li>
<li class="chapter" data-level="12.3.2" data-path=""><a href="#msigdb-gsea"><i class="fa fa-check"></i><b>12.3.2</b> MSigDb gene set enrichment analysis</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path=""><a href="#chipseeker-enrichment"><i class="fa fa-check"></i><b>13</b> Functional enrichment analysis of genomic coordinations</a></li>
<li class="chapter" data-level="14" data-path=""><a href="#clusterprofiler-comparecluster"><i class="fa fa-check"></i><b>14</b> Biological theme comparison</a>
<ul>
<li class="chapter" data-level="14.1" data-path=""><a href="#comparing-multiple-gene-lists"><i class="fa fa-check"></i><b>14.1</b> Comparing multiple gene lists</a></li>
<li class="chapter" data-level="14.2" data-path=""><a href="#formula-interface-of-comparecluster"><i class="fa fa-check"></i><b>14.2</b> Formula interface of compareCluster</a></li>
<li class="chapter" data-level="14.3" data-path=""><a href="#visualization-of-profile-comparison"><i class="fa fa-check"></i><b>14.3</b> Visualization of profile comparison</a></li>
<li class="chapter" data-level="14.4" data-path=""><a href="#summary-1"><i class="fa fa-check"></i><b>14.4</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path=""><a href="#enrichplot"><i class="fa fa-check"></i><b>15</b> Visualization of functional enrichment result</a>
<ul>
<li class="chapter" data-level="15.1" data-path=""><a href="#bar-plot"><i class="fa fa-check"></i><b>15.1</b> Bar Plot</a></li>
<li class="chapter" data-level="15.2" data-path=""><a href="#dot-plot"><i class="fa fa-check"></i><b>15.2</b> Dot plot</a></li>
<li class="chapter" data-level="15.3" data-path=""><a href="#gene-concept-network"><i class="fa fa-check"></i><b>15.3</b> Gene-Concept Network</a></li>
<li class="chapter" data-level="15.4" data-path=""><a href="#heatmap-like-functional-classification"><i class="fa fa-check"></i><b>15.4</b> Heatmap-like functional classification</a></li>
<li class="chapter" data-level="15.5" data-path=""><a href="#enrichment-map"><i class="fa fa-check"></i><b>15.5</b> Enrichment Map</a></li>
<li class="chapter" data-level="15.6" data-path=""><a href="#biological-theme-comparison"><i class="fa fa-check"></i><b>15.6</b> Biological theme comparison</a></li>
<li class="chapter" data-level="15.7" data-path=""><a href="#upset-plot"><i class="fa fa-check"></i><b>15.7</b> UpSet Plot</a></li>
<li class="chapter" data-level="15.8" data-path=""><a href="#ridgeline-plot-for-expression-distribution-of-gsea-result"><i class="fa fa-check"></i><b>15.8</b> ridgeline plot for expression distribution of GSEA result</a></li>
<li class="chapter" data-level="15.9" data-path=""><a href="#running-score-and-preranked-list-of-gsea-result"><i class="fa fa-check"></i><b>15.9</b> running score and preranked list of GSEA result</a></li>
<li class="chapter" data-level="15.10" data-path=""><a href="#pubmed-trend-of-enriched-terms"><i class="fa fa-check"></i><b>15.10</b> pubmed trend of enriched terms</a></li>
</ul></li>
<li class="part"><span><b>Part III: Miscellaneous topics</b></span></li>
<li class="chapter" data-level="16" data-path=""><a href="#clusterProfiler-dplyr"><i class="fa fa-check"></i><b>16</b> dplyr verbs for manipulating enrichment result</a>
<ul>
<li class="chapter" data-level="16.1" data-path=""><a href="#filter"><i class="fa fa-check"></i><b>16.1</b> filter</a></li>
<li class="chapter" data-level="16.2" data-path=""><a href="#arrange"><i class="fa fa-check"></i><b>16.2</b> arrange</a></li>
<li class="chapter" data-level="16.3" data-path=""><a href="#select"><i class="fa fa-check"></i><b>16.3</b> select</a></li>
<li class="chapter" data-level="16.4" data-path=""><a href="#mutate"><i class="fa fa-check"></i><b>16.4</b> mutate</a></li>
<li class="chapter" data-level="16.5" data-path=""><a href="#slice"><i class="fa fa-check"></i><b>16.5</b> slice</a></li>
<li class="chapter" data-level="16.6" data-path=""><a href="#summarise"><i class="fa fa-check"></i><b>16.6</b> summarise</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path=""><a href="#useful-utilities"><i class="fa fa-check"></i><b>17</b> Useful utilities</a>
<ul>
<li class="chapter" data-level="17.1" data-path=""><a href="#id-convert"><i class="fa fa-check"></i><b>17.1</b> Translating Biological ID</a>
<ul>
<li class="chapter" data-level="17.1.1" data-path=""><a href="#bitr"><i class="fa fa-check"></i><b>17.1.1</b> <code>bitr</code>: Biological Id TranslatoR</a></li>
<li class="chapter" data-level="17.1.2" data-path=""><a href="#bitr_kegg"><i class="fa fa-check"></i><b>17.1.2</b> <code>bitr_kegg</code>: converting biological IDs using KEGG API</a></li>
</ul></li>
<li class="chapter" data-level="17.2" data-path=""><a href="#setReadable"><i class="fa fa-check"></i><b>17.2</b> <code>setReadable</code>: translating gene IDs to human readable symbols</a></li>
<li class="chapter" data-level="17.3" data-path=""><a href="#read-gmt"><i class="fa fa-check"></i><b>17.3</b> Parsing GMT files</a></li>
<li class="chapter" data-level="17.4" data-path=""><a href="#go-utilities"><i class="fa fa-check"></i><b>17.4</b> GO utilities</a>
<ul>
<li class="chapter" data-level="17.4.1" data-path=""><a href="#drop-specific-go-terms-or-level"><i class="fa fa-check"></i><b>17.4.1</b> drop specific GO terms or level</a></li>
<li class="chapter" data-level="17.4.2" data-path=""><a href="#test-go-at-sepcific-level"><i class="fa fa-check"></i><b>17.4.2</b> test GO at sepcific level</a></li>
<li class="chapter" data-level="17.4.3" data-path=""><a href="#reduce-redundancy-of-enriched-go-terms"><i class="fa fa-check"></i><b>17.4.3</b> reduce redundancy of enriched GO terms</a></li>
</ul></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path=""><a href="#faq"><i class="fa fa-check"></i><b>A</b> Frequently asked questions</a>
<ul>
<li class="chapter" data-level="A.1" data-path=""><a href="#genelist"><i class="fa fa-check"></i><b>A.1</b> How to prepare your own geneList</a></li>
<li class="chapter" data-level="A.2" data-path=""><a href="#no-gene-can-be-mapped"><i class="fa fa-check"></i><b>A.2</b> No gene can be mapped</a></li>
<li class="chapter" data-level="A.3" data-path=""><a href="#showing-specific-pathways"><i class="fa fa-check"></i><b>A.3</b> Showing specific pathways</a></li>
<li class="chapter" data-level="A.4" data-path=""><a href="#how-to-extract-genes-of-a-specific-termpathway"><i class="fa fa-check"></i><b>A.4</b> How to extract genes of a specific term/pathway</a></li>
</ul></li>
<li class="chapter" data-level="" data-path=""><a href="#references"><i class="fa fa-check"></i>References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Biomedical Knowledge Mining using GOSemSim and clusterProfiler</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="header">
<h1 class="title">Biomedical Knowledge Mining using GOSemSim and clusterProfiler</h1>
<p class="author"><em>Guangchuang Yu</em></p>
<p class="date" style="margin-top: 1.5em;"><em>2021-08-16</em></p>
</div>
<div id="introduction" class="section level1 unnumbered">
<h1>üìñ Introduction</h1>
<p><a href="https://github.com/YuLab-SMU/ChIPseeker"><img src="https://img.shields.io/badge/ChIPseeker-1.28.3-green.svg" /></a>
<a href="https://github.com/YuLab-SMU/clusterProfiler"><img src="https://img.shields.io/badge/clusterProfiler-4.1.1-green.svg" /></a>
<a href="https://github.com/YuLab-SMU/DOSE"><img src="https://img.shields.io/badge/DOSE-3.19.3-green.svg" /></a>
<a href="https://github.com/YuLab-SMU/enrichplot"><img src="https://img.shields.io/badge/enrichplot-1.13.1-green.svg" /></a>
<a href="https://github.com/YuLab-SMU/GOSemSim"><img src="https://img.shields.io/badge/GOSemSim-2.19.0-green.svg" /></a>
<a href="https://github.com/YuLab-SMU/meshes"><img src="https://img.shields.io/badge/meshes-1.19.0.991-green.svg" /></a>
<a href="https://github.com/YuLab-SMU/ReactomePA"><img src="https://img.shields.io/badge/ReactomePA-1.36.0-green.svg" /></a></p>
<p>hello</p>
<p><img src="figures/clusterProfiler-diagram.png" width="95%" /></p>
<!--
In recently years, high-throughput experimental techniques such as
microarray, RNA-Seq and mass spectrometry can detect cellular
molecules at systems-level. These kinds of analyses generate huge
quantitaties of data, which need to be given a biological
interpretation. A commonly used approach is via clustering in the gene
dimension for grouping different genes based on their similarities[@yu2010].

To search for shared functions among genes, a common way is to
incorporate the biological knowledge, such as Gene Ontology (GO) and
Kyoto Encyclopedia of Genes and Genomes (KEGG), for identifying
predominant biological themes of a collection of genes.

After clustering analysis, researchers not only want to determine
whether there is a common theme of a particular gene cluster, but also
to compare the biological themes among gene clusters. The manual step
to choose interesting clusters followed by enrichment analysis on each
selected cluster is slow and tedious. To bridge this gap, we designed
[clusterProfiler](https://www.bioconductor.org/packages/clusterProfiler)[@yu2012], for comparing and visualizing functional
profiles among gene clusters.

-->
<div id="motivation" class="section level2 unnumbered">
<h2>üéØ Motivation</h2>
<p>The book is meant as a guide for mining biological knowledge to elucidate or interpret molecular mechanisms using a suite of R packages, including <a href="http://bioconductor.org/packages/ChIPseeker">ChIPseeker</a>, <a href="http://bioconductor.org/packages/clusterProfiler">clusterProfiler</a>, <a href="http://bioconductor.org/packages/DOSE">DOSE</a>, <a href="http://bioconductor.org/packages/enrichplot">enrichplot</a>, <a href="http://bioconductor.org/packages/GOSemSim">GOSemSim</a>, <a href="http://bioconductor.org/packages/meshes">meshes</a> and <a href="http://bioconductor.org/packages/ReactomePA">ReactomePA</a>. Hence, if you are starting to read this book, we assume you have a working knowledge of how to use R.</p>
</div>
<div id="citation" class="section level2 unnumbered">
<h2>üìù Citation</h2>
<p>If you use the software suite in published research, please cite the most appropriate paper(s) from this list:</p>
<ol style="list-style-type: decimal">
<li>T Wu<sup>#</sup>, E Hu<sup>#</sup>, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan, X Fu, S Liu, X Bo<sup>*</sup>, <strong>G Yu</strong><sup>*</sup>. clusterProfiler 4.0: A universal enrichment tool for interpreting omics data. <strong><em>The Innovation</em></strong>. 2021, 2(3):100141.
doi: <a href="https://doi.org/10.1016/j.xinn.2021.100141">10.1016/j.xinn.2021.100141</a></li>
<li><strong>G Yu</strong><sup>*</sup>. Gene Ontology Semantic Similarity Analysis Using GOSemSim. In: Kidder B. (eds) Stem Cell Transcriptional Networks. <strong><em>Methods in Molecular Biology</em></strong>. 2020, 2117:207-215. Humana, New York, NY.
doi: <a href="https://doi.org/10.1007/978-1-0716-0301-7_11">10.1007/978-1-0716-0301-7_11</a></li>
<li><strong>G Yu</strong><sup>*</sup>. Using meshes for MeSH term enrichment and semantic analyses. <strong><em>Bioinformatics</em></strong>. 2018, 34(21):3766‚Äì3767.
doi: <a href="https://doi.org/10.1093/bioinformatics/bty410">10.1093/bioinformatics/bty410</a></li>
<li><strong>G Yu</strong>, QY He<sup>*</sup>. ReactomePA: an R/Bioconductor package for reactome pathway analysis and visualization. <strong><em>Molecular BioSystems</em></strong>. 2016, 12(2):477-479.
doi: <a href="https://doi.org/10.1039/C5MB00663E">10.1039/C5MB00663E</a></li>
<li><strong>G Yu</strong><sup>*</sup>, LG Wang, and QY He<sup>*</sup>. ChIPseeker: an R/Bioconductor package for ChIP peak annotation, comparison and visualization. <strong><em>Bioinformatics</em></strong>. 2015, 31(14):2382-2383.
doi: <a href="https://doi.org/10.1093/bioinformatics/btv145">10.1093/bioinformatics/btv145</a></li>
<li><strong>G Yu</strong><sup>*</sup>, LG Wang, GR Yan, QY He<sup>*</sup>. DOSE: an R/Bioconductor package for Disease Ontology Semantic and Enrichment analysis. <strong><em>Bioinformatics</em></strong>. 2015, 31(4):608-609.
doi: <a href="https://doi.org/10.1093/bioinformatics/btu684">10.1093/bioinformatics/btu684</a></li>
<li><strong>G Yu</strong>, LG Wang, Y Han and QY He<sup>*</sup>. clusterProfiler: an R package for comparing biological themes among gene clusters. <strong><em>OMICS: A Journal of Integrative Biology</em></strong>. 2012, 16(5):284-287.
doi: <a href="https://doi.org/10.1089/omi.2011.0118">10.1089/omi.2011.0118</a></li>
<li><strong>G Yu</strong>, F Li, Y Qin, X Bo<sup>*</sup>, Y Wu, S Wang<sup>*</sup>. GOSemSim: an R package for measuring semantic similarity among GO terms and gene products. <strong><em>Bioinformatics</em></strong>. 2010, 26(7):976-978.
doi: <a href="https://doi.org/10.1093/bioinformatics/btq064">10.1093/bioinformatics/btq064</a></li>
</ol>
</div>
<div id="book-structure" class="section level2 unnumbered">
<h2>üìö Book structure</h2>
<ul>
<li>Part 1 (Semantic similarity analysis) describes <a href="http://bioconductor.org/packages/GOSemSim">GOSemSim</a>, <a href="http://bioconductor.org/packages/DOSE">DOSE</a> and <a href="http://bioconductor.org/packages/meshes">meshes</a> packages for measuring semantic similarity of genes or gene products based on Gene Ontology, Disease Ontology and Medical Subject Headings.</li>
<li>Part 2 (Enrichment analysis) introduces over-reprensentation analysis and gene set enrichment analysis using <a href="http://bioconductor.org/packages/clusterProfiler">clusterProfiler</a> (supports GO, KEGG, MSigDb, WikiPathway, and many others via universal interface), <a href="http://bioconductor.org/packages/DOSE">DOSE</a> (DO, Disease-Gene Network, Network of Cancer Genes), <a href="http://bioconductor.org/packages/meshes">meshes</a> (MeSH), and <a href="http://bioconductor.org/packages/ReactomePA">ReactomePA</a> (Reactome pathway). Functional enrichment analysis of Genomic coordination is supported via <a href="http://bioconductor.org/packages/ChIPseeker">ChIPseeker</a> and comparison among multiple conditions is also supported by <a href="http://bioconductor.org/packages/clusterProfiler">clusterProfiler</a>. We implemented a number of visualization methods in the <a href="http://bioconductor.org/packages/enrichplot">enrichplot</a> package to help users to interpret their results.</li>
<li>Part 3 (Miscellaneous topics) describes useful utilities including translating gene IDs and manipulating enrichment results.</li>
</ul>
</div>
<div id="want-to-help" class="section level2 unnumbered">
<h2>üíñ Want to help?</h2>
<p>The book‚Äôs source code is hosted on GitHub, at <a href="https://github.com/YuLab-SMU/biomedical-knowledge-mining-book" class="uri">https://github.com/YuLab-SMU/biomedical-knowledge-mining-book</a>. Any feedback on the book is very welcome. Feel free to <a href="https://github.com/YuLab-SMU/biomedical-knowledge-mining-book/issues/new">open an issue</a> on GitHub or send me a pull request if you notice typos or other issues (I‚Äôm not a native English speaker ;) ).</p>

</div>
</div>



<div id="semantic-similarity-overview" class="section level1" number="1">
<h1><span class="header-section-number">Chapter 1</span> Overview of semantic similarity analysis</h1>
<p>Functional similarity of gene products can be estimated by controlled
biological vocabularies, such as Gene Ontology (GO), Disease Ontology (DO) and Medical Subject Headings (MeSH).</p>
<p>Four methods including Resnik <span class="citation">(<a href="#ref-philip_semantic_1999" role="doc-biblioref">Philip 1999</a>)</span>, Jiang <span class="citation">(<a href="#ref-jiang_semantic_1997" role="doc-biblioref">Jiang and Conrath 1997</a>)</span>, Lin <span class="citation">(<a href="#ref-lin_information-theoretic_1998" role="doc-biblioref">Lin 1998</a>)</span> and Schlicker <span class="citation">(<a href="#ref-schlicker_new_2006" role="doc-biblioref">Schlicker et al. 2006</a>)</span> have been presented to determine the semantic similarity of two GO terms based on the annotation statistics of their common ancestor terms. Wang <span class="citation">(<a href="#ref-wang_new_2007" role="doc-biblioref">Wang et al. 2007</a>)</span>
proposed a method to measure the similarity based on the graph structure of GO. Each of these methods has its own advantages and
weaknesses and can be applied to other ontologies that have similar structure (i.e.¬†directed acyclic graph).</p>
<div id="information-content-based-methods" class="section level2" number="1.1">
<h2><span class="header-section-number">1.1</span> Information content-based methods</h2>
<p>Four methods proposed by Resnik <span class="citation">(<a href="#ref-philip_semantic_1999" role="doc-biblioref">Philip 1999</a>)</span>,
Jiang <span class="citation">(<a href="#ref-jiang_semantic_1997" role="doc-biblioref">Jiang and Conrath 1997</a>)</span>, Lin <span class="citation">(<a href="#ref-lin_information-theoretic_1998" role="doc-biblioref">Lin 1998</a>)</span>
and Schlicker <span class="citation">(<a href="#ref-schlicker_new_2006" role="doc-biblioref">Schlicker et al. 2006</a>)</span> are information content (IC) based, which depend on the frequencies of two GO terms involved and that of their closest common ancestor term in a specific corpus of GO
annotations. The information content of a GO term is computed by the
negative log probability of the term occurring in GO corpus. A rarely used term contains a greater amount of information.</p>
<p>The frequency of a term t is defined as:</p>
<p><span class="math display">\[p(t) = \frac{n_{t&#39;}}{N} | t&#39; \in \left\{t, \; children\: of\: t \right\}\]</span></p>
<p>where <span class="math inline">\(n_{t&#39;}\)</span> is the number of term <span class="math inline">\(t&#39;\)</span>, and <span class="math inline">\(N\)</span> is the total number of terms in GO corpus.</p>
<p>Thus the information content is defined as:</p>
<p><span class="math display">\[IC(t) = -\log(p(t))\]</span></p>
<p>As GO allow multiple parents for each concept, two terms can share
parents by multiple paths. IC-based methods calculate similarity of two GO terms based on the information content of their closest common ancestor term, which was also called most informative common ancestor (MICA).</p>
<div id="resnik-method" class="section level3" number="1.1.1">
<h3><span class="header-section-number">1.1.1</span> Resnik method</h3>
<p>The Resnik method is defined as:</p>
<p><span class="math display">\[sim_{Resnik}(t_1,t_2) = IC(MICA)\]</span></p>
</div>
<div id="lin-method" class="section level3" number="1.1.2">
<h3><span class="header-section-number">1.1.2</span> Lin method</h3>
<p>The Lin method is defined as:</p>
<p><span class="math display">\[sim_{Lin}(t_1,t_2) = \frac{2IC(MICA)}{IC(t_1)+IC(t_2)}\]</span></p>
</div>
<div id="rel-method" class="section level3" number="1.1.3">
<h3><span class="header-section-number">1.1.3</span> Rel method</h3>
<p>The Relevance method, which was proposed by Schlicker, combine Resnik‚Äôs and Lin‚Äôs method and is defined as:</p>
<p><span class="math display">\[sim_{Rel}(t_1,t_2) = \frac{2IC(MICA)(1-p(MICA))}{IC(t_1)+IC(t_2)}\]</span></p>
</div>
<div id="jiang-method" class="section level3" number="1.1.4">
<h3><span class="header-section-number">1.1.4</span> Jiang method</h3>
<p>The Jiang and Conrath‚Äôs method is defined as:</p>
<p><span class="math display">\[sim_{Jiang}(t_1,t_2) = 1-\min(1, IC(t_1) + IC(t_2) - 2IC(MICA))\]</span></p>
</div>
</div>
<div id="graph-based-method" class="section level2" number="1.2">
<h2><span class="header-section-number">1.2</span> Graph-based method</h2>
<p>Graph-based methods using the topology of GO graph structure to
compute semantic similarity. Formally, a GO term A can be represented
as <span class="math inline">\(DAG_{A}=(A,T_{A},E_{A})\)</span> where <span class="math inline">\(T_{A}\)</span> is the set of GO terms in
<span class="math inline">\(DAG_{A}\)</span>, including term A and all of its ancestor terms in the GO
graph, and <span class="math inline">\(E_{A}\)</span> is the set of edges connecting the GO terms in
<span class="math inline">\(DAG_{A}\)</span>.</p>
<div id="wang-method" class="section level3" number="1.2.1">
<h3><span class="header-section-number">1.2.1</span> Wang method</h3>
<p>To encode the semantic of a GO term in a measurable format to enable a quantitative comparison, Wang<span class="citation">(<a href="#ref-wang_new_2007" role="doc-biblioref">Wang et al. 2007</a>)</span> firstly defined the semantic value of term A as the aggregate contribution of all terms in <span class="math inline">\(DAG_{A}\)</span> to the semantics of term A, terms closer to term A in <span class="math inline">\(DAG_{A}\)</span> contribute more to its semantics. Thus, defined the contribution of a GO term <span class="math inline">\(t\)</span> to the semantic of GO term <span class="math inline">\(A\)</span> as the S-value of GO term <span class="math inline">\(t\)</span> related to term <span class="math inline">\(A\)</span>.</p>
<p>For any of term <span class="math inline">\(t\)</span> in <span class="math inline">\(DAG_{A}\)</span>, its S-value related to term <span class="math inline">\(A\)</span>, <span class="math inline">\(S_{A}(\textit{t})\)</span> is defined as:</p>
<p><span class="math display">\[\left\{\begin{array}{l} S_{A}(A)=1 \\ S_{A}(\textit{t})=\max\{w_{e} \times S_{A}(\textit{t}&#39;) | \textit{t}&#39; \in children \: of(\textit{t}) \} \; if \: \textit{t} \ne A \end{array} \right.\]</span></p>
<p>where <span class="math inline">\(w_{e}\)</span> is the semantic contribution factor for edge <span class="math inline">\(e \in E_{A}\)</span> linking term <span class="math inline">\(t\)</span> with its child term <span class="math inline">\(t&#39;\)</span>.
Term <span class="math inline">\(A\)</span> contributes to its own is defined as 1. After obtaining the S-values for all terms in <span class="math inline">\(DAG_{A}\)</span>,
the semantic value of DO term A, <span class="math inline">\(SV(A)\)</span>, is calculated as:</p>
<p><span class="math display">\[SV(A)=\displaystyle\sum_{t \in T_{A}} S_{A}(t)\]</span></p>
<p>Thus given two GO terms A and B, the semantic similarity between these two terms is defined as:</p>
<p><span class="math display">\[sim_{Wang}(A, B) = \frac{\displaystyle\sum_{t \in T_{A} \cap T_{B}}{S_{A}(t) + S_{B}(t)}}{SV(A) + SV(B)}\]</span></p>
<p>where <span class="math inline">\(S_{A}(\textit{t})\)</span> is the S-value of GO term <span class="math inline">\(t\)</span> related to term <span class="math inline">\(A\)</span>
and <span class="math inline">\(S_{B}(\textit{t})\)</span> is the S-value of GO term <span class="math inline">\(t\)</span> related to term <span class="math inline">\(B\)</span>.</p>
<p>This method proposed by Wang <span class="citation">(<a href="#ref-wang_new_2007" role="doc-biblioref">Wang et al. 2007</a>)</span> determines the semantic
similarity of two GO terms based on both the locations of these terms
in the GO graph and their relations with their ancestor terms.</p>
</div>
</div>
<div id="combine-methods" class="section level2" number="1.3">
<h2><span class="header-section-number">1.3</span> Combine methods</h2>
<p>Since a gene product can be annotated by multiple GO terms, semantic similarity among gene products needs to be aggregated from different semantic similarity scores of multiple GO terms associated with genes, including <code>max</code>, <code>avg</code>, <code>rcmax</code> and <code>BMA</code>.</p>
<div id="max" class="section level3" number="1.3.1">
<h3><span class="header-section-number">1.3.1</span> max</h3>
<p>The <code>max</code> method calculates the maximum semantic similarity score over all pairs of GO terms between these two GO term sets.</p>
<p><span class="math display">\[sim_{max}(g_1, g_2) = \displaystyle\max_{1 \le i \le m, 1 \le j \le n} sim(go_{1i}, go_{2j})\]</span></p>
</div>
<div id="avg" class="section level3" number="1.3.2">
<h3><span class="header-section-number">1.3.2</span> avg</h3>
<p>The <code>avg</code> calculates the average semantic similarity score over all pairs of GO terms.</p>
<p><span class="math display">\[sim_{avg}(g_1, g_2) = \frac{\displaystyle\sum_{i=1}^m\sum_{j=1}^nsim(go_{1i}, go_{2j})}{m \times n}\]</span></p>
</div>
<div id="rcmax" class="section level3" number="1.3.3">
<h3><span class="header-section-number">1.3.3</span> rcmax</h3>
<p>Similarities among two sets of GO terms form a matrix, the <code>rcmax</code> method uses the maximum of <code>RowScore</code> and <code>ColumnScore</code>, where <code>RowScore</code> (or <code>ColumnScore</code>) is the average of maximum similarity on each row (or column).</p>
<p><span class="math display">\[sim_{rcmax}(g_1, g_2) = \max(\frac{\displaystyle\sum_{i=1}^m \max_{1 \le j \le n} sim(go_{1i}, go_{2j})}{m},\frac{\displaystyle\sum_{j=1}^n \max_{1 \le i \le m} sim(go_{1i},go_{2j})}{n})\]</span></p>
</div>
<div id="bma" class="section level3" number="1.3.4">
<h3><span class="header-section-number">1.3.4</span> BMA</h3>
<p>The <code>BMA</code> method, used the <strong>B</strong>est-<strong>M</strong>atch <strong>A</strong>verage strategy, calculates the average of all maximum similarities on each row and column, and is defined as:</p>
<p><span class="math display">\[sim_{BMA}(g_1, g_2) = \frac{\displaystyle\sum_{1=i}^m \max_{1 \le j \le n}sim(go_{1i}, go_{2j}) + \displaystyle\sum_{1=j}^n \max_{1 \le i \le m}sim(go_{1i}, go_{2j})} {m+n}\]</span></p>
</div>
</div>
<div id="summary" class="section level2" number="1.4">
<h2><span class="header-section-number">1.4</span> Summary</h2>
<p>The idea behind semantic similarity measurement is the notion that genes with similar function should have similar annotation vocabulary and have a close relationship in the ontology strucutre. Measuring similarity is critical for expanding knownledge, since similar objects tend to behave similarly, which supports many bioinformatics applications to infer gene/protein functions, miRNA function, genetic interaction, protein-protein interaction, miRNA-mRNA interaction and celluar localization.</p>
<p>We developed several Bioconductor packages, including <a href="http://bioconductor.org/packages/GOSemSim">GOSemSim</a> <span class="citation">(<a href="#ref-yu2010" role="doc-biblioref">Yu et al. 2010</a>; <a href="#ref-yu_gosemsim_2020" role="doc-biblioref">Yu 2020</a>)</span> for computing semantic similarity among GO terms, sets of GO terms, gene products and gene clusters (see also <a href="#GOSemSim">Chapter 2</a>), <a href="http://bioconductor.org/packages/DOSE">DOSE</a> <span class="citation">(<a href="#ref-yu_dose_2015" role="doc-biblioref">Yu et al. 2015</a>)</span> for Disease Ontology (DO) (see also <a href="#DOSE-semantic-similarity">Chapter 3</a>) and <a href="http://bioconductor.org/packages/meshes">meshes</a> <span class="citation">(<a href="#ref-yu_meshes_2018" role="doc-biblioref">Yu 2018</a>)</span> that based on Medical Subject Headings (MeSH) (see also <a href="#meshes-semantic-similarity">Chapter 4</a>).</p>

</div>
</div>
<div id="GOSemSim" class="section level1" number="2">
<h1><span class="header-section-number">Chapter 2</span> GO semantic similarity analysis</h1>
<p><a href="http://bioconductor.org/packages/GOSemSim">GOSemSim</a> implemented all methods described in <a href="#semantic-similarity-overview">Chapter 1</a>, including four IC-based methods and one graph-based method.</p>
<div id="semantic-data" class="section level2" number="2.1">
<h2><span class="header-section-number">2.1</span> Semantic data</h2>
<p>To measure semantic similarity, we need to prepare GO annotations including GO structure (i.e.¬†GO term relationships) and gene to GO mapping. For IC-based methods, information of GO term is species specific. We need to calculate <code>IC</code> for all GO terms of a species before we measure semantic similarity.</p>
<p><a href="http://bioconductor.org/packages/GOSemSim">GOSemSim</a> provides the <code>godata()</code> function to prepare semantic data to support measuring GO and gene simiarlity. It internally used the <a href="http://bioconductor.org/packages/GO.db">GO.db</a> package to obtain GO strucuture and <code>OrgDb</code> for gene to GO mapping.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GOSemSim)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>hsGO <span class="ot">&lt;-</span> <span class="fu">godata</span>(<span class="st">&#39;org.Hs.eg.db&#39;</span>, <span class="at">ont=</span><span class="st">&quot;MF&quot;</span>)</span></code></pre></div>
<p>User can set <code>computeIC=FALSE</code> if they only want to use Wang‚Äôs method.</p>
</div>
<div id="gosemsim-supported-organisms" class="section level2" number="2.2">
<h2><span class="header-section-number">2.2</span> Supported organisms</h2>
<p><a href="http://bioconductor.org/packages/GOSemSim">GOSemSim</a> supports all organisms that have an <code>OrgDb</code> object available.</p>
<p>Bioconductor have already provided <code>OrgDb</code> for <a href="http://bioconductor.org/packages/release/BiocViews.html#___OrgDb">about 20 species</a>.</p>
<p>We can query <code>OrgDb</code> online via the <a href="http://bioconductor.org/packages/AnnotationHub">AnnotationHub</a> package. For example:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(AnnotationHub)</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>hub <span class="ot">&lt;-</span> <span class="fu">AnnotationHub</span>()</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>q <span class="ot">&lt;-</span> <span class="fu">query</span>(hub, <span class="st">&quot;Cricetulus&quot;</span>)</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>id <span class="ot">&lt;-</span> q<span class="sc">$</span>ah_id[<span class="fu">length</span>(q)]</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>Cgriseus <span class="ot">&lt;-</span> hub[[id]]</span></code></pre></div>
<p>If an organism is not supported by <a href="http://bioconductor.org/packages/AnnotationHub">AnnotationHub</a>, user can use the <a href="http://bioconductor.org/packages/AnnotationForge">AnnotationForge</a> package to build <code>OrgDb</code> manually.</p>
<p>Once we have <code>OrgDb</code>, we can build annotation data needed by <a href="http://bioconductor.org/packages/GOSemSim">GOSemSim</a> via <code>godata()</code> function described previously.</p>
</div>
<div id="go-semantic-simiarlity" class="section level2" number="2.3">
<h2><span class="header-section-number">2.3</span> GO semantic similarity measurement</h2>
<p>The <code>goSim()</code> function calculates semantic similarity between two GO terms, while the <code>mgoSim()</code> function calculates semantic similarity between two sets of GO terms.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">goSim</span>(<span class="st">&quot;GO:0004022&quot;</span>, <span class="st">&quot;GO:0005515&quot;</span>, <span class="at">semData=</span>hsGO, <span class="at">measure=</span><span class="st">&quot;Jiang&quot;</span>)</span></code></pre></div>
<pre><code>## [1] 0.16</code></pre>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">goSim</span>(<span class="st">&quot;GO:0004022&quot;</span>, <span class="st">&quot;GO:0005515&quot;</span>, <span class="at">semData=</span>hsGO, <span class="at">measure=</span><span class="st">&quot;Wang&quot;</span>)</span></code></pre></div>
<pre><code>## [1] 0.116</code></pre>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>go1 <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;GO:0004022&quot;</span>,<span class="st">&quot;GO:0004024&quot;</span>,<span class="st">&quot;GO:0004174&quot;</span>)</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>go2 <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;GO:0009055&quot;</span>,<span class="st">&quot;GO:0005515&quot;</span>)</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="fu">mgoSim</span>(go1, go2, <span class="at">semData=</span>hsGO, <span class="at">measure=</span><span class="st">&quot;Wang&quot;</span>, <span class="at">combine=</span><span class="cn">NULL</span>)</span></code></pre></div>
<pre><code>##            GO:0009055 GO:0005515
## GO:0004022      0.368      0.116
## GO:0004024      0.335      0.107
## GO:0004174      0.663      0.119</code></pre>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="fu">mgoSim</span>(go1, go2, <span class="at">semData=</span>hsGO, <span class="at">measure=</span><span class="st">&quot;Wang&quot;</span>, <span class="at">combine=</span><span class="st">&quot;BMA&quot;</span>)</span></code></pre></div>
<pre><code>## [1] 0.43</code></pre>
</div>
<div id="gene-go-semantic-similarity" class="section level2" number="2.4">
<h2><span class="header-section-number">2.4</span> Gene semantic similarity measurement</h2>
<p>On the basis of semantic similarity between GO terms, <a href="https://www.bioconductor.org/packages/GOSemSim">GOSemSim</a> can
also compute semantic similarity among sets of GO terms, gene products, and gene clusters.</p>
<p>Suppose we have gene <span class="math inline">\(g_1\)</span> annotated by GO terms sets <span class="math inline">\(GO_{1}=\{go_{11},go_{12} \cdots go_{1m}\}\)</span>
and <span class="math inline">\(g_2\)</span> annotated by <span class="math inline">\(GO_{2}=\{go_{21},go_{22} \cdots go_{2n}\}\)</span>, <a href="http://bioconductor.org/packages/GOSemSim">GOSemSim</a> implemented four combine methods, including <strong><em>max</em></strong>, <strong><em>avg</em></strong>, <strong><em>rcmax</em></strong>, and <strong><em>BMA</em></strong>, to aggregate semantic similarity scores of multiple GO terms (see also <a href="#combine-methods">session 1.3</a>). The similarities
among gene products and gene clusters which annotated by multiple GO
terms are also calculated by the these combine methods.</p>
<p><a href="http://bioconductor.org/packages/GOSemSim">GOSemSim</a> provides <code>geneSim()</code> to calculate semantic similarity between two gene products, and <code>mgeneSim()</code> to calculate semantic similarity among multiple gene products.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="fu">geneSim</span>(<span class="st">&quot;241&quot;</span>, <span class="st">&quot;251&quot;</span>, <span class="at">semData=</span>hsGO, <span class="at">measure=</span><span class="st">&quot;Wang&quot;</span>, <span class="at">combine=</span><span class="st">&quot;BMA&quot;</span>)</span></code></pre></div>
<pre><code>## $geneSim
## [1] 0.149
## 
## $GO1
## [1] &quot;GO:0004364&quot; &quot;GO:0004464&quot; &quot;GO:0004602&quot; &quot;GO:0005515&quot; &quot;GO:0047485&quot;
## [6] &quot;GO:0050544&quot;
## 
## $GO2
## [1] &quot;GO:0004035&quot;</code></pre>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="fu">mgeneSim</span>(<span class="at">genes=</span><span class="fu">c</span>(<span class="st">&quot;835&quot;</span>, <span class="st">&quot;5261&quot;</span>,<span class="st">&quot;241&quot;</span>, <span class="st">&quot;994&quot;</span>),</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>         <span class="at">semData=</span>hsGO, <span class="at">measure=</span><span class="st">&quot;Wang&quot;</span>,<span class="at">verbose=</span><span class="cn">FALSE</span>)</span></code></pre></div>
<pre><code>##        835  5261   241   994
## 835  1.000 0.465 0.448 0.562
## 5261 0.465 1.000 0.433 0.499
## 241  0.448 0.433 1.000 0.452
## 994  0.562 0.499 0.452 1.000</code></pre>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="fu">mgeneSim</span>(<span class="at">genes=</span><span class="fu">c</span>(<span class="st">&quot;835&quot;</span>, <span class="st">&quot;5261&quot;</span>,<span class="st">&quot;241&quot;</span>, <span class="st">&quot;994&quot;</span>),</span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>       <span class="at">semData=</span>hsGO, <span class="at">measure=</span><span class="st">&quot;Rel&quot;</span>,<span class="at">verbose=</span><span class="cn">FALSE</span>)</span></code></pre></div>
<pre><code>##        835  5261   241   994
## 835  0.945 0.413 0.257 0.482
## 5261 0.413 0.907 0.332 0.473
## 241  0.257 0.332 0.940 0.274
## 994  0.482 0.473 0.274 0.906</code></pre>
<p>By default, <code>godata</code> function use <code>ENTREZID</code> as keytype, and the input ID type is <code>ENTREZID</code>. User can use other ID types such as <code>ENSEMBL</code>, <code>UNIPROT</code>, <code>REFSEQ</code>, <code>ACCNUM</code>, <code>SYMBOL</code> <em>et al</em>.</p>
<p>Here as an example, we use <code>SYMBOL</code> as <code>keytype</code> and calculate semantic similarities among several genes by using their gene symbol as input.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>hsGO2 <span class="ot">&lt;-</span> <span class="fu">godata</span>(<span class="st">&#39;org.Hs.eg.db&#39;</span>, <span class="at">keytype =</span> <span class="st">&quot;SYMBOL&quot;</span>, <span class="at">ont=</span><span class="st">&quot;MF&quot;</span>, <span class="at">computeIC=</span><span class="cn">FALSE</span>) </span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>genes <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;CDC45&quot;</span>, <span class="st">&quot;MCM10&quot;</span>, <span class="st">&quot;CDC20&quot;</span>, <span class="st">&quot;NMU&quot;</span>, <span class="st">&quot;MMP1&quot;</span>)</span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a><span class="fu">mgeneSim</span>(genes, <span class="at">semData=</span>hsGO2, <span class="at">measure=</span><span class="st">&quot;Wang&quot;</span>, <span class="at">combine=</span><span class="st">&quot;BMA&quot;</span>, <span class="at">verbose=</span><span class="cn">FALSE</span>)</span></code></pre></div>
<pre><code>##       CDC45 MCM10 CDC20   NMU  MMP1
## CDC45 1.000 0.892 0.578 0.592 0.113
## MCM10 0.892 1.000 0.574 0.587 0.105
## CDC20 0.578 0.574 1.000 0.640 0.117
## NMU   0.592 0.587 0.640 1.000 0.104
## MMP1  0.113 0.105 0.117 0.104 1.000</code></pre>
<p>Users can also use <a href="#bitr"><code>clusterProfiler::bitr</code></a> to translate biological IDs.</p>
</div>
<div id="gene-cluster-go-semantic-similarity" class="section level2" number="2.5">
<h2><span class="header-section-number">2.5</span> Gene cluster semantic similarity measurement</h2>
<p><a href="http://bioconductor.org/packages/GOSemSim">GOSemSim</a> also supports calculating semantic similarity between two gene clusters using <code>clusterSim()</code> function and measuring semantic similarity among multiple gene clusters using <code>mclusterSim()</code> function.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>gs1 <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;835&quot;</span>, <span class="st">&quot;5261&quot;</span>,<span class="st">&quot;241&quot;</span>, <span class="st">&quot;994&quot;</span>, <span class="st">&quot;514&quot;</span>, <span class="st">&quot;533&quot;</span>)</span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>gs2 <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;578&quot;</span>,<span class="st">&quot;582&quot;</span>, <span class="st">&quot;400&quot;</span>, <span class="st">&quot;409&quot;</span>, <span class="st">&quot;411&quot;</span>)</span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a><span class="fu">clusterSim</span>(gs1, gs2, <span class="at">semData=</span>hsGO, <span class="at">measure=</span><span class="st">&quot;Wang&quot;</span>, <span class="at">combine=</span><span class="st">&quot;BMA&quot;</span>)</span></code></pre></div>
<pre><code>## [1] 0.609</code></pre>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(org.Hs.eg.db)</span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>x <span class="ot">&lt;-</span> org.Hs.egGO</span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>hsEG <span class="ot">&lt;-</span> <span class="fu">mappedkeys</span>(x)</span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>set.seed <span class="ot">&lt;-</span> <span class="dv">123</span></span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>clusters <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">a=</span><span class="fu">sample</span>(hsEG, <span class="dv">20</span>), <span class="at">b=</span><span class="fu">sample</span>(hsEG, <span class="dv">20</span>), <span class="at">c=</span><span class="fu">sample</span>(hsEG, <span class="dv">20</span>))</span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a><span class="fu">mclusterSim</span>(clusters, <span class="at">semData=</span>hsGO, <span class="at">measure=</span><span class="st">&quot;Wang&quot;</span>, <span class="at">combine=</span><span class="st">&quot;BMA&quot;</span>)</span></code></pre></div>
<pre><code>##       a     b     c
## a 1.000 0.720 0.722
## b 0.720 1.000 0.709
## c 0.722 0.709 1.000</code></pre>
<!--


## Applications

[GOSemSim](https://www.bioconductor.org/packages/GOSemSim) was cited by more than [200 papers](https://scholar.google.com.hk/scholar?oi=bibs&hl=en&cites=9484177541993722322,17633835198940746971,18126401808149291947) and had been applied to many research domains, including:

+ [Disease or Drug analysis](https://guangchuangyu.github.io/software/GOSemSim/featuredArticles/#diease-or-drug-analysis)
+ [Gene/Protein functional analysis](https://guangchuangyu.github.io/software/GOSemSim/featuredArticles/#geneprotein-functional-analysis)
+ [Protein-Protein interaction](https://guangchuangyu.github.io/software/GOSemSim/featuredArticles/#protein-protein-interaction)
+ [miRNA-mRNA interaction](https://guangchuangyu.github.io/software/GOSemSim/featuredArticles/#mirna-mrna-interaction)
+ [sRNA regulation](https://guangchuangyu.github.io/software/GOSemSim/featuredArticles/#srna-regulation)
+ [Evolution](https://guangchuangyu.github.io/software/GOSemSim/featuredArticles/#evolution)

Find out more on <https://guangchuangyu.github.io/software/GOSemSim/featuredArticles/>.


# GO enrichment analysis

GO enrichment analysis can be supported by our package [clusterProfiler](https://www.bioconductor.org/packages/clusterProfiler)[@yu2012], which supports hypergeometric test and Gene Set Enrichment Analysis (GSEA). Enrichment results across different gene clusters can be compared using __*compareCluster*__ function.

# Disease Ontology Semantic and Enrichment analysis

Disease Ontology (DO) annotates human genes in the context of disease. DO is an important annotation in translating molecular findings from high-throughput data to clinical relevance.
[DOSE](https://www.bioconductor.org/packages/DOSE)[@yu_dose_2015] supports semantic similarity computation among DO terms and genes.
Enrichment analysis including hypergeometric model and GSEA are also implemented to support discovering disease associations of high-throughput biological data.

# MeSH enrichment and semantic analyses

MeSH (Medical Subject Headings) is the NLM controlled vocabulary used to manually index articles for MEDLINE/PubMed. [meshes](https://www.bioconductor.org/packages/meshes) supports enrichment (hypergeometric test and GSEA) and semantic similarity analyses for more than 70 species.


-->

</div>
</div>
<div id="DOSE-semantic-similarity" class="section level1" number="3">
<h1><span class="header-section-number">Chapter 3</span> DO semantic similarity analysis</h1>
<p>Public health is an important driving force behind biological and medical research.
A major challenge of the post-genomic era is bridging the gap between fundamental biological research and its clinical applications.
Recent research has increasingly demonstrated that many seemingly dissimilar diseases have common molecular mechanisms.
Understanding similarities among disease aids in early diagnosis and new drug development.</p>
<p>Formal knowledge representation of gene-disease association is demanded for this purpose.
Ontologies, such as Gene Ontology (GO), have been successfully applied to represent biological knowledge, and many related techniques have been adopted to extract information.
Disease Ontology (DO) <span class="citation">(<a href="#ref-schriml_disease_2011" role="doc-biblioref">Schriml et al. 2011</a>)</span> was developed to create a consistent description of gene products with disease perspectives, and is essential for supporting functional genomics in disease context.
Accurate disease descriptions can discover new relationships between genes and disease, and new functions for previous uncharacteried genes and alleles.</p>
<p>Disease Ontology (DO) <span class="citation">(<a href="#ref-schriml_disease_2011" role="doc-biblioref">Schriml et al. 2011</a>)</span> aims to provide an open source ontology for the integration of biomedical data that is associated with human disease. Unlike other clinical vocabularies that defined disease related concepts disparately, DO is organized as a directed acyclic graph, laying the foundation for quantitative computation of disease knowledge.</p>
<p>We developed an <code>R</code> package <a href="http://bioconductor.org/packages/DOSE">DOSE</a> <span class="citation">(<a href="#ref-yu_dose_2015" role="doc-biblioref">Yu et al. 2015</a>)</span> for analyzing semantic similarities among DO terms and gene products annotated with DO terms.</p>
<div id="do-semantic-similarity" class="section level2" number="3.1">
<h2><span class="header-section-number">3.1</span> DO term semantic similarity measurement</h2>
<p>Four methods determine the semantic similarity of two terms based on the Information Content of their common ancestor term were proposed by Resnik <span class="citation">(<a href="#ref-philip_semantic_1999" role="doc-biblioref">Philip 1999</a>)</span>, Jiang <span class="citation">(<a href="#ref-jiang_semantic_1997" role="doc-biblioref">Jiang and Conrath 1997</a>)</span>, Lin <span class="citation">(<a href="#ref-lin_information-theoretic_1998" role="doc-biblioref">Lin 1998</a>)</span> and Schlicker <span class="citation">(<a href="#ref-schlicker_new_2006" role="doc-biblioref">Schlicker et al. 2006</a>)</span>. Wang <span class="citation">(<a href="#ref-wang_new_2007" role="doc-biblioref">Wang et al. 2007</a>)</span> presented a method to measure the similarity based on the graph structure. Each of these methods has its own advantage and weakness. <a href="http://bioconductor.org/packages/DOSE">DOSE</a> supports all these methods to compute semantic similarity among DO terms and gene products. For algorithm details, please refer to <a href="#semantic-similarity-overview">chapter 1</a>.</p>
<p>In <a href="http://bioconductor.org/packages/DOSE">DOSE</a>, we implemented <code>doSim</code> for calculating semantic similarity between two DO terms and two set of DO terms.</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DOSE)</span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a>a <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;DOID:14095&quot;</span>, <span class="st">&quot;DOID:5844&quot;</span>, <span class="st">&quot;DOID:2044&quot;</span>, <span class="st">&quot;DOID:8432&quot;</span>, <span class="st">&quot;DOID:9146&quot;</span>,</span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a>       <span class="st">&quot;DOID:10588&quot;</span>, <span class="st">&quot;DOID:3209&quot;</span>, <span class="st">&quot;DOID:848&quot;</span>, <span class="st">&quot;DOID:3341&quot;</span>, <span class="st">&quot;DOID:252&quot;</span>)</span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a>b <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;DOID:9409&quot;</span>, <span class="st">&quot;DOID:2491&quot;</span>, <span class="st">&quot;DOID:4467&quot;</span>, <span class="st">&quot;DOID:3498&quot;</span>, <span class="st">&quot;DOID:11256&quot;</span>)</span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a><span class="fu">doSim</span>(a[<span class="dv">1</span>], b[<span class="dv">1</span>], <span class="at">measure=</span><span class="st">&quot;Wang&quot;</span>)</span></code></pre></div>
<pre><code>## [1] 0.07142995</code></pre>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="fu">doSim</span>(a[<span class="dv">1</span>], b[<span class="dv">1</span>], <span class="at">measure=</span><span class="st">&quot;Resnik&quot;</span>)</span></code></pre></div>
<pre><code>## [1] 0</code></pre>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="fu">doSim</span>(a[<span class="dv">1</span>], b[<span class="dv">1</span>], <span class="at">measure=</span><span class="st">&quot;Lin&quot;</span>)</span></code></pre></div>
<pre><code>## [1] 0</code></pre>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a>s <span class="ot">&lt;-</span> <span class="fu">doSim</span>(a, b, <span class="at">measure=</span><span class="st">&quot;Wang&quot;</span>)</span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a>s</span></code></pre></div>
<pre><code>##             DOID:9409  DOID:2491  DOID:4467  DOID:3498 DOID:11256
## DOID:14095 0.07142995 0.05714393 0.03676194 0.03676194 0.52749870
## DOID:5844  0.14897652 0.11564838 0.02801328 0.02801328 0.06134327
## DOID:2044  0.14897652 0.11564838 0.02801328 0.02801328 0.06134327
## DOID:8432  0.17347273 0.13877811 0.03676194 0.03676194 0.07142995
## DOID:9146  0.07142995 0.05714393 0.03676194 0.03676194 0.17347273
## DOID:10588 0.13240905 0.18401515 0.02208240 0.02208240 0.05452137
## DOID:3209  0.14897652 0.11564838 0.02801328 0.02801328 0.06134327
## DOID:848   0.14897652 0.11564838 0.02801328 0.02801328 0.06134327
## DOID:3341  0.13240905 0.09998997 0.02208240 0.02208240 0.05452137
## DOID:252   0.06134327 0.04761992 0.02801328 0.02801328 0.06134327</code></pre>
<p>The <code>doSim</code> function requires three parameter <code>DOID1</code>, <code>DOID2</code> and <code>measure</code>. <code>DOID1</code> and <code>DOID2</code> should be a vector of DO terms, while <code>measure</code> should be one of <em>Resnik</em>, <em>Jiang</em>, <em>Lin</em>, <em>Rel</em>, and <em>Wang</em>.</p>
<p>We also implement a plot function <code>simplot()</code> to visualize the similarity result as demonstrated in Figure <a href="#fig:dose-simplot">3.1</a>.</p>


<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="fu">simplot</span>(s,</span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a>        <span class="at">color.low=</span><span class="st">&quot;white&quot;</span>, <span class="at">color.high=</span><span class="st">&quot;red&quot;</span>,</span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">labs=</span><span class="cn">TRUE</span>, <span class="at">digits=</span><span class="dv">2</span>, <span class="at">labs.size=</span><span class="dv">5</span>,</span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">font.size=</span><span class="dv">14</span>, <span class="at">xlab=</span><span class="st">&quot;&quot;</span>, <span class="at">ylab=</span><span class="st">&quot;&quot;</span>)</span></code></pre></div>
<div class="figure"><span id="fig:dose-simplot"></span>
<img src="biomedicalKnowledge_files/figure-html/dose-simplot-1.png" alt="Similarity matrix visualization. The simplot() function visualize similarity matrix as a heatmap." width="95%"  />
<p class="caption">
Figure 3.1: <strong>Similarity matrix visualization</strong>. The <code>simplot()</code> function visualize similarity matrix as a heatmap.
</p>
</div>
<p>Parameter <code>color.low</code> and <code>colow.high</code> are used to setting the color gradient; <code>labs</code> is a logical parameter indicating whether to show the similarity values or not, <code>digits</code> to indicate the number of decimal places to be used and <code>labs.size</code> control the font size of similarity values; <code>font.size</code> setting the font size of axis and label of the coordinate system.</p>
</div>
<div id="gene-do-semantic-similarity" class="section level2" number="3.2">
<h2><span class="header-section-number">3.2</span> Gene semantic similarity measurement</h2>
<p>On the basis of semantic similarity between DO terms, <a href="http://bioconductor.org/packages/DOSE">DOSE</a> can also compute semantic similarity among gene products. <a href="http://bioconductor.org/packages/DOSE">DOSE</a> provides four methods which called <code>max</code>, <code>avg</code>, <code>rcmax</code> and <code>BMA</code> to combine semantic similarity scores of multiple DO terms (as described in <a href="combine-methods">session 1.3</a>). The similarities among genes and gene clusters which annotated by multiple DO terms were also calculated by these combine methods.</p>
<p><a href="http://bioconductor.org/packages/DOSE">DOSE</a> implemented <code>geneSim</code> to measure semantic similarities among genes.</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a>g1 <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;84842&quot;</span>, <span class="st">&quot;2524&quot;</span>, <span class="st">&quot;10590&quot;</span>, <span class="st">&quot;3070&quot;</span>, <span class="st">&quot;91746&quot;</span>)</span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a>g2 <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;84289&quot;</span>, <span class="st">&quot;6045&quot;</span>, <span class="st">&quot;56999&quot;</span>, <span class="st">&quot;9869&quot;</span>)</span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a><span class="fu">geneSim</span>(g1[<span class="dv">1</span>], g2[<span class="dv">1</span>], <span class="at">measure=</span><span class="st">&quot;Wang&quot;</span>, <span class="at">combine=</span><span class="st">&quot;BMA&quot;</span>)</span></code></pre></div>
<pre><code>## [1] 0.051</code></pre>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a>gs <span class="ot">&lt;-</span> <span class="fu">geneSim</span>(g1, g2, <span class="at">measure=</span><span class="st">&quot;Wang&quot;</span>, <span class="at">combine=</span><span class="st">&quot;BMA&quot;</span>)</span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a>gs</span></code></pre></div>
<pre><code>##       84289  6045 56999  9869
## 84842 0.051 0.135 0.355 0.103
## 2524  0.284 0.172 0.517 0.517
## 10590 0.150 0.173 0.242 0.262
## 3070  0.573 0.517 1.000 1.000
## 91746 0.351 0.308 0.527 0.496</code></pre>
<p>The <code>geneSim</code> requires four parameter <code>geneID1</code>, <code>geneID2</code>, <code>measure</code> and <code>combine</code>. <code>geneID1</code> and <code>geneID2</code> should be a vector of entrez gene IDs; <code>measure</code> should be one of <em>Resnik</em>, <em>Jiang</em>, <em>Lin</em>, <em>Rel</em>, and <em>Wang</em>, while <code>combine</code> should be one of <em>max</em>, <em>avg</em>, <em>rcmax</em> and <em>BMA</em>.</p>
</div>
<div id="gene-cluster-do-semantic-similarity" class="section level2" number="3.3">
<h2><span class="header-section-number">3.3</span> Gene cluster semantic similarity measurement</h2>
<p><a href="http://bioconductor.org/packages/DOSE">DOSE</a> also implemented <code>clusterSim</code> for calculating semantic similarity between two gene clusters and <code>mclusterSim</code> for calculating semantic similarities among multiple gene clusters.</p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="fu">clusterSim</span>(g1, g2, <span class="at">measure=</span><span class="st">&quot;Wang&quot;</span>, <span class="at">combine=</span><span class="st">&quot;BMA&quot;</span>)</span></code></pre></div>
<pre><code>## [1] 0.549</code></pre>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a>g3 <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;57491&quot;</span>, <span class="st">&quot;6296&quot;</span>, <span class="st">&quot;51438&quot;</span>, <span class="st">&quot;5504&quot;</span>, <span class="st">&quot;27319&quot;</span>, <span class="st">&quot;1643&quot;</span>)</span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a>clusters <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">a=</span>g1, <span class="at">b=</span>g2, <span class="at">c=</span>g3)</span>
<span id="cb38-3"><a href="#cb38-3" aria-hidden="true" tabindex="-1"></a><span class="fu">mclusterSim</span>(clusters, <span class="at">measure=</span><span class="st">&quot;Wang&quot;</span>, <span class="at">combine=</span><span class="st">&quot;BMA&quot;</span>)</span></code></pre></div>
<pre><code>##       a     b     c
## a 1.000 0.549 0.425
## b 0.549 1.000 0.645
## c 0.425 0.645 1.000</code></pre>
<!--

# GO semantic similarity calculation

GO Semantic similarity can be calculated by [GOSemSim](http://bioconductor.org/packages/GOSemSim)[@yu2010].

We have developed another package [GOSemSim](http://bioconductor.org/packages/GOSemSim)[@yu2010] to explore the functional similarity at GO perspective, including molecular function (MF), biological process (BP) and cellular component (CC).


# MeSH semantic analysis

MeSH (Medical Subject Headings) is the NLM controlled vocabulary used
to manually index articles for MEDLINE/PubMed. [meshes](http://bioconductor.org/packages/meshes)
supports enrichment (hypergeometric test and GSEA) and semantic
similarity analyses for more than 70 species.

-->

</div>
</div>
<div id="meshes-semantic-similarity" class="section level1" number="4">
<h1><span class="header-section-number">Chapter 4</span> MeSH semantic similarity analysis</h1>
<p>MeSH (Medical Subject Headings) is the NLM (U.S. National Library of
Medicine) controlled vocabulary used to manually index articles for
MEDLINE/PubMed. MeSH is a comprehensive life science vocabulary. MeSH has
19 categories and <code>MeSH.db</code> contains 16 of them. That is:</p>
<table>
<thead>
<tr class="header">
<th align="left">Abbreviation</th>
<th align="left">Category</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">A</td>
<td align="left">Anatomy</td>
</tr>
<tr class="even">
<td align="left">B</td>
<td align="left">Organisms</td>
</tr>
<tr class="odd">
<td align="left">C</td>
<td align="left">Diseases</td>
</tr>
<tr class="even">
<td align="left">D</td>
<td align="left">Chemicals and Drugs</td>
</tr>
<tr class="odd">
<td align="left">E</td>
<td align="left">Analytical, Diagnostic and Therapeutic Techniques and Equipment</td>
</tr>
<tr class="even">
<td align="left">F</td>
<td align="left">Psychiatry and Psychology</td>
</tr>
<tr class="odd">
<td align="left">G</td>
<td align="left">Phenomena and Processes</td>
</tr>
<tr class="even">
<td align="left">H</td>
<td align="left">Disciplines and Occupations</td>
</tr>
<tr class="odd">
<td align="left">I</td>
<td align="left">Anthropology, Education, Sociology and Social Phenomena</td>
</tr>
<tr class="even">
<td align="left">J</td>
<td align="left">Technology and Food and Beverages</td>
</tr>
<tr class="odd">
<td align="left">K</td>
<td align="left">Humanities</td>
</tr>
<tr class="even">
<td align="left">L</td>
<td align="left">Information Science</td>
</tr>
<tr class="odd">
<td align="left">M</td>
<td align="left">Persons</td>
</tr>
<tr class="even">
<td align="left">N</td>
<td align="left">Health Care</td>
</tr>
<tr class="odd">
<td align="left">V</td>
<td align="left">Publication Type</td>
</tr>
<tr class="even">
<td align="left">Z</td>
<td align="left">Geographical Locations</td>
</tr>
</tbody>
</table>
<p>MeSH terms were associated with Entrez Gene ID by three methods,
<code>gendoo</code>, <code>gene2pubmed</code> and <code>RBBH</code> (Reciprocal Blast Best Hit).</p>
<table>
<thead>
<tr class="header">
<th>Method</th>
<th>Way of corresponding Entrez Gene IDs and MeSH IDs</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Gendoo</td>
<td>Text-mining</td>
</tr>
<tr class="even">
<td>gene2pubmed</td>
<td>Manual curation by NCBI teams</td>
</tr>
<tr class="odd">
<td>RBBH</td>
<td>sequence homology with BLASTP search (E-value&lt;10<sup>-50</sup>)</td>
</tr>
</tbody>
</table>
<div id="meshes-supported-organisms" class="section level2" number="4.1">
<h2><span class="header-section-number">4.1</span> Supported organisms</h2>
<p>The <a href="http://bioconductor.org/packages/meshes">meshes</a> package <span class="citation">(<a href="#ref-yu_meshes_2018" role="doc-biblioref">Yu 2018</a>)</span> relies on <code>MeSHDb</code> to prepare semantic data for measuring simiarlity. A list of <code>MeSHDb</code> can be found on Bioconductor, <a href="https://bioconductor.org/packages/release/BiocViews.html#___MeSHDb" class="uri">https://bioconductor.org/packages/release/BiocViews.html#___MeSHDb</a> and all the species (more than 70) that have MeSHDb available are supported by the <a href="http://bioconductor.org/packages/meshes">meshes</a> package.</p>
<p>The semantic data is prepare by the <code>meshdata()</code> function:</p>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(meshes)</span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a>hsamd <span class="ot">&lt;-</span> <span class="fu">meshdata</span>(<span class="st">&quot;MeSH.Hsa.eg.db&quot;</span>, <span class="at">category=</span><span class="st">&#39;A&#39;</span>, <span class="at">computeIC=</span>T, <span class="at">database=</span><span class="st">&quot;gendoo&quot;</span>)</span>
<span id="cb40-3"><a href="#cb40-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-4"><a href="#cb40-4" aria-hidden="true" tabindex="-1"></a><span class="do">## you may want to save the result for future usage</span></span>
<span id="cb40-5"><a href="#cb40-5" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb40-6"><a href="#cb40-6" aria-hidden="true" tabindex="-1"></a><span class="co"># save(hsamd, file = &quot;hsamd.rda&quot;)</span></span>
<span id="cb40-7"><a href="#cb40-7" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span></code></pre></div>
</div>
<div id="meshes-semantic-simiarlity" class="section level2" number="4.2">
<h2><span class="header-section-number">4.2</span> MeSH semantic similarity measurement</h2>
<p>The <a href="http://bioconductor.org/packages/meshes">meshes</a> package <span class="citation">(<a href="#ref-yu_meshes_2018" role="doc-biblioref">Yu 2018</a>)</span> implemented four IC-based methods (i.e.¬†Resnik <span class="citation">(<a href="#ref-philip_semantic_1999" role="doc-biblioref">Philip 1999</a>)</span>, Jiang <span class="citation">(<a href="#ref-jiang_semantic_1997" role="doc-biblioref">Jiang and Conrath 1997</a>)</span>,
Lin <span class="citation">(<a href="#ref-lin_information-theoretic_1998" role="doc-biblioref">Lin 1998</a>)</span> and Schlicker <span class="citation">(<a href="#ref-schlicker_new_2006" role="doc-biblioref">Schlicker et al. 2006</a>)</span>) and one graph-structure based method (i.e.¬†Wang <span class="citation">(<a href="#ref-wang_new_2007" role="doc-biblioref">Wang et al. 2007</a>)</span>), to measure MeSH term semantic similarity. For algorithm details, please refer to <a href="#semantic-similarity-overview">Chapter 1</a>.</p>
<p>The <code>meshSim()</code> function is designed to measure semantic similarity between two MeSH term vectors.</p>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a><span class="fu">meshSim</span>(<span class="st">&quot;D000009&quot;</span>, <span class="st">&quot;D009130&quot;</span>, <span class="at">semData=</span>hsamd, <span class="at">measure=</span><span class="st">&quot;Resnik&quot;</span>)</span></code></pre></div>
<pre><code>## [1] 0.2910261</code></pre>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a><span class="fu">meshSim</span>(<span class="st">&quot;D000009&quot;</span>, <span class="st">&quot;D009130&quot;</span>, <span class="at">semData=</span>hsamd, <span class="at">measure=</span><span class="st">&quot;Rel&quot;</span>)</span></code></pre></div>
<pre><code>## [1] 0.521396</code></pre>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a><span class="fu">meshSim</span>(<span class="st">&quot;D000009&quot;</span>, <span class="st">&quot;D009130&quot;</span>, <span class="at">semData=</span>hsamd, <span class="at">measure=</span><span class="st">&quot;Jiang&quot;</span>)</span></code></pre></div>
<pre><code>## [1] 0.4914785</code></pre>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="fu">meshSim</span>(<span class="st">&quot;D000009&quot;</span>, <span class="st">&quot;D009130&quot;</span>, <span class="at">semData=</span>hsamd, <span class="at">measure=</span><span class="st">&quot;Wang&quot;</span>)</span></code></pre></div>
<pre><code>## [1] 0.5557103</code></pre>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a><span class="fu">meshSim</span>(<span class="fu">c</span>(<span class="st">&quot;D001369&quot;</span>, <span class="st">&quot;D002462&quot;</span>), <span class="fu">c</span>(<span class="st">&quot;D017629&quot;</span>, <span class="st">&quot;D002890&quot;</span>, <span class="st">&quot;D008928&quot;</span>), <span class="at">semData=</span>hsamd, <span class="at">measure=</span><span class="st">&quot;Wang&quot;</span>)</span></code></pre></div>
<pre><code>##           D017629   D002890   D008928
## D001369 0.2886598 0.1923711 0.2193326
## D002462 0.6521739 0.2381925 0.2809552</code></pre>
</div>
<div id="gene-meshes-semantic-similarity" class="section level2" number="4.3">
<h2><span class="header-section-number">4.3</span> Gene semantic similarity measurement</h2>
<p>The <code>geneSim()</code> function is designed to measure semantic similarity among two gene vectors.</p>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a><span class="fu">geneSim</span>(<span class="st">&quot;241&quot;</span>, <span class="st">&quot;251&quot;</span>, <span class="at">semData=</span>hsamd, <span class="at">measure=</span><span class="st">&quot;Wang&quot;</span>, <span class="at">combine=</span><span class="st">&quot;BMA&quot;</span>)</span></code></pre></div>
<pre><code>## [1] 0.487</code></pre>
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a><span class="fu">geneSim</span>(<span class="fu">c</span>(<span class="st">&quot;241&quot;</span>, <span class="st">&quot;251&quot;</span>), <span class="fu">c</span>(<span class="st">&quot;835&quot;</span>, <span class="st">&quot;5261&quot;</span>,<span class="st">&quot;241&quot;</span>, <span class="st">&quot;994&quot;</span>), <span class="at">semData=</span>hsamd, <span class="at">measure=</span><span class="st">&quot;Wang&quot;</span>, <span class="at">combine=</span><span class="st">&quot;BMA&quot;</span>)</span></code></pre></div>
<pre><code>##       835  5261   241   994
## 241 0.732 0.337 1.000 0.438
## 251 0.526 0.588 0.487 0.597</code></pre>

</div>
</div>



<div id="enrichment-overview" class="section level1" number="5">
<h1><span class="header-section-number">Chapter 5</span> Overview of enrichment analysis</h1>
<div id="terminology" class="section level2" number="5.1">
<h2><span class="header-section-number">5.1</span> Terminology</h2>
<div id="gene-sets-and-pathway" class="section level3" number="5.1.1">
<h3><span class="header-section-number">5.1.1</span> Gene sets and pathway</h3>
<p>A gene set is an unordered collection of genes that are functional related. A pathway can be interpreted as a gene set by ignoring functional relationships among genes.</p>
</div>
<div id="gene-ontology-go" class="section level3" number="5.1.2">
<h3><span class="header-section-number">5.1.2</span> Gene Ontology (GO)</h3>
<p><a href="http://www.geneontology.org/">Gene Ontology</a> defines concepts/classes used to describe gene function, and relationships between these concepts. It classifies functions along three aspects:</p>
<ul>
<li>MF: Molecular Function
<ul>
<li>molecular activities of gene products</li>
</ul></li>
<li>CC: Cellular Component
<ul>
<li>where gene products are active</li>
</ul></li>
<li>BP: Biological Process
<ul>
<li>pathways and larger processes made up of the activities of multiple gene products</li>
</ul></li>
</ul>
<p>GO terms are organized in a directed acyclic graph, where edge between the terms represent parent-child relationship.</p>
</div>
<div id="kyoto-encyclopedia-of-genes-and-genomes-kegg" class="section level3" number="5.1.3">
<h3><span class="header-section-number">5.1.3</span> Kyoto Encyclopedia of Genes and Genomes (KEGG)</h3>
<p><a href="https://www.genome.jp/kegg/">KEGG</a> is a collection of manually drawn pathway maps representing molecular interaction and reaction networks. These pathways cover a wide range of biochemical processes that can be divided in <a href="https://www.genome.jp/kegg/pathway.html">7 broad categories</a>:</p>
<ol style="list-style-type: decimal">
<li>Metabolism</li>
<li>Genetic information processing</li>
<li>Environmental information processing</li>
<li>Cellular processes</li>
<li>Organismal systems</li>
<li>Human diseases</li>
<li>Drug development.</li>
</ol>
<!-- https://pathview.uncc.edu/data/khier.tsv -->
</div>
<div id="other-gene-sets" class="section level3" number="5.1.4">
<h3><span class="header-section-number">5.1.4</span> Other gene sets</h3>
<p>GO and KEGG are most frequently used for the functional analysis. They are typically the first choice because their long-standing curation and availability for a wide range of species.</p>
<p>Other gene sets including but not limited to Disease Ontology (<a href="http://disease-ontology.org/">DO</a>), Disease Gene Network (<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4397996/">DisGeNET</a>), <a href="https://www.wikipathways.org">wikiPathways</a>, Molecular Signatures Database (<a href="http://software.broadinstitute.org/gsea/msigdb">MSigDb</a>).</p>
</div>
</div>
<div id="ora-algorithm" class="section level2" number="5.2">
<h2><span class="header-section-number">5.2</span> Over Representation Analysis</h2>
<p>Over Representation Analysis (ORA) <span class="citation">(<a href="#ref-boyle2004" role="doc-biblioref">Boyle et al. 2004</a>)</span> is a widely used approach to determine whether known biological functions or processes are over-represented (= enriched) in an experimentally-derived gene list, <em>e.g.</em> a list of differentially expressed genes (DEGs).</p>
<p>The <em>p</em>-value can be calculated by hypergeometric distribution.</p>
<p><span class="math inline">\(p = 1 - \displaystyle\sum_{i = 0}^{k-1}\frac{{M \choose i}{{N-M} \choose {n-i}}} {{N \choose n}}\)</span></p>
<p>In this equation, <code>N</code> is the total number of genes in the background distribution,
<code>M</code> is the number of genes within that distribution that are annotated (either directly or indirectly) to the gene set of interest,
<code>n</code> is the size of the list of genes of interest and <code>k</code> is the number of genes within that list which
are annotated to the gene set. The background distribution by default is
all the genes that have annotation. <em>P</em>-values should be adjusted for <a href="https://en.wikipedia.org/wiki/Multiple_comparisons_problem">multiple comparison</a>.</p>
<p><strong>Example:</strong> Suppose we have 17,980 genes detected in a Microarray study and 57 genes were differentially expressed. Among the differential expressed genes, 28 are annotated to a gene set<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a>.</p>
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a>d <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">gene.not.interest=</span><span class="fu">c</span>(<span class="dv">2613</span>, <span class="dv">15310</span>), <span class="at">gene.in.interest=</span><span class="fu">c</span>(<span class="dv">28</span>, <span class="dv">29</span>))</span>
<span id="cb55-2"><a href="#cb55-2" aria-hidden="true" tabindex="-1"></a><span class="fu">row.names</span>(d) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;In_category&quot;</span>, <span class="st">&quot;not_in_category&quot;</span>)</span>
<span id="cb55-3"><a href="#cb55-3" aria-hidden="true" tabindex="-1"></a>d</span></code></pre></div>
<pre><code>##                 gene.not.interest gene.in.interest
## In_category                  2613               28
## not_in_category             15310               29</code></pre>
<p>whether the overlap of 25 genes are significantly over represented in the gene set can be assessed using hypergeometric distribution. This corresponding to a one-sided version of Fisher‚Äôs exact test.</p>
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1" aria-hidden="true" tabindex="-1"></a><span class="fu">fisher.test</span>(d, <span class="at">alternative =</span> <span class="st">&quot;greater&quot;</span>)</span></code></pre></div>
<pre><code>## 
##  Fisher&#39;s Exact Test for Count Data
## 
## data:  d
## p-value = 1
## alternative hypothesis: true odds ratio is greater than 1
## 95 percent confidence interval:
##  0.110242      Inf
## sample estimates:
## odds ratio 
##  0.1767937</code></pre>
</div>
<div id="gsea-algorithm" class="section level2" number="5.3">
<h2><span class="header-section-number">5.3</span> Gene Set Enrichment Analysis</h2>
<p>A common approach in analyzing gene expression profiles was identifying differential expressed genes that are deemed interesting. The <a href="#ora-algorithm">ORA enrichment analysis</a> is based on these differential expressed genes. This approach will find genes where the difference is large, but it will not detect a situation where the difference is small, but evidenced in coordinated way in a set of related genes. Gene Set Enrichment Analysis (GSEA)<span class="citation">(<a href="#ref-subramanian_gene_2005" role="doc-biblioref">Subramanian et al. 2005</a>)</span> directly addresses this limitation.
All genes can be used in GSEA; GSEA aggregates the per gene statistics across genes within a gene set, therefore making it possible to detect situations where all genes in a predefined set change in a small but coordinated way. Since it is likely that many relevant phenotypic differences are manifested by small but consistent changes in a set of genes.</p>
<p>Genes are ranked based on their phenotypes. Given a priori defined set of gene <em>S</em> (e.g., genes shareing the same <em>DO</em> category), the goal of GSEA is to determine whether the members of <em>S</em> are randomly distributed throughout the ranked gene list (<em>L</em>) or primarily found at the top or bottom.</p>
<p>There are three key elements of the GSEA method:</p>
<ul>
<li>Calculation of an Enrichment Score.
<ul>
<li>The enrichment score (<em>ES</em>) represent the degree to which a set <em>S</em> is over-represented at the top or bottom of the ranked list <em>L</em>. The score is calculated by walking down the list <em>L</em>, increasing a running-sum statistic when we encounter a gene in <em>S</em> and decreasing when it is not. The magnitude of the increment depends on the gene statistics (e.g., correlation of the gene with phenotype). The <em>ES</em> is the maximum deviation from zero encountered in the random walk; it corresponds to a weighted Kolmogorov-Smirnov-like statistic <span class="citation">(<a href="#ref-subramanian_gene_2005" role="doc-biblioref">Subramanian et al. 2005</a>)</span>.</li>
</ul></li>
<li>Esimation of Significance Level of <em>ES</em>.
<ul>
<li>The <em>p</em>-value of the <em>ES</em> is calculated using permutation test. Specifically, we permute the gene labels of the gene list <em>L</em> and recompute the <em>ES</em> of the gene set for the permutated data, which generate a null distribution for the <em>ES</em>. The <em>p</em>-value of the observed <em>ES</em> is then calculated relative to this null distribution.</li>
</ul></li>
<li>Adjustment for Multiple Hypothesis Testing.
<ul>
<li>When the entire gene sets were evaluated, the estimated significance level was adjusted to account for multiple hypothesis testing and also <em>q</em>-values were calculated for FDR control.</li>
</ul></li>
</ul>
<p>We implemented GSEA algorithm proposed by Subramanian <span class="citation">(<a href="#ref-subramanian_gene_2005" role="doc-biblioref">Subramanian et al. 2005</a>)</span>. Alexey Sergushichev implemented an algorithm for fast GSEA calculation in the <a href="http://bioconductor.org/packages/fgsea">fgsea</a> <span class="citation">(<a href="#ref-korotkevich_fast_2019" role="doc-biblioref">Korotkevich, Sukhov, and Sergushichev 2019</a>)</span> package. In our packages (<a href="http://bioconductor.org/packages/clusterProfiler">clusterProfiler</a>, <a href="http://bioconductor.org/packages/DOSE">DOSE</a>, <a href="http://bioconductor.org/packages/meshes">meshes</a> and <a href="http://bioconductor.org/packages/ReactomePA">ReactomePA</a>), users can use GSEA algorithm implemented in <code>DOSE</code> or <code>fgsea</code> by specifying the parameter <code>by="DOSE"</code> or <code>by="fgsea"</code>. By default, the <code>fgsea</code> method will be used since it is much more faster.</p>
</div>
<div id="leading-edge-analysis-and-core-enriched-genes" class="section level2" number="5.4">
<h2><span class="header-section-number">5.4</span> Leading edge analysis and core enriched genes</h2>
<p>Leading edge analysis reports <code>Tags</code> to indicate the percentage of genes contributing to the enrichment score, <code>List</code> to indicate where in the list the enrichment score is attained and <code>Signal</code> for enrichment signal strength.</p>
<p>It would also be very interesting to get the core enriched genes that contribute to the enrichment. Our packages (<a href="http://bioconductor.org/packages/clusterProfiler">clusterProfiler</a>, <a href="http://bioconductor.org/packages/DOSE">DOSE</a>, <a href="http://bioconductor.org/packages/meshes">meshes</a> and <a href="http://bioconductor.org/packages/ReactomePA">ReactomePA</a>) support leading edge analysis and report core enriched genes in GSEA analysis.</p>

</div>
</div>
<div id="clusterprofiler-go" class="section level1" number="6">
<h1><span class="header-section-number">Chapter 6</span> GO enrichment analysis</h1>
<p>GO comprises of three orthogonal ontologies, i.e.¬†molecular function (MF), biological
process (BP), and cellular component (CC).</p>
<div id="clusterProfiler-go-supported-organisms" class="section level2" number="6.1">
<h2><span class="header-section-number">6.1</span> Supported organisms</h2>
<p>GO analyses (<code>groupGO()</code>, <code>enrichGO()</code> and <code>gseGO()</code>) support organisms that have an <code>OrgDb</code> object available (see also <a href="#gosemsim-supported-organisms">session 2.2</a>).</p>
<p>If user have GO annotation data (in data.frame format with first column of gene ID and second column of GO ID), they can use <code>enricher()</code> and <code>gseGO()</code> functions to perform over-representation test and gene set enrichment analysis.</p>
<p>If genes are annotated by direction annotation, it should also annotated by its ancestor GO nodes (indirect annation). If user only has direct annotation, they can pass their annotation to <code>buildGOmap</code> function, which will infer indirection annotation and generate a <code>data.frame</code> that suitable for both <code>enricher()</code> and <code>gseGO()</code>.</p>
</div>
<div id="go-classification" class="section level2" number="6.2">
<h2><span class="header-section-number">6.2</span> GO classification</h2>
<p>In <a href="http://bioconductor.org/packages/clusterProfiler">clusterProfiler</a>, the <code>groupGO()</code> is designed for gene classification based on GO distribution at a specific level. Here we use the dataset <code>geneList</code> provided by <a href="http://bioconductor.org/packages/DOSE">DOSE</a>.</p>
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(clusterProfiler)</span>
<span id="cb59-2"><a href="#cb59-2" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(geneList, <span class="at">package=</span><span class="st">&quot;DOSE&quot;</span>)</span>
<span id="cb59-3"><a href="#cb59-3" aria-hidden="true" tabindex="-1"></a>gene <span class="ot">&lt;-</span> <span class="fu">names</span>(geneList)[<span class="fu">abs</span>(geneList) <span class="sc">&gt;</span> <span class="dv">2</span>]</span>
<span id="cb59-4"><a href="#cb59-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-5"><a href="#cb59-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Entrez gene ID</span></span>
<span id="cb59-6"><a href="#cb59-6" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(gene)</span></code></pre></div>
<pre><code>## [1] &quot;4312&quot;  &quot;8318&quot;  &quot;10874&quot; &quot;55143&quot; &quot;55388&quot; &quot;991&quot;</code></pre>
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb61-1"><a href="#cb61-1" aria-hidden="true" tabindex="-1"></a>ggo <span class="ot">&lt;-</span> <span class="fu">groupGO</span>(<span class="at">gene     =</span> gene,</span>
<span id="cb61-2"><a href="#cb61-2" aria-hidden="true" tabindex="-1"></a>               <span class="at">OrgDb    =</span> org.Hs.eg.db,</span>
<span id="cb61-3"><a href="#cb61-3" aria-hidden="true" tabindex="-1"></a>               <span class="at">ont      =</span> <span class="st">&quot;CC&quot;</span>,</span>
<span id="cb61-4"><a href="#cb61-4" aria-hidden="true" tabindex="-1"></a>               <span class="at">level    =</span> <span class="dv">3</span>,</span>
<span id="cb61-5"><a href="#cb61-5" aria-hidden="true" tabindex="-1"></a>               <span class="at">readable =</span> <span class="cn">TRUE</span>)</span>
<span id="cb61-6"><a href="#cb61-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-7"><a href="#cb61-7" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(ggo)</span></code></pre></div>
<pre><code>##                    ID                                        Description Count
## GO:0000131 GO:0000131                        incipient cellular bud site     0
## GO:0000151 GO:0000151                           ubiquitin ligase complex     4
## GO:0000159 GO:0000159                protein phosphatase type 2A complex     0
## GO:0000178 GO:0000178                            exosome (RNase complex)     0
## GO:0000307 GO:0000307 cyclin-dependent protein kinase holoenzyme complex     4
## GO:0000408 GO:0000408                                  EKC/KEOPS complex     0
##            GeneRatio                 geneID
## GO:0000131     0/207                       
## GO:0000151     4/207  CDC20/UBE2C/UBE2S/DTL
## GO:0000159     0/207                       
## GO:0000178     0/207                       
## GO:0000307     4/207 CCNB2/CCNA2/CDK1/CCNB1
## GO:0000408     0/207</code></pre>
<p>The <code>gene</code> parameter is a vector of gene IDs (can be any ID type that supported by corresponding <code>OrgDb</code>, see also <a href="#id-convert">session 16.1</a>). If <code>readable</code> is set to <code>TRUE</code>, the input gene IDs will be converted to gene symbols.</p>
</div>
<div id="clusterprofiler-go-ora" class="section level2" number="6.3">
<h2><span class="header-section-number">6.3</span> GO over-representation analysis</h2>
<p>The <a href="http://bioconductor.org/packages/clusterProfiler">clusterProfiler</a> package implemented <code>enrichGO()</code> function for gene ontology over-representation test.</p>
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb63-1"><a href="#cb63-1" aria-hidden="true" tabindex="-1"></a>ego <span class="ot">&lt;-</span> <span class="fu">enrichGO</span>(<span class="at">gene          =</span> gene,</span>
<span id="cb63-2"><a href="#cb63-2" aria-hidden="true" tabindex="-1"></a>                <span class="at">universe      =</span> <span class="fu">names</span>(geneList),</span>
<span id="cb63-3"><a href="#cb63-3" aria-hidden="true" tabindex="-1"></a>                <span class="at">OrgDb         =</span> org.Hs.eg.db,</span>
<span id="cb63-4"><a href="#cb63-4" aria-hidden="true" tabindex="-1"></a>                <span class="at">ont           =</span> <span class="st">&quot;CC&quot;</span>,</span>
<span id="cb63-5"><a href="#cb63-5" aria-hidden="true" tabindex="-1"></a>                <span class="at">pAdjustMethod =</span> <span class="st">&quot;BH&quot;</span>,</span>
<span id="cb63-6"><a href="#cb63-6" aria-hidden="true" tabindex="-1"></a>                <span class="at">pvalueCutoff  =</span> <span class="fl">0.01</span>,</span>
<span id="cb63-7"><a href="#cb63-7" aria-hidden="true" tabindex="-1"></a>                <span class="at">qvalueCutoff  =</span> <span class="fl">0.05</span>,</span>
<span id="cb63-8"><a href="#cb63-8" aria-hidden="true" tabindex="-1"></a>        <span class="at">readable      =</span> <span class="cn">TRUE</span>)</span>
<span id="cb63-9"><a href="#cb63-9" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(ego)</span></code></pre></div>
<pre><code>##                    ID                              Description GeneRatio
## GO:0005819 GO:0005819                                  spindle    25/201
## GO:0072686 GO:0072686                          mitotic spindle    16/201
## GO:0005876 GO:0005876                      spindle microtubule    12/201
## GO:0000779 GO:0000779 condensed chromosome, centromeric region    15/201
## GO:0000775 GO:0000775           chromosome, centromeric region    18/201
## GO:0000776 GO:0000776                              kinetochore    15/201
##              BgRatio       pvalue     p.adjust       qvalue
## GO:0005819 288/11841 1.770094e-11 4.167650e-09 3.831313e-09
## GO:0072686 110/11841 4.495145e-11 4.167650e-09 3.831313e-09
## GO:0005876  52/11841 4.691637e-11 4.167650e-09 3.831313e-09
## GO:0000779  95/11841 5.556867e-11 4.167650e-09 3.831313e-09
## GO:0000775 158/11841 1.712374e-10 1.027424e-08 9.445095e-09
## GO:0000776 109/11841 4.151093e-10 1.841211e-08 1.692622e-08
##                                                                                                                                                         geneID
## GO:0005819 CDCA8/CDC20/KIF23/CENPE/ASPM/DLGAP5/SKA1/NUSAP1/TPX2/TACC3/NEK2/CDK1/MAD2L1/KIF18A/BIRC5/KIF11/TTK/AURKB/PRC1/KIFC1/KIF18B/KIF20A/AURKA/CCNB1/KIF4A
## GO:0072686                                                      KIF23/CENPE/ASPM/SKA1/NUSAP1/TPX2/TACC3/CDK1/MAD2L1/KIF18A/KIF11/AURKB/PRC1/KIFC1/KIF18B/AURKA
## GO:0005876                                                                             CENPE/SKA1/NUSAP1/CDK1/KIF18A/BIRC5/KIF11/AURKB/PRC1/KIF18B/AURKA/KIF4A
## GO:0000779                                                            CENPE/NDC80/HJURP/SKA1/NEK2/CENPM/CENPN/ERCC6L/MAD2L1/CDT1/BIRC5/NCAPG/AURKB/AURKA/CCNB1
## GO:0000775                                           CDCA8/CENPE/NDC80/HJURP/SKA1/NEK2/CENPM/CENPN/ERCC6L/MAD2L1/KIF18A/CDT1/BIRC5/TTK/NCAPG/AURKB/AURKA/CCNB1
## GO:0000776                                                             CENPE/NDC80/HJURP/SKA1/NEK2/CENPM/CENPN/ERCC6L/MAD2L1/KIF18A/CDT1/BIRC5/TTK/AURKB/CCNB1
##            Count
## GO:0005819    25
## GO:0072686    16
## GO:0005876    12
## GO:0000779    15
## GO:0000775    18
## GO:0000776    15</code></pre>
<p>Any gene ID type that supported in <code>OrgDb</code> can be directly used in GO analyses. User need to specify the <code>keyType</code> parameter to specify the input gene ID type.</p>
<div class="sourceCode" id="cb65"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb65-1"><a href="#cb65-1" aria-hidden="true" tabindex="-1"></a>gene.df <span class="ot">&lt;-</span> <span class="fu">bitr</span>(gene, <span class="at">fromType =</span> <span class="st">&quot;ENTREZID&quot;</span>,</span>
<span id="cb65-2"><a href="#cb65-2" aria-hidden="true" tabindex="-1"></a>        <span class="at">toType =</span> <span class="fu">c</span>(<span class="st">&quot;ENSEMBL&quot;</span>, <span class="st">&quot;SYMBOL&quot;</span>),</span>
<span id="cb65-3"><a href="#cb65-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">OrgDb =</span> org.Hs.eg.db)</span>
<span id="cb65-4"><a href="#cb65-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb65-5"><a href="#cb65-5" aria-hidden="true" tabindex="-1"></a>ego2 <span class="ot">&lt;-</span> <span class="fu">enrichGO</span>(<span class="at">gene         =</span> gene.df<span class="sc">$</span>ENSEMBL,</span>
<span id="cb65-6"><a href="#cb65-6" aria-hidden="true" tabindex="-1"></a>                <span class="at">OrgDb         =</span> org.Hs.eg.db,</span>
<span id="cb65-7"><a href="#cb65-7" aria-hidden="true" tabindex="-1"></a>                <span class="at">keyType       =</span> <span class="st">&#39;ENSEMBL&#39;</span>,</span>
<span id="cb65-8"><a href="#cb65-8" aria-hidden="true" tabindex="-1"></a>                <span class="at">ont           =</span> <span class="st">&quot;CC&quot;</span>,</span>
<span id="cb65-9"><a href="#cb65-9" aria-hidden="true" tabindex="-1"></a>                <span class="at">pAdjustMethod =</span> <span class="st">&quot;BH&quot;</span>,</span>
<span id="cb65-10"><a href="#cb65-10" aria-hidden="true" tabindex="-1"></a>                <span class="at">pvalueCutoff  =</span> <span class="fl">0.01</span>,</span>
<span id="cb65-11"><a href="#cb65-11" aria-hidden="true" tabindex="-1"></a>                <span class="at">qvalueCutoff  =</span> <span class="fl">0.05</span>)</span>
<span id="cb65-12"><a href="#cb65-12" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(ego2, <span class="dv">3</span>)                </span></code></pre></div>
<pre><code>##                    ID         Description GeneRatio   BgRatio       pvalue
## GO:0072686 GO:0072686     mitotic spindle    20/233 149/21872 1.456640e-16
## GO:0005819 GO:0005819             spindle    29/233 395/21872 2.896081e-16
## GO:0005876 GO:0005876 spindle microtubule    13/233  68/21872 3.039661e-13
##                p.adjust       qvalue
## GO:0072686 4.213799e-14 3.200932e-14
## GO:0005819 4.213799e-14 3.200932e-14
## GO:0005876 2.948471e-11 2.239750e-11
##                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     geneID
## GO:0072686                                                                                                                                                 ENSG00000137807/ENSG00000138778/ENSG00000066279/ENSG00000154839/ENSG00000262634/ENSG00000137804/ENSG00000088325/ENSG00000013810/ENSG00000170312/ENSG00000164109/ENSG00000121621/ENSG00000138160/ENSG00000178999/ENSG00000198901/ENSG00000237649/ENSG00000056678/ENSG00000204197/ENSG00000233450/ENSG00000186185/ENSG00000087586
## GO:0005819 ENSG00000134690/ENSG00000117399/ENSG00000137807/ENSG00000138778/ENSG00000066279/ENSG00000126787/ENSG00000154839/ENSG00000262634/ENSG00000137804/ENSG00000088325/ENSG00000013810/ENSG00000117650/ENSG00000170312/ENSG00000164109/ENSG00000121621/ENSG00000089685/ENSG00000138160/ENSG00000112742/ENSG00000178999/ENSG00000198901/ENSG00000237649/ENSG00000056678/ENSG00000204197/ENSG00000233450/ENSG00000186185/ENSG00000112984/ENSG00000087586/ENSG00000134057/ENSG00000090889
## GO:0005876                                                                                                                                                                                                                                                                 ENSG00000138778/ENSG00000154839/ENSG00000262634/ENSG00000137804/ENSG00000170312/ENSG00000121621/ENSG00000089685/ENSG00000138160/ENSG00000178999/ENSG00000198901/ENSG00000186185/ENSG00000087586/ENSG00000090889
##            Count
## GO:0072686    20
## GO:0005819    29
## GO:0005876    13</code></pre>
<p>Gene ID can be mapped to gene Symbol by using paramter <code>readable=TRUE</code> or <a href="#setReadable"><code>setReadable()</code> function</a>.</p>
</div>
<div id="go-gene-set-enrichment-analysis" class="section level2" number="6.4">
<h2><span class="header-section-number">6.4</span> GO Gene Set Enrichment Analysis</h2>
<p>The <a href="http://bioconductor.org/packages/clusterProfiler">clusterProfiler</a> package provides <code>gseGO()</code> function for <a href="#gsea-algorithm">gene set enrichment analysis</a> using gene ontology.</p>
<div class="sourceCode" id="cb67"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb67-1"><a href="#cb67-1" aria-hidden="true" tabindex="-1"></a>ego3 <span class="ot">&lt;-</span> <span class="fu">gseGO</span>(<span class="at">geneList     =</span> geneList,</span>
<span id="cb67-2"><a href="#cb67-2" aria-hidden="true" tabindex="-1"></a>              <span class="at">OrgDb        =</span> org.Hs.eg.db,</span>
<span id="cb67-3"><a href="#cb67-3" aria-hidden="true" tabindex="-1"></a>              <span class="at">ont          =</span> <span class="st">&quot;CC&quot;</span>,</span>
<span id="cb67-4"><a href="#cb67-4" aria-hidden="true" tabindex="-1"></a>              <span class="at">minGSSize    =</span> <span class="dv">100</span>,</span>
<span id="cb67-5"><a href="#cb67-5" aria-hidden="true" tabindex="-1"></a>              <span class="at">maxGSSize    =</span> <span class="dv">500</span>,</span>
<span id="cb67-6"><a href="#cb67-6" aria-hidden="true" tabindex="-1"></a>              <span class="at">pvalueCutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb67-7"><a href="#cb67-7" aria-hidden="true" tabindex="-1"></a>              <span class="at">verbose      =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<p>The format of input data, <code>geneList</code>, was documented in the <a href="#genelist">FAQ</a>. Beware that only gene Set size in <code>[minGSSize, maxGSSize]</code> will be tested.</p>
</div>
<div id="clusterprofiler-go-non-model" class="section level2" number="6.5">
<h2><span class="header-section-number">6.5</span> GO analysis for non-model organisms</h2>
<p>Both <code>enrichGO()</code> and <code>gseGO()</code> functions require an <code>OrgDb</code> object as background annotation. For organisms that don‚Äôt have <code>OrgDb</code> provided by Bioconductor, users can query one (if available) online via <a href="http://bioconductor.org/packages/AnnotationHub">AnnotationHub</a>. If there is no <code>OrgDb</code> available, users can obtain GO annotation from other sources, e.g.¬†from <a href="http://bioconductor.org/packages/biomaRt">biomaRt</a> or <a href="https://www.blast2go.com/">Blast2GO</a>. Then the <code>enricher()</code> or <code>GSEA()</code> functions can be used to perform GO analysis for these organisms, similar to the examples using wikiPathways and MSigDB. Another solution is to create <code>OrgDb</code> by your own using <a href="http://bioconductor.org/packages/AnnotationForge">AnnotationForge</a> package.</p>
</div>
<div id="visualize-enriched-go-terms-as-a-directed-acyclic-graph" class="section level2" number="6.6">
<h2><span class="header-section-number">6.6</span> Visualize enriched GO terms as a directed acyclic graph</h2>
<p>The <code>goplot()</code> can accept output of <code>enrichGO</code> and visualized the enriched GO induced graph.</p>


<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb68-1"><a href="#cb68-1" aria-hidden="true" tabindex="-1"></a><span class="fu">goplot</span>(ego)</span></code></pre></div>
<div class="figure"><span id="fig:goplot"></span>
<img src="biomedicalKnowledge_files/figure-html/goplot-1.png" alt="Goplot of enrichment analysis." width="95%" />
<p class="caption">
Figure 6.1: <strong>Goplot of enrichment analysis.</strong>
</p>
</div>
</div>
<div id="clusterprofiler-go-summary" class="section level2" number="6.7">
<h2><span class="header-section-number">6.7</span> Summary</h2>
<p>GO semantic similarity can be calculated by <a href="http://bioconductor.org/packages/GOSemSim">GOSemSim</a> <span class="citation">(<a href="#ref-yu2010" role="doc-biblioref">Yu et al. 2010</a>)</span>. We can use it to cluster genes/proteins into different clusters based on their functional similarity and can also use it to measure the similarities among GO terms to reduce the redundancy of GO enrichment results.</p>

</div>
</div>
<div id="clusterprofiler-kegg" class="section level1" number="7">
<h1><span class="header-section-number">Chapter 7</span> KEGG enrichment analysis</h1>
<p>The KEGG FTP service is not freely available for academic since 2012, and there are many software packages using out-dated KEGG annotation data. The <a href="http://bioconductor.org/packages/clusterProfiler">clusterProfiler</a> package supports downloading latest online version of KEGG data using the <a href="https://www.kegg.jp">KEGG website</a>, which is freely available for academic users. Both KEGG pathway and module are supported in <a href="http://bioconductor.org/packages/clusterProfiler">clusterProfiler</a>.</p>
<div id="clusterProfiler-kegg-supported-organisms" class="section level2" number="7.1">
<h2><span class="header-section-number">7.1</span> Supported organisms</h2>
<p>The <a href="http://bioconductor.org/packages/clusterProfiler">clusterProfiler</a> package supports all organisms that have KEGG annotation data available in KEGG database. User should pass abbreviation of academic name to the <code>organism</code> parameter. The full list of KEGG supported organisms can be accessed via <a href="http://www.genome.jp/kegg/catalog/org_list.html" class="uri">http://www.genome.jp/kegg/catalog/org_list.html</a>. <a href="https://www.genome.jp/kegg/ko.html">KEGG Orthology</a> (KO) Database is also supported by specifying <code>organism = "ko"</code>.</p>
<p>The <a href="http://bioconductor.org/packages/clusterProfiler">clusterProfiler</a> package provides <code>search_kegg_organism()</code> function to help searching supported organisms.</p>
<div class="sourceCode" id="cb69"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb69-1"><a href="#cb69-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(clusterProfiler)</span>
<span id="cb69-2"><a href="#cb69-2" aria-hidden="true" tabindex="-1"></a><span class="fu">search_kegg_organism</span>(<span class="st">&#39;ece&#39;</span>, <span class="at">by=</span><span class="st">&#39;kegg_code&#39;</span>)</span></code></pre></div>
<pre><code>##     kegg_code                        scientific_name common_name
## 366       ece Escherichia coli O157:H7 EDL933 (EHEC)        &lt;NA&gt;</code></pre>
<div class="sourceCode" id="cb71"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb71-1"><a href="#cb71-1" aria-hidden="true" tabindex="-1"></a>ecoli <span class="ot">&lt;-</span> <span class="fu">search_kegg_organism</span>(<span class="st">&#39;Escherichia coli&#39;</span>, <span class="at">by=</span><span class="st">&#39;scientific_name&#39;</span>)</span>
<span id="cb71-2"><a href="#cb71-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(ecoli)</span></code></pre></div>
<pre><code>## [1] 65  3</code></pre>
<div class="sourceCode" id="cb73"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb73-1"><a href="#cb73-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(ecoli)</span></code></pre></div>
<pre><code>##     kegg_code                        scientific_name common_name
## 361       eco           Escherichia coli K-12 MG1655        &lt;NA&gt;
## 362       ecj            Escherichia coli K-12 W3110        &lt;NA&gt;
## 363       ecd            Escherichia coli K-12 DH10B        &lt;NA&gt;
## 364       ebw                Escherichia coli BW2952        &lt;NA&gt;
## 365      ecok            Escherichia coli K-12 MDS42        &lt;NA&gt;
## 366       ece Escherichia coli O157:H7 EDL933 (EHEC)        &lt;NA&gt;</code></pre>
</div>
<div id="clusterprofiler-kegg-pathway-ora" class="section level2" number="7.2">
<h2><span class="header-section-number">7.2</span> KEGG pathway over-representation analysis</h2>
<div class="sourceCode" id="cb75"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb75-1"><a href="#cb75-1" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(geneList, <span class="at">package=</span><span class="st">&quot;DOSE&quot;</span>)</span>
<span id="cb75-2"><a href="#cb75-2" aria-hidden="true" tabindex="-1"></a>gene <span class="ot">&lt;-</span> <span class="fu">names</span>(geneList)[<span class="fu">abs</span>(geneList) <span class="sc">&gt;</span> <span class="dv">2</span>]</span>
<span id="cb75-3"><a href="#cb75-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb75-4"><a href="#cb75-4" aria-hidden="true" tabindex="-1"></a>kk <span class="ot">&lt;-</span> <span class="fu">enrichKEGG</span>(<span class="at">gene         =</span> gene,</span>
<span id="cb75-5"><a href="#cb75-5" aria-hidden="true" tabindex="-1"></a>                 <span class="at">organism     =</span> <span class="st">&#39;hsa&#39;</span>,</span>
<span id="cb75-6"><a href="#cb75-6" aria-hidden="true" tabindex="-1"></a>                 <span class="at">pvalueCutoff =</span> <span class="fl">0.05</span>)</span>
<span id="cb75-7"><a href="#cb75-7" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(kk)</span></code></pre></div>
<pre><code>##                ID                                                   Description
## hsa04110 hsa04110                                                    Cell cycle
## hsa04114 hsa04114                                                Oocyte meiosis
## hsa04218 hsa04218                                           Cellular senescence
## hsa04061 hsa04061 Viral protein interaction with cytokine and cytokine receptor
## hsa03320 hsa03320                                        PPAR signaling pathway
## hsa04914 hsa04914                       Progesterone-mediated oocyte maturation
##          GeneRatio  BgRatio       pvalue     p.adjust       qvalue
## hsa04110     11/95 124/8087 1.853094e-07 3.817373e-05 0.0000374520
## hsa04114     10/95 129/8087 2.409878e-06 2.482174e-04 0.0002435245
## hsa04218     10/95 156/8087 1.323532e-05 9.088250e-04 0.0008916423
## hsa04061      8/95 100/8087 2.066338e-05 1.064164e-03 0.0010440443
## hsa03320      7/95  76/8087 2.790313e-05 1.149609e-03 0.0011278737
## hsa04914      7/95 100/8087 1.625327e-04 5.580289e-03 0.0054747850
##                                                      geneID Count
## hsa04110 8318/991/9133/890/983/4085/7272/1111/891/4174/9232    11
## hsa04114    991/9133/983/4085/51806/6790/891/9232/3708/5241    10
## hsa04218     2305/4605/9133/890/983/51806/1111/891/776/3708    10
## hsa04061           3627/10563/6373/4283/6362/6355/9547/1524     8
## hsa03320                 4312/9415/9370/5105/2167/3158/5346     7
## hsa04914                    9133/890/983/4085/6790/891/5241     7</code></pre>
<p>Input ID type can be <code>kegg</code>, <code>ncbi-geneid</code>, <code>ncbi-proteinid</code> or <code>uniprot</code> (see also <a href="#bitr_kegg">session 16.1.2</a>). Unlike <code>enrichGO()</code>, there is no <code>readable</code> parameter for <code>enrichKEGG()</code>. However, users can use the <a href="#setReadable"><code>setReadable()</code> function</a> if there is an <code>OrgDb</code> available for the species.</p>
</div>
<div id="clusterprofiler-kegg-pathway-gsea" class="section level2" number="7.3">
<h2><span class="header-section-number">7.3</span> KEGG pathway gene set enrichment analysis</h2>
<div class="sourceCode" id="cb77"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb77-1"><a href="#cb77-1" aria-hidden="true" tabindex="-1"></a>kk2 <span class="ot">&lt;-</span> <span class="fu">gseKEGG</span>(<span class="at">geneList     =</span> geneList,</span>
<span id="cb77-2"><a href="#cb77-2" aria-hidden="true" tabindex="-1"></a>               <span class="at">organism     =</span> <span class="st">&#39;hsa&#39;</span>,</span>
<span id="cb77-3"><a href="#cb77-3" aria-hidden="true" tabindex="-1"></a>               <span class="at">minGSSize    =</span> <span class="dv">120</span>,</span>
<span id="cb77-4"><a href="#cb77-4" aria-hidden="true" tabindex="-1"></a>               <span class="at">pvalueCutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb77-5"><a href="#cb77-5" aria-hidden="true" tabindex="-1"></a>               <span class="at">verbose      =</span> <span class="cn">FALSE</span>)</span>
<span id="cb77-6"><a href="#cb77-6" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(kk2)</span></code></pre></div>
<pre><code>##                ID                             Description setSize
## hsa05169 hsa05169            Epstein-Barr virus infection     193
## hsa04613 hsa04613 Neutrophil extracellular trap formation     130
## hsa05166 hsa05166 Human T-cell leukemia virus 1 infection     200
## hsa04218 hsa04218                     Cellular senescence     141
## hsa04510 hsa04510                          Focal adhesion     190
## hsa03013 hsa03013                           RNA transport     135
##          enrichmentScore       NES       pvalue     p.adjust      qvalues rank
## hsa05169       0.4335010  1.939504 1.689197e-07 1.233114e-05 8.890510e-06 2820
## hsa04613       0.4496569  1.910343 3.594203e-06 1.311884e-04 9.458428e-05 2575
## hsa05166       0.3926594  1.766423 8.756111e-06 2.130654e-04 1.536160e-04 1955
## hsa04218       0.4115945  1.755570 6.112069e-05 9.133145e-04 6.584820e-04 1155
## hsa04510      -0.4169068 -1.697998 6.255579e-05 9.133145e-04 6.584820e-04 2183
## hsa03013       0.4103684  1.749128 1.240059e-04 1.256361e-03 9.058120e-04 3383
##                            leading_edge
## hsa05169 tags=39%, list=23%, signal=31%
## hsa04613 tags=37%, list=21%, signal=30%
## hsa05166 tags=26%, list=16%, signal=23%
## hsa04218  tags=17%, list=9%, signal=16%
## hsa04510 tags=27%, list=17%, signal=22%
## hsa03013 tags=39%, list=27%, signal=29%
##                                                                                                                                                                                                                                                                                                                                                                                core_enrichment
## hsa05169 3627/890/6890/9636/898/9134/6502/6772/3126/3112/4609/917/5709/1869/3654/919/915/4067/4938/864/4940/5713/5336/11047/3066/54205/1871/578/1019/637/916/3383/4939/10213/23586/4793/5603/7979/7128/6891/930/5714/3452/6850/5702/4794/7124/3569/7097/5708/2208/8772/3119/5704/7186/5971/3135/1380/958/5610/4792/10018/8819/3134/10379/9641/1147/5718/6300/3109/811/5606/2923/3108/5707/1432
## hsa04613                                                                                                                                      820/366/51311/64581/3015/85236/55506/8970/8357/1535/2359/5336/4688/92815/3066/8336/292/1991/3689/8345/5603/4689/5880/10105/1184/6404/3018/6850/5604/3014/7097/1378/8290/1536/834/5605/1183/728/2215/8335/5594/9734/3674/5578/5582/7417/8331/6300
## hsa05166                                                                                                                      991/9133/890/4085/7850/1111/9232/8061/701/9700/898/4316/9134/3932/3559/3126/3112/4609/3561/917/1869/1029/915/114/2005/5902/55697/1871/1031/2224/292/1019/3689/916/3383/11200/706/3600/6513/3601/468/5604/7124/1030/3569/4049/4055/10393/3119/5901/5971/1959/3135
## hsa04218                                                                                                                                                                                                                                                                 2305/4605/9133/890/983/51806/1111/891/993/3576/1978/898/9134/4609/1869/1029/22808/1871/5499/91860/292/1019/11200/1875
## hsa04510                                                                                                                   5595/5228/7424/1499/4636/83660/7059/5295/1288/23396/3910/3371/3082/1291/394/3791/7450/596/3685/1280/3675/595/2318/3912/1793/1278/1277/1293/10398/55742/2317/7058/25759/56034/3693/3480/5159/857/1292/3908/3909/63923/3913/1287/3679/7060/3479/10451/80310/1311/1101
## hsa03013                                                                                  10460/1978/55110/54913/9688/8894/11260/10799/9631/4116/5042/8761/6396/23165/8662/10248/55706/79833/9775/29107/23636/5905/9513/5901/10775/10557/4927/79902/1981/26986/11171/10762/8480/8891/11097/26019/55308/10940/4686/9972/81929/10556/3646/9470/387082/1977/57122/8563/7514/79023/3837/9818/56000</code></pre>
</div>
<div id="clusterprofiler-kegg-module-ora" class="section level2" number="7.4">
<h2><span class="header-section-number">7.4</span> KEGG module over-representation analysis</h2>
<p><a href="http://www.genome.jp/kegg/module.html">KEGG Module</a> is a collection of manually defined function units. In some situation, KEGG Modules have a more straightforward interpretation.</p>
<div class="sourceCode" id="cb79"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb79-1"><a href="#cb79-1" aria-hidden="true" tabindex="-1"></a>mkk <span class="ot">&lt;-</span> <span class="fu">enrichMKEGG</span>(<span class="at">gene =</span> gene,</span>
<span id="cb79-2"><a href="#cb79-2" aria-hidden="true" tabindex="-1"></a>                   <span class="at">organism =</span> <span class="st">&#39;hsa&#39;</span>,</span>
<span id="cb79-3"><a href="#cb79-3" aria-hidden="true" tabindex="-1"></a>                   <span class="at">pvalueCutoff =</span> <span class="dv">1</span>,</span>
<span id="cb79-4"><a href="#cb79-4" aria-hidden="true" tabindex="-1"></a>                   <span class="at">qvalueCutoff =</span> <span class="dv">1</span>)</span>
<span id="cb79-5"><a href="#cb79-5" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(mkk)                   </span></code></pre></div>
<pre><code>##            ID
## M00912 M00912
## M00095 M00095
## M00053 M00053
## M00003 M00003
## M00049 M00049
##                                                                       Description
## M00912                         NAD biosynthesis, tryptophan =&gt; quinolinate =&gt; NAD
## M00095                             C5 isoprenoid biosynthesis, mevalonate pathway
## M00053 Pyrimidine deoxyribonuleotide biosynthesis, CDP/CTP =&gt; dCDP/dCTP,dTDP/dTTP
## M00003                               Gluconeogenesis, oxaloacetate =&gt; fructose-6P
## M00049                        Adenine ribonucleotide biosynthesis, IMP =&gt; ADP,ATP
##        GeneRatio BgRatio     pvalue  p.adjust     qvalue     geneID Count
## M00912       2/9  12/830 0.00652644 0.0326322 0.02747975 23475/3620     2
## M00095       1/9  10/830 0.10382973 0.2156987 0.18164098       3158     1
## M00053       1/9  14/830 0.14260060 0.2156987 0.18164098       6241     1
## M00003       1/9  18/830 0.17987328 0.2156987 0.18164098       5105     1
## M00049       1/9  22/830 0.21569867 0.2156987 0.18164098      26289     1</code></pre>
</div>
<div id="clusterprofiler-kegg-module-gsea" class="section level2" number="7.5">
<h2><span class="header-section-number">7.5</span> KEGG module gene set enrichment analysis</h2>
<div class="sourceCode" id="cb81"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb81-1"><a href="#cb81-1" aria-hidden="true" tabindex="-1"></a>mkk2 <span class="ot">&lt;-</span> <span class="fu">gseMKEGG</span>(<span class="at">geneList =</span> geneList,</span>
<span id="cb81-2"><a href="#cb81-2" aria-hidden="true" tabindex="-1"></a>                 <span class="at">organism =</span> <span class="st">&#39;hsa&#39;</span>,</span>
<span id="cb81-3"><a href="#cb81-3" aria-hidden="true" tabindex="-1"></a>                 <span class="at">pvalueCutoff =</span> <span class="dv">1</span>)</span>
<span id="cb81-4"><a href="#cb81-4" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(mkk2)</span></code></pre></div>
<pre><code>##            ID
## M00001 M00001
## M00035 M00035
## M00002 M00002
## M00053 M00053
## M00104 M00104
## M00009 M00009
##                                                                       Description
## M00001                  Glycolysis (Embden-Meyerhof pathway), glucose =&gt; pyruvate
## M00035                                                     Methionine degradation
## M00002                   Glycolysis, core module involving three-carbon compounds
## M00053 Pyrimidine deoxyribonuleotide biosynthesis, CDP/CTP =&gt; dCDP/dCTP,dTDP/dTTP
## M00104           Bile acid biosynthesis, cholesterol =&gt; cholate/chenodeoxycholate
## M00009                                     Citrate cycle (TCA cycle, Krebs cycle)
##        setSize enrichmentScore       NES      pvalue  p.adjust   qvalues rank
## M00001      24       0.5739036  1.711643 0.006803306 0.2058382 0.1895878 2886
## M00035      10       0.6784636  1.613869 0.019936410 0.2058382 0.1895878 1555
## M00002      11       0.6421781  1.584692 0.023577638 0.2058382 0.1895878 1381
## M00053      10       0.6648004  1.581368 0.025729772 0.2058382 0.1895878  648
## M00104      10      -0.5876900 -1.345261 0.122557726 0.7529412 0.6934985  961
## M00009      22       0.4504911  1.324526 0.141176471 0.7529412 0.6934985 3514
##                          leading_edge
## M00001 tags=54%, list=23%, signal=42%
## M00035 tags=50%, list=12%, signal=44%
## M00002 tags=55%, list=11%, signal=49%
## M00053  tags=40%, list=5%, signal=38%
## M00104  tags=50%, list=8%, signal=46%
## M00009 tags=50%, list=28%, signal=36%
##                                                         core_enrichment
## M00001 5214/3101/2821/7167/2597/5230/2023/5223/5315/3099/5232/2027/5211
## M00035                                           875/1789/191/1788/1786
## M00002                                    7167/2597/5230/2023/5223/5315
## M00053                                              6241/7298/4830/1841
## M00104                                        6342/10998/1581/3295/8309
## M00009            3418/50/4190/3419/2271/3421/55753/3417/1431/6389/4191</code></pre>
</div>
<div id="visualize-enriched-kegg-pathways" class="section level2" number="7.6">
<h2><span class="header-section-number">7.6</span> Visualize enriched KEGG pathways</h2>
<p>The <a href="http://bioconductor.org/packages/enrichplot">enrichplot</a> package implemented <a href="#enrichplot">several methods</a> to visualize enriched terms. Most of them are general methods that can be used to GO, KEGG, MSigDb and other gene set annotations. Here, we introduce <code>clusterProfiler::browseKEGG()</code> and <code>pathview::pathview()</code> functions to help users exploring enriched KEGG pathway with interested genes.</p>
<p>To view the KEGG pathway, user can use <code>browseKEGG</code> function, which will open web browser and highlight enriched genes.</p>
<div class="sourceCode" id="cb83"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb83-1"><a href="#cb83-1" aria-hidden="true" tabindex="-1"></a><span class="fu">browseKEGG</span>(kk, <span class="st">&#39;hsa04110&#39;</span>)</span></code></pre></div>


<div class="figure"><span id="fig:browseKEGG"></span>
<img src="figures/browseKEGG.png" alt="Explore selected KEGG pathway. Differentially expressed genes that enriched in the selected pathway will be highlighted." width="100%" />
<p class="caption">
Figure 7.1: <strong>Explore selected KEGG pathway.</strong> Differentially expressed genes that enriched in the selected pathway will be highlighted.
</p>
</div>
<p>Users can also use the <code>pathview()</code> function from the <a href="http://bioconductor.org/packages/pathview">pathview</a> <span class="citation">(<a href="#ref-luo_pathview" role="doc-biblioref">Luo and Brouwer 2013</a>)</span> to visualize enriched KEGG pathways identify by the <a href="http://bioconductor.org/packages/clusterProfiler">clusterProfiler</a> package <span class="citation">(<a href="#ref-yu2012" role="doc-biblioref">Yu et al. 2012</a>)</span>.</p>
<p>The following example illustrate how to visualize ‚Äúhsa04110‚Äù pathway, which was enriched in our previous analysis.</p>
<div class="sourceCode" id="cb84"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb84-1"><a href="#cb84-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;pathview&quot;</span>)</span>
<span id="cb84-2"><a href="#cb84-2" aria-hidden="true" tabindex="-1"></a>hsa04110 <span class="ot">&lt;-</span> <span class="fu">pathview</span>(<span class="at">gene.data  =</span> geneList,</span>
<span id="cb84-3"><a href="#cb84-3" aria-hidden="true" tabindex="-1"></a>                     <span class="at">pathway.id =</span> <span class="st">&quot;hsa04110&quot;</span>,</span>
<span id="cb84-4"><a href="#cb84-4" aria-hidden="true" tabindex="-1"></a>                     <span class="at">species    =</span> <span class="st">&quot;hsa&quot;</span>,</span>
<span id="cb84-5"><a href="#cb84-5" aria-hidden="true" tabindex="-1"></a>                     <span class="at">limit      =</span> <span class="fu">list</span>(<span class="at">gene=</span><span class="fu">max</span>(<span class="fu">abs</span>(geneList)), <span class="at">cpd=</span><span class="dv">1</span>))</span></code></pre></div>


<div class="figure"><span id="fig:pathview"></span>
<img src="figures/hsa04110_pathview.png" alt="Visualize selected KEGG pathway by pathview(). Gene expression values can be mapped to gradient color scale." width="100%" />
<p class="caption">
Figure 7.2: <strong>Visualize selected KEGG pathway by <code>pathview()</code>.</strong> Gene expression values can be mapped to gradient color scale.
</p>
</div>

</div>
</div>
<div id="wikipathways-analysis" class="section level1" number="8">
<h1><span class="header-section-number">Chapter 8</span> WikiPathways analysis</h1>
<p><a href="https://www.wikipathways.org">WikiPathways</a> is a continuously updated pathway database curated by a community of researchers and pathway enthusiasts. WikiPathways produces monthly releases of gmt files for supported organisms at <a href="http://data.wikipathways.org/current/gmt/">data.wikipathways.org</a>. The <a href="http://bioconductor.org/packages/clusterProfiler">clusterProfiler</a> package <span class="citation">(<a href="#ref-yu2012" role="doc-biblioref">Yu et al. 2012</a>)</span> supports enrichment analysis (either ORA or GSEA) for WikiPathways using the <code>enrichWP()</code> and <code>gseWP()</code> functions. These functions will automatically download and parse latest WikiPathways GMT file for selected organism.</p>
<p>Supported organisms can be listed by:</p>
<div class="sourceCode" id="cb85"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb85-1"><a href="#cb85-1" aria-hidden="true" tabindex="-1"></a><span class="fu">get_wp_organisms</span>()</span></code></pre></div>
<pre><code>##  [1] &quot;Equus caballus&quot;           &quot;Populus trichocarpa&quot;     
##  [3] &quot;Sus scrofa&quot;               &quot;Anopheles gambiae&quot;       
##  [5] &quot;Arabidopsis thaliana&quot;     &quot;Bos taurus&quot;              
##  [7] &quot;Caenorhabditis elegans&quot;   &quot;Canis familiaris&quot;        
##  [9] &quot;Danio rerio&quot;              &quot;Drosophila melanogaster&quot; 
## [11] &quot;Gallus gallus&quot;            &quot;Homo sapiens&quot;            
## [13] &quot;Mus musculus&quot;             &quot;Oryza sativa&quot;            
## [15] &quot;Pan troglodytes&quot;          &quot;Rattus norvegicus&quot;       
## [17] &quot;Saccharomyces cerevisiae&quot;</code></pre>
<!--


Download the appropriate gmt file and then generate `TERM2GENE` and `TERM2NAME` to use `enricher()` and `GSEA()` functions. 

Use can download wikiPathways manually:

```r
###################################
# download file manually
# 1. visit the website, http://data.wikipathways.org/current/gmt/, to get the url
# 2. use the following code to download it
####################################
url <- "http://data.wikipathways.org/current/gmt/wikipathways-20200810-gmt-Homo_sapiens.gmt"
wpgmtfile <- "wikiPathways-HS.gmt"
download.file(url, destfile = wpgmtfile)
```

As an alternative to manually downloading gmt files, install the [rWikiPathways](http://bioconductor.org/packages/rWikiPathways) to gain scripting access to the latest gmt files using the `downloadPathwayArchive()` function.


```r
## supported organisms can be accessed via the following command:
## rWikiPathways::listOrganisms()

wpgmtfile <- rWikiPathways::downloadPathwayArchive(organism="Homo sapiens", format = "gmt")
```

Once the GMT file was downloaded, we can use `read.gmt.wp()` to parse it. Note that the `read.gmt.wp()` function is designed for WikiPathways GMT file. For ordinary GMT file, please use the `read.gmt()` function.

```r
wp2gene <- read.gmt.wp(wpgmtfile)
#TERM2GENE
wpid2gene <- wp2gene %>% dplyr::select(wpid, gene) 
#TERM2NAME
wpid2name <- wp2gene %>% dplyr::select(wpid, name) 

ewp <- enricher(gene, TERM2GENE = wpid2gene, TERM2NAME = wpid2name)
head(ewp)

ewp2 <- GSEA(geneList, TERM2GENE = wpid2gene, TERM2NAME = wpid2name, verbose=FALSE)
head(ewp2)
```
-->
<div id="clusterprofiler-wikipathway-ora" class="section level2" number="8.1">
<h2><span class="header-section-number">8.1</span> WikiPathways over-representation analysis</h2>
<div class="sourceCode" id="cb87"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb87-1"><a href="#cb87-1" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(geneList, <span class="at">package=</span><span class="st">&quot;DOSE&quot;</span>)</span>
<span id="cb87-2"><a href="#cb87-2" aria-hidden="true" tabindex="-1"></a>gene <span class="ot">&lt;-</span> <span class="fu">names</span>(geneList)[<span class="fu">abs</span>(geneList) <span class="sc">&gt;</span> <span class="dv">2</span>]</span>
<span id="cb87-3"><a href="#cb87-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb87-4"><a href="#cb87-4" aria-hidden="true" tabindex="-1"></a><span class="fu">enrichWP</span>(gene, <span class="at">organism =</span> <span class="st">&quot;Homo sapiens&quot;</span>) </span></code></pre></div>
<pre><code>## #
## # over-representation test
## #
## #...@organism     Homo sapiens 
## #...@ontology     WikiPathways 
## #...@keytype      ENTREZID 
## #...@gene     chr [1:207] &quot;4312&quot; &quot;8318&quot; &quot;10874&quot; &quot;55143&quot; &quot;55388&quot; &quot;991&quot; &quot;6280&quot; &quot;2305&quot; ...
## #...pvalues adjusted by &#39;BH&#39; with cutoff &lt;0.05 
## #...6 enriched terms found
## &#39;data.frame&#39;:    6 obs. of  9 variables:
##  $ ID         : chr  &quot;WP2446&quot; &quot;WP2361&quot; &quot;WP179&quot; &quot;WP3942&quot; ...
##  $ Description: chr  &quot;Retinoblastoma Gene in Cancer&quot; &quot;Gastric Cancer Network 1&quot; &quot;Cell Cycle&quot; &quot;PPAR signaling pathway&quot; ...
##  $ GeneRatio  : chr  &quot;11/105&quot; &quot;6/105&quot; &quot;10/105&quot; &quot;7/105&quot; ...
##  $ BgRatio    : chr  &quot;90/7477&quot; &quot;29/7477&quot; &quot;120/7477&quot; &quot;67/7477&quot; ...
##  $ pvalue     : num  4.07e-08 2.43e-06 6.22e-06 3.84e-05 4.45e-05 ...
##  $ p.adjust   : num  8.96e-06 2.67e-04 4.56e-04 1.96e-03 1.96e-03 ...
##  $ qvalue     : num  8.19e-06 2.44e-04 4.17e-04 1.79e-03 1.79e-03 ...
##  $ geneID     : chr  &quot;8318/9133/7153/6241/890/983/81620/7272/1111/891/24137&quot; &quot;4605/7153/11065/22974/6286/6790&quot; &quot;8318/991/9133/890/983/7272/1111/891/4174/9232&quot; &quot;4312/9415/9370/5105/2167/3158/5346&quot; ...
##  $ Count      : int  11 6 10 7 4 7
## #...Citation
##   Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.
##   clusterProfiler: an R package for comparing biological themes among
##   gene clusters. OMICS: A Journal of Integrative Biology
##   2012, 16(5):284-287</code></pre>
</div>
<div id="clusterprofiler-wikipathway-gsea" class="section level2" number="8.2">
<h2><span class="header-section-number">8.2</span> WikiPathways gene set enrichment analysis</h2>
<div class="sourceCode" id="cb89"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb89-1"><a href="#cb89-1" aria-hidden="true" tabindex="-1"></a><span class="fu">gseWP</span>(geneList, <span class="at">organism =</span> <span class="st">&quot;Homo sapiens&quot;</span>)</span></code></pre></div>
<pre><code>## #
## # Gene Set Enrichment Analysis
## #
## #...@organism     Homo sapiens 
## #...@setType      WikiPathways 
## #...@keytype      ENTREZID 
## #...@geneList     Named num [1:12495] 4.57 4.51 4.42 4.14 3.88 ...
##  - attr(*, &quot;names&quot;)= chr [1:12495] &quot;4312&quot; &quot;8318&quot; &quot;10874&quot; &quot;55143&quot; ...
## #...nPerm     
## #...pvalues adjusted by &#39;BH&#39; with cutoff &lt;0.05 
## #...59 enriched terms found
## &#39;data.frame&#39;:    59 obs. of  11 variables:
##  $ ID             : chr  &quot;WP179&quot; &quot;WP2446&quot; &quot;WP466&quot; &quot;WP45&quot; ...
##  $ Description    : chr  &quot;Cell Cycle&quot; &quot;Retinoblastoma Gene in Cancer&quot; &quot;DNA Replication&quot; &quot;G1 to S cell cycle control&quot; ...
##  $ setSize        : int  111 84 42 62 23 41 76 61 36 14 ...
##  $ enrichmentScore: num  0.663 0.731 0.792 0.665 0.837 ...
##  $ NES            : num  2.74 2.84 2.73 2.48 2.5 ...
##  $ pvalue         : num  1.00e-10 1.00e-10 1.00e-10 2.93e-10 3.49e-09 ...
##  $ p.adjust       : num  1.69e-08 1.69e-08 1.69e-08 3.71e-08 3.54e-07 ...
##  $ qvalues        : num  1.42e-08 1.42e-08 1.42e-08 3.12e-08 2.97e-07 ...
##  $ rank           : num  1234 1333 1002 1111 854 ...
##  $ leading_edge   : chr  &quot;tags=40%, list=10%, signal=36%&quot; &quot;tags=51%, list=11%, signal=46%&quot; &quot;tags=55%, list=8%, signal=51%&quot; &quot;tags=42%, list=9%, signal=38%&quot; ...
##  $ core_enrichment: chr  &quot;8318/991/9133/890/983/7272/1111/891/4174/9232/4171/993/990/5347/9700/898/23594/4998/9134/4175/4173/10926/6502/9&quot;| __truncated__ &quot;8318/9133/7153/6241/890/983/81620/7272/1111/891/24137/993/898/4998/10733/9134/4175/4173/6502/5984/994/7298/3015&quot;| __truncated__ &quot;8318/55388/81620/4174/4171/990/23594/4998/4175/4173/10926/5984/5111/51053/8317/5427/23649/4176/5982/5557/5558/4172/5424&quot; &quot;8318/983/891/4174/4171/993/898/23594/4998/9134/4175/4173/4609/5111/1869/1029/5427/23649/4176/5557/1871/1031/101&quot;| __truncated__ ...
## #...Citation
##   Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.
##   clusterProfiler: an R package for comparing biological themes among
##   gene clusters. OMICS: A Journal of Integrative Biology
##   2012, 16(5):284-287</code></pre>
<p>If your input gene ID type is not Entrez gene ID, you can use the <a href="#bitr"><code>bitr()</code></a> function to convert gene ID. If you want to convert the gene IDs in output result to gene symbols, you can use the <a href="#setReadable"><code>setReadable()</code></a> function.</p>

</div>
</div>
<div id="reactomepa" class="section level1" number="9">
<h1><span class="header-section-number">Chapter 9</span> Reactome enrichment analysis</h1>
<p><a href="http://bioconductor.org/packages/ReactomePA">ReactomePA</a> is designed for reactome pathway based analysis <span class="citation">(<a href="#ref-yu_reactomepa_2016" role="doc-biblioref">Yu and He 2016</a>)</span>. Reactome is an open-source, open access, manually curated and peer-reviewed pathway database.</p>
<div id="reactomepa-supported-organisms" class="section level2" number="9.1">
<h2><span class="header-section-number">9.1</span> Supported organisms</h2>
<p>Currently <a href="http://bioconductor.org/packages/ReactomePA">ReactomePA</a> supports several model organisms, including ‚Äòcelegans,‚Äô ‚Äòfly,‚Äô ‚Äòhuman,‚Äô ‚Äòmouse,‚Äô ‚Äòrat,‚Äô ‚Äòyeast‚Äô and ‚Äòzebrafish.‚Äô The input gene ID should be Entrez gene ID. We recommend using <a href="#bitr"><code>clusterProfiler::bitr()</code></a> to convert biological IDs.</p>
</div>
<div id="reactomepa-ora" class="section level2" number="9.2">
<h2><span class="header-section-number">9.2</span> Reactome pathway over-representation analysis</h2>
<p>Enrichment analysis is a widely used approach to identify biological
themes. <a href="http://bioconductor.org/packages/ReactomePA">ReactomePA</a> implemented <code>enrichPathway()</code> that uses <a href="#ora-algorithm">hypergeometric model</a> to assess whether the number of selected genes associated with a reactome pathway is larger than expected.</p>
<div class="sourceCode" id="cb91"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb91-1"><a href="#cb91-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ReactomePA)</span>
<span id="cb91-2"><a href="#cb91-2" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(geneList)</span>
<span id="cb91-3"><a href="#cb91-3" aria-hidden="true" tabindex="-1"></a>de <span class="ot">&lt;-</span> <span class="fu">names</span>(geneList)[<span class="fu">abs</span>(geneList) <span class="sc">&gt;</span> <span class="fl">1.5</span>]</span>
<span id="cb91-4"><a href="#cb91-4" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(de)</span></code></pre></div>
<pre><code>## [1] &quot;4312&quot;  &quot;8318&quot;  &quot;10874&quot; &quot;55143&quot; &quot;55388&quot; &quot;991&quot;</code></pre>
<div class="sourceCode" id="cb93"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb93-1"><a href="#cb93-1" aria-hidden="true" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="fu">enrichPathway</span>(<span class="at">gene=</span>de, <span class="at">pvalueCutoff =</span> <span class="fl">0.05</span>, <span class="at">readable=</span><span class="cn">TRUE</span>)</span>
<span id="cb93-2"><a href="#cb93-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(x)</span></code></pre></div>
<pre><code>##                          ID
## R-HSA-69620     R-HSA-69620
## R-HSA-69618     R-HSA-69618
## R-HSA-2500257 R-HSA-2500257
## R-HSA-141424   R-HSA-141424
## R-HSA-141444   R-HSA-141444
## R-HSA-9648025 R-HSA-9648025
##                                                                                        Description
## R-HSA-69620                                                                 Cell Cycle Checkpoints
## R-HSA-69618                                                             Mitotic Spindle Checkpoint
## R-HSA-2500257                                              Resolution of Sister Chromatid Cohesion
## R-HSA-141424                                         Amplification of signal from the kinetochores
## R-HSA-141444  Amplification  of signal from unattached  kinetochores via a MAD2  inhibitory signal
## R-HSA-9648025                                           EML4 and NUDC in mitotic spindle formation
##               GeneRatio   BgRatio       pvalue     p.adjust       qvalue
## R-HSA-69620      38/326 294/10704 2.720718e-14 2.035097e-11 1.807129e-11
## R-HSA-69618      22/326 113/10704 2.569512e-12 8.279535e-10 7.352078e-10
## R-HSA-2500257    23/326 126/10704 3.320669e-12 8.279535e-10 7.352078e-10
## R-HSA-141424     20/326  96/10704 6.983701e-12 1.044762e-09 9.277295e-10
## R-HSA-141444     20/326  96/10704 6.983701e-12 1.044762e-09 9.277295e-10
## R-HSA-9648025    20/326 117/10704 3.063131e-10 3.818703e-08 3.390939e-08
##                                                                                                                                                                                                                                       geneID
## R-HSA-69620   CDC45/CDCA8/MCM10/CDC20/CENPE/CCNB2/NDC80/UBE2C/SKA1/CENPM/CENPN/UBE2S/CCNA2/CDK1/ERCC6L/MAD2L1/KIF18A/BIRC5/AURKB/CHEK1/CCNB1/MCM5/MCM2/KIF2C/CDC25A/CDC6/PLK1/BUB1B/GTSE1/EXO1/ZWINT/CENPU/SPC25/CENPI/CCNE1/ORC6/ORC1/TAOK1
## R-HSA-69618                                                                                             CDCA8/CDC20/CENPE/NDC80/UBE2C/SKA1/CENPM/CENPN/UBE2S/ERCC6L/MAD2L1/KIF18A/BIRC5/AURKB/KIF2C/PLK1/BUB1B/ZWINT/CENPU/SPC25/CENPI/TAOK1
## R-HSA-2500257                                                                                      CDCA8/CDC20/CENPE/CCNB2/NDC80/SKA1/CENPM/CENPN/CDK1/ERCC6L/MAD2L1/KIF18A/BIRC5/AURKB/CCNB1/KIF2C/PLK1/BUB1B/ZWINT/CENPU/SPC25/CENPI/TAOK1
## R-HSA-141424                                                                                                        CDCA8/CDC20/CENPE/NDC80/SKA1/CENPM/CENPN/ERCC6L/MAD2L1/KIF18A/BIRC5/AURKB/KIF2C/PLK1/BUB1B/ZWINT/CENPU/SPC25/CENPI/TAOK1
## R-HSA-141444                                                                                                        CDCA8/CDC20/CENPE/NDC80/SKA1/CENPM/CENPN/ERCC6L/MAD2L1/KIF18A/BIRC5/AURKB/KIF2C/PLK1/BUB1B/ZWINT/CENPU/SPC25/CENPI/TAOK1
## R-HSA-9648025                                                                                                       CDCA8/CDC20/CENPE/NDC80/SKA1/CENPM/CENPN/ERCC6L/MAD2L1/KIF18A/BIRC5/AURKB/KIF2C/PLK1/BUB1B/ZWINT/CENPU/SPC25/CENPI/TAOK1
##               Count
## R-HSA-69620      38
## R-HSA-69618      22
## R-HSA-2500257    23
## R-HSA-141424     20
## R-HSA-141444     20
## R-HSA-9648025    20</code></pre>
</div>
<div id="reactomepa-gsea" class="section level2" number="9.3">
<h2><span class="header-section-number">9.3</span> Reactome pathway gene set enrichment analysis</h2>
<div class="sourceCode" id="cb95"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb95-1"><a href="#cb95-1" aria-hidden="true" tabindex="-1"></a>y <span class="ot">&lt;-</span> <span class="fu">gsePathway</span>(geneList, </span>
<span id="cb95-2"><a href="#cb95-2" aria-hidden="true" tabindex="-1"></a>                <span class="at">pvalueCutoff =</span> <span class="fl">0.2</span>,</span>
<span id="cb95-3"><a href="#cb95-3" aria-hidden="true" tabindex="-1"></a>                <span class="at">pAdjustMethod =</span> <span class="st">&quot;BH&quot;</span>, </span>
<span id="cb95-4"><a href="#cb95-4" aria-hidden="true" tabindex="-1"></a>                <span class="at">verbose =</span> <span class="cn">FALSE</span>)</span>
<span id="cb95-5"><a href="#cb95-5" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(y)</span></code></pre></div>
<pre><code>##                        ID
## R-HSA-141424 R-HSA-141424
## R-HSA-141444 R-HSA-141444
## R-HSA-162909 R-HSA-162909
## R-HSA-174143 R-HSA-174143
## R-HSA-174178 R-HSA-174178
## R-HSA-174184 R-HSA-174184
##                                                                                                           Description
## R-HSA-141424                                                            Amplification of signal from the kinetochores
## R-HSA-141444                     Amplification  of signal from unattached  kinetochores via a MAD2  inhibitory signal
## R-HSA-162909                                                                         Host Interactions of HIV factors
## R-HSA-174143                                                        APC/C-mediated degradation of cell cycle proteins
## R-HSA-174178 APC/C:Cdh1 mediated degradation of Cdc20 and other APC/C:Cdh1 targeted proteins in late mitosis/early G1
## R-HSA-174184                                                     Cdc20:Phospho-APC/C mediated degradation of Cyclin A
##              setSize enrichmentScore      NES pvalue     p.adjust      qvalues
## R-HSA-141424      81       0.7111667 2.790129  1e-10 3.994286e-09 2.986466e-09
## R-HSA-141444      81       0.7111667 2.790129  1e-10 3.994286e-09 2.986466e-09
## R-HSA-162909     122       0.5860187 2.442364  1e-10 3.994286e-09 2.986466e-09
## R-HSA-174143      79       0.6940241 2.713148  1e-10 3.994286e-09 2.986466e-09
## R-HSA-174178      65       0.6868145 2.570550  1e-10 3.994286e-09 2.986466e-09
## R-HSA-174184      64       0.6941964 2.569925  1e-10 3.994286e-09 2.986466e-09
##              rank                   leading_edge
## R-HSA-141424  759  tags=31%, list=6%, signal=29%
## R-HSA-141444  759  tags=31%, list=6%, signal=29%
## R-HSA-162909 3321 tags=57%, list=27%, signal=43%
## R-HSA-174143 1851 tags=47%, list=15%, signal=40%
## R-HSA-174178 1851 tags=46%, list=15%, signal=40%
## R-HSA-174184 1851 tags=45%, list=15%, signal=39%
##                                                                                                                                                                                                                                                                                                                                                                 core_enrichment
## R-HSA-141424                                                                                                                                                                                                                             55143/991/1062/10403/220134/79019/55839/54821/4085/81930/332/9212/11004/5347/701/11130/79682/57405/2491/1058/699/1063/55055/79980/9735
## R-HSA-141444                                                                                                                                                                                                                             55143/991/1062/10403/220134/79019/55839/54821/4085/81930/332/9212/11004/5347/701/11130/79682/57405/2491/1058/699/1063/55055/79980/9735
## R-HSA-162909 3159/3932/5688/9688/3055/5709/5698/919/5693/5902/5713/5721/5691/292/1104/5685/1794/9631/5690/5684/5686/5695/11168/10213/940/23198/7979/6396/23165/5699/5714/6921/55706/8815/5702/23636/9978/5905/5708/1174/5692/5704/5901/164/60489/4927/79902/5683/10762/1175/5694/8480/11097/5718/5682/5716/9972/904/81929/162/5707/57122/4869/926/5696/7514/79023/3837/920/9818
## R-HSA-174143                                                                                                                                                                          991/11065/4751/27338/890/983/4085/9212/6790/891/9232/5347/701/6502/5688/5709/26271/5698/5693/5713/5721/5691/5685/5690/5684/5686/5695/10213/23198/7979/5699/5714/5702/5708/5692/10393/5704
## R-HSA-174178                                                                                                                                                                                                          991/11065/27338/9212/6790/9232/5347/6502/5688/5709/5698/5693/5713/5721/5691/5685/5690/5684/5686/5695/10213/23198/7979/5699/5714/5702/5708/5692/10393/5704
## R-HSA-174184                                                                                                                                                                                                                  991/11065/27338/890/983/4085/701/5688/5709/5698/5693/5713/5721/5691/5685/5690/5684/5686/5695/10213/23198/7979/5699/5714/5702/5708/5692/10393/5704</code></pre>
</div>
<div id="pathway-visualization" class="section level2" number="9.4">
<h2><span class="header-section-number">9.4</span> Pathway Visualization</h2>
<p><a href="http://bioconductor.org/packages/ReactomePA">ReactomePA</a> implemented the <code>viewPathway()</code> to visualized selected reactome pathway. More general purpose of visualization methods for ORA and GSEA results are provided in the <a href="http://bioconductor.org/packages/enrichplot">enrichplot</a> package and are documented on <a href="#enrichplot">Chapter 14</a>.</p>


<div class="sourceCode" id="cb97"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb97-1"><a href="#cb97-1" aria-hidden="true" tabindex="-1"></a><span class="fu">viewPathway</span>(<span class="st">&quot;E2F mediated regulation of DNA replication&quot;</span>, </span>
<span id="cb97-2"><a href="#cb97-2" aria-hidden="true" tabindex="-1"></a>            <span class="at">readable =</span> <span class="cn">TRUE</span>, </span>
<span id="cb97-3"><a href="#cb97-3" aria-hidden="true" tabindex="-1"></a>            <span class="at">foldChange =</span> geneList)</span></code></pre></div>
<div class="figure"><span id="fig:viewpathway"></span>
<img src="biomedicalKnowledge_files/figure-html/viewpathway-1.png" alt="Visualize reactome pathway." width="95%" />
<p class="caption">
Figure 9.1: <strong>Visualize reactome pathway.</strong>
</p>
</div>
<!--

## Pathway analysis of NGS data

Pathway analysis using NGS data (eg, RNA-Seq and ChIP-Seq) can be performed by linking coding and non-coding regions to coding genes via [ChIPseeker](http://bioconductor.org/packages/ChIPseeker) package, which can annotates genomic regions to their nearest genes, host genes, and flanking genes respectivly. In addtion, it provides a function, __*seq2gene*__, that simultaneously considering host genes, promoter region and flanking gene from intergenic region that may under control via cis-regulation. This function maps genomic regions to genes in a many-to-many manner and facilitate functional analysis. For more details, please refer to [ChIPseeker](http://bioconductor.org/packages/ChIPseeker)[@yu_chipseeker_2015].


## Visualize enrichment result

We implement barplot, dotplot enrichment map and category-gene-network for visualization. It is very common to visualize the enrichment result in bar or pie chart. We believe the pie chart is misleading and only provide bar chart.
```r fig.height=6, fig.width=12}
barplot(x, showCategory=8)
```
```r fig.height=6, fig.width=12}
dotplot(x, showCategory=15)
```

Enrichment map can be viusalized by __*enrichMap*__:
```r fig.height=10, fig.width=10}
emapplot(x)
```



In order to consider the potentially biological complexities in which a gene may belong to multiple annotation categories, we developed __*cnetplot*__ function to extract the complex association between genes and diseases.
```r fig.height=8, fig.width=8}
cnetplot(x, categorySize="pvalue", foldChange=geneList)
```


## Visualize GSEA result

```r fig.height=8, fig.width=8}
emapplot(y, color="pvalue")
```


```r fig.height=7, fig.width=10}
gseaplot(y, geneSetID = "R-HSA-69242")
```


## Comparing enriched reactome pathways among gene clusters with clusterProfiler

We have developed an `R` package [clusterProfiler](http://bioconductor.org/packages/clusterProfiler)[@yu_clusterprofiler:_2012] for comparing biological themes among gene clusters. [ReactomePA](http://bioconductor.org/packages/ReactomePA) works fine with [clusterProfiler](http://bioconductor.org/packages/clusterProfiler) and can compare biological themes at reactome pathway perspective.

```r fig.height=8, fig.width=13, eval=FALSE}
require(clusterProfiler)
data(gcSample)
res <- compareCluster(gcSample, fun="enrichPathway")
dotplot(res)
```


-->

</div>
</div>
<div id="dose-enrichment" class="section level1" number="10">
<h1><span class="header-section-number">Chapter 10</span> Disease enrichment analysis</h1>
<p>We developed <a href="http://bioconductor.org/packages/DOSE">DOSE</a> <span class="citation">(<a href="#ref-yu_dose_2015" role="doc-biblioref">Yu et al. 2015</a>)</span> package to promote the investigation of diseases. <a href="http://bioconductor.org/packages/DOSE">DOSE</a> provides five methods for <a href="#DOSE-semantic-similarity">measuring semantic similarities among DO terms and gene products</a>, hypergeometric model and gene set enrichment analysis (GSEA) for associating disease with gene list and extracting disease association insight from genome wide expression profiles.</p>
<div id="dose-ora" class="section level2" number="10.1">
<h2><span class="header-section-number">10.1</span> Disease over-representation analysis</h2>
<p><a href="http://bioconductor.org/packages/DOSE">DOSE</a> supports enrichment analysis of Disease Ontology (DO) <span class="citation">(<a href="#ref-schriml_disease_2011" role="doc-biblioref">Schriml et al. 2011</a>)</span>, <a href="http://ncg.kcl.ac.uk/">Network of Cancer Gene</a> <span class="citation">(<a href="#ref-omer_ncg" role="doc-biblioref">A. et al. 2016</a>)</span> and <a href="http://disgenet.org/">Disease Gene Network</a> (DisGeNET) <span class="citation">(<a href="#ref-janet_disgenet" role="doc-biblioref">Janet et al. 2015</a>)</span>. In addition, several visualization methods were provided by <a href="http://bioconductor.org/packages/enrichplot">enrichplot</a> to help interpreting semantic and enrichment results.</p>
<div id="dose-do-ora" class="section level3" number="10.1.1">
<h3><span class="header-section-number">10.1.1</span> Over-representation analysis for disease ontology</h3>
<p>In the following example, we selected fold change above 1.5 as the differential genes and analyzing their disease association.</p>
<div class="sourceCode" id="cb98"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb98-1"><a href="#cb98-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DOSE)</span>
<span id="cb98-2"><a href="#cb98-2" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(geneList)</span>
<span id="cb98-3"><a href="#cb98-3" aria-hidden="true" tabindex="-1"></a>gene <span class="ot">&lt;-</span> <span class="fu">names</span>(geneList)[<span class="fu">abs</span>(geneList) <span class="sc">&gt;</span> <span class="fl">1.5</span>]</span>
<span id="cb98-4"><a href="#cb98-4" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(gene)</span></code></pre></div>
<pre><code>## [1] &quot;4312&quot;  &quot;8318&quot;  &quot;10874&quot; &quot;55143&quot; &quot;55388&quot; &quot;991&quot;</code></pre>
<div class="sourceCode" id="cb100"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb100-1"><a href="#cb100-1" aria-hidden="true" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="fu">enrichDO</span>(<span class="at">gene          =</span> gene,</span>
<span id="cb100-2"><a href="#cb100-2" aria-hidden="true" tabindex="-1"></a>              <span class="at">ont           =</span> <span class="st">&quot;DO&quot;</span>,</span>
<span id="cb100-3"><a href="#cb100-3" aria-hidden="true" tabindex="-1"></a>              <span class="at">pvalueCutoff  =</span> <span class="fl">0.05</span>,</span>
<span id="cb100-4"><a href="#cb100-4" aria-hidden="true" tabindex="-1"></a>              <span class="at">pAdjustMethod =</span> <span class="st">&quot;BH&quot;</span>,</span>
<span id="cb100-5"><a href="#cb100-5" aria-hidden="true" tabindex="-1"></a>              <span class="at">universe      =</span> <span class="fu">names</span>(geneList),</span>
<span id="cb100-6"><a href="#cb100-6" aria-hidden="true" tabindex="-1"></a>              <span class="at">minGSSize     =</span> <span class="dv">5</span>,</span>
<span id="cb100-7"><a href="#cb100-7" aria-hidden="true" tabindex="-1"></a>              <span class="at">maxGSSize     =</span> <span class="dv">500</span>,</span>
<span id="cb100-8"><a href="#cb100-8" aria-hidden="true" tabindex="-1"></a>              <span class="at">qvalueCutoff  =</span> <span class="fl">0.05</span>,</span>
<span id="cb100-9"><a href="#cb100-9" aria-hidden="true" tabindex="-1"></a>              <span class="at">readable      =</span> <span class="cn">FALSE</span>)</span>
<span id="cb100-10"><a href="#cb100-10" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(x)</span></code></pre></div>
<pre><code>##                    ID                    Description GeneRatio  BgRatio
## DOID:170     DOID:170         endocrine gland cancer    48/331 472/6268
## DOID:10283 DOID:10283                prostate cancer    40/331 394/6268
## DOID:3459   DOID:3459               breast carcinoma    37/331 357/6268
## DOID:3856   DOID:3856 male reproductive organ cancer    40/331 404/6268
## DOID:824     DOID:824                  periodontitis    16/331 109/6268
## DOID:3905   DOID:3905                 lung carcinoma    43/331 465/6268
##                  pvalue    p.adjust      qvalue
## DOID:170   5.662129e-06 0.004784499 0.003826407
## DOID:10283 3.859157e-05 0.013921739 0.011133923
## DOID:3459  4.942629e-05 0.013921739 0.011133923
## DOID:3856  6.821467e-05 0.014410349 0.011524689
## DOID:824   1.699304e-04 0.018859464 0.015082872
## DOID:3905  1.749754e-04 0.018859464 0.015082872
##                                                                                                                                                                                                                                                   geneID
## DOID:170   10874/7153/1381/6241/11065/10232/332/6286/2146/10112/891/9232/4171/993/5347/4318/3576/1515/4821/8836/3159/7980/5888/333/898/9768/4288/3551/2152/9590/185/7043/3357/2952/5327/3667/1634/1287/4582/7122/3479/4680/6424/80310/652/8839/9547/1524
## DOID:10283                                          4312/6280/6279/597/3627/332/6286/2146/4321/4521/891/5347/4102/4318/701/3576/79852/10321/6352/4288/3551/2152/247/2952/3487/367/3667/4128/4582/563/3679/4117/7031/3479/6424/10451/80310/652/4036/10551
## DOID:3459                                                          4312/6280/6279/7153/4751/890/4085/332/6286/6790/891/9232/10855/4171/5347/4318/701/2633/3576/9636/898/8792/4288/2952/4982/4128/4582/7031/3479/771/4250/2066/3169/10647/5304/5241/10551
## DOID:3856                                           4312/6280/6279/597/3627/332/6286/2146/4321/4521/891/5347/4102/4318/701/3576/79852/10321/6352/4288/3551/2152/247/2952/3487/367/3667/4128/4582/563/3679/4117/7031/3479/6424/10451/80310/652/4036/10551
## DOID:824                                                                                                                                                                   4312/6279/820/7850/4321/3595/4318/4069/3576/1493/6352/8842/185/2952/5327/4982
## DOID:3905                          4312/6280/2305/9133/6279/7153/6278/6241/55165/11065/8140/10232/332/6286/3002/9212/4521/891/4171/9928/8061/4318/3576/1978/1894/7980/7083/898/6352/8842/4288/2152/2697/2952/3572/4582/7049/563/3479/1846/3117/2532/2922
##            Count
## DOID:170      48
## DOID:10283    40
## DOID:3459     37
## DOID:3856     40
## DOID:824      16
## DOID:3905     43</code></pre>
<p>The <code>enrichDO()</code> function requires an entrezgene ID vector as input, mostly is the differential gene list of gene expression profile studies. Please refer to <a href="#id-convert">session 16.1</a> if you need to conver other gene ID types to entrezgene ID.</p>
<p>The <code>ont</code> parameter can be ‚ÄúDO‚Äù or ‚ÄúDOLite,‚Äù DOLite <span class="citation">(<a href="#ref-Du15062009" role="doc-biblioref">Du et al. 2009</a>)</span> was constructed to aggregate the redundant DO terms. The DOLite data is not updated, we recommend user use <code>ont="DO"</code>. <code>pvalueCutoff</code> setting the cutoff value of <em>p</em> value and adjusted <em>p</em> value; <code>pAdjustMethod</code> setting the <em>p</em> value correction methods, include the Bonferroni correction (‚Äúbonferroni‚Äù), Holm (‚Äúholm‚Äù), Hochberg (‚Äúhochberg‚Äù), Hommel (‚Äúhommel‚Äù), Benjamini &amp; Hochberg (‚ÄúBH‚Äù) and Benjamini &amp; Yekutieli (‚ÄúBY‚Äù) while <code>qvalueCutoff</code> is used to control <em>q</em>-values.</p>
<p>The <code>universe</code> setting the background gene universe for testing. If user do not explicitly setting this parameter, <code>enrichDO()</code> will set the universe to all human genes that have DO annotation.</p>
<p>The <code>minGSSize</code> (and <code>maxGSSize</code>) indicates that only those DO terms that have more than <code>minGSSize</code> (and less than <code>maxGSSize</code>) genes annotated will be tested.</p>
<p>The <code>readable</code> is a logical parameter, indicates whether the entrezgene IDs will mapping to gene symbols or not (see also <a href="#setReadable">setReadable</a>).</p>
</div>
<div id="dose-ncg-ora" class="section level3" number="10.1.2">
<h3><span class="header-section-number">10.1.2</span> Over-representation analysis for the network of cancer gene</h3>
<p><a href="http://ncg.kcl.ac.uk/">Network of Cancer Gene</a> (NCG) <span class="citation">(<a href="#ref-omer_ncg" role="doc-biblioref">A. et al. 2016</a>)</span> is a manually curated repository of cancer genes. NCG release 5.0 (Aug.¬†2015) collects 1,571 cancer genes from 175 published studies. <a href="http://bioconductor.org/packages/DOSE">DOSE</a> supports analyzing gene list and determine whether they are enriched in genes known to be mutated in a given cancer type.</p>
<div class="sourceCode" id="cb102"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb102-1"><a href="#cb102-1" aria-hidden="true" tabindex="-1"></a>gene2 <span class="ot">&lt;-</span> <span class="fu">names</span>(geneList)[<span class="fu">abs</span>(geneList) <span class="sc">&lt;</span> <span class="dv">3</span>]</span>
<span id="cb102-2"><a href="#cb102-2" aria-hidden="true" tabindex="-1"></a>ncg <span class="ot">&lt;-</span> <span class="fu">enrichNCG</span>(gene2) </span>
<span id="cb102-3"><a href="#cb102-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(ncg)</span></code></pre></div>
<pre><code>##                                                                      ID
## pan-cancer_paediatric                             pan-cancer_paediatric
## triple_negative_breast_cancer             triple_negative_breast_cancer
## breast_cancer                                             breast_cancer
## soft_tissue_sarcoma                                 soft_tissue_sarcoma
## paediatric_high-grade_glioma               paediatric_high-grade_glioma
## pancreatic_cancer_(all_histologies) pancreatic_cancer_(all_histologies)
##                                                             Description
## pan-cancer_paediatric                             pan-cancer_paediatric
## triple_negative_breast_cancer             triple_negative_breast_cancer
## breast_cancer                                             breast_cancer
## soft_tissue_sarcoma                                 soft_tissue_sarcoma
## paediatric_high-grade_glioma               paediatric_high-grade_glioma
## pancreatic_cancer_(all_histologies) pancreatic_cancer_(all_histologies)
##                                     GeneRatio  BgRatio       pvalue
## pan-cancer_paediatric                161/1782 182/2372 2.748816e-06
## triple_negative_breast_cancer         71/1782  75/2372 6.564667e-06
## breast_cancer                        146/1782 171/2372 5.249102e-04
## soft_tissue_sarcoma                   26/1782  26/2372 5.633144e-04
## paediatric_high-grade_glioma          25/1782  25/2372 7.524752e-04
## pancreatic_cancer_(all_histologies)   39/1782  41/2372 7.825494e-04
##                                         p.adjust       qvalue
## pan-cancer_paediatric               0.0002226541 0.0001504615
## triple_negative_breast_cancer       0.0002658690 0.0001796646
## breast_cancer                       0.0105644165 0.0071390469
## soft_tissue_sarcoma                 0.0105644165 0.0071390469
## paediatric_high-grade_glioma        0.0105644165 0.0071390469
## pancreatic_cancer_(all_histologies) 0.0105644165 0.0071390469
##                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           geneID
## pan-cancer_paediatric               2146/55353/4609/1029/3575/22806/3418/3066/2120/30012/867/7468/7545/3195/865/64109/4613/613/11177/7490/238/10736/10054/5771/4893/140885/1785/9760/3417/6597/6476/9126/4869/10320/7307/80204/1050/8028/2312/6608/896/894/2196/4849/7023/5093/5079/5293/5727/55181/171017/51322/5781/3718/55294/60/673/8085/5897/4851/51176/1108/7764/10664/6098/2332/2201/6495/3845/7015/1441/2782/64919/4298/23512/8239/29102/6929/8021/6134/6598/4209/5290/22941/8726/207/3717/2033/10716/4928/6932/694/5156/10019/6886/9968/7080/2623/7874/1654/4149/3020/23219/55252/55729/10735/5728/4853/23451/51341/387/3206/6146/79718/2624/63035/3815/171023/23269/25/9839/23592/5896/7403/2260/54880/3716/9203/57178/6777/5789/4297/29072/90/546/120/25836/8289/4345/9611/5925/4763/1997/1499/7157/3399/5295/1387/4602/51564/1027/4005/2322/2078/678/6403/55709/1277/7494/64061/2625
## triple_negative_breast_cancer                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 6790/898/4609/1029/1789/4436/2120/867/7128/1788/1030/7490/2271/238/675/2047/4914/1316/5291/5293/5781/55294/8085/4851/4170/3845/355/1616/4854/5290/207/2033/4233/29110/2903/5979/5728/4853/2624/3815/10000/7403/2260/55193/472/5789/4297/2065/4286/8626/8405/8289/10499/55164/5925/4763/23405/1499/4921/7157/5295/1387/2078/324/7248/7048/22894/3480/2045/2066/2625
## breast_cancer                                                                                                                     4751/701/898/639/29028/4609/7399/1029/1520/4436/83990/11200/10849/2072/4771/865/999/1788/26191/1030/10801/83737/6262/1956/672/8590/675/4893/6597/8202/2778/208/51412/896/2132/677/4849/4221/65220/2854/55294/673/4193/8085/4851/57127/841/3265/7764/10664/9721/3845/3956/868/9175/6602/11174/8239/9860/6954/5290/1523/207/2033/2334/3782/8312/9514/5156/186/54897/71/79728/545/143/2064/4089/8471/8314/91/5289/1021/10735/5979/5728/4853/23451/9439/6738/387/55770/79718/4301/171023/23013/51135/80243/4292/149076/10983/6103/7403/54880/4916/55193/9203/1635/1495/2309/472/5076/2909/5789/4297/2065/29072/2263/546/8289/2874/9611/5925/6416/4763/7157/4088/23152/5295/6794/1387/4602/1027/5737/324/595/7188/4681/4214/7494/2099/3480/4485/2891/6926/3169/2625
## soft_tissue_sarcoma                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             999/6850/4914/4342/2185/55294/2041/4851/2044/4058/5290/4486/5297/5728/3815/2324/7403/546/5925/4763/1499/7157/5159/2045/3667/2066
## paediatric_high-grade_glioma                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             4609/1029/1019/4613/1030/1956/4914/896/894/673/8493/5290/4233/5156/1021/63035/54880/4916/90/546/4763/7157/5295/595/4915
## pancreatic_cancer_(all_histologies)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      1029/4771/8997/7159/2011/6597/7307/3710/6710/55294/7091/3845/23654/7046/3096/4089/91/8241/54549/92/23451/63035/7403/55193/23309/472/800/29072/23077/23499/8289/54894/6416/7157/4088/182/7048/2199/26960
##                                     Count
## pan-cancer_paediatric                 161
## triple_negative_breast_cancer          71
## breast_cancer                         146
## soft_tissue_sarcoma                    26
## paediatric_high-grade_glioma           25
## pancreatic_cancer_(all_histologies)    39</code></pre>
</div>
<div id="dose-dgn-ora" class="section level3" number="10.1.3">
<h3><span class="header-section-number">10.1.3</span> Over-representation analysis for the disease gene network</h3>
<p><a href="http://disgenet.org/">DisGeNET</a><span class="citation">(<a href="#ref-janet_disgenet" role="doc-biblioref">Janet et al. 2015</a>)</span> is an integrative and comprehensive resources of gene-disease associations from several public data sources and the literature. It contains gene-disease associations and snp-gene-disease associations.</p>
<p>The enrichment analysis of disease-gene associations is supported by the <code>enrichDGN</code> function and analysis of snp-gene-disease associations is supported by the <code>enrichDGNv</code> function.</p>
<div class="sourceCode" id="cb104"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb104-1"><a href="#cb104-1" aria-hidden="true" tabindex="-1"></a>dgn <span class="ot">&lt;-</span> <span class="fu">enrichDGN</span>(gene) </span>
<span id="cb104-2"><a href="#cb104-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(dgn)</span></code></pre></div>
<pre><code>##                ID                              Description GeneRatio   BgRatio
## C0010278 C0010278                         Craniosynostosis    43/497 488/21671
## C0853879 C0853879             Invasive carcinoma of breast    42/497 473/21671
## C4733092 C4733092 estrogen receptor-negative breast cancer    34/497 356/21671
## C3642347 C3642347              Basal-Like Breast Carcinoma    28/497 245/21671
## C3642345 C3642345               Luminal A Breast Carcinoma    22/497 153/21671
## C0036202 C0036202                              Sarcoidosis    36/497 413/21671
##                pvalue     p.adjust       qvalue
## C0010278 4.609534e-14 2.267976e-10 1.636811e-10
## C0853879 7.105190e-14 2.267976e-10 1.636811e-10
## C4733092 2.446675e-12 4.864593e-09 3.510804e-09
## C3642347 3.047991e-12 4.864593e-09 3.510804e-09
## C3642345 7.034749e-12 8.438458e-09 6.090082e-09
## C0036202 7.930882e-12 8.438458e-09 6.090082e-09
##                                                                                                                                                                                                                           geneID
## C0010278 4312/8318/6280/1062/6279/6278/3627/820/27299/6362/81620/2146/3002/29968/990/4318/4069/3576/6890/23594/26279/1493/6352/4998/2152/2697/185/4330/5327/4982/1300/3667/2200/9607/3572/563/7031/3479/6424/1846/3117/1308/2625
## C0853879        4312/7153/6278/9787/9582/51203/890/983/5080/2146/1111/9232/10855/4171/6664/4102/2173/4318/701/3576/1978/8836/53335/1894/7980/8792/8842/2151/185/2952/367/4982/4582/6926/3479/1602/23158/2066/3169/5304/2625/5241
## C4733092                                         2305/6278/79733/6241/81930/81620/2146/3620/29968/11004/8061/3576/1894/2491/7083/8792/214/5327/367/4982/3667/4582/27324/3479/1846/80129/4137/8839/3169/1408/5304/2625/5241/10551
## C3642347                                                                          2305/1062/4605/9833/7368/11065/10232/55765/5163/2146/2568/3620/6790/6664/29127/2173/4318/3576/3159/8792/6663/27324/3479/1846/18/3169/2625/5241
## C3642345                                                                                                         2305/9833/7153/55355/1111/3161/4318/3576/2001/6663/4288/2152/185/4128/4582/27324/80129/3169/5304/8614/2625/5241
## C0036202                               4312/6280/6279/10403/3627/6373/4283/27299/6362/3002/4321/6355/6364/29851/4318/5004/4069/3576/26227/6890/6352/4485/23541/185/7043/6863/2952/4982/25802/4582/2053/3479/3117/2167/80736/1524
##          Count
## C0010278    43
## C0853879    42
## C4733092    34
## C3642347    28
## C3642345    22
## C0036202    36</code></pre>
<div class="sourceCode" id="cb106"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb106-1"><a href="#cb106-1" aria-hidden="true" tabindex="-1"></a>snp <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;rs1401296&quot;</span>, <span class="st">&quot;rs9315050&quot;</span>, <span class="st">&quot;rs5498&quot;</span>, <span class="st">&quot;rs1524668&quot;</span>, <span class="st">&quot;rs147377392&quot;</span>,</span>
<span id="cb106-2"><a href="#cb106-2" aria-hidden="true" tabindex="-1"></a>         <span class="st">&quot;rs841&quot;</span>, <span class="st">&quot;rs909253&quot;</span>, <span class="st">&quot;rs7193343&quot;</span>, <span class="st">&quot;rs3918232&quot;</span>, <span class="st">&quot;rs3760396&quot;</span>,</span>
<span id="cb106-3"><a href="#cb106-3" aria-hidden="true" tabindex="-1"></a>         <span class="st">&quot;rs2231137&quot;</span>, <span class="st">&quot;rs10947803&quot;</span>, <span class="st">&quot;rs17222919&quot;</span>, <span class="st">&quot;rs386602276&quot;</span>, <span class="st">&quot;rs11053646&quot;</span>,</span>
<span id="cb106-4"><a href="#cb106-4" aria-hidden="true" tabindex="-1"></a>         <span class="st">&quot;rs1805192&quot;</span>, <span class="st">&quot;rs139564723&quot;</span>, <span class="st">&quot;rs2230806&quot;</span>, <span class="st">&quot;rs20417&quot;</span>, <span class="st">&quot;rs966221&quot;</span>)</span>
<span id="cb106-5"><a href="#cb106-5" aria-hidden="true" tabindex="-1"></a>dgnv <span class="ot">&lt;-</span> <span class="fu">enrichDGNv</span>(snp)</span>
<span id="cb106-6"><a href="#cb106-6" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(dgnv)</span></code></pre></div>
<pre><code>##                ID               Description GeneRatio    BgRatio       pvalue
## C0010054 C0010054 Coronary Arteriosclerosis      6/17 440/194515 1.568917e-12
## C0151744 C0151744       Myocardial Ischemia      4/17 103/194515 1.754840e-10
## C0031099 C0031099             Periodontitis      4/17 116/194515 2.839985e-10
## C0007785 C0007785       Cerebral Infarction      4/17 123/194515 3.599531e-10
## C0003850 C0003850          Arteriosclerosis      4/17 267/194515 8.145389e-09
## C0004153 C0004153           Atherosclerosis      4/17 281/194515 9.996713e-09
##              p.adjust qvalue
## C0010054 2.761295e-10     NA
## C0151744 1.544259e-08     NA
## C0031099 1.583793e-08     NA
## C0007785 1.583793e-08     NA
## C0003850 2.867177e-07     NA
## C0004153 2.932369e-07     NA
##                                                             geneID Count
## C0010054 rs5498/rs147377392/rs11053646/rs1805192/rs2230806/rs20417     6
## C0151744                   rs5498/rs147377392/rs11053646/rs1805192     4
## C0031099                         rs5498/rs909253/rs1805192/rs20417     4
## C0007785                rs147377392/rs11053646/rs1805192/rs2230806     4
## C0003850                        rs5498/rs1805192/rs2230806/rs20417     4
## C0004153                        rs5498/rs1805192/rs2230806/rs20417     4</code></pre>
</div>
</div>
<div id="dose-gsea" class="section level2" number="10.2">
<h2><span class="header-section-number">10.2</span> Disease gene set enrichment analysis</h2>
<div id="gsedo-fuction" class="section level3" number="10.2.1">
<h3><span class="header-section-number">10.2.1</span> <code>gseDO</code> fuction</h3>
<p>In the following example, in order to speedup the compilation of this document, only gene sets with size above 120 were tested and only 100 permutations were performed.</p>
<div class="sourceCode" id="cb108"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb108-1"><a href="#cb108-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DOSE)</span>
<span id="cb108-2"><a href="#cb108-2" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(geneList)</span>
<span id="cb108-3"><a href="#cb108-3" aria-hidden="true" tabindex="-1"></a>y <span class="ot">&lt;-</span> <span class="fu">gseDO</span>(geneList,</span>
<span id="cb108-4"><a href="#cb108-4" aria-hidden="true" tabindex="-1"></a>           <span class="at">minGSSize     =</span> <span class="dv">120</span>,</span>
<span id="cb108-5"><a href="#cb108-5" aria-hidden="true" tabindex="-1"></a>           <span class="at">pvalueCutoff  =</span> <span class="fl">0.2</span>,</span>
<span id="cb108-6"><a href="#cb108-6" aria-hidden="true" tabindex="-1"></a>           <span class="at">pAdjustMethod =</span> <span class="st">&quot;BH&quot;</span>,</span>
<span id="cb108-7"><a href="#cb108-7" aria-hidden="true" tabindex="-1"></a>           <span class="at">verbose       =</span> <span class="cn">FALSE</span>)</span>
<span id="cb108-8"><a href="#cb108-8" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(y, <span class="dv">3</span>)</span></code></pre></div>
<pre><code>##                        ID                          Description setSize
## DOID:0050338 DOID:0050338 primary bacterial infectious disease     214
## DOID:104         DOID:104         bacterial infectious disease     243
## DOID:399         DOID:399                         tuberculosis     140
##              enrichmentScore      NES       pvalue     p.adjust      qvalues
## DOID:0050338       0.4569856 2.042460 1.363997e-09 2.141476e-07 8.183983e-08
## DOID:104           0.4286201 1.954745 6.508681e-09 5.109314e-07 1.952604e-07
## DOID:399           0.4999969 2.105697 2.492603e-08 1.304462e-06 4.985205e-07
##              rank                   leading_edge
## DOID:0050338 1850 tags=35%, list=15%, signal=30%
## DOID:104     1808 tags=32%, list=14%, signal=28%
## DOID:399     1808 tags=36%, list=14%, signal=31%
##                                                                                                                                                                                                                                                                                                                                                                                            core_enrichment
## DOID:0050338                 4312/597/3627/6373/820/3620/6364/29851/4318/3576/26227/6890/952/1493/6352/3934/54210/3932/5551/3559/6347/6402/639/94025/3126/3001/6351/1236/5698/3948/919/3458/959/7296/79139/3804/4159/942/3329/9235/1234/7096/3383/4068/6367/5806/100/3659/4360/939/6891/4210/671/7422/929/26191/6504/27087/4282/7124/5027/5329/3569/4049/7097/56244/7852/1378/5133/5743/348/1118/3119/7415
## DOID:104     4312/597/3627/6373/820/3620/6364/29851/4318/3576/26227/6890/952/1493/6352/3934/54210/10663/3932/5551/3559/6772/6347/6402/639/94025/3126/3001/6351/1236/3654/5698/3948/919/3458/959/7296/79139/3804/4159/942/3329/9235/3689/1234/7096/3383/4068/6367/5806/100/3659/4360/939/6891/4210/671/7422/929/26191/6504/27087/4282/7124/5027/5329/3569/4049/7097/56244/7852/1378/5133/5743/348/1118/3119
## DOID:399                                                                                                                                             4312/597/3627/820/6364/29851/4318/3576/26227/1493/6352/3934/5551/3559/6347/94025/3001/6351/3948/919/3458/959/4159/942/9235/7096/3383/4068/6367/5806/100/3659/4360/939/671/929/26191/27087/4282/7124/5027/3569/7097/56244/7852/1378/5133/348/1118/3119</code></pre>
</div>
<div id="gsencg-fuction" class="section level3" number="10.2.2">
<h3><span class="header-section-number">10.2.2</span> <code>gseNCG</code> fuction</h3>
<div class="sourceCode" id="cb110"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb110-1"><a href="#cb110-1" aria-hidden="true" tabindex="-1"></a>ncg <span class="ot">&lt;-</span> <span class="fu">gseNCG</span>(geneList,</span>
<span id="cb110-2"><a href="#cb110-2" aria-hidden="true" tabindex="-1"></a>              <span class="at">minGSSize     =</span> <span class="dv">120</span>,</span>
<span id="cb110-3"><a href="#cb110-3" aria-hidden="true" tabindex="-1"></a>              <span class="at">pvalueCutoff  =</span> <span class="fl">0.2</span>,</span>
<span id="cb110-4"><a href="#cb110-4" aria-hidden="true" tabindex="-1"></a>              <span class="at">pAdjustMethod =</span> <span class="st">&quot;BH&quot;</span>,</span>
<span id="cb110-5"><a href="#cb110-5" aria-hidden="true" tabindex="-1"></a>              <span class="at">verbose       =</span> <span class="cn">FALSE</span>)</span>
<span id="cb110-6"><a href="#cb110-6" aria-hidden="true" tabindex="-1"></a>ncg <span class="ot">&lt;-</span> <span class="fu">setReadable</span>(ncg, <span class="st">&#39;org.Hs.eg.db&#39;</span>)</span>
<span id="cb110-7"><a href="#cb110-7" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(ncg, <span class="dv">3</span>) </span></code></pre></div>
<pre><code>## [1] ID              Description     setSize         enrichmentScore
## [5] NES             pvalue          p.adjust        qvalues        
## &lt;0 rows&gt; (or 0-length row.names)</code></pre>
</div>
<div id="gsedgn-fuction" class="section level3" number="10.2.3">
<h3><span class="header-section-number">10.2.3</span> <code>gseDGN</code> fuction</h3>
<div class="sourceCode" id="cb112"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb112-1"><a href="#cb112-1" aria-hidden="true" tabindex="-1"></a>dgn <span class="ot">&lt;-</span> <span class="fu">gseDGN</span>(geneList,</span>
<span id="cb112-2"><a href="#cb112-2" aria-hidden="true" tabindex="-1"></a>              <span class="at">minGSSize     =</span> <span class="dv">120</span>,</span>
<span id="cb112-3"><a href="#cb112-3" aria-hidden="true" tabindex="-1"></a>              <span class="at">pvalueCutoff  =</span> <span class="fl">0.2</span>,</span>
<span id="cb112-4"><a href="#cb112-4" aria-hidden="true" tabindex="-1"></a>              <span class="at">pAdjustMethod =</span> <span class="st">&quot;BH&quot;</span>,</span>
<span id="cb112-5"><a href="#cb112-5" aria-hidden="true" tabindex="-1"></a>              <span class="at">verbose       =</span> <span class="cn">FALSE</span>)</span>
<span id="cb112-6"><a href="#cb112-6" aria-hidden="true" tabindex="-1"></a>dgn <span class="ot">&lt;-</span> <span class="fu">setReadable</span>(dgn, <span class="st">&#39;org.Hs.eg.db&#39;</span>)</span>
<span id="cb112-7"><a href="#cb112-7" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(dgn, <span class="dv">3</span>) </span></code></pre></div>
<pre><code>##                ID                  Description setSize enrichmentScore
## C0024266 C0024266 Lymphocytic Choriomeningitis     120       0.5712593
## C0151779 C0151779           Cutaneous Melanoma     415       0.3963772
## C0205682 C0205682              Waist-Hip Ratio     401      -0.4425633
##                NES pvalue p.adjust  qvalues rank                   leading_edge
## C0024266  2.380210  1e-10 2.69e-08 1.75e-08 2579 tags=48%, list=21%, signal=38%
## C0151779  1.942344  1e-10 2.69e-08 1.75e-08 1864 tags=27%, list=15%, signal=24%
## C0205682 -1.935937  1e-10 2.69e-08 1.75e-08 2011 tags=28%, list=16%, signal=24%
##                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 core_enrichment
## C0024266                                                                                                                                                                                                                                                                                                                                                                                S100A9/CXCL10/CXCL9/EZH2/GZMB/ICOS/USP18/CXCL8/CTLA4/TREM1/PRF1/ADM/CA9/STAT1/CCL2/SELL/CDKN2A/IL7/IL7R/IFNG/CCR5/IL27RA/SH2D1A/FCER1G/CDK2AP2/CPVL/CD27/PSMB10/PTPN22/SLAMF1/KDM1A/TNF/IL6/FGL2/TLR2/RPAIN/NELFCD/PDCD1/WAS/HIF1A/ATP5F1B/FCGR2B/EGR2/STX11/CXCR3/TYROBP/YME1L1/SOSTDC1/PTPN2/TRAF1/HNF1A/IRF9/PML/NR0B2/IL2/TOX/AGFG1
## C0151779                                                          MMP1/CDCA8/S100A8/TOP2A/PBK/ISG20/NEK2/CCL18/PDK1/EZH2/AURKB/CHAF1B/KIF18B/MMP12/LAG3/DTL/PTTG1/SOX11/CDKN3/PRAME/PLK1/FABP7/MMP9/GBP1/CXCL8/CTLA4/SOX10/PROM1/MKI67/FANCA/MCM6/MCM4/PITX1/DBF4/IL2RA/TYMS/PLA2G2A/MYC/PCNA/NDRG1/CDKAL1/BLM/E2F1/OCA2/CD163/CDKN2A/NME1/TCF21/KMO/VCAM1/RUNX3/IFNG/MCM7/KYNU/STIP1/CCDC88C/GSTP1/CHAF1A/MSH2/CYP27B1/MAGEA1/CDKN2C/PRDM16/MX2/CDK4/GAPDH/MCM3/ICAM1/CCL22/CHEK2/CXCL5/ERCC4/SPP1/TEAD4/MTAP/HCP5/MRC1/TUBB3/CD27/S100B/CSPG4/PLP2/NBN/VEGFA/IFNW1/CDK16/CDH1/CCNF/CBLIF/SOX4/PARP1/SLC16A1/DNMT1/MAP2K1/ATG5/TNF/NXT1/CDKN2B/RBX1/IL6/TLR2/CXCR4/PDCD1/PTGS2/HIF1A/JMJD6/XRCC3/HLA-DQB1/PVR/CASP10/TGFA/EGFR
## C0205682 RETREG3/JMJD1C/SH2B1/BDNF/ARHGEF26/PDE5A/BPTF/SMAD3/TTC39A/ATP2B1/ARID4A/HOXC4/NID1/LAMA4/LRRC36/NUDT18/ANKRD28/HECTD4/COL11A1/MEIS1/INSR/CUL9/NRIP3/BCL2/CD34/EZH1/DYM/NDST1/COL15A1/VGLL3/CCND1/ZCCHC10/HOXC6/RAB26/QTRT1/MEIS2/ARID5B/AHNAK/FGF1/CAPRIN2/LAMB1/CPEB3/ELOVL4/CDADC1/PDE8B/ZNF268/NRP1/SYTL2/NR5A2/DGLUCY/SEMA3B/NID2/SIK3/PRR5L/FGF2/COL8A1/RAPGEF3/RBM6/CDH13/JCAD/NAV3/TRIM8/PPIEL/PTPRG/NBL1/CALCRL/PPL/LPL/BCKDHB/MAPKBP1/CNTLN/BBS4/P4HTM/FTO/PDZRN4/PDGFC/SGCD/NRXN3/AFF3/IGF1R/ABCC8/MPPED2/COL5A1/COL6A2/LOXL1/CYP21A2/LTBP2/TTC28/PATJ/PCSK5/WNT4/TTC12/NISCH/ASTN2/TCEA2/MN1/SETBP1/TAOK1/MAST4/ITGA7/ITGBL1/COL14A1/C1QTNF3/ZNF423/IQCH/MYH11/ADH1B/ABLIM3/MAPT/STC2/TFAP2B/CYBRD1/SCUBE2</code></pre>

</div>
</div>
</div>
<div id="meshes-enrichment" class="section level1" number="11">
<h1><span class="header-section-number">Chapter 11</span> MeSH enrichment analysis</h1>
<p>The <a href="http://bioconductor.org/packages/meshes">meshes</a> package <span class="citation">(<a href="#ref-yu_meshes_2018" role="doc-biblioref">Yu 2018</a>)</span> supports enrichment analysis (over-representation analysis and gene set
enrichment analysis) of gene list or whole expression profile using MeSH
annotation. Data source from <code>gendoo</code>, <code>gene2pubmed</code> and <code>RBBH</code> are all
supported. User can select interesting category to test. All 16
categories of MeSH are supported. The analysis supports &gt;70 species listed in <a href="https://bioconductor.org/packages/release/BiocViews.html#___MeSHDb">MeSHDb BiocView</a> (see also <a href="#meshes-semantic-similarity">Chapter 4</a> for more details).</p>
<div id="meshes-ora" class="section level2" number="11.1">
<h2><span class="header-section-number">11.1</span> MeSH over-representation analysis</h2>
<p>In this example, we use data source from <code>gendoo</code> and <code>C</code> (Diseases) category.</p>
<div class="sourceCode" id="cb114"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb114-1"><a href="#cb114-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(meshes)</span>
<span id="cb114-2"><a href="#cb114-2" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(geneList, <span class="at">package=</span><span class="st">&quot;DOSE&quot;</span>)</span>
<span id="cb114-3"><a href="#cb114-3" aria-hidden="true" tabindex="-1"></a>de <span class="ot">&lt;-</span> <span class="fu">names</span>(geneList)[<span class="dv">1</span><span class="sc">:</span><span class="dv">100</span>]</span>
<span id="cb114-4"><a href="#cb114-4" aria-hidden="true" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="fu">enrichMeSH</span>(de, <span class="at">MeSHDb =</span> <span class="st">&quot;MeSH.Hsa.eg.db&quot;</span>, <span class="at">database=</span><span class="st">&#39;gendoo&#39;</span>, <span class="at">category =</span> <span class="st">&#39;C&#39;</span>)</span>
<span id="cb114-5"><a href="#cb114-5" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(x)</span></code></pre></div>
<pre><code>##              ID              Description GeneRatio   BgRatio       pvalue
## D043171 D043171  Chromosomal Instability     16/96 198/16528 2.794765e-14
## D000782 D000782               Aneuploidy     17/96 320/16528 3.866830e-12
## D042822 D042822      Genomic Instability     16/96 312/16528 3.007419e-11
## D012595 D012595    Scleroderma, Systemic     11/96 279/16528 6.449334e-07
## D009303 D009303 Nasopharyngeal Neoplasms     11/96 314/16528 2.049315e-06
## D019698 D019698     Hepatitis C, Chronic     11/96 317/16528 2.246856e-06
##             p.adjust       qvalue
## D043171 2.459394e-11 1.815127e-11
## D000782 1.701405e-09 1.255702e-09
## D042822 8.821761e-09 6.510798e-09
## D012595 1.418854e-04 1.047168e-04
## D009303 3.295389e-04 2.432123e-04
## D019698 3.295389e-04 2.432123e-04
##                                                                                      geneID
## D043171    4312/991/2305/1062/4605/10403/7153/55355/4751/4085/81620/332/7272/9212/1111/6790
## D000782 4312/55143/991/1062/7153/4751/79019/55839/890/983/4085/332/7272/9212/8208/1111/6790
## D042822     55143/991/1062/4605/7153/1381/9787/4751/10635/890/4085/81620/332/9212/1111/6790
## D012595                              4312/6280/1062/4605/7153/3627/4283/6362/7850/3002/4321
## D009303                                4312/7153/3627/6241/983/4085/5918/332/3002/4321/6790
## D019698                               4312/3627/10563/6373/4283/983/6362/7850/332/3002/3620
##         Count
## D043171    16
## D000782    17
## D042822    16
## D012595    11
## D009303    11
## D019698    11</code></pre>
</div>
<div id="meshes-gsea" class="section level2" number="11.2">
<h2><span class="header-section-number">11.2</span> MeSH gene set enrichment analysis</h2>
<p>In the following example, we use data source from <code>gene2pubmed</code> and test category <code>G</code> (Phenomena and Processes).</p>
<div class="sourceCode" id="cb116"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb116-1"><a href="#cb116-1" aria-hidden="true" tabindex="-1"></a>y <span class="ot">&lt;-</span> <span class="fu">gseMeSH</span>(geneList, <span class="at">MeSHDb =</span> <span class="st">&quot;MeSH.Hsa.eg.db&quot;</span>, <span class="at">database =</span> <span class="st">&#39;gene2pubmed&#39;</span>, <span class="at">category =</span> <span class="st">&quot;G&quot;</span>)</span>
<span id="cb116-2"><a href="#cb116-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(y)</span></code></pre></div>
<pre><code>##              ID        Description setSize enrichmentScore       NES pvalue
## D000782 D000782         Aneuploidy     391       0.4154424  1.991027  1e-10
## D007156 D007156 Immunologic Memory     444       0.3983485  1.927594  1e-10
## D011123 D011123         Polyploidy     188       0.5235442  2.324306  1e-10
## D011418 D011418           Prophase      80       0.6553019  2.568616  1e-10
## D014690 D014690    Venous Pressure     140      -0.5849507 -2.309944  1e-10
## D017735 D017735      Virus Latency     415       0.3891947  1.884672  1e-10
##             p.adjust      qvalues rank                   leading_edge
## D000782 7.085714e-09 5.255639e-09 1826 tags=27%, list=15%, signal=24%
## D007156 7.085714e-09 5.255639e-09 2111 tags=31%, list=17%, signal=27%
## D011123 7.085714e-09 5.255639e-09 1905 tags=33%, list=15%, signal=28%
## D011418 7.085714e-09 5.255639e-09  437  tags=22%, list=3%, signal=22%
## D014690 7.085714e-09 5.255639e-09 1824 tags=51%, list=15%, signal=44%
## D017735 7.085714e-09 5.255639e-09 1862 tags=25%, list=15%, signal=22%
##                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        core_enrichment
## D000782                                                                                                                                         4312/991/2305/1062/10403/7153/23397/55165/11065/22974/10460/4751/79019/55839/890/983/4085/332/50852/7272/9212/8208/1111/9319/9055/3833/6790/891/24137/9232/4171/11004/5347/701/11130/79682/2491/9700/5888/7083/898/4288/10733/2237/6502/8538/9918/1058/699/4609/5111/10397/84823/55506/51053/1869/1029/1763/136/4904/55159/5613/5471/2950/54512/23310/3066/9496/79915/995/10606/51471/5885/5616/2762/79017/7428/22827/3195/7283/6636/5440/54908/1445/7422/5315/999/6573/138046/10376/142/54584/5129/9775/4613/9978/3619/11335/613/5133/81552/4363/3091/7517/6629/26047
## D007156 6280/8685/3627/10563/6373/4283/6362/3002/3620/891/10578/3695/6364/23532/29851/3576/7083/952/1493/6352/4288/10663/80380/5551/3559/6772/3507/6402/639/914/64581/57823/50802/4609/3561/6351/1237/6614/917/1236/51053/10859/1029/55423/3574/3575/919/915/22806/4067/6354/608/3458/959/1789/3903/11151/8525/3162/3702/925/11314/50615/942/1880/3329/969/867/3783/4064/25939/6361/1794/3689/5788/1234/916/7096/3383/4068/962/6367/30009/100/7374/3600/6513/923/940/939/5588/3965/9308/4683/6404/3738/930/6318/929/8115/26191/3601/27087/6850/7124/912/5027/10875/7097/7852/1378/2208/5133/3692/7454/51293/5817/951/3820/843/27033/5142/2213/7490/1536/5795/3135/60489/6376/3596/2000/7293/2833/655/356/1380/958/1672
## D011123                                                                                                                                                                                                                                                                                                                                                                                             2305/4605/9133/7153/79733/890/983/4085/81620/332/2146/7272/9212/51659/1111/6790/891/9232/990/5347/701/1978/9700/898/23594/6502/699/4609/5111/10397/26271/51053/1869/1029/4830/3925/5693/8565/2810/8607/58/1019/11200/2072/7283/7422/5315/865/86/63967/999/5604/3014/9978/4049/3619/7454/58477/1000/1956/28985/6119
## D011418                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      991/1062/259266/22974/4751/890/983/4085/3832/6790/891/5347/701/79682/10615/1058/699/26271
## D014690                                                                                                                                                                                                                                                                                                                                         4811/3910/3371/2662/1291/3490/7450/80781/1280/1306/8425/10085/4054/4256/3912/2817/1278/1511/2934/4060/1277/7078/5549/22795/10516/1293/1295/1012/8076/5118/2192/1281/1290/7058/2202/2199/11117/3339/1462/1289/1292/3908/4016/3909/4053/6678/633/1191/165/5654/10631/1805/2331/347/1359/3913/8404/2200/1634/4059/7177/563/7373/1307/9370/79148/1311/8483/54829/4239/4969
## D017735                                                                                                                                                                           6280/4605/6279/3627/6373/4283/332/2146/9212/6355/6364/9768/4599/54210/54962/5551/200315/6772/639/914/3015/699/4609/85236/5111/10397/917/55506/330/6590/1029/1051/3574/8970/8357/919/915/51191/22806/3458/2957/1789/5435/92815/3066/5499/8336/9088/6283/9235/25939/5788/1234/916/467/3383/5885/9188/54496/11200/11168/7374/940/8345/6873/1024/5440/5588/3965/8914/865/10148/6404/3276/3018/929/142/6504/6839/23028/10969/5467/1786/27087/4781/3014/7124/3569/6749/11335/83737/7852/51497/2208/5133/5743/5433/4363/3091/3119/8290/9681</code></pre>
<p>User can use visualization methods implemented in <a href="http://bioconductor.org/packages/enrichplot">enrichplot</a> to visualize these enrichment results. With <a href="#enrichplot">these visualization methods</a>, it‚Äôs much easier to interpret enriched results.</p>

</div>
</div>
<div id="universal-api" class="section level1" number="12">
<h1><span class="header-section-number">Chapter 12</span> Universal enrichment analysis</h1>
<p>The <a href="http://bioconductor.org/packages/clusterProfiler">clusterProfiler</a> package <span class="citation">(<a href="#ref-yu2012" role="doc-biblioref">Yu et al. 2012</a>)</span> supports both hypergeometric test and gene set enrichment analyses of many ontology/pathway, but it‚Äôs still not enough for users may want to analyze their data with unsupported organisms, slim version of GO, novel functional annotation (e.g.¬†GO via BlastGO or KEGG via KAAS), unsupported ontologies/pathways or customized annotations.</p>
<p>The <a href="http://bioconductor.org/packages/clusterProfiler">clusterProfiler</a> package provides <code>enricher()</code> function for hypergeometric test and <code>GSEA()</code> function for gene set enrichment analysis that are designed to accept user defined annotation. They accept two additional parameters <code>TERM2GENE</code> and <code>TERM2NAME</code>. As indicated in the parameter names, <code>TERM2GENE</code> is a data.frame with first column of term ID and second column of corresponding mapped gene and <code>TERM2NAME</code> is a <code>data.frame</code> with first column of term ID and second column of corresponding term name. <code>TERM2NAME</code> is optional.</p>
<div id="input-data" class="section level2" number="12.1">
<h2><span class="header-section-number">12.1</span> Input data</h2>
<p>For over representation analysis, all we need is a gene vector, that is a vector of gene IDs. These gene IDs can be obtained by differential expression analysis (<em>e.g.</em> with the <a href="http://bioconductor.org/packages/DESeq2">DESeq2</a> package).</p>
<p>For gene set enrichment analysis, we need a ranked list of genes. <a href="http://bioconductor.org/packages/DOSE">DOSE</a> provides an example dataset <code>geneList</code> which was derived from <code>R</code> package <a href="http://bioconductor.org/packages/breastCancerMAINZ">breastCancerMAINZ</a> that contained 200 samples, including 29 samples in grade I, 136 samples in grade II and 35 samples in grade III. We computed the ratios of geometric means of grade III samples versus geometric means of grade I samples. Logarithm of these ratios (base 2) were stored in <code>geneList</code> dataset. If you want to prepare your own <code>geneList</code>, please refer to the <a href="#genelist">FAQ</a>.</p>
<p>We can load the sample data into R via:</p>
<div class="sourceCode" id="cb118"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb118-1"><a href="#cb118-1" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(geneList, <span class="at">package=</span><span class="st">&quot;DOSE&quot;</span>)</span>
<span id="cb118-2"><a href="#cb118-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(geneList)</span></code></pre></div>
<pre><code>##     4312     8318    10874    55143    55388      991 
## 4.572613 4.514594 4.418218 4.144075 3.876258 3.677857</code></pre>
<p>Suppose we define fold change greater than 2 as DEGs:</p>
<div class="sourceCode" id="cb120"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb120-1"><a href="#cb120-1" aria-hidden="true" tabindex="-1"></a>gene <span class="ot">&lt;-</span> <span class="fu">names</span>(geneList)[<span class="fu">abs</span>(geneList) <span class="sc">&gt;</span> <span class="dv">2</span>]</span>
<span id="cb120-2"><a href="#cb120-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(gene)</span></code></pre></div>
<pre><code>## [1] &quot;4312&quot;  &quot;8318&quot;  &quot;10874&quot; &quot;55143&quot; &quot;55388&quot; &quot;991&quot;</code></pre>
</div>
<div id="cell-marker" class="section level2" number="12.2">
<h2><span class="header-section-number">12.2</span> Cell Marker</h2>
<!--     # tidyr::unite("cellMarker", tissueType, cancerType, cellName, sep=", ")  -->
<div class="sourceCode" id="cb122"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb122-1"><a href="#cb122-1" aria-hidden="true" tabindex="-1"></a>cell_marker_data <span class="ot">&lt;-</span> vroom<span class="sc">::</span><span class="fu">vroom</span>(<span class="st">&#39;http://bio-bigdata.hrbmu.edu.cn/CellMarker/download/Human_cell_markers.txt&#39;</span>)</span>
<span id="cb122-2"><a href="#cb122-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb122-3"><a href="#cb122-3" aria-hidden="true" tabindex="-1"></a><span class="do">## instead of `cellName`, users can use other features (e.g. `cancerType`)</span></span>
<span id="cb122-4"><a href="#cb122-4" aria-hidden="true" tabindex="-1"></a>cells <span class="ot">&lt;-</span> cell_marker_data <span class="sc">%&gt;%</span></span>
<span id="cb122-5"><a href="#cb122-5" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(cellName, geneID) <span class="sc">%&gt;%</span></span>
<span id="cb122-6"><a href="#cb122-6" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">geneID =</span> <span class="fu">strsplit</span>(geneID, <span class="st">&#39;, &#39;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb122-7"><a href="#cb122-7" aria-hidden="true" tabindex="-1"></a>    tidyr<span class="sc">::</span><span class="fu">unnest</span>()</span></code></pre></div>
<div id="cell-marker-ora" class="section level3" number="12.2.1">
<h3><span class="header-section-number">12.2.1</span> Cell Marker over-presentaton analysis</h3>
<div class="sourceCode" id="cb123"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb123-1"><a href="#cb123-1" aria-hidden="true" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="fu">enricher</span>(gene, <span class="at">TERM2GENE =</span> cells)</span>
<span id="cb123-2"><a href="#cb123-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(x)</span></code></pre></div>
<pre><code>##                                            ID            Description GeneRatio
## Neural progenitor cell Neural progenitor cell Neural progenitor cell    33/153
## MKI67+ progenitor cell MKI67+ progenitor cell MKI67+ progenitor cell    20/153
## DCLK1+ progenitor cell DCLK1+ progenitor cell DCLK1+ progenitor cell     7/153
##                          BgRatio       pvalue     p.adjust       qvalue
## Neural progenitor cell 176/11582 3.395466e-29 3.259648e-27 2.716373e-27
## MKI67+ progenitor cell 111/11582 1.426062e-17 6.845100e-16 5.704250e-16
## DCLK1+ progenitor cell  99/11582 3.298755e-04 1.055601e-02 8.796679e-03
##                                                                                                                                                                                              geneID
## Neural progenitor cell 991/9493/9833/9133/10403/7153/259266/6241/9787/11065/55355/55872/51203/83461/22974/10460/79019/55839/890/983/4085/5080/332/3832/9212/51659/9055/891/4174/9232/1602/2018/4137
## MKI67+ progenitor cell                                                                       991/9133/10403/7153/259266/6241/9787/11065/51203/22974/10460/890/983/332/9212/9055/891/9232/2167/51705
## DCLK1+ progenitor cell                                                                                                                                        1307/730/54829/10234/79901/57758/4969
##                        Count
## Neural progenitor cell    33
## MKI67+ progenitor cell    20
## DCLK1+ progenitor cell     7</code></pre>
</div>
<div id="cell-marker-gsea" class="section level3" number="12.2.2">
<h3><span class="header-section-number">12.2.2</span> Cell Marker gene set enrichment analysis</h3>
<div class="sourceCode" id="cb125"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb125-1"><a href="#cb125-1" aria-hidden="true" tabindex="-1"></a>y <span class="ot">&lt;-</span> <span class="fu">GSEA</span>(geneList, <span class="at">TERM2GENE =</span> cells)</span>
<span id="cb125-2"><a href="#cb125-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(y)</span></code></pre></div>
<pre><code>##                                                            ID
## Ciliated epithelial cell             Ciliated epithelial cell
## FOXN4+ cell                                       FOXN4+ cell
## Leydig precursor cell                   Leydig precursor cell
## Neural progenitor cell                 Neural progenitor cell
## Paneth cell                                       Paneth cell
## FGFR1HighNME5- epithelial cell FGFR1HighNME5- epithelial cell
##                                                   Description setSize
## Ciliated epithelial cell             Ciliated epithelial cell     217
## FOXN4+ cell                                       FOXN4+ cell      63
## Leydig precursor cell                   Leydig precursor cell     184
## Neural progenitor cell                 Neural progenitor cell     149
## Paneth cell                                       Paneth cell     261
## FGFR1HighNME5- epithelial cell FGFR1HighNME5- epithelial cell     131
##                                enrichmentScore       NES       pvalue
## Ciliated epithelial cell            -0.5012031 -2.067533 1.000000e-10
## FOXN4+ cell                          0.7136090  2.707612 1.000000e-10
## Leydig precursor cell               -0.5710583 -2.307920 1.000000e-10
## Neural progenitor cell               0.7160571  3.059446 1.000000e-10
## Paneth cell                          0.5381280  2.482673 1.000000e-10
## FGFR1HighNME5- epithelial cell      -0.5739472 -2.241150 2.159136e-10
##                                    p.adjust      qvalues rank
## Ciliated epithelial cell       3.300000e-09 1.936842e-09 1440
## FOXN4+ cell                    3.300000e-09 1.936842e-09 1419
## Leydig precursor cell          3.300000e-09 1.936842e-09 2609
## Neural progenitor cell         3.300000e-09 1.936842e-09  684
## Paneth cell                    3.300000e-09 1.936842e-09 2187
## FGFR1HighNME5- epithelial cell 5.937623e-09 3.484921e-09 1404
##                                                  leading_edge
## Ciliated epithelial cell       tags=29%, list=12%, signal=27%
## FOXN4+ cell                    tags=51%, list=11%, signal=45%
## Leydig precursor cell          tags=45%, list=21%, signal=36%
## Neural progenitor cell          tags=47%, list=5%, signal=45%
## Paneth cell                    tags=45%, list=18%, signal=38%
## FGFR1HighNME5- epithelial cell tags=39%, list=11%, signal=35%
##                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         core_enrichment
## Ciliated epithelial cell                                                                                                                                                                                                                                                          9666/1490/9284/7840/51281/126070/5021/55779/80823/79781/11001/81890/79645/79411/8777/4753/2701/2259/9854/8100/323/55152/51364/9940/23345/216/146542/22832/582/57728/4246/7220/55064/79819/54681/5002/79884/10610/26018/79659/6542/53832/7033/8938/23245/26960/55638/9576/114327/54585/1287/9358/55259/643314/64799/6038/80129/4857/80736/2066/79083/79846/11122/79838
## FOXN4+ cell                                                                                                                                                                                                                                                                                                                                                                                                                                                                   79733/4751/79019/983/81620/4069/8836/3070/10733/81831/3932/10926/56938/7298/6491/262/1029/1719/8894/5902/10950/5496/2023/124222/10263/51142/11168/3638/51491/8751/6637/86
## Leydig precursor cell                                                                                                                                                                                   7169/120/2949/22871/79822/7070/3624/10761/3671/80114/6326/25959/2013/9902/8654/4017/4811/26010/1291/92689/126393/7474/7168/27303/51454/1306/678/8425/7837/590/5919/23635/8829/1831/30008/1277/7078/4747/6591/56944/8406/1281/11167/1290/9037/2946/283298/1009/4313/6320/3075/6387/5159/1289/1292/116039/3908/5101/1909/83716/5376/633/6857/10631/54674/54587/30846/83468/3487/8404/2200/1634/1287/3479/5348/2006/7373/730/10234/4239/79901/4969
## Neural progenitor cell                                                                                                                                                                                                                                            991/9493/9833/9133/10403/7153/259266/6241/9787/11065/55355/55872/51203/83461/22974/10460/79019/55839/890/983/4085/5080/332/3832/9212/51659/9055/891/4174/9232/4171/1033/1164/11004/51512/11130/79682/26227/57405/10615/3149/23594/56992/9768/11339/3070/4288/1163/4175/4173/2237/29028/3838/54478/3015/699/81611/1063/5111/26271/51053/10535/1719/8543/9289/4176/1841/7112/58516/3148
## Paneth cell                    597/3627/8140/1844/4283/7850/2921/7941/4318/4069/3576/1075/10797/56833/713/1230/200315/712/51338/64581/57823/10437/8767/5341/91543/6351/3055/6614/10859/330/9332/313/51765/1051/22797/4067/80896/864/5226/9046/5336/8942/3162/4688/29887/5641/11314/79888/942/6916/1439/1116/10288/7127/4064/1794/483/963/3689/5788/3383/962/23250/3937/2207/3394/5552/54504/1514/79879/64092/409/7805/4360/3936/4689/5880/7128/10105/9308/51296/10148/10287/8477/6503/929/945/10261/4794/9775/8013/5912/7124/5329/10875/7097/7852/1378/5743/6398/7454/3119/951/6039/1240/5142/1536/3071/834/8676/7305/7133/3310/958/5724/728/11309/3059
## FGFR1HighNME5- epithelial cell                                                                                                                                                                                                                                                                                                                                            3931/6586/23284/7106/5919/316/8829/4060/10559/8642/2846/58494/79987/8835/8076/64221/5118/83700/79776/50509/1513/7058/4313/55314/91851/10536/2674/3305/5159/443/3908/5101/2690/55733/2550/6857/10631/81035/4330/83468/576/2200/5157/7049/51380/1307/196740/2532/1308/8483/4239</code></pre>
</div>
</div>
<div id="msigdb-analysis" class="section level2" number="12.3">
<h2><span class="header-section-number">12.3</span> MSigDb analysis</h2>
<p><a href="http://software.broadinstitute.org/gsea/msigdb">Molecular Signatures Database</a> is a collection of annotated gene sets. It contains 8 major collections:</p>
<ul>
<li>H: hallmark gene sets</li>
<li>C1: positional gene sets</li>
<li>C2: curated gene sets</li>
<li>C3: motif gene sets</li>
<li>C4: computational gene sets</li>
<li>C5: GO gene sets</li>
<li>C6: oncogenic signatures</li>
<li>C7: immunologic signatures</li>
</ul>
<p>Users can download <a href="www.broadinstitute.org/cancer/software/gsea/wiki/index.php/Data_formats#GMT:_Gene_Matrix_Transposed_file_format_.28.2A.gmt.29">GMT files</a> from <a href="http://software.broadinstitute.org/gsea/msigdb">Broad Institute</a> and use the <code>read.gmt()</code> function to parse the file to be used in <code>enricher()</code> and <code>GSEA()</code>.</p>
<p>There is an R package, <a href="https://cran.r-project.org/package=msigdbr">msigdbr</a>, that already packed the MSigDB gene sets in tidy data format that can be used directly with <a href="http://bioconductor.org/packages/clusterProfiler">clusterProfiler</a> <span class="citation">(<a href="#ref-yu2012" role="doc-biblioref">Yu et al. 2012</a>)</span>.</p>
<p>It supports several specices:</p>
<div class="sourceCode" id="cb127"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb127-1"><a href="#cb127-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(msigdbr)</span>
<span id="cb127-2"><a href="#cb127-2" aria-hidden="true" tabindex="-1"></a><span class="fu">msigdbr_show_species</span>()</span></code></pre></div>
<pre><code>##  [1] &quot;Bos taurus&quot;               &quot;Caenorhabditis elegans&quot;  
##  [3] &quot;Canis lupus familiaris&quot;   &quot;Danio rerio&quot;             
##  [5] &quot;Drosophila melanogaster&quot;  &quot;Gallus gallus&quot;           
##  [7] &quot;Homo sapiens&quot;             &quot;Mus musculus&quot;            
##  [9] &quot;Rattus norvegicus&quot;        &quot;Saccharomyces cerevisiae&quot;
## [11] &quot;Sus scrofa&quot;</code></pre>
<p>We can retrieve all human gene sets:</p>
<div class="sourceCode" id="cb129"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb129-1"><a href="#cb129-1" aria-hidden="true" tabindex="-1"></a>m_df <span class="ot">&lt;-</span> <span class="fu">msigdbr</span>(<span class="at">species =</span> <span class="st">&quot;Homo sapiens&quot;</span>)</span>
<span id="cb129-2"><a href="#cb129-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(m_df, <span class="dv">2</span>) <span class="sc">%&gt;%</span> as.data.frame</span></code></pre></div>
<pre><code>##   gs_cat      gs_subcat        gs_name entrez_gene gene_symbol
## 1     C3 MIR:MIR_Legacy AAACCAC_MIR140       10257       ABCC4
## 2     C3 MIR:MIR_Legacy AAACCAC_MIR140       23172    ABRAXAS2
##   human_entrez_gene human_gene_symbol  gs_id gs_pmid gs_geoid gs_exact_source
## 1             10257             ABCC4 M12609                                 
## 2             23172          ABRAXAS2 M12609                                 
##   gs_url
## 1       
## 2       
##                                                                                                                                                                                           gs_description
## 1 Genes having at least one occurence of the motif AAACCAC in their 3&#39; untranslated region. The motif represents putative target (that is, seed match) of human mature miRNA hsa-miR-140 (v7.1 miRBase).
## 2 Genes having at least one occurence of the motif AAACCAC in their 3&#39; untranslated region. The motif represents putative target (that is, seed match) of human mature miRNA hsa-miR-140 (v7.1 miRBase).
##   species_name species_common_name ortholog_sources num_ortholog_sources
## 1 Homo sapiens               human             &lt;NA&gt;                   NA
## 2 Homo sapiens               human             &lt;NA&gt;                   NA</code></pre>
<p>Or specific collection. Here we use C6, oncogenic gene sets as an example:</p>
<div class="sourceCode" id="cb131"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb131-1"><a href="#cb131-1" aria-hidden="true" tabindex="-1"></a>m_t2g <span class="ot">&lt;-</span> <span class="fu">msigdbr</span>(<span class="at">species =</span> <span class="st">&quot;Homo sapiens&quot;</span>, <span class="at">category =</span> <span class="st">&quot;C6&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb131-2"><a href="#cb131-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(gs_name, entrez_gene)</span>
<span id="cb131-3"><a href="#cb131-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(m_t2g)</span></code></pre></div>
<pre><code>## # A tibble: 6 x 2
##   gs_name              entrez_gene
##   &lt;chr&gt;                      &lt;int&gt;
## 1 AKT_UP_MTOR_DN.V1_DN       25864
## 2 AKT_UP_MTOR_DN.V1_DN          95
## 3 AKT_UP_MTOR_DN.V1_DN      137872
## 4 AKT_UP_MTOR_DN.V1_DN         134
## 5 AKT_UP_MTOR_DN.V1_DN       55326
## 6 AKT_UP_MTOR_DN.V1_DN         271</code></pre>
<div id="msigdb-ora" class="section level3" number="12.3.1">
<h3><span class="header-section-number">12.3.1</span> MSigDb over-presentaton analysis</h3>
<div class="sourceCode" id="cb133"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb133-1"><a href="#cb133-1" aria-hidden="true" tabindex="-1"></a>em <span class="ot">&lt;-</span> <span class="fu">enricher</span>(gene, <span class="at">TERM2GENE=</span>m_t2g)</span>
<span id="cb133-2"><a href="#cb133-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(em)</span></code></pre></div>
<pre><code>##                                            ID            Description GeneRatio
## RPS14_DN.V1_DN                 RPS14_DN.V1_DN         RPS14_DN.V1_DN    22/183
## GCNP_SHH_UP_LATE.V1_UP GCNP_SHH_UP_LATE.V1_UP GCNP_SHH_UP_LATE.V1_UP    16/183
## PRC2_EZH2_UP.V1_DN         PRC2_EZH2_UP.V1_DN     PRC2_EZH2_UP.V1_DN    15/183
## VEGF_A_UP.V1_DN               VEGF_A_UP.V1_DN        VEGF_A_UP.V1_DN    15/183
## RB_P107_DN.V1_UP             RB_P107_DN.V1_UP       RB_P107_DN.V1_UP    10/183
## E2F1_UP.V1_UP                   E2F1_UP.V1_UP          E2F1_UP.V1_UP    12/183
##                          BgRatio       pvalue     p.adjust       qvalue
## RPS14_DN.V1_DN         186/10913 4.733372e-13 7.999398e-11 6.527071e-11
## GCNP_SHH_UP_LATE.V1_UP 181/10913 5.829947e-08 4.926305e-06 4.019595e-06
## PRC2_EZH2_UP.V1_DN     192/10913 7.652392e-07 3.454763e-05 2.818897e-05
## VEGF_A_UP.V1_DN        193/10913 8.176953e-07 3.454763e-05 2.818897e-05
## RB_P107_DN.V1_UP       130/10913 6.438296e-05 2.027918e-03 1.654670e-03
## E2F1_UP.V1_UP          188/10913 7.535630e-05 2.027918e-03 1.654670e-03
##                                                                                                                                        geneID
## RPS14_DN.V1_DN         10874/55388/991/9493/1062/4605/9133/23397/79733/9787/55872/83461/54821/51659/9319/9055/10112/4174/5105/2532/7021/79901
## GCNP_SHH_UP_LATE.V1_UP                                      55388/7153/79733/6241/9787/51203/983/9212/1111/9319/9055/3833/6790/4174/3169/1580
## PRC2_EZH2_UP.V1_DN                                          8318/55388/4605/23397/9787/55355/10460/6362/81620/2146/7272/9212/11182/3887/24137
## VEGF_A_UP.V1_DN                                                   8318/9493/1062/9133/10403/6241/9787/4085/332/3832/7272/891/23362/2167/10234
## RB_P107_DN.V1_UP                                                                         8318/23397/79733/6241/4085/8208/9055/24137/4174/1307
## E2F1_UP.V1_UP                                                                  55388/7153/23397/79733/9787/2146/2842/9212/8208/1111/9055/3833
##                        Count
## RPS14_DN.V1_DN            22
## GCNP_SHH_UP_LATE.V1_UP    16
## PRC2_EZH2_UP.V1_DN        15
## VEGF_A_UP.V1_DN           15
## RB_P107_DN.V1_UP          10
## E2F1_UP.V1_UP             12</code></pre>
</div>
<div id="msigdb-gsea" class="section level3" number="12.3.2">
<h3><span class="header-section-number">12.3.2</span> MSigDb gene set enrichment analysis</h3>
<p>In over-presentaton analysis, we use oncogenic gene sets (i.e.¬†C6) to test whether the DE genes are involved in the process that leads to cancer. In this example, we will use the C3 category to test whether genes are up/down-regulated by sharing specific motif using GSEA approach.</p>
<div class="sourceCode" id="cb135"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb135-1"><a href="#cb135-1" aria-hidden="true" tabindex="-1"></a>C3_t2g <span class="ot">&lt;-</span> <span class="fu">msigdbr</span>(<span class="at">species =</span> <span class="st">&quot;Homo sapiens&quot;</span>, <span class="at">category =</span> <span class="st">&quot;C3&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb135-2"><a href="#cb135-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(gs_name, entrez_gene)</span>
<span id="cb135-3"><a href="#cb135-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(C3_t2g)</span></code></pre></div>
<pre><code>## # A tibble: 6 x 2
##   gs_name        entrez_gene
##   &lt;chr&gt;                &lt;int&gt;
## 1 AAACCAC_MIR140       10257
## 2 AAACCAC_MIR140       23172
## 3 AAACCAC_MIR140          81
## 4 AAACCAC_MIR140          90
## 5 AAACCAC_MIR140        8754
## 6 AAACCAC_MIR140       11096</code></pre>
<div class="sourceCode" id="cb137"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb137-1"><a href="#cb137-1" aria-hidden="true" tabindex="-1"></a>em2 <span class="ot">&lt;-</span> <span class="fu">GSEA</span>(geneList, <span class="at">TERM2GENE =</span> C3_t2g)</span>
<span id="cb137-2"><a href="#cb137-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(em2)</span></code></pre></div>
<pre><code>##                                        ID          Description setSize
## E2F1_Q3                           E2F1_Q3              E2F1_Q3     187
## HSD17B8_TARGET_GENES HSD17B8_TARGET_GENES HSD17B8_TARGET_GENES     401
## SGCGSSAAA_E2F1DP2_01 SGCGSSAAA_E2F1DP2_01 SGCGSSAAA_E2F1DP2_01     129
## E2F1_Q6                           E2F1_Q6              E2F1_Q6     184
## E2F_Q4                             E2F_Q4               E2F_Q4     192
## E2F_Q6                             E2F_Q6               E2F_Q6     189
##                      enrichmentScore      NES       pvalue     p.adjust
## E2F1_Q3                    0.4983913 2.217898 1.000000e-10 1.606000e-07
## HSD17B8_TARGET_GENES       0.6368099 3.092388 1.000000e-10 1.606000e-07
## SGCGSSAAA_E2F1DP2_01       0.5408182 2.259504 1.666457e-10 1.784220e-07
## E2F1_Q6                    0.4937839 2.187073 2.324233e-10 1.866359e-07
## E2F_Q4                     0.4713651 2.099329 8.030102e-10 5.158537e-07
## E2F_Q6                     0.4700783 2.095930 1.044638e-09 5.592297e-07
##                           qvalues rank                   leading_edge
## E2F1_Q3              1.262632e-07 2045 tags=36%, list=16%, signal=30%
## HSD17B8_TARGET_GENES 1.262632e-07 1042  tags=36%, list=8%, signal=34%
## SGCGSSAAA_E2F1DP2_01 1.402747e-07 1797 tags=38%, list=14%, signal=33%
## E2F1_Q6              1.467325e-07 1819 tags=34%, list=15%, signal=30%
## E2F_Q4               4.055624e-07 1663 tags=29%, list=13%, signal=26%
## E2F_Q6               4.396644e-07 1663 tags=29%, list=13%, signal=26%
##                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             core_enrichment
## E2F1_Q3                                                                                                                                                                                                                                                                                                                                                                                                                                                 79733/6241/983/5080/2146/8715/4171/993/990/56992/9768/4998/10733/7037/4175/54962/54954/29028/3015/2643/85236/5111/64785/26271/51053/1869/3925/114/5427/4176/7112/4436/1871/79915/9088/6632/284403/7290/2597/7398/4172/5424/124222/1633/10849/9824/4678/6646/29902/63967/56886/6059/23234/1786/204/51087/23636/2048/83463/10393/79173/688/3007/79677/9462/56946/9221
## HSD17B8_TARGET_GENES 55143/55388/991/2305/9493/1062/4605/9833/9133/10403/7153/23397/79733/259266/6241/55165/9787/11065/55355/9582/220134/55872/83461/10460/4751/10635/55839/890/9415/983/54821/4085/9837/5080/332/7272/64151/2842/9212/51659/9319/9055/3833/146909/10112/51514/4174/9232/1033/9928/3161/11004/993/4603/79801/990/5347/55215/701/55723/51512/55635/9156/11130/10024/57405/10615/1894/2491/8438/9700/5888/7083/898/3149/56992/9768/4288/10733/1163/4175/5307/2237/29899/55010/59336/56942/5984/29028/3838/1058/54478/3015/699/6491/81611/1063/27346/64785/9401/26271/51026/641/1869/10535/1029/28998/1763/8970/54892/55159/864/93594/1789/4176/7111/3148/116832/9585/9735/55732/81624/23310/1871/1031/79915/995/10051/30012/1104/80178/78995/7468/284403/5230/8549/7884/5558/4172/5424/79866/3182/56902/83990
## SGCGSSAAA_E2F1DP2_01                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     79733/6241/4171/993/990/3159/9768/4175/4173/29028/3015/4609/85236/5111/64785/26271/51053/1869/5427/4176/5902/4436/1871/79915/9088/6632/284403/7290/4172/5424/124222/1633/7374/2189/4678/2289/50/63967/56886/23234/6839/1786/204/51087/23636/1174/83463/10492/79173
## E2F1_Q6                                                                                                                                                                                                                                                                                                                                                                                                                                                                          79733/6241/983/5080/81620/2146/4171/993/990/3159/9768/4998/4175/4173/29028/3015/4609/85236/5111/64785/26271/51053/1869/3925/5427/4176/7112/5902/4436/1871/79915/9088/6632/284403/7290/2597/4172/5424/124222/1633/10849/7374/9824/2189/4678/2289/6646/29902/50/63967/56886/23234/6839/1786/204/51087/23636/11335/1174/83463/10492/79173/688
## E2F_Q4                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 8318/79733/6241/983/5080/81620/2146/4171/993/990/5888/9768/4175/4173/29028/3015/85236/5111/64785/26271/51053/1869/3925/5427/23649/4176/7112/5902/4436/1871/79915/9088/6632/284403/7290/2597/4172/5424/124222/1633/10849/7473/7374/9824/7353/4678/6646/50/63967/56886/23234/1786/204/6427/51087/23636
## E2F_Q6                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      8318/79733/6241/983/5080/81620/2146/4171/993/990/5888/9768/4175/4173/29028/3015/85236/5111/64785/26271/51053/1869/3925/5427/23649/4176/7112/5902/4436/1871/79915/9088/6632/284403/7290/2597/4172/5424/124222/1633/10849/7374/9824/7353/4678/6646/50/63967/56886/23234/1786/204/6427/51087/23636</code></pre>

</div>
</div>
</div>
<div id="chipseeker-enrichment" class="section level1" number="13">
<h1><span class="header-section-number">Chapter 13</span> Functional enrichment analysis of genomic coordinations</h1>
<p>Functional analysis using NGS data (eg, RNA-Seq and ChIP-Seq) can be performed by linking coding and non-coding regions to coding genes via <a href="http://bioconductor.org/packages/ChIPseeker">ChIPseeker</a> <span class="citation">(<a href="#ref-yu_chipseeker_2015" role="doc-biblioref">Yu, Wang, and He 2015</a>)</span> package, which can annotates genomic regions to their nearest genes, host genes, and flanking genes respectivly. In addtion, it provides a function, <code>seq2gene</code>, that simultaneously considering host genes, promoter region and flanking gene from intergenic region that may under control via cis-regulation. This function maps genomic regions to genes in a many-to-many manner and facilitate functional analysis. For more details, please refer to <a href="http://bioconductor.org/packages/ChIPseeker">ChIPseeker</a> <span class="citation">(<a href="#ref-yu_chipseeker_2015" role="doc-biblioref">Yu, Wang, and He 2015</a>)</span>.</p>

</div>
<div id="clusterprofiler-comparecluster" class="section level1" number="14">
<h1><span class="header-section-number">Chapter 14</span> Biological theme comparison</h1>
<p>The <a href="http://bioconductor.org/packages/clusterProfiler">clusterProfiler</a> package was developed for biological theme comparison <span class="citation">(<a href="#ref-yu2012" role="doc-biblioref">Yu et al. 2012</a>; <a href="#ref-wu_clusterprofiler_2021" role="doc-biblioref">Wu et al. 2021</a>)</span>, and it provides a function, <code>compareCluster</code>, to automatically calculate enriched functional profiles of each gene clusters and aggregate the results into a single object. Comparing functional profiles can reveal functional consensus and differences among different experiments and helps in identifying differential functional modules in omics datasets.</p>
<div id="comparing-multiple-gene-lists" class="section level2" number="14.1">
<h2><span class="header-section-number">14.1</span> Comparing multiple gene lists</h2>
<p>The <code>compareCluster()</code> function applies selected function (via the <code>fun</code> parameter) to perform enrichment analysis for each gene list.</p>
<div class="sourceCode" id="cb139"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb139-1"><a href="#cb139-1" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(gcSample)</span>
<span id="cb139-2"><a href="#cb139-2" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(gcSample) </span></code></pre></div>
<pre><code>## List of 8
##  $ X1: chr [1:216] &quot;4597&quot; &quot;7111&quot; &quot;5266&quot; &quot;2175&quot; ...
##  $ X2: chr [1:805] &quot;23450&quot; &quot;5160&quot; &quot;7126&quot; &quot;26118&quot; ...
##  $ X3: chr [1:392] &quot;894&quot; &quot;7057&quot; &quot;22906&quot; &quot;3339&quot; ...
##  $ X4: chr [1:838] &quot;5573&quot; &quot;7453&quot; &quot;5245&quot; &quot;23450&quot; ...
##  $ X5: chr [1:929] &quot;5982&quot; &quot;7318&quot; &quot;6352&quot; &quot;2101&quot; ...
##  $ X6: chr [1:585] &quot;5337&quot; &quot;9295&quot; &quot;4035&quot; &quot;811&quot; ...
##  $ X7: chr [1:582] &quot;2621&quot; &quot;2665&quot; &quot;5690&quot; &quot;3608&quot; ...
##  $ X8: chr [1:237] &quot;2665&quot; &quot;4735&quot; &quot;1327&quot; &quot;3192&quot; ...</code></pre>
<p>Users can use a named list of gene IDs as the input that passed to the <code>geneCluster</code> parameter.</p>
<div class="sourceCode" id="cb141"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb141-1"><a href="#cb141-1" aria-hidden="true" tabindex="-1"></a>ck <span class="ot">&lt;-</span> <span class="fu">compareCluster</span>(<span class="at">geneCluster =</span> gcSample, <span class="at">fun =</span> enrichKEGG)</span>
<span id="cb141-2"><a href="#cb141-2" aria-hidden="true" tabindex="-1"></a>ck <span class="ot">&lt;-</span> <span class="fu">setReadable</span>(ck, <span class="at">OrgDb =</span> org.Hs.eg.db, <span class="at">keyType=</span><span class="st">&quot;ENTREZID&quot;</span>)</span>
<span id="cb141-3"><a href="#cb141-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(ck) </span></code></pre></div>
<pre><code>##   Cluster       ID                             Description GeneRatio  BgRatio
## 1      X2 hsa04110                              Cell cycle    18/380 124/8087
## 2      X2 hsa05169            Epstein-Barr virus infection    23/380 202/8087
## 3      X2 hsa05340                Primary immunodeficiency     8/380  38/8087
## 4      X2 hsa05166 Human T-cell leukemia virus 1 infection    22/380 219/8087
## 5      X3 hsa04512                ECM-receptor interaction     9/186  88/8087
## 6      X3 hsa04060  Cytokine-cytokine receptor interaction    17/186 295/8087
##         pvalue    p.adjust      qvalue
## 1 1.747566e-05 0.005242697 0.004874788
## 2 6.970114e-05 0.010455171 0.009721475
## 3 3.117722e-04 0.031177224 0.028989348
## 4 5.993266e-04 0.044949493 0.041795143
## 5 1.772661e-04 0.042517822 0.039650400
## 6 4.326582e-04 0.042517822 0.039650400
##                                                                                                                          geneID
## 1                          CDC20/E2F1/CCNA2/E2F3/BUB1B/CDC6/DBF4/PKMYT1/CDC7/ESPL1/CCNE2/CDKN2A/SFN/BUB1/CHEK2/ORC6/CDC14B/MCM4
## 2 LYN/ICAM1/TNFAIP3/E2F1/CCNA2/E2F3/BAK1/RUNX3/BID/IKBKE/TAP2/FAS/CCNE2/RELB/CD3E/ENTPD3/SYK/TRAF3/IKBKB/CD247/NEDD4/CD40/STAT1
## 3                                                                                     ADA/TAP2/LCK/CD79A/CD3E/CD8A/CD40/DCLRE1C
## 4 ICAM1/CDC20/E2F1/CCNA2/E2F3/ADCY7/BUB1B/ESPL1/LCK/CCNE2/RELB/CD3E/MSX1/IL15/CDKN2A/IKBKB/RANBP3/MSX2/CHEK2/NFATC4/MAP3K1/CD40
## 5                                                                         THBS1/HSPG2/COL9A3/ITGB7/CHAD/ITGA7/LAMA4/ITGB8/ITGB5
## 6                CXCL1/TNFRSF11B/LIFR/XCL1/GDF5/TNFRSF17/TNFRSF11A/IL5RA/EPOR/CSF2RA/TNFRSF25/BMP7/BMP4/GDF11/IL21R/IL23A/TGFB2
##   Count
## 1    18
## 2    23
## 3     8
## 4    22
## 5     9
## 6    17</code></pre>
</div>
<div id="formula-interface-of-comparecluster" class="section level2" number="14.2">
<h2><span class="header-section-number">14.2</span> Formula interface of compareCluster</h2>
<p>As an alternaitve to using named list, the <code>compareCluster()</code> function also supports passing a formula to describe more complicated experimental designs (<em>e.g.</em>, <span class="math inline">\(Gene \sim time + treatment\)</span>).</p>
<div class="sourceCode" id="cb143"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb143-1"><a href="#cb143-1" aria-hidden="true" tabindex="-1"></a>mydf <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Entrez=</span><span class="fu">names</span>(geneList), <span class="at">FC=</span>geneList)</span>
<span id="cb143-2"><a href="#cb143-2" aria-hidden="true" tabindex="-1"></a>mydf <span class="ot">&lt;-</span> mydf[<span class="fu">abs</span>(mydf<span class="sc">$</span>FC) <span class="sc">&gt;</span> <span class="dv">1</span>,]</span>
<span id="cb143-3"><a href="#cb143-3" aria-hidden="true" tabindex="-1"></a>mydf<span class="sc">$</span>group <span class="ot">&lt;-</span> <span class="st">&quot;upregulated&quot;</span></span>
<span id="cb143-4"><a href="#cb143-4" aria-hidden="true" tabindex="-1"></a>mydf<span class="sc">$</span>group[mydf<span class="sc">$</span>FC <span class="sc">&lt;</span> <span class="dv">0</span>] <span class="ot">&lt;-</span> <span class="st">&quot;downregulated&quot;</span></span>
<span id="cb143-5"><a href="#cb143-5" aria-hidden="true" tabindex="-1"></a>mydf<span class="sc">$</span>othergroup <span class="ot">&lt;-</span> <span class="st">&quot;A&quot;</span></span>
<span id="cb143-6"><a href="#cb143-6" aria-hidden="true" tabindex="-1"></a>mydf<span class="sc">$</span>othergroup[<span class="fu">abs</span>(mydf<span class="sc">$</span>FC) <span class="sc">&gt;</span> <span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="st">&quot;B&quot;</span></span>
<span id="cb143-7"><a href="#cb143-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb143-8"><a href="#cb143-8" aria-hidden="true" tabindex="-1"></a>formula_res <span class="ot">&lt;-</span> <span class="fu">compareCluster</span>(Entrez<span class="sc">~</span>group<span class="sc">+</span>othergroup, <span class="at">data=</span>mydf, <span class="at">fun=</span><span class="st">&quot;enrichKEGG&quot;</span>)</span>
<span id="cb143-9"><a href="#cb143-9" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb143-10"><a href="#cb143-10" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(formula_res)</span></code></pre></div>
<pre><code>##           Cluster         group othergroup       ID
## 1 downregulated.A downregulated          A hsa04974
## 2 downregulated.A downregulated          A hsa04510
## 3 downregulated.A downregulated          A hsa04151
## 4 downregulated.A downregulated          A hsa04512
## 5 downregulated.B downregulated          B hsa03320
## 6   upregulated.A   upregulated          A hsa04110
##                        Description GeneRatio  BgRatio       pvalue     p.adjust
## 1 Protein digestion and absorption    16/316 103/8087 2.072453e-06 5.699245e-04
## 2                   Focal adhesion    20/316 201/8087 1.083631e-04 1.489993e-02
## 3       PI3K-Akt signaling pathway    28/316 354/8087 2.804784e-04 2.571052e-02
## 4         ECM-receptor interaction    11/316  88/8087 5.920352e-04 4.070242e-02
## 5           PPAR signaling pathway      5/43  76/8087 4.675122e-05 7.199689e-03
## 6                       Cell cycle    20/219 124/8087 9.477515e-11 2.331469e-08
##         qvalue
## 1 5.388377e-04
## 2 1.408721e-02
## 3 2.430813e-02
## 4 3.848229e-02
## 5 6.791231e-03
## 6 2.085053e-08
##                                                                                                                                             geneID
## 1                                                                 1281/50509/1290/477/1294/1360/1289/1292/1296/23428/1359/1300/1287/6505/2006/7373
## 2                                         55742/2317/7058/25759/56034/3693/3480/5159/857/1292/3908/3909/63923/3913/1287/3679/7060/3479/10451/80310
## 3 55970/5618/7058/10161/56034/3693/4254/3480/4908/5159/1292/3908/2690/3909/8817/9223/4915/3551/2791/63923/3913/9863/3667/1287/3679/7060/3479/80310
## 4                                                                                          7058/3693/3339/1292/3908/3909/63923/3913/1287/3679/7060
## 5                                                                                                                         9370/5105/2167/3158/5346
## 6                                                  4171/993/990/5347/701/9700/898/23594/4998/9134/4175/4173/10926/6502/994/699/4609/5111/1869/1029
##   Count
## 1    16
## 2    20
## 3    28
## 4    11
## 5     5
## 6    20</code></pre>
</div>
<div id="visualization-of-profile-comparison" class="section level2" number="14.3">
<h2><span class="header-section-number">14.3</span> Visualization of profile comparison</h2>
<p>We can visualize the result using the <code>dotplot()</code> method.</p>
<div class="sourceCode" id="cb145"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb145-1"><a href="#cb145-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dotplot</span>(ck)</span>
<span id="cb145-2"><a href="#cb145-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dotplot</span>(formula_res)</span></code></pre></div>


<div class="figure"><span id="fig:ccn"></span>
<img src="biomedicalKnowledge_files/figure-html/ccn-1.png" alt="Comparing enrichment results of multiple gene lists. (A) Using a named list of gene clusters, the results were displayed as multiple columns with each one represents an enrichment result of a gene cluster. (B) Using formula interface, the columns represent gene clusters defined by the formula." width="95%" />
<p class="caption">
Figure 14.1: <strong>Comparing enrichment results of multiple gene lists.</strong> (A) Using a named list of gene clusters, the results were displayed as multiple columns with each one represents an enrichment result of a gene cluster. (B) Using formula interface, the columns represent gene clusters defined by the formula.
</p>
</div>
<p>The fomula interface allows more complicated gene cluster definition. In Figure <a href="#fig:ccn">14.1</a>B, the gene clusters were defined by two variables (i.e.¬†<code>group</code> that divides genes into <code>upregulated</code> and <code>downregulated</code> and <code>othergroup</code> that divides the genes into two categories of <code>A</code> and <code>B</code>.). The <code>dotplot()</code> function allows us to use one variable to divide the result into different facet and plot the result with other variables in each facet panel (Figure <a href="#fig:ccf">14.2</a>).</p>


<div class="sourceCode" id="cb146"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb146-1"><a href="#cb146-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dotplot</span>(formula_res, <span class="at">x=</span><span class="st">&quot;group&quot;</span>) <span class="sc">+</span> <span class="fu">facet_grid</span>(<span class="sc">~</span>othergroup)</span></code></pre></div>
<div class="figure"><span id="fig:ccf"></span>
<img src="biomedicalKnowledge_files/figure-html/ccf-1.png" alt="Comparing enrichment results of multiple gene lists defined by multiple variable. The results were visualized as a dot plot with an x-axis representing one level of conditions (the group variable) and a facet panel indicating another level of conditions (the othergroup variable)." width="95%" />
<p class="caption">
Figure 14.2: <strong>Comparing enrichment results of multiple gene lists defined by multiple variable.</strong> The results were visualized as a dot plot with an x-axis representing one level of conditions (the <em>group</em> variable) and a facet panel indicating another level of conditions (the <em>othergroup</em> variable).
</p>
</div>
<p>By default, only top 5 (most significant) categories of each cluster
was plotted. User can changes the parameter <code>showCategory</code> to
specify how many categories of each cluster to be plotted, and if
<code>showCategory</code> was set to <code>NULL</code>, the whole result will
be plotted.</p>
<p>The <code>dotplot()</code> function tries to make the comparison among different clusters more informative and reasonable. After extracting <em>e.g.</em> 10 categories for each clusters, <a href="http://bioconductor.org/packages/clusterProfiler">clusterProfiler</a> tries to collect overlap of these categories among clusters. For example, <code>term A</code> is enriched in all the gene clusters (<em>e.g.</em>, <code>g1</code> and <code>g2</code>) and is in the 10 most significant categories of <code>g1</code> but not <code>g2</code>. <a href="http://bioconductor.org/packages/clusterProfiler">clusterProfiler</a> will capture this information and include <code>term A</code> in <code>g2</code> cluster to make the comparison in <code>dotplot</code> more reasonable. If users want to ignore these information, they can set <code>includeAll = FALSE</code> in <code>dotplot()</code>, which is not recommended.</p>
<p>The <code>dotplot()</code> function accepts a parameter <code>by</code> for setting the scale of dot sizes. The default parameter <code>by</code> is setting to <code>geneRatio</code>, which corresponding to the <code>GeneRatio</code> column of the output. If it was setting to <code>count</code>, the comparison will be based on gene counts, while if setting to <code>rowPercentage</code>, the dot sizes will be normalized by <code>count/(sum of each row)</code>. Users can also map the dot size to other derived variable (see <a href="#clusterProfiler-dplyr">Chapter 16</a>).</p>
<p>To provide the full information, we also provide number of identified genes in each category (numbers in parentheses) when <code>by</code> is setting to <code>rowPercentage</code> and number of gene clusters in each cluster label (numbers in parentheses) when <code>by</code> is setting to <code>geneRatio</code>, as shown in Figure <a href="#fig:ccn">14.1</a>.</p>
<p>The p-values indicate that which categories are more likely to have biological meanings. The dots in the plot are color-coded based on their corresponding adjusted p-values. Color gradient ranging from red to blue correspond to the order of increasing adjusted p-values. That is, red indicate low p-values (high enrichment), and blue indicate high p-values (low enrichment). Adjusted p-values were filtered out by the threshold giving by the parameter <code>pvalueCutoff</code>, and FDR can be estimated by <code>qvalue</code>.</p>
</div>
<div id="summary-1" class="section level2" number="14.4">
<h2><span class="header-section-number">14.4</span> Summary</h2>
<p>The comparison function was designed as a framework for comparing gene clusters of any kind of ontology associations, not only <a href="#go-classification">groupGO</a>, <a href="#clusterprofiler-go-ora">enrichGO</a>, <a href="#clusterprofiler-kegg-pathway-ora">enrichKEGG</a>, <a href="#clusterprofiler-kegg-module-ora">enrichMKEGG</a>, <a href="#clusterprofiler-wikipathway-ora">enrichWP</a> and <a href="#universal-api">enricher</a> that were provided in this package, but also other biological and biomedical ontologies, including but not limited to <a href="#reactomepa-ora">enrichPathway</a>, <a href="#dose-do-ora">enrichDO</a>, <a href="#dose-ncg-ora">enrichNCG</a>, <a href="#dose-dgn-ora">enrichDGN</a> and <a href="#meshes-ora">enrichMeSH</a>.</p>
<p>In <span class="citation">(<a href="#ref-yu2012" role="doc-biblioref">Yu et al. 2012</a>)</span>, we analyzed the publicly available expression dataset of breast tumour tissues from 200 patients (GSE11121, Gene Expression Omnibus) <span class="citation">(<a href="#ref-schmidt2008" role="doc-biblioref">Schmidt et al. 2008</a>)</span>. We identified 8 gene clusters from differentially expressed genes, and using the <code>compareCluster()</code> function to compare these gene clusters by their enriched biological process. In <span class="citation">(<a href="#ref-wu_clusterprofiler_2021" role="doc-biblioref">Wu et al. 2021</a>)</span>, we analyzed the GSE8057 dataset which contains expression data from ovarian cancer cells at multiple time points and under two treatment conditions. Eight groups of DEG lists were analyzed simultaneously using <code>compareCluster()</code> with WikiPathways. The result indicates that the two drgus have distinct effects at the begining but consistent effects in the later stages (Fig. 4 of <span class="citation">(<a href="#ref-wu_clusterprofiler_2021" role="doc-biblioref">Wu et al. 2021</a>)</span>).</p>

</div>
</div>
<div id="enrichplot" class="section level1" number="15">
<h1><span class="header-section-number">Chapter 15</span> Visualization of functional enrichment result</h1>
<p>The <a href="http://bioconductor.org/packages/enrichplot">enrichplot</a> package implements several visualization methods to help interpreting enrichment results. It supports visualizing enrichment results obtained from <a href="http://bioconductor.org/packages/DOSE">DOSE</a> <span class="citation">(<a href="#ref-yu_dose_2015" role="doc-biblioref">Yu et al. 2015</a>)</span>, <a href="http://bioconductor.org/packages/clusterProfiler">clusterProfiler</a> <span class="citation">(<a href="#ref-yu2012" role="doc-biblioref">Yu et al. 2012</a>)</span>,
<a href="http://bioconductor.org/packages/ReactomePA">ReactomePA</a> <span class="citation">(<a href="#ref-yu_reactomepa_2016" role="doc-biblioref">Yu and He 2016</a>)</span> and <a href="http://bioconductor.org/packages/meshes">meshes</a> <span class="citation">(<a href="#ref-yu_meshes_2018" role="doc-biblioref">Yu 2018</a>)</span>. Both over representation analysis (ORA) and gene set enrichment analysis (GSEA) are supported.</p>
<p>Several visualization methods were first implemented in <a href="http://bioconductor.org/packages/DOSE">DOSE</a> and rewrote from scratch using <a href="https://CRAN.R-project.org/package=ggplot2">ggplot2</a>. If you want to use the <a href="https://www.biostars.org/p/375555">old methods</a>, you can use the <a href="https://github.com/GuangchuangYu/doseplot">doseplot</a> package.</p>
<div id="bar-plot" class="section level2" number="15.1">
<h2><span class="header-section-number">15.1</span> Bar Plot</h2>
<p>Bar plot is the most widely used method to visualize enriched terms. It depicts the enrichment scores (<em>e.g.</em> p values) and gene count or ratio as bar height
and color.</p>
<div class="sourceCode" id="cb147"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb147-1"><a href="#cb147-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DOSE)</span>
<span id="cb147-2"><a href="#cb147-2" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(geneList)</span>
<span id="cb147-3"><a href="#cb147-3" aria-hidden="true" tabindex="-1"></a>de <span class="ot">&lt;-</span> <span class="fu">names</span>(geneList)[<span class="fu">abs</span>(geneList) <span class="sc">&gt;</span> <span class="dv">2</span>]</span>
<span id="cb147-4"><a href="#cb147-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb147-5"><a href="#cb147-5" aria-hidden="true" tabindex="-1"></a>edo <span class="ot">&lt;-</span> <span class="fu">enrichDGN</span>(de)</span></code></pre></div>


<div class="sourceCode" id="cb148"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb148-1"><a href="#cb148-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(enrichplot)</span>
<span id="cb148-2"><a href="#cb148-2" aria-hidden="true" tabindex="-1"></a><span class="fu">barplot</span>(edo, <span class="at">showCategory=</span><span class="dv">20</span>) </span></code></pre></div>
<div class="figure"><span id="fig:Barplot"></span>
<img src="biomedicalKnowledge_files/figure-html/Barplot-1.png" alt="Bar plot of enriched terms." width="95%" />
<p class="caption">
Figure 15.1: <strong>Bar plot of enriched terms.</strong>
</p>
</div>
</div>
<div id="dot-plot" class="section level2" number="15.2">
<h2><span class="header-section-number">15.2</span> Dot plot</h2>
<p>Dot plot is similar to bar plot with the capability to encode another score as dot size.</p>
<div class="sourceCode" id="cb149"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb149-1"><a href="#cb149-1" aria-hidden="true" tabindex="-1"></a>edo2 <span class="ot">&lt;-</span> <span class="fu">gseDO</span>(geneList)</span>
<span id="cb149-2"><a href="#cb149-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dotplot</span>(edo, <span class="at">showCategory=</span><span class="dv">30</span>) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">&quot;dotplot for ORA&quot;</span>)</span>
<span id="cb149-3"><a href="#cb149-3" aria-hidden="true" tabindex="-1"></a><span class="fu">dotplot</span>(edo2, <span class="at">showCategory=</span><span class="dv">30</span>) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">&quot;dotplot for GSEA&quot;</span>)</span></code></pre></div>


<div class="figure"><span id="fig:Dotplotcap"></span>
<img src="biomedicalKnowledge_files/figure-html/Dotplotcap-1.png" alt="Dot plot of enriched terms." width="95%" />
<p class="caption">
Figure 15.2: <strong>Dot plot of enriched terms.</strong>
</p>
</div>
</div>
<div id="gene-concept-network" class="section level2" number="15.3">
<h2><span class="header-section-number">15.3</span> Gene-Concept Network</h2>
<p>Both the <code>barplot()</code> and <code>dotplot()</code> only displayed most significant enriched terms,
while users may want to know which genes are involved in these significant
terms.
In order to consider the potentially biological complexities in which a gene may belong to multiple annotation categories and provide information of numeric changes if available, we developed <code>cnetplot()</code> function to extract the complex association.
The <code>cnetplot()</code> depicts the linkages of genes and biological concepts (<em>e.g.</em> GO terms or KEGG pathways) as a network. GSEA result is also supported
with only core enriched genes displayed.</p>


<div class="sourceCode" id="cb150"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb150-1"><a href="#cb150-1" aria-hidden="true" tabindex="-1"></a><span class="do">## convert gene ID to Symbol</span></span>
<span id="cb150-2"><a href="#cb150-2" aria-hidden="true" tabindex="-1"></a>edox <span class="ot">&lt;-</span> <span class="fu">setReadable</span>(edo, <span class="st">&#39;org.Hs.eg.db&#39;</span>, <span class="st">&#39;ENTREZID&#39;</span>)</span>
<span id="cb150-3"><a href="#cb150-3" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">cnetplot</span>(edox, <span class="at">foldChange=</span>geneList)</span>
<span id="cb150-4"><a href="#cb150-4" aria-hidden="true" tabindex="-1"></a><span class="do">## categorySize can be scaled by &#39;pvalue&#39; or &#39;geneNum&#39;</span></span>
<span id="cb150-5"><a href="#cb150-5" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">cnetplot</span>(edox, <span class="at">categorySize=</span><span class="st">&quot;pvalue&quot;</span>, <span class="at">foldChange=</span>geneList)</span>
<span id="cb150-6"><a href="#cb150-6" aria-hidden="true" tabindex="-1"></a>p3 <span class="ot">&lt;-</span> <span class="fu">cnetplot</span>(edox, <span class="at">foldChange=</span>geneList, <span class="at">circular =</span> <span class="cn">TRUE</span>, <span class="at">colorEdge =</span> <span class="cn">TRUE</span>)</span>
<span id="cb150-7"><a href="#cb150-7" aria-hidden="true" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(p1, p2, p3, <span class="at">ncol=</span><span class="dv">3</span>, <span class="at">labels=</span>LETTERS[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>], <span class="at">rel_widths=</span><span class="fu">c</span>(.<span class="dv">8</span>, .<span class="dv">8</span>, <span class="fl">1.2</span>))</span></code></pre></div>
<div class="figure"><span id="fig:Networkplot"></span>
<img src="biomedicalKnowledge_files/figure-html/Networkplot-1.png" alt="Network plot of enriched terms." width="95%" />
<p class="caption">
Figure 15.3: <strong>Network plot of enriched terms.</strong>
</p>
</div>
<p>If you would like label subset of the nodes, you can use the <code>node_label</code> parameter, which supports 4 possible selections (i.e.¬†‚Äúcategory,‚Äù ‚Äúgene,‚Äù ‚Äúall‚Äù and ‚Äúnone‚Äù), as demonstrated in Figure <a href="#fig:cnetNodeLabel">15.4</a>.</p>


<div class="sourceCode" id="cb151"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb151-1"><a href="#cb151-1" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">cnetplot</span>(edox, <span class="at">node_label=</span><span class="st">&quot;category&quot;</span>) </span>
<span id="cb151-2"><a href="#cb151-2" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">cnetplot</span>(edox, <span class="at">node_label=</span><span class="st">&quot;gene&quot;</span>) </span>
<span id="cb151-3"><a href="#cb151-3" aria-hidden="true" tabindex="-1"></a>p3 <span class="ot">&lt;-</span> <span class="fu">cnetplot</span>(edox, <span class="at">node_label=</span><span class="st">&quot;all&quot;</span>) </span>
<span id="cb151-4"><a href="#cb151-4" aria-hidden="true" tabindex="-1"></a>p4 <span class="ot">&lt;-</span> <span class="fu">cnetplot</span>(edox, <span class="at">node_label=</span><span class="st">&quot;none&quot;</span>) </span>
<span id="cb151-5"><a href="#cb151-5" aria-hidden="true" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(p1, p2, p3, p4, <span class="at">ncol=</span><span class="dv">2</span>, <span class="at">labels=</span>LETTERS[<span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>])</span></code></pre></div>
<div class="figure"><span id="fig:cnetNodeLabel"></span>
<img src="biomedicalKnowledge_files/figure-html/cnetNodeLabel-1.png" alt="Labelling nodes by selected subset. gene category (A), gene name (B), both gene category and gene name (C, default) and not to label at all (D)." width="95%" />
<p class="caption">
Figure 15.4: <strong>Labelling nodes by selected subset.</strong> gene category (A), gene name (B), both gene category and gene name (C, default) and not to label at all (D).
</p>
</div>
</div>
<div id="heatmap-like-functional-classification" class="section level2" number="15.4">
<h2><span class="header-section-number">15.4</span> Heatmap-like functional classification</h2>
<p>The <code>heatplot</code> is similar to <code>cnetplot</code>, while displaying the relationships as a
heatmap. The gene-concept network may become too complicated if user want to
show a large number significant terms. The <code>heatplot</code> can simplify the result
and more easy to identify expression patterns.</p>


<div class="sourceCode" id="cb152"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb152-1"><a href="#cb152-1" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">heatplot</span>(edox)</span>
<span id="cb152-2"><a href="#cb152-2" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">heatplot</span>(edox, <span class="at">foldChange=</span>geneList)</span>
<span id="cb152-3"><a href="#cb152-3" aria-hidden="true" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(p1, p2, <span class="at">ncol=</span><span class="dv">1</span>, <span class="at">labels=</span>LETTERS[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>])</span></code></pre></div>
<div class="figure"><span id="fig:Heatplot"></span>
<img src="biomedicalKnowledge_files/figure-html/Heatplot-1.png" alt="Heatmap plot of enriched terms. default (A), foldChange=geneList (B)" width="95%" />
<p class="caption">
Figure 15.5: <strong>Heatmap plot of enriched terms.</strong> default (A), <code>foldChange=geneList</code> (B)
</p>
</div>
</div>
<div id="enrichment-map" class="section level2" number="15.5">
<h2><span class="header-section-number">15.5</span> Enrichment Map</h2>
<p>Enrichment map organizes enriched terms into a network with edges connecting
overlapping gene sets. In this way, mutually overlapping gene sets are tend to
cluster together, making it easy to identify functional module.</p>
<p>The <code>emapplot</code> function supports results obtained from hypergeometric test and gene set enrichment analysis. The <code>cex_category</code> parameter can be used to resize nodes, as demonstrated in Figure <a href="#fig:Enrichment">15.6</a> B, and the <code>layout</code> parameter can adjust the layout, as demonstrated in Figure <a href="#fig:Enrichment">15.6</a> C and D.</p>


<div class="sourceCode" id="cb153"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb153-1"><a href="#cb153-1" aria-hidden="true" tabindex="-1"></a>edo <span class="ot">&lt;-</span> <span class="fu">pairwise_termsim</span>(edo)</span>
<span id="cb153-2"><a href="#cb153-2" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">emapplot</span>(edo)</span>
<span id="cb153-3"><a href="#cb153-3" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">emapplot</span>(edo, <span class="at">cex_category=</span><span class="fl">1.5</span>)</span>
<span id="cb153-4"><a href="#cb153-4" aria-hidden="true" tabindex="-1"></a>p3 <span class="ot">&lt;-</span> <span class="fu">emapplot</span>(edo, <span class="at">layout=</span><span class="st">&quot;kk&quot;</span>)</span>
<span id="cb153-5"><a href="#cb153-5" aria-hidden="true" tabindex="-1"></a>p4 <span class="ot">&lt;-</span> <span class="fu">emapplot</span>(edo, <span class="at">cex_category=</span><span class="fl">1.5</span>,<span class="at">layout=</span><span class="st">&quot;kk&quot;</span>) </span>
<span id="cb153-6"><a href="#cb153-6" aria-hidden="true" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(p1, p2, p3, p4, <span class="at">ncol=</span><span class="dv">2</span>, <span class="at">labels=</span>LETTERS[<span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>])</span></code></pre></div>
<div class="figure"><span id="fig:Enrichment"></span>
<img src="biomedicalKnowledge_files/figure-html/Enrichment-1.png" alt="Plot for results obtained from hypergeometric test and gene set enrichment analysis. default (A), cex_category=1.5 (B), layout="kk" (C) and cex_category=1.5,layout="kk" (D)." width="95%" />
<p class="caption">
Figure 15.6: <strong>Plot for results obtained from hypergeometric test and gene set enrichment analysis.</strong> default (A), <code>cex_category=1.5</code> (B), <code>layout="kk"</code> (C) and <code>cex_category=1.5,layout="kk"</code> (D).
</p>
</div>
</div>
<div id="biological-theme-comparison" class="section level2" number="15.6">
<h2><span class="header-section-number">15.6</span> Biological theme comparison</h2>
<p>The <code>emapplot</code> function also supports results obtained from <code>compareCluster</code> function of <code>clusterProfiler</code> package. In addition to <code>cex_category</code> and <code>layout</code> parameters, the number of circles in the bottom left corner can be adjusted using the <code>legend_n</code> parameteras, as demonstrated in Figure <a href="#fig:Enrichment2">15.7</a> B. And proportion of clusters in the pie chart can be adjusted using the <code>pie</code> parameter, when <code>pie="count"</code>, the proportion of clusters in the pie chart is determined by the number of genes, as demonstrated in Figure <a href="#fig:Enrichment2">15.7</a> C and D.</p>


<div class="sourceCode" id="cb154"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb154-1"><a href="#cb154-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(clusterProfiler)</span>
<span id="cb154-2"><a href="#cb154-2" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(gcSample)</span>
<span id="cb154-3"><a href="#cb154-3" aria-hidden="true" tabindex="-1"></a>xx <span class="ot">&lt;-</span> <span class="fu">compareCluster</span>(gcSample, <span class="at">fun=</span><span class="st">&quot;enrichKEGG&quot;</span>,</span>
<span id="cb154-4"><a href="#cb154-4" aria-hidden="true" tabindex="-1"></a>                     <span class="at">organism=</span><span class="st">&quot;hsa&quot;</span>, <span class="at">pvalueCutoff=</span><span class="fl">0.05</span>)</span>
<span id="cb154-5"><a href="#cb154-5" aria-hidden="true" tabindex="-1"></a>xx <span class="ot">&lt;-</span> <span class="fu">pairwise_termsim</span>(xx)                     </span>
<span id="cb154-6"><a href="#cb154-6" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">emapplot</span>(xx)</span>
<span id="cb154-7"><a href="#cb154-7" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">emapplot</span>(xx, <span class="at">legend_n=</span><span class="dv">2</span>) </span>
<span id="cb154-8"><a href="#cb154-8" aria-hidden="true" tabindex="-1"></a>p3 <span class="ot">&lt;-</span> <span class="fu">emapplot</span>(xx, <span class="at">pie=</span><span class="st">&quot;count&quot;</span>)</span>
<span id="cb154-9"><a href="#cb154-9" aria-hidden="true" tabindex="-1"></a>p4 <span class="ot">&lt;-</span> <span class="fu">emapplot</span>(xx, <span class="at">pie=</span><span class="st">&quot;count&quot;</span>, <span class="at">cex_category=</span><span class="fl">1.5</span>, <span class="at">layout=</span><span class="st">&quot;kk&quot;</span>)</span>
<span id="cb154-10"><a href="#cb154-10" aria-hidden="true" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(p1, p2, p3, p4, <span class="at">ncol=</span><span class="dv">2</span>, <span class="at">labels=</span>LETTERS[<span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>])</span></code></pre></div>
<div class="figure"><span id="fig:Enrichment2"></span>
<img src="biomedicalKnowledge_files/figure-html/Enrichment2-1.png" alt="Plot for results obtained from compareCluster function of clusterProfiler package. default (A), legend_n=2 (B), pie="count" (C) and pie="count", cex_category=1.5, layout="kk" (D)." width="95%" />
<p class="caption">
Figure 15.7: <strong>Plot for results obtained from <code>compareCluster</code> function of <code>clusterProfiler</code> package.</strong> default (A), <code>legend_n=2</code> (B), <code>pie="count"</code> (C) and <code>pie="count", cex_category=1.5, layout="kk"</code> (D).
</p>
</div>
</div>
<div id="upset-plot" class="section level2" number="15.7">
<h2><span class="header-section-number">15.7</span> UpSet Plot</h2>
<p>The <code>upsetplot</code> is an alternative to <code>cnetplot</code> for visualizing the complex
association between genes and gene sets. It emphasizes the gene overlapping
among different gene sets.</p>


<div class="sourceCode" id="cb155"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb155-1"><a href="#cb155-1" aria-hidden="true" tabindex="-1"></a><span class="fu">upsetplot</span>(edo)</span></code></pre></div>
<div class="figure"><span id="fig:upsetORA"></span>
<img src="biomedicalKnowledge_files/figure-html/upsetORA-1.png" alt="Upsetplot for over-representation analysis." width="95%" />
<p class="caption">
Figure 15.8: <strong>Upsetplot for over-representation analysis.</strong>
</p>
</div>
<p>For over-representation analysis, <code>upsetplot</code> will calculate the overlaps among different gene sets as demonstrated in Figure <a href="#fig:upsetORA">15.8</a>. For GSEA result, it will plot the fold change distributions of different categories (e.g.¬†unique to pathway, overlaps among different pathways).</p>


<div class="sourceCode" id="cb156"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb156-1"><a href="#cb156-1" aria-hidden="true" tabindex="-1"></a><span class="fu">upsetplot</span>(kk2) </span></code></pre></div>
<div class="figure"><span id="fig:upsetGSEA"></span>
<img src="biomedicalKnowledge_files/figure-html/upsetGSEA-1.png" alt="Upsetplot for gene set enrichment analysis." width="95%" />
<p class="caption">
Figure 15.9: <strong>Upsetplot for gene set enrichment analysis.</strong>
</p>
</div>
</div>
<div id="ridgeline-plot-for-expression-distribution-of-gsea-result" class="section level2" number="15.8">
<h2><span class="header-section-number">15.8</span> ridgeline plot for expression distribution of GSEA result</h2>
<p>The <code>ridgeplot</code> will visualize expression distributions of core enriched genes
for GSEA enriched categories. It helps users to interpret up/down-regulated pathways.</p>


<div class="sourceCode" id="cb157"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb157-1"><a href="#cb157-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ridgeplot</span>(edo2)</span></code></pre></div>
<div class="figure"><span id="fig:ridgeplot"></span>
<img src="biomedicalKnowledge_files/figure-html/ridgeplot-1.png" alt="Ridgeplot for gene set enrichment analysis." width="95%" />
<p class="caption">
Figure 15.10: <strong>Ridgeplot for gene set enrichment analysis.</strong>
</p>
</div>
</div>
<div id="running-score-and-preranked-list-of-gsea-result" class="section level2" number="15.9">
<h2><span class="header-section-number">15.9</span> running score and preranked list of GSEA result</h2>
<p>Running score and preranked list are traditional methods for visualizing GSEA
result. The <a href="http://bioconductor.org/packages/enrichplot">enrichplot</a> package supports both of them to visualize
the distribution of the gene set and the enrichment score.</p>


<div class="sourceCode" id="cb158"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb158-1"><a href="#cb158-1" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">gseaplot</span>(edo2, <span class="at">geneSetID =</span> <span class="dv">1</span>, <span class="at">by =</span> <span class="st">&quot;runningScore&quot;</span>, <span class="at">title =</span> edo2<span class="sc">$</span>Description[<span class="dv">1</span>])</span>
<span id="cb158-2"><a href="#cb158-2" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">gseaplot</span>(edo2, <span class="at">geneSetID =</span> <span class="dv">1</span>, <span class="at">by =</span> <span class="st">&quot;preranked&quot;</span>, <span class="at">title =</span> edo2<span class="sc">$</span>Description[<span class="dv">1</span>])</span>
<span id="cb158-3"><a href="#cb158-3" aria-hidden="true" tabindex="-1"></a>p3 <span class="ot">&lt;-</span> <span class="fu">gseaplot</span>(edo2, <span class="at">geneSetID =</span> <span class="dv">1</span>, <span class="at">title =</span> edo2<span class="sc">$</span>Description[<span class="dv">1</span>])</span>
<span id="cb158-4"><a href="#cb158-4" aria-hidden="true" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(p1, p2, p3, <span class="at">ncol=</span><span class="dv">1</span>, <span class="at">labels=</span>LETTERS[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>])</span></code></pre></div>
<div class="figure"><span id="fig:gseaplot"></span>
<img src="biomedicalKnowledge_files/figure-html/gseaplot-1.png" alt="gseaplot for GSEA result(by = "runningScore"). by = "runningScore" (A), by = "preranked" (B), default (C)" width="95%" />
<p class="caption">
Figure 15.11: <strong>gseaplot for GSEA result(<code>by = "runningScore"</code>).</strong> <code>by = "runningScore"</code> (A), <code>by = "preranked"</code> (B), default (C)
</p>
</div>
<p>Another method to plot GSEA result is the <code>gseaplot2</code> function:</p>


<div class="sourceCode" id="cb159"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb159-1"><a href="#cb159-1" aria-hidden="true" tabindex="-1"></a><span class="fu">gseaplot2</span>(edo2, <span class="at">geneSetID =</span> <span class="dv">1</span>, <span class="at">title =</span> edo2<span class="sc">$</span>Description[<span class="dv">1</span>])</span></code></pre></div>
<div class="figure"><span id="fig:gseaplot2"></span>
<img src="biomedicalKnowledge_files/figure-html/gseaplot2-1.png" alt="Gseaplot2 for GSEA result." width="95%" />
<p class="caption">
Figure 15.12: <strong>Gseaplot2 for GSEA result.</strong>
</p>
</div>
<p>The <code>gseaplot2</code> also supports multile gene sets to be displayed on the same figure:</p>


<div class="sourceCode" id="cb160"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb160-1"><a href="#cb160-1" aria-hidden="true" tabindex="-1"></a><span class="fu">gseaplot2</span>(edo2, <span class="at">geneSetID =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>)</span></code></pre></div>
<div class="figure"><span id="fig:gseaplot22"></span>
<img src="biomedicalKnowledge_files/figure-html/gseaplot22-1.png" alt="Gseaplot2 for GSEA result of multile gene sets." width="95%" />
<p class="caption">
Figure 15.13: <strong>Gseaplot2 for GSEA result of multile gene sets.</strong>
</p>
</div>
<p>User can also displaying the pvalue table on the plot via <code>pvalue_table</code>
parameter:</p>


<div class="sourceCode" id="cb161"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb161-1"><a href="#cb161-1" aria-hidden="true" tabindex="-1"></a><span class="fu">gseaplot2</span>(edo2, <span class="at">geneSetID =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>, <span class="at">pvalue_table =</span> <span class="cn">TRUE</span>,</span>
<span id="cb161-2"><a href="#cb161-2" aria-hidden="true" tabindex="-1"></a>          <span class="at">color =</span> <span class="fu">c</span>(<span class="st">&quot;#E495A5&quot;</span>, <span class="st">&quot;#86B875&quot;</span>, <span class="st">&quot;#7DB0DD&quot;</span>), <span class="at">ES_geom =</span> <span class="st">&quot;dot&quot;</span>)</span></code></pre></div>
<div class="figure"><span id="fig:gseaplot23"></span>
<img src="biomedicalKnowledge_files/figure-html/gseaplot23-1.png" alt="Gseaplot2 for GSEA result of multile gene sets(add pvalue_table)." width="95%" />
<p class="caption">
Figure 15.14: <strong>Gseaplot2 for GSEA result of multile gene sets(add pvalue_table).</strong>
</p>
</div>
<p>User can specify <code>subplots</code> to only display a subset of plots:</p>


<div class="sourceCode" id="cb162"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb162-1"><a href="#cb162-1" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">gseaplot2</span>(edo2, <span class="at">geneSetID =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>, <span class="at">subplots =</span> <span class="dv">1</span>)</span>
<span id="cb162-2"><a href="#cb162-2" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">gseaplot2</span>(edo2, <span class="at">geneSetID =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>, <span class="at">subplots =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>)</span>
<span id="cb162-3"><a href="#cb162-3" aria-hidden="true" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(p1, p2, <span class="at">ncol=</span><span class="dv">1</span>, <span class="at">labels=</span>LETTERS[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>])</span></code></pre></div>
<div class="figure"><span id="fig:gseaplot24"></span>
<img src="biomedicalKnowledge_files/figure-html/gseaplot24-1.png" alt="Gseaplot2 for GSEA result of multile gene sets(add subplots). subplots = 1 (A),subplots = 1:2 (B)" width="95%" />
<p class="caption">
Figure 15.15: <strong>Gseaplot2 for GSEA result of multile gene sets(add subplots).</strong> <code>subplots = 1</code> (A),<code>subplots = 1:2</code> (B)
</p>
</div>
<p>The <code>gsearank</code> function plot the ranked list of genes belong to the specific
gene set.</p>


<div class="sourceCode" id="cb163"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb163-1"><a href="#cb163-1" aria-hidden="true" tabindex="-1"></a><span class="fu">gsearank</span>(edo2, <span class="dv">1</span>, <span class="at">title =</span> edo2[<span class="dv">1</span>, <span class="st">&quot;Description&quot;</span>])</span></code></pre></div>
<div class="figure"><span id="fig:gsearank"></span>
<img src="biomedicalKnowledge_files/figure-html/gsearank-1.png" alt="Ranked list of genes belong to the specific gene set." width="95%" />
<p class="caption">
Figure 15.16: <strong>Ranked list of genes belong to the specific gene set.</strong>
</p>
</div>
<p>Multiple gene sets can be aligned using <code>cowplot</code>:</p>


<div class="sourceCode" id="cb164"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb164-1"><a href="#cb164-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb164-2"><a href="#cb164-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(cowplot)</span>
<span id="cb164-3"><a href="#cb164-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb164-4"><a href="#cb164-4" aria-hidden="true" tabindex="-1"></a>pp <span class="ot">&lt;-</span> <span class="fu">lapply</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>, <span class="cf">function</span>(i) {</span>
<span id="cb164-5"><a href="#cb164-5" aria-hidden="true" tabindex="-1"></a>    anno <span class="ot">&lt;-</span> edo2[i, <span class="fu">c</span>(<span class="st">&quot;NES&quot;</span>, <span class="st">&quot;pvalue&quot;</span>, <span class="st">&quot;p.adjust&quot;</span>)]</span>
<span id="cb164-6"><a href="#cb164-6" aria-hidden="true" tabindex="-1"></a>    lab <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="fu">names</span>(anno), <span class="st">&quot;=&quot;</span>,  <span class="fu">round</span>(anno, <span class="dv">3</span>), <span class="at">collapse=</span><span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb164-7"><a href="#cb164-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb164-8"><a href="#cb164-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">gsearank</span>(edo2, i, edo2[i, <span class="dv">2</span>]) <span class="sc">+</span> <span class="fu">xlab</span>(<span class="cn">NULL</span>) <span class="sc">+</span><span class="fu">ylab</span>(<span class="cn">NULL</span>) <span class="sc">+</span></span>
<span id="cb164-9"><a href="#cb164-9" aria-hidden="true" tabindex="-1"></a>        <span class="fu">annotate</span>(<span class="st">&quot;text&quot;</span>, <span class="dv">10000</span>, edo2[i, <span class="st">&quot;enrichmentScore&quot;</span>] <span class="sc">*</span> .<span class="dv">75</span>, <span class="at">label =</span> lab, <span class="at">hjust=</span><span class="dv">0</span>, <span class="at">vjust=</span><span class="dv">0</span>)</span>
<span id="cb164-10"><a href="#cb164-10" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb164-11"><a href="#cb164-11" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_grid</span>(<span class="at">plotlist=</span>pp, <span class="at">ncol=</span><span class="dv">1</span>)</span></code></pre></div>
<div class="figure"><span id="fig:gsearank2"></span>
<img src="biomedicalKnowledge_files/figure-html/gsearank2-1.png" alt="Gsearank for multiple gene sets." width="95%" />
<p class="caption">
Figure 15.17: <strong>Gsearank for multiple gene sets.</strong>
</p>
</div>
</div>
<div id="pubmed-trend-of-enriched-terms" class="section level2" number="15.10">
<h2><span class="header-section-number">15.10</span> pubmed trend of enriched terms</h2>
<p>One of the problem of enrichment analysis is to find pathways for further
investigation. Here, we provide <code>pmcplot</code> function to plot the number/proportion
of publications trend based on the query result from PubMed Central. Of course,
users can use <code>pmcplot</code> in other scenarios. All text that can be queried on PMC
is valid as input of <code>pmcplot</code>.</p>


<div class="sourceCode" id="cb165"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb165-1"><a href="#cb165-1" aria-hidden="true" tabindex="-1"></a>terms <span class="ot">&lt;-</span> edo<span class="sc">$</span>Description[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span>
<span id="cb165-2"><a href="#cb165-2" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">pmcplot</span>(terms, <span class="dv">2010</span><span class="sc">:</span><span class="dv">2020</span>)</span>
<span id="cb165-3"><a href="#cb165-3" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">pmcplot</span>(terms, <span class="dv">2010</span><span class="sc">:</span><span class="dv">2020</span>, <span class="at">proportion=</span><span class="cn">FALSE</span>)</span>
<span id="cb165-4"><a href="#cb165-4" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_grid</span>(p, p2, <span class="at">ncol=</span><span class="dv">2</span>)</span></code></pre></div>
<div class="figure"><span id="fig:pmcplot"></span>
<img src="biomedicalKnowledge_files/figure-html/pmcplot-1.png" alt="Pmcplot of enrichment analysis." width="95%" />
<p class="caption">
Figure 15.18: <strong>Pmcplot of enrichment analysis.</strong>
</p>
</div>
<!--

using ggplot2 api

`[` accessor


ggplot(egobp[1:20], aes(x=reorder(Description, -pvalue), y=Count, fill=-p.adjust)) +
    geom_bar(stat = "identity") +
    coord_flip() +
    scale_fill_continuous(low="blue", high="red") +
    labs(x = "", y = "", fill = "p.adjust") +
    theme(axis.text=element_text(size=11))

-->

</div>
</div>



<div id="clusterProfiler-dplyr" class="section level1" number="16">
<h1><span class="header-section-number">Chapter 16</span> dplyr verbs for manipulating enrichment result</h1>
<div class="sourceCode" id="cb166"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb166-1"><a href="#cb166-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DOSE)</span>
<span id="cb166-2"><a href="#cb166-2" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(geneList)</span>
<span id="cb166-3"><a href="#cb166-3" aria-hidden="true" tabindex="-1"></a>de <span class="ot">=</span> <span class="fu">names</span>(geneList)[<span class="dv">1</span><span class="sc">:</span><span class="dv">100</span>]</span>
<span id="cb166-4"><a href="#cb166-4" aria-hidden="true" tabindex="-1"></a>x <span class="ot">=</span> <span class="fu">enrichDO</span>(de)</span></code></pre></div>
<div id="filter" class="section level2" number="16.1">
<h2><span class="header-section-number">16.1</span> filter</h2>
<div class="sourceCode" id="cb167"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb167-1"><a href="#cb167-1" aria-hidden="true" tabindex="-1"></a><span class="fu">filter</span>(x, p.adjust <span class="sc">&lt;</span> .<span class="dv">05</span>, qvalue <span class="sc">&lt;</span> <span class="fl">0.2</span>)</span></code></pre></div>
<pre><code>## #
## # over-representation test
## #
## #...@organism     Homo sapiens 
## #...@ontology     DO 
## #...@keytype      ENTREZID 
## #...@gene     chr [1:100] &quot;4312&quot; &quot;8318&quot; &quot;10874&quot; &quot;55143&quot; &quot;55388&quot; &quot;991&quot; &quot;6280&quot; &quot;2305&quot; ...
## #...pvalues adjusted by &#39;BH&#39; with cutoff &lt;0.05 
## #...28 enriched terms found
## &#39;data.frame&#39;:    28 obs. of  9 variables:
##  $ ID         : chr  &quot;DOID:0060071&quot; &quot;DOID:5295&quot; &quot;DOID:8719&quot; &quot;DOID:3007&quot; ...
##  $ Description: chr  &quot;pre-malignant neoplasm&quot; &quot;intestinal disease&quot; &quot;in situ carcinoma&quot; &quot;breast ductal carcinoma&quot; ...
##  $ GeneRatio  : chr  &quot;5/77&quot; &quot;9/77&quot; &quot;4/77&quot; &quot;4/77&quot; ...
##  $ BgRatio    : chr  &quot;22/8007&quot; &quot;157/8007&quot; &quot;18/8007&quot; &quot;29/8007&quot; ...
##  $ pvalue     : num  1.67e-06 1.76e-05 2.18e-05 1.56e-04 2.08e-04 ...
##  $ p.adjust   : num  0.00064 0.00279 0.00279 0.0136 0.0136 ...
##  $ qvalue     : num  0.000461 0.002008 0.002008 0.009796 0.009796 ...
##  $ geneID     : chr  &quot;6280/6278/10232/332/4321&quot; &quot;4312/6279/3627/10563/4283/890/366/4902/3620&quot; &quot;6280/6278/10232/332&quot; &quot;6280/6279/4751/6286&quot; ...
##  $ Count      : int  5 9 4 4 13 6 13 5 5 6 ...
## #...Citation
##   Guangchuang Yu, Li-Gen Wang, Guang-Rong Yan, Qing-Yu He. DOSE: an
##   R/Bioconductor package for Disease Ontology Semantic and Enrichment
##   analysis. Bioinformatics 2015, 31(4):608-609</code></pre>
</div>
<div id="arrange" class="section level2" number="16.2">
<h2><span class="header-section-number">16.2</span> arrange</h2>
<div class="sourceCode" id="cb169"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb169-1"><a href="#cb169-1" aria-hidden="true" tabindex="-1"></a><span class="fu">mutate</span>(x, <span class="at">geneRatio =</span> <span class="fu">parse_ratio</span>(GeneRatio)) <span class="sc">%&gt;%</span></span>
<span id="cb169-2"><a href="#cb169-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(geneRatio))</span></code></pre></div>
<pre><code>## #
## # over-representation test
## #
## #...@organism     Homo sapiens 
## #...@ontology     DO 
## #...@keytype      ENTREZID 
## #...@gene     chr [1:100] &quot;4312&quot; &quot;8318&quot; &quot;10874&quot; &quot;55143&quot; &quot;55388&quot; &quot;991&quot; &quot;6280&quot; &quot;2305&quot; ...
## #...pvalues adjusted by &#39;BH&#39; with cutoff &lt;0.05 
## #...28 enriched terms found
## &#39;data.frame&#39;:    28 obs. of  10 variables:
##  $ ID         : chr  &quot;DOID:3908&quot; &quot;DOID:120&quot; &quot;DOID:2394&quot; &quot;DOID:3459&quot; ...
##  $ Description: chr  &quot;non-small cell lung carcinoma&quot; &quot;female reproductive organ cancer&quot; &quot;ovarian cancer&quot; &quot;breast carcinoma&quot; ...
##  $ GeneRatio  : chr  &quot;13/77&quot; &quot;13/77&quot; &quot;10/77&quot; &quot;10/77&quot; ...
##  $ BgRatio    : chr  &quot;431/8007&quot; &quot;455/8007&quot; &quot;312/8007&quot; &quot;383/8007&quot; ...
##  $ pvalue     : num  2.08e-04 3.52e-04 7.54e-04 3.48e-03 1.76e-05 ...
##  $ p.adjust   : num  0.0136 0.01928 0.02625 0.04756 0.00279 ...
##  $ qvalue     : num  0.0098 0.01389 0.0189 0.03424 0.00201 ...
##  $ geneID     : chr  &quot;6280/2305/9133/6279/7153/6278/6241/11065/10232/332/6286/3002/9212&quot; &quot;4312/6279/7153/3627/820/983/10232/6362/332/6286/9212/4321/6790&quot; &quot;4312/820/983/10232/6362/332/6286/9212/4321/6790&quot; &quot;4312/6280/6279/7153/4751/890/4085/332/6286/6790&quot; ...
##  $ Count      : int  13 13 10 10 9 7 7 6 6 6 ...
##  $ geneRatio  : num  0.169 0.169 0.13 0.13 0.117 ...
## #...Citation
##   Guangchuang Yu, Li-Gen Wang, Guang-Rong Yan, Qing-Yu He. DOSE: an
##   R/Bioconductor package for Disease Ontology Semantic and Enrichment
##   analysis. Bioinformatics 2015, 31(4):608-609</code></pre>
</div>
<div id="select" class="section level2" number="16.3">
<h2><span class="header-section-number">16.3</span> select</h2>
<div class="sourceCode" id="cb171"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb171-1"><a href="#cb171-1" aria-hidden="true" tabindex="-1"></a><span class="fu">select</span>(x, <span class="sc">-</span>geneID) <span class="sc">%&gt;%</span> head</span></code></pre></div>
<pre><code>##                        ID                   Description GeneRatio  BgRatio
## DOID:0060071 DOID:0060071        pre-malignant neoplasm      5/77  22/8007
## DOID:5295       DOID:5295            intestinal disease      9/77 157/8007
## DOID:8719       DOID:8719             in situ carcinoma      4/77  18/8007
## DOID:3007       DOID:3007       breast ductal carcinoma      4/77  29/8007
## DOID:3908       DOID:3908 non-small cell lung carcinoma     13/77 431/8007
## DOID:0050589 DOID:0050589    inflammatory bowel disease      6/77  90/8007
##                    pvalue     p.adjust       qvalue Count
## DOID:0060071 1.671524e-06 0.0006401937 0.0004609887     5
## DOID:5295    1.759049e-05 0.0027885022 0.0020079362     9
## DOID:8719    2.184205e-05 0.0027885022 0.0020079362     4
## DOID:3007    1.564603e-04 0.0136037018 0.0097957122     4
## DOID:3908    2.075001e-04 0.0136037018 0.0097957122    13
## DOID:0050589 2.131128e-04 0.0136037018 0.0097957122     6</code></pre>
</div>
<div id="mutate" class="section level2" number="16.4">
<h2><span class="header-section-number">16.4</span> mutate</h2>


<div class="sourceCode" id="cb173"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb173-1"><a href="#cb173-1" aria-hidden="true" tabindex="-1"></a><span class="co"># k/M</span></span>
<span id="cb173-2"><a href="#cb173-2" aria-hidden="true" tabindex="-1"></a>y <span class="ot">&lt;-</span> <span class="fu">mutate</span>(x, <span class="at">richFactor =</span> Count <span class="sc">/</span> <span class="fu">as.numeric</span>(<span class="fu">sub</span>(<span class="st">&quot;/</span><span class="sc">\\</span><span class="st">d+&quot;</span>, <span class="st">&quot;&quot;</span>, BgRatio)))</span>
<span id="cb173-3"><a href="#cb173-3" aria-hidden="true" tabindex="-1"></a>y</span></code></pre></div>
<pre><code>## #
## # over-representation test
## #
## #...@organism     Homo sapiens 
## #...@ontology     DO 
## #...@keytype      ENTREZID 
## #...@gene     chr [1:100] &quot;4312&quot; &quot;8318&quot; &quot;10874&quot; &quot;55143&quot; &quot;55388&quot; &quot;991&quot; &quot;6280&quot; &quot;2305&quot; ...
## #...pvalues adjusted by &#39;BH&#39; with cutoff &lt;0.05 
## #...28 enriched terms found
## &#39;data.frame&#39;:    28 obs. of  10 variables:
##  $ ID         : chr  &quot;DOID:0060071&quot; &quot;DOID:5295&quot; &quot;DOID:8719&quot; &quot;DOID:3007&quot; ...
##  $ Description: chr  &quot;pre-malignant neoplasm&quot; &quot;intestinal disease&quot; &quot;in situ carcinoma&quot; &quot;breast ductal carcinoma&quot; ...
##  $ GeneRatio  : chr  &quot;5/77&quot; &quot;9/77&quot; &quot;4/77&quot; &quot;4/77&quot; ...
##  $ BgRatio    : chr  &quot;22/8007&quot; &quot;157/8007&quot; &quot;18/8007&quot; &quot;29/8007&quot; ...
##  $ pvalue     : num  1.67e-06 1.76e-05 2.18e-05 1.56e-04 2.08e-04 ...
##  $ p.adjust   : num  0.00064 0.00279 0.00279 0.0136 0.0136 ...
##  $ qvalue     : num  0.000461 0.002008 0.002008 0.009796 0.009796 ...
##  $ geneID     : chr  &quot;6280/6278/10232/332/4321&quot; &quot;4312/6279/3627/10563/4283/890/366/4902/3620&quot; &quot;6280/6278/10232/332&quot; &quot;6280/6279/4751/6286&quot; ...
##  $ Count      : int  5 9 4 4 13 6 13 5 5 6 ...
##  $ richFactor : num  0.2273 0.0573 0.2222 0.1379 0.0302 ...
## #...Citation
##   Guangchuang Yu, Li-Gen Wang, Guang-Rong Yan, Qing-Yu He. DOSE: an
##   R/Bioconductor package for Disease Ontology Semantic and Enrichment
##   analysis. Bioinformatics 2015, 31(4):608-609</code></pre>
<div class="sourceCode" id="cb175"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb175-1"><a href="#cb175-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb175-2"><a href="#cb175-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(forcats)</span>
<span id="cb175-3"><a href="#cb175-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(enrichplot)</span>
<span id="cb175-4"><a href="#cb175-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb175-5"><a href="#cb175-5" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(y, <span class="at">showCategory =</span> <span class="dv">20</span>, </span>
<span id="cb175-6"><a href="#cb175-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">aes</span>(richFactor, <span class="fu">fct_reorder</span>(Description, richFactor))) <span class="sc">+</span> </span>
<span id="cb175-7"><a href="#cb175-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_segment</span>(<span class="fu">aes</span>(<span class="at">xend=</span><span class="dv">0</span>, <span class="at">yend =</span> Description)) <span class="sc">+</span></span>
<span id="cb175-8"><a href="#cb175-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">color=</span>p.adjust, <span class="at">size =</span> Count)) <span class="sc">+</span></span>
<span id="cb175-9"><a href="#cb175-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_viridis_c</span>(<span class="at">guide=</span><span class="fu">guide_colorbar</span>(<span class="at">reverse=</span><span class="cn">TRUE</span>)) <span class="sc">+</span></span>
<span id="cb175-10"><a href="#cb175-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_size_continuous</span>(<span class="at">range=</span><span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">10</span>)) <span class="sc">+</span></span>
<span id="cb175-11"><a href="#cb175-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span> </span>
<span id="cb175-12"><a href="#cb175-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">&quot;rich factor&quot;</span>) <span class="sc">+</span></span>
<span id="cb175-13"><a href="#cb175-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="cn">NULL</span>) <span class="sc">+</span> </span>
<span id="cb175-14"><a href="#cb175-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Enriched Disease Ontology&quot;</span>)</span></code></pre></div>
<div class="figure"><span id="fig:dot-richfactor"></span>
<img src="biomedicalKnowledge_files/figure-html/dot-richfactor-1.png" alt="Visualizing rich factor of enriched terms using lolliplot." width="95%" />
<p class="caption">
Figure 16.1: <strong>Visualizing rich factor of enriched terms using lolliplot.</strong>
</p>
</div>
<p>A very similar concept is Fold Enrichment, which is defined as the ratio of two proportions, (k/n) / (M/N). Using <code>mutate</code> to add the fold enrichment variable is also easy:</p>
<div class="sourceCode" id="cb176"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb176-1"><a href="#cb176-1" aria-hidden="true" tabindex="-1"></a><span class="fu">mutate</span>(x, <span class="at">FoldEnrichment =</span> <span class="fu">parse_ratio</span>(GeneRatio) <span class="sc">/</span> <span class="fu">parse_ratio</span>(BgRatio))</span></code></pre></div>
</div>
<div id="slice" class="section level2" number="16.5">
<h2><span class="header-section-number">16.5</span> slice</h2>
<p>We can use <code>slice</code> to choose rows by their ordinal position in the enrichment result. Grouped result use the ordinal position with the group.</p>
<p>In the following example, a GSEA result of Reactome pathway was sorted by the absolute values of NES and the result was grouped by the sign of NES. We then extracted first 5 rows of each groups. The result was displayed in Figure <a href="#fig:sliceNES">16.2</a>.</p>


<div class="sourceCode" id="cb177"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb177-1"><a href="#cb177-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ReactomePA)</span>
<span id="cb177-2"><a href="#cb177-2" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(geneList)</span>
<span id="cb177-3"><a href="#cb177-3" aria-hidden="true" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="fu">gsePathway</span>(geneList)</span>
<span id="cb177-4"><a href="#cb177-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb177-5"><a href="#cb177-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb177-6"><a href="#cb177-6" aria-hidden="true" tabindex="-1"></a>y <span class="ot">&lt;-</span> <span class="fu">arrange</span>(x, <span class="fu">abs</span>(NES)) <span class="sc">%&gt;%</span> </span>
<span id="cb177-7"><a href="#cb177-7" aria-hidden="true" tabindex="-1"></a>        <span class="fu">group_by</span>(<span class="fu">sign</span>(NES)) <span class="sc">%&gt;%</span> </span>
<span id="cb177-8"><a href="#cb177-8" aria-hidden="true" tabindex="-1"></a>        <span class="fu">slice</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>)</span>
<span id="cb177-9"><a href="#cb177-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb177-10"><a href="#cb177-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(forcats)</span>
<span id="cb177-11"><a href="#cb177-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb177-12"><a href="#cb177-12" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggstance)</span>
<span id="cb177-13"><a href="#cb177-13" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(enrichplot)</span>
<span id="cb177-14"><a href="#cb177-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb177-15"><a href="#cb177-15" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(y, <span class="fu">aes</span>(NES, <span class="fu">fct_reorder</span>(Description, NES), <span class="at">fill=</span>qvalues), <span class="at">showCategory=</span><span class="dv">10</span>) <span class="sc">+</span> </span>
<span id="cb177-16"><a href="#cb177-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_barh</span>(<span class="at">stat=</span><span class="st">&#39;identity&#39;</span>) <span class="sc">+</span> </span>
<span id="cb177-17"><a href="#cb177-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_continuous</span>(<span class="at">low=</span><span class="st">&#39;red&#39;</span>, <span class="at">high=</span><span class="st">&#39;blue&#39;</span>, <span class="at">guide=</span><span class="fu">guide_colorbar</span>(<span class="at">reverse=</span><span class="cn">TRUE</span>)) <span class="sc">+</span> </span>
<span id="cb177-18"><a href="#cb177-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_minimal</span>() <span class="sc">+</span> <span class="fu">ylab</span>(<span class="cn">NULL</span>)</span></code></pre></div>
<div class="figure"><span id="fig:sliceNES"></span>
<img src="biomedicalKnowledge_files/figure-html/sliceNES-1.png" alt="Choose pathways by ordinal positions." width="95%" />
<p class="caption">
Figure 16.2: <strong>Choose pathways by ordinal positions.</strong>
</p>
</div>
</div>
<div id="summarise" class="section level2" number="16.6">
<h2><span class="header-section-number">16.6</span> summarise</h2>


<div class="sourceCode" id="cb178"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb178-1"><a href="#cb178-1" aria-hidden="true" tabindex="-1"></a>pbar <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {</span>
<span id="cb178-2"><a href="#cb178-2" aria-hidden="true" tabindex="-1"></a>  pi<span class="ot">=</span><span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="at">length.out=</span><span class="dv">11</span>)</span>
<span id="cb178-3"><a href="#cb178-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb178-4"><a href="#cb178-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(x, <span class="at">pp =</span> <span class="fu">cut</span>(p.adjust, pi)) <span class="sc">%&gt;%</span></span>
<span id="cb178-5"><a href="#cb178-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(pp) <span class="sc">%&gt;%</span> </span>
<span id="cb178-6"><a href="#cb178-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">cnt =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb178-7"><a href="#cb178-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(pp, cnt)) <span class="sc">+</span> <span class="fu">geom_col</span>() <span class="sc">+</span> </span>
<span id="cb178-8"><a href="#cb178-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb178-9"><a href="#cb178-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="st">&quot;p value intervals&quot;</span>) <span class="sc">+</span></span>
<span id="cb178-10"><a href="#cb178-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ylab</span>(<span class="st">&quot;Frequency&quot;</span>) <span class="sc">+</span> </span>
<span id="cb178-11"><a href="#cb178-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="st">&quot;p value distribution&quot;</span>)</span>
<span id="cb178-12"><a href="#cb178-12" aria-hidden="true" tabindex="-1"></a>}    </span>
<span id="cb178-13"><a href="#cb178-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb178-14"><a href="#cb178-14" aria-hidden="true" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="fu">enrichDO</span>(de, <span class="at">pvalueCutoff=</span><span class="dv">1</span>, <span class="at">qvalueCutoff=</span><span class="dv">1</span>)</span>
<span id="cb178-15"><a href="#cb178-15" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">2020-09-10</span>)</span>
<span id="cb178-16"><a href="#cb178-16" aria-hidden="true" tabindex="-1"></a>random_genes <span class="ot">&lt;-</span> <span class="fu">sample</span>(<span class="fu">names</span>(geneList), <span class="dv">100</span>)</span>
<span id="cb178-17"><a href="#cb178-17" aria-hidden="true" tabindex="-1"></a>y <span class="ot">&lt;-</span> <span class="fu">enrichDO</span>(random_genes, <span class="at">pvalueCutoff=</span><span class="dv">1</span>, <span class="at">qvalueCutoff=</span><span class="dv">1</span>)</span>
<span id="cb178-18"><a href="#cb178-18" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">pbar</span>(x)</span>
<span id="cb178-19"><a href="#cb178-19" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">pbar</span>(y)</span>
<span id="cb178-20"><a href="#cb178-20" aria-hidden="true" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(p1, p2, <span class="at">ncol=</span><span class="dv">1</span>, <span class="at">labels =</span> LETTERS[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>])</span></code></pre></div>
<div class="figure"><span id="fig:p-bar"></span>
<img src="biomedicalKnowledge_files/figure-html/p-bar-1.png" alt="Distribution of adjusted p-values." width="95%" />
<p class="caption">
Figure 16.3: <strong>Distribution of adjusted p-values.</strong>
</p>
</div>

</div>
</div>
<div id="useful-utilities" class="section level1" number="17">
<h1><span class="header-section-number">Chapter 17</span> Useful utilities</h1>
<div id="id-convert" class="section level2" number="17.1">
<h2><span class="header-section-number">17.1</span> Translating Biological ID</h2>
<div id="bitr" class="section level3" number="17.1.1">
<h3><span class="header-section-number">17.1.1</span> <code>bitr</code>: Biological Id TranslatoR</h3>
<p>The <a href="http://bioconductor.org/packages/clusterProfiler">clusterProfiler</a> package provides the <code>bitr()</code> and <code>bitr_kegg()</code> functions for converting ID types. Both <code>bitr()</code> and <code>bitr_kegg()</code> support many species including model and many non-model organisms.</p>
<div class="sourceCode" id="cb179"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb179-1"><a href="#cb179-1" aria-hidden="true" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;GPX3&quot;</span>,  <span class="st">&quot;GLRX&quot;</span>,   <span class="st">&quot;LBP&quot;</span>,   <span class="st">&quot;CRYAB&quot;</span>, <span class="st">&quot;DEFB1&quot;</span>, <span class="st">&quot;HCLS1&quot;</span>,   <span class="st">&quot;SOD2&quot;</span>,   <span class="st">&quot;HSPA2&quot;</span>,</span>
<span id="cb179-2"><a href="#cb179-2" aria-hidden="true" tabindex="-1"></a>       <span class="st">&quot;ORM1&quot;</span>,  <span class="st">&quot;IGFBP1&quot;</span>, <span class="st">&quot;PTHLH&quot;</span>, <span class="st">&quot;GPC3&quot;</span>,  <span class="st">&quot;IGFBP3&quot;</span>,<span class="st">&quot;TOB1&quot;</span>,    <span class="st">&quot;MITF&quot;</span>,   <span class="st">&quot;NDRG1&quot;</span>,</span>
<span id="cb179-3"><a href="#cb179-3" aria-hidden="true" tabindex="-1"></a>       <span class="st">&quot;NR1H4&quot;</span>, <span class="st">&quot;FGFR3&quot;</span>,  <span class="st">&quot;PVR&quot;</span>,   <span class="st">&quot;IL6&quot;</span>,   <span class="st">&quot;PTPRM&quot;</span>, <span class="st">&quot;ERBB2&quot;</span>,   <span class="st">&quot;NID2&quot;</span>,   <span class="st">&quot;LAMB1&quot;</span>,</span>
<span id="cb179-4"><a href="#cb179-4" aria-hidden="true" tabindex="-1"></a>       <span class="st">&quot;COMP&quot;</span>,  <span class="st">&quot;PLS3&quot;</span>,   <span class="st">&quot;MCAM&quot;</span>,  <span class="st">&quot;SPP1&quot;</span>,  <span class="st">&quot;LAMC1&quot;</span>, <span class="st">&quot;COL4A2&quot;</span>,  <span class="st">&quot;COL4A1&quot;</span>, <span class="st">&quot;MYOC&quot;</span>,</span>
<span id="cb179-5"><a href="#cb179-5" aria-hidden="true" tabindex="-1"></a>       <span class="st">&quot;ANXA4&quot;</span>, <span class="st">&quot;TFPI2&quot;</span>,  <span class="st">&quot;CST6&quot;</span>,  <span class="st">&quot;SLPI&quot;</span>,  <span class="st">&quot;TIMP2&quot;</span>, <span class="st">&quot;CPM&quot;</span>,     <span class="st">&quot;GGT1&quot;</span>,   <span class="st">&quot;NNMT&quot;</span>,</span>
<span id="cb179-6"><a href="#cb179-6" aria-hidden="true" tabindex="-1"></a>       <span class="st">&quot;MAL&quot;</span>,   <span class="st">&quot;EEF1A2&quot;</span>, <span class="st">&quot;HGD&quot;</span>,   <span class="st">&quot;TCN2&quot;</span>,  <span class="st">&quot;CDA&quot;</span>,   <span class="st">&quot;PCCA&quot;</span>,    <span class="st">&quot;CRYM&quot;</span>,   <span class="st">&quot;PDXK&quot;</span>,</span>
<span id="cb179-7"><a href="#cb179-7" aria-hidden="true" tabindex="-1"></a>       <span class="st">&quot;STC1&quot;</span>,  <span class="st">&quot;WARS&quot;</span>,  <span class="st">&quot;HMOX1&quot;</span>, <span class="st">&quot;FXYD2&quot;</span>, <span class="st">&quot;RBP4&quot;</span>,   <span class="st">&quot;SLC6A12&quot;</span>, <span class="st">&quot;KDELR3&quot;</span>, <span class="st">&quot;ITM2B&quot;</span>)</span>
<span id="cb179-8"><a href="#cb179-8" aria-hidden="true" tabindex="-1"></a>eg <span class="ot">=</span> <span class="fu">bitr</span>(x, <span class="at">fromType=</span><span class="st">&quot;SYMBOL&quot;</span>, <span class="at">toType=</span><span class="st">&quot;ENTREZID&quot;</span>, <span class="at">OrgDb=</span><span class="st">&quot;org.Hs.eg.db&quot;</span>)</span>
<span id="cb179-9"><a href="#cb179-9" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(eg)</span></code></pre></div>
<pre><code>##   SYMBOL ENTREZID
## 1   GPX3     2878
## 2   GLRX     2745
## 3    LBP     3929
## 4  CRYAB     1410
## 5  DEFB1     1672
## 6  HCLS1     3059</code></pre>
<p>User should provides an annotation package, both <code>fromType</code> and <code>toType</code> can accept any types that supported.</p>
<p>User can use <code>keytypes</code> to list all supporting types.</p>
<div class="sourceCode" id="cb181"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb181-1"><a href="#cb181-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(org.Hs.eg.db)</span>
<span id="cb181-2"><a href="#cb181-2" aria-hidden="true" tabindex="-1"></a><span class="fu">keytypes</span>(org.Hs.eg.db)</span></code></pre></div>
<pre><code>##  [1] &quot;ACCNUM&quot;       &quot;ALIAS&quot;        &quot;ENSEMBL&quot;      &quot;ENSEMBLPROT&quot;  &quot;ENSEMBLTRANS&quot;
##  [6] &quot;ENTREZID&quot;     &quot;ENZYME&quot;       &quot;EVIDENCE&quot;     &quot;EVIDENCEALL&quot;  &quot;GENENAME&quot;    
## [11] &quot;GENETYPE&quot;     &quot;GO&quot;           &quot;GOALL&quot;        &quot;IPI&quot;          &quot;MAP&quot;         
## [16] &quot;OMIM&quot;         &quot;ONTOLOGY&quot;     &quot;ONTOLOGYALL&quot;  &quot;PATH&quot;         &quot;PFAM&quot;        
## [21] &quot;PMID&quot;         &quot;PROSITE&quot;      &quot;REFSEQ&quot;       &quot;SYMBOL&quot;       &quot;UCSCKG&quot;      
## [26] &quot;UNIPROT&quot;</code></pre>
<p>We can translate from one type to other types.</p>
<div class="sourceCode" id="cb183"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb183-1"><a href="#cb183-1" aria-hidden="true" tabindex="-1"></a>ids <span class="ot">&lt;-</span> <span class="fu">bitr</span>(x, <span class="at">fromType=</span><span class="st">&quot;SYMBOL&quot;</span>, <span class="at">toType=</span><span class="fu">c</span>(<span class="st">&quot;UNIPROT&quot;</span>, <span class="st">&quot;ENSEMBL&quot;</span>), <span class="at">OrgDb=</span><span class="st">&quot;org.Hs.eg.db&quot;</span>)</span></code></pre></div>
<pre><code>## Warning in bitr(x, fromType = &quot;SYMBOL&quot;, toType = c(&quot;UNIPROT&quot;, &quot;ENSEMBL&quot;), :
## 1.79% of input gene IDs are fail to map...</code></pre>
<div class="sourceCode" id="cb185"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb185-1"><a href="#cb185-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(ids)</span></code></pre></div>
<pre><code>##   SYMBOL    UNIPROT         ENSEMBL
## 1   GPX3     P22352 ENSG00000211445
## 2   GLRX A0A024RAM2 ENSG00000173221
## 3   GLRX     P35754 ENSG00000173221
## 4    LBP     P18428 ENSG00000129988
## 5    LBP     Q8TCF0 ENSG00000129988
## 6  CRYAB     P02511 ENSG00000109846</code></pre>
<p>For GO analysis, user don‚Äôt need to convert ID, all ID type provided by <code>OrgDb</code> can be used in <code>groupGO</code>, <code>enrichGO</code> and <code>gseGO</code> by specifying <code>keyType</code> parameter.</p>
<p>Users can use the <code>bitr</code> function to convert IDs using any ID types that available in the <code>OrbDb</code> object. For example, users may want to know the genes in background that are belong to a specific GO term. Such information can be easily accessed using <code>bitr</code>.</p>
<div class="sourceCode" id="cb187"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb187-1"><a href="#cb187-1" aria-hidden="true" tabindex="-1"></a>m <span class="ot">&lt;-</span> <span class="fu">bitr</span>(<span class="st">&quot;GO:0006805&quot;</span>, <span class="at">fromType=</span><span class="st">&quot;GO&quot;</span>, <span class="at">toType =</span> <span class="st">&quot;SYMBOL&quot;</span>, <span class="at">OrgDb=</span>org.Hs.eg.db)</span>
<span id="cb187-2"><a href="#cb187-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(m)</span></code></pre></div>
<pre><code>## [1] 111   4</code></pre>
<div class="sourceCode" id="cb189"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb189-1"><a href="#cb189-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(m)</span></code></pre></div>
<pre><code>##           GO EVIDENCE ONTOLOGY SYMBOL
## 1 GO:0006805      TAS       BP   NAT1
## 2 GO:0006805      TAS       BP   NAT2
## 3 GO:0006805      TAS       BP  AADAC
## 4 GO:0006805      TAS       BP   AOC1
## 5 GO:0006805      TAS       BP   ACY1
## 6 GO:0006805      TAS       BP    AHR</code></pre>
<p>Note: If you want to extract genes in your input gene list that are belong to a specific term/pathway, you can use the <a href="#how-to-extract-genes-of-a-specific-termpathway">geneInCategory</a> function.</p>
</div>
<div id="bitr_kegg" class="section level3" number="17.1.2">
<h3><span class="header-section-number">17.1.2</span> <code>bitr_kegg</code>: converting biological IDs using KEGG API</h3>
<div class="sourceCode" id="cb191"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb191-1"><a href="#cb191-1" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(gcSample)</span>
<span id="cb191-2"><a href="#cb191-2" aria-hidden="true" tabindex="-1"></a>hg <span class="ot">&lt;-</span> gcSample[[<span class="dv">1</span>]]</span>
<span id="cb191-3"><a href="#cb191-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(hg)</span></code></pre></div>
<pre><code>## [1] &quot;4597&quot;  &quot;7111&quot;  &quot;5266&quot;  &quot;2175&quot;  &quot;755&quot;   &quot;23046&quot;</code></pre>
<div class="sourceCode" id="cb193"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb193-1"><a href="#cb193-1" aria-hidden="true" tabindex="-1"></a>eg2np <span class="ot">&lt;-</span> <span class="fu">bitr_kegg</span>(hg, <span class="at">fromType=</span><span class="st">&#39;kegg&#39;</span>, <span class="at">toType=</span><span class="st">&#39;ncbi-proteinid&#39;</span>, <span class="at">organism=</span><span class="st">&#39;hsa&#39;</span>)</span></code></pre></div>
<pre><code>## Warning in bitr_kegg(hg, fromType = &quot;kegg&quot;, toType = &quot;ncbi-proteinid&quot;, organism
## = &quot;hsa&quot;): 3.7% of input gene IDs are fail to map...</code></pre>
<div class="sourceCode" id="cb195"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb195-1"><a href="#cb195-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(eg2np)</span></code></pre></div>
<pre><code>##    kegg ncbi-proteinid
## 1 10001      NP_005457
## 2 10209      NP_005792
## 3 10232      NP_037536
## 4 10324      NP_006054
## 5 10411   NP_001092002
## 6 10614      NP_006451</code></pre>
<p>The ID type (both <code>fromType</code> &amp; <code>toType</code>) should be one of ‚Äòkegg,‚Äô ‚Äòncbi-geneid,‚Äô ‚Äòncbi-proteinid‚Äô or ‚Äòuniprot.‚Äô The ‚Äòkegg‚Äô is the primary ID used in KEGG database. The data source of KEGG was from NCBI. A rule of thumb for the ‚Äòkegg‚Äô ID is <code>entrezgene</code> ID for eukaryote species and <code>Locus</code> ID for prokaryotes.</p>
<p>Many prokaryote species don‚Äôt have entrezgene ID available. For example we can check the gene information of <code>ece:Z5100</code> in <a href="http://www.genome.jp/dbget-bin/www_bget?ece:Z5100" class="uri">http://www.genome.jp/dbget-bin/www_bget?ece:Z5100</a>, which have <code>NCBI-ProteinID</code> and <code>UnitProt</code> links in the <code>Other DBs</code> Entry, but not <code>NCBI-GeneID</code>.</p>
<p>If we try to convert <code>Z5100</code> to <code>ncbi-geneid</code>, <code>bitr_kegg</code> will throw error of <code>ncbi-geneid is not supported</code>.</p>
<div class="sourceCode" id="cb197"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb197-1"><a href="#cb197-1" aria-hidden="true" tabindex="-1"></a><span class="fu">bitr_kegg</span>(<span class="st">&quot;Z5100&quot;</span>, <span class="at">fromType=</span><span class="st">&quot;kegg&quot;</span>, <span class="at">toType=</span><span class="st">&#39;ncbi-geneid&#39;</span>, <span class="at">organism=</span><span class="st">&#39;ece&#39;</span>)</span>
<span id="cb197-2"><a href="#cb197-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb197-3"><a href="#cb197-3" aria-hidden="true" tabindex="-1"></a><span class="do">## Error in KEGG_convert(fromType, toType, organism) :</span></span>
<span id="cb197-4"><a href="#cb197-4" aria-hidden="true" tabindex="-1"></a><span class="do">## ncbi-geneid is not supported for ece ...</span></span></code></pre></div>
<p>We can of course convert it to <code>ncbi-proteinid</code> and <code>uniprot</code>:</p>
<div class="sourceCode" id="cb198"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb198-1"><a href="#cb198-1" aria-hidden="true" tabindex="-1"></a><span class="fu">bitr_kegg</span>(<span class="st">&quot;Z5100&quot;</span>, <span class="at">fromType=</span><span class="st">&quot;kegg&quot;</span>, <span class="at">toType=</span><span class="st">&#39;ncbi-proteinid&#39;</span>, <span class="at">organism=</span><span class="st">&#39;ece&#39;</span>)</span></code></pre></div>
<pre><code>##    kegg ncbi-proteinid
## 1 Z5100       AAG58814</code></pre>
<div class="sourceCode" id="cb200"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb200-1"><a href="#cb200-1" aria-hidden="true" tabindex="-1"></a><span class="fu">bitr_kegg</span>(<span class="st">&quot;Z5100&quot;</span>, <span class="at">fromType=</span><span class="st">&quot;kegg&quot;</span>, <span class="at">toType=</span><span class="st">&#39;uniprot&#39;</span>, <span class="at">organism=</span><span class="st">&#39;ece&#39;</span>)</span></code></pre></div>
<pre><code>##    kegg    uniprot
## 1 Z5100 A0A7Z3D5K5</code></pre>
</div>
</div>
<div id="setReadable" class="section level2" number="17.2">
<h2><span class="header-section-number">17.2</span> <code>setReadable</code>: translating gene IDs to human readable symbols</h2>
<p>Some of the functions, especially those internally supported for <a href="#chapter4">DO</a>, <a href="#chapter5">GO</a>, and <a href="#chapter8">Reactome Pathway</a>, support a parameter, <code>readable</code>. If <code>readable = TRUE</code>, all the gene IDs will be translated to gene symbols. The <code>readable</code> parameter is not available for enrichment analysis of KEGG or using user‚Äôs own annotation. KEGG analysis using <code>enrichKEGG</code> and <code>gseKEGG</code>, which internally query annotation information from KEEGG database and thus support all species if it is available in the KEGG database. However, KEGG database doesn‚Äôt provide gene ID to symbol mapping information. For analysis using user‚Äôs own annotation data, we even don‚Äôt know what species is in analyzed. Translating gene IDs to gene symbols is partly supported using the <code>setReadable</code> function if and only if there is an <code>OrgDb</code> available.</p>
<div class="sourceCode" id="cb202"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb202-1"><a href="#cb202-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(org.Hs.eg.db)</span>
<span id="cb202-2"><a href="#cb202-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(clusterProfiler)</span>
<span id="cb202-3"><a href="#cb202-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb202-4"><a href="#cb202-4" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(geneList, <span class="at">package=</span><span class="st">&quot;DOSE&quot;</span>)</span>
<span id="cb202-5"><a href="#cb202-5" aria-hidden="true" tabindex="-1"></a>de <span class="ot">&lt;-</span> <span class="fu">names</span>(geneList)[<span class="dv">1</span><span class="sc">:</span><span class="dv">100</span>]</span>
<span id="cb202-6"><a href="#cb202-6" aria-hidden="true" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="fu">enrichKEGG</span>(de)</span>
<span id="cb202-7"><a href="#cb202-7" aria-hidden="true" tabindex="-1"></a><span class="do">## The geneID column is ENTREZID</span></span>
<span id="cb202-8"><a href="#cb202-8" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(x, <span class="dv">3</span>)</span></code></pre></div>
<pre><code>##                ID         Description GeneRatio  BgRatio       pvalue
## hsa04110 hsa04110          Cell cycle      8/48 124/8087 5.499447e-07
## hsa04218 hsa04218 Cellular senescence      7/48 156/8087 3.292600e-05
## hsa04114 hsa04114      Oocyte meiosis      6/48 129/8087 1.039030e-04
##              p.adjust       qvalue                               geneID Count
## hsa04110 6.489348e-05 0.0000590467 8318/991/9133/890/983/4085/7272/1111     8
## hsa04218 1.942634e-03 0.0017676064    2305/4605/9133/890/983/51806/1111     7
## hsa04114 4.086853e-03 0.0037186351         991/9133/983/4085/51806/6790     6</code></pre>
<div class="sourceCode" id="cb204"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb204-1"><a href="#cb204-1" aria-hidden="true" tabindex="-1"></a>y <span class="ot">&lt;-</span> <span class="fu">setReadable</span>(x, <span class="at">OrgDb =</span> org.Hs.eg.db, <span class="at">keyType=</span><span class="st">&quot;ENTREZID&quot;</span>)</span>
<span id="cb204-2"><a href="#cb204-2" aria-hidden="true" tabindex="-1"></a><span class="do">## The geneID column is translated to symbol</span></span>
<span id="cb204-3"><a href="#cb204-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(y, <span class="dv">3</span>)</span></code></pre></div>
<pre><code>##                ID         Description GeneRatio  BgRatio       pvalue
## hsa04110 hsa04110          Cell cycle      8/48 124/8087 5.499447e-07
## hsa04218 hsa04218 Cellular senescence      7/48 156/8087 3.292600e-05
## hsa04114 hsa04114      Oocyte meiosis      6/48 129/8087 1.039030e-04
##              p.adjust       qvalue
## hsa04110 6.489348e-05 0.0000590467
## hsa04218 1.942634e-03 0.0017676064
## hsa04114 4.086853e-03 0.0037186351
##                                                 geneID Count
## hsa04110 CDC45/CDC20/CCNB2/CCNA2/CDK1/MAD2L1/TTK/CHEK1     8
## hsa04218     FOXM1/MYBL2/CCNB2/CCNA2/CDK1/CALML5/CHEK1     7
## hsa04114          CDC20/CCNB2/CDK1/MAD2L1/CALML5/AURKA     6</code></pre>
<p>For those functions that internally support <code>readable</code> parameter, user can also use <code>setReadable</code> for translating gene IDs.</p>
</div>
<div id="read-gmt" class="section level2" number="17.3">
<h2><span class="header-section-number">17.3</span> Parsing GMT files</h2>
<p>The GMT (Gene Matrix Transposed) file format is a tab delimited file format that is widely used to describe gene sets. Each row in the GMT format represents one gene set and each gene set is described by a name (or ID), a description and the genes in the gene set as illustrated in Figure <a href="#fig:gmtFormat">17.1</a>.</p>


<div class="figure"><span id="fig:gmtFormat"></span>
<img src="figures/Gmt_format_snapshot.gif" alt="The GMT format. Figure taken from https://software.broadinstitute.org/cancer/software/gsea/wiki/index.php/Data_formats." width="100%" />
<p class="caption">
Figure 17.1: <strong>The GMT format.</strong> Figure taken from <a href="https://software.broadinstitute.org/cancer/software/gsea/wiki/index.php/Data_formats" class="uri">https://software.broadinstitute.org/cancer/software/gsea/wiki/index.php/Data_formats</a>.
</p>
</div>
<p>The <a href="http://bioconductor.org/packages/clusterProfiler">clusterProfiler</a> package implemented a function, <code>read.gmt()</code>, to parse GMT file into a <code>data.frame</code>. The WikiPathway GMT file encodes information of <code>version</code>, <code>wpid</code> and <code>species</code> into the <code>Description</code> column. The <a href="http://bioconductor.org/packages/clusterProfiler">clusterProfiler</a> provides the <code>read.gmt.wp()</code> function to parse WikiPathway GMT file and supports parsing information that encoded in the <code>Description</code> column.</p>
<div class="sourceCode" id="cb206"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb206-1"><a href="#cb206-1" aria-hidden="true" tabindex="-1"></a>url <span class="ot">&lt;-</span> <span class="st">&quot;https://maayanlab.cloud/Enrichr/geneSetLibrary?mode=text&amp;libraryName=COVID-19_Related_Gene_Sets&quot;</span></span>
<span id="cb206-2"><a href="#cb206-2" aria-hidden="true" tabindex="-1"></a><span class="fu">download.file</span>(url, <span class="at">destfile =</span> <span class="st">&quot;COVID19_GeneSets.gmt&quot;</span>)</span>
<span id="cb206-3"><a href="#cb206-3" aria-hidden="true" tabindex="-1"></a>covid19_gs <span class="ot">&lt;-</span> <span class="fu">read.gmt</span>(<span class="st">&quot;COVID19_GeneSets.gmt&quot;</span>)</span>
<span id="cb206-4"><a href="#cb206-4" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(covid19_gs)</span></code></pre></div>
<pre><code>##                                     term    gene
## 1 COVID19-E protein host PPI from Krogan    BRD4
## 2 COVID19-E protein host PPI from Krogan    BRD2
## 3 COVID19-E protein host PPI from Krogan SLC44A2
## 4 COVID19-E protein host PPI from Krogan  ZC3H18
## 5 COVID19-E protein host PPI from Krogan   AP3B1
## 6 COVID19-E protein host PPI from Krogan   CWC27</code></pre>
<p>Gene set resources:</p>
<ul>
<li><a href="https://maayanlab.cloud/Enrichr/#stats" class="uri">https://maayanlab.cloud/Enrichr/#stats</a></li>
<li><a href="http://ge-lab.org/#/data" class="uri">http://ge-lab.org/#/data</a></li>
</ul>
<p>There are many gene sets available online. After parsing by the <code>read.gmt()</code> function, the data can be directly used to perform enrichment analysis using <code>enricher()</code> and <code>GSEA()</code> functions (see also <a href="#universal-api">Chapter 12</a>).</p>
</div>
<div id="go-utilities" class="section level2" number="17.4">
<h2><span class="header-section-number">17.4</span> GO utilities</h2>
<div id="drop-specific-go-terms-or-level" class="section level3" number="17.4.1">
<h3><span class="header-section-number">17.4.1</span> drop specific GO terms or level</h3>
<p>The <code>enrichGO()</code> function tests the whole GO corpus and enriched result may contains very general terms. With <code>dropGO</code> function, user can remove specific GO terms or GO level from results obtained from both <code>enrichGO()</code> and <code>compareCluster()</code>.</p>
</div>
<div id="test-go-at-sepcific-level" class="section level3" number="17.4.2">
<h3><span class="header-section-number">17.4.2</span> test GO at sepcific level</h3>
<p>The <code>enrichGO()</code> doesn‚Äôt contain parameter to restrict the test at specific GO level. Instead, we provide a function, <code>gofilter()</code>, to restrict the result at specific GO level. It works with results obtained from both <code>enrichGO()</code> and <code>compareCluster()</code>.</p>
</div>
<div id="reduce-redundancy-of-enriched-go-terms" class="section level3" number="17.4.3">
<h3><span class="header-section-number">17.4.3</span> reduce redundancy of enriched GO terms</h3>
<p>GO is organized in parent-child structure, thus a parent term can be overlapped with a large proportion with all its child terms. This can result in redundant findings. To solve this issue, <a href="http://bioconductor.org/packages/clusterProfiler">clusterProfiler</a> implements <a href="https://github.com/GuangchuangYu/clusterProfiler/issues/28"><code>simplify</code></a> method to reduce redundant GO terms from the outputs of <code>enrichGO()</code> and <code>gseGO()</code>. The function internally called <a href="http://bioconductor.org/packages/GOSemSim">GOSemSim</a> <span class="citation">(<a href="#ref-yu2010" role="doc-biblioref">Yu et al. 2010</a>)</span> to calculate semantic similarity among GO terms and remove those highly similar terms by keeping one representative term. The <code>simplify()</code> method apply <code>select_fun</code> (which can be a user defined function) to feature <code>by</code> to select one representative term from redundant terms (which have similarity higher than <code>cutoff</code>).</p>
<p>In Figure <a href="#fig:simplify-emapplot">17.2</a>A, we can found that there are many redundant terms form a highly condense network. After removing redundant terms using the <code>simplify()</code> method, the result is more clear to view the whole story.</p>


<div class="sourceCode" id="cb208"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb208-1"><a href="#cb208-1" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(geneList, <span class="at">package=</span><span class="st">&quot;DOSE&quot;</span>)</span>
<span id="cb208-2"><a href="#cb208-2" aria-hidden="true" tabindex="-1"></a>de <span class="ot">&lt;-</span> <span class="fu">names</span>(geneList)[<span class="fu">abs</span>(geneList) <span class="sc">&gt;</span> <span class="dv">2</span>]</span>
<span id="cb208-3"><a href="#cb208-3" aria-hidden="true" tabindex="-1"></a>bp <span class="ot">&lt;-</span> <span class="fu">enrichGO</span>(de, <span class="at">ont=</span><span class="st">&quot;BP&quot;</span>, <span class="at">OrgDb =</span> <span class="st">&#39;org.Hs.eg.db&#39;</span>)</span>
<span id="cb208-4"><a href="#cb208-4" aria-hidden="true" tabindex="-1"></a>bp <span class="ot">&lt;-</span> <span class="fu">pairwise_termsim</span>(bp)</span>
<span id="cb208-5"><a href="#cb208-5" aria-hidden="true" tabindex="-1"></a>bp2 <span class="ot">&lt;-</span> <span class="fu">simplify</span>(bp, <span class="at">cutoff=</span><span class="fl">0.7</span>, <span class="at">by=</span><span class="st">&quot;p.adjust&quot;</span>, <span class="at">select_fun=</span>min)</span>
<span id="cb208-6"><a href="#cb208-6" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">emapplot</span>(bp)</span>
<span id="cb208-7"><a href="#cb208-7" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">emapplot</span>(bp2)</span>
<span id="cb208-8"><a href="#cb208-8" aria-hidden="true" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(p1, p2, <span class="at">ncol=</span><span class="dv">2</span>, <span class="at">labels =</span> LETTERS[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>])</span></code></pre></div>
<div class="figure"><span id="fig:simplify-emapplot"></span>
<img src="biomedicalKnowledge_files/figure-html/simplify-emapplot-1.png" alt="Visualize enriched terms by EnrichmentMap using the emapplot() function. (A) original result. (B) simplify result." width="95%" />
<p class="caption">
Figure 17.2: <strong>Visualize enriched terms by EnrichmentMap using the <code>emapplot()</code> function.</strong> (A) original result. (B) simplify result.
</p>
</div>
<p>Alternatively, users can use slim version of GO and use the <code>enricher()</code> o r <code>gseGO()</code> functions to analyze.</p>

</div>
</div>
</div>



<div id="faq" class="section level1" number="18">
<h1><span class="header-section-number">A</span> Frequently asked questions</h1>
<div id="genelist" class="section level2" number="18.1">
<h2><span class="header-section-number">A.1</span> How to prepare your own geneList</h2>
<p>GSEA analysis requires a ranked gene list, which contains three features:</p>
<ul>
<li>numeric vector: fold change or other type of numerical variable</li>
<li>named vector: every number has a name, the corresponding gene ID</li>
<li>sorted vector: number should be sorted in decreasing order</li>
</ul>
<p>If you import your data from a <code>csv</code> file, the file should contains two columns, one for gene ID (no duplicated ID allowed) and another one for fold change. You can prepare your own <code>geneList</code> via the following command:</p>
<div class="sourceCode" id="cb209"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb209-1"><a href="#cb209-1" aria-hidden="true" tabindex="-1"></a>d <span class="ot">=</span> <span class="fu">read.csv</span>(your_csv_file)</span>
<span id="cb209-2"><a href="#cb209-2" aria-hidden="true" tabindex="-1"></a><span class="do">## assume 1st column is ID</span></span>
<span id="cb209-3"><a href="#cb209-3" aria-hidden="true" tabindex="-1"></a><span class="do">## 2nd column is FC</span></span>
<span id="cb209-4"><a href="#cb209-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb209-5"><a href="#cb209-5" aria-hidden="true" tabindex="-1"></a><span class="do">## feature 1: numeric vector</span></span>
<span id="cb209-6"><a href="#cb209-6" aria-hidden="true" tabindex="-1"></a>geneList <span class="ot">=</span> d[,<span class="dv">2</span>]</span>
<span id="cb209-7"><a href="#cb209-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb209-8"><a href="#cb209-8" aria-hidden="true" tabindex="-1"></a><span class="do">## feature 2: named vector</span></span>
<span id="cb209-9"><a href="#cb209-9" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(geneList) <span class="ot">=</span> <span class="fu">as.character</span>(d[,<span class="dv">1</span>])</span>
<span id="cb209-10"><a href="#cb209-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb209-11"><a href="#cb209-11" aria-hidden="true" tabindex="-1"></a><span class="do">## feature 3: decreasing orde</span></span>
<span id="cb209-12"><a href="#cb209-12" aria-hidden="true" tabindex="-1"></a>geneList <span class="ot">=</span> <span class="fu">sort</span>(geneList, <span class="at">decreasing =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<!--


# DAVID functional analysis

[clusterProfiler](https://www.bioconductor.org/packages/clusterProfiler) provides enrichment and GSEA analysis with GO, KEGG, DO and Reactome pathway supported internally, some user may prefer GO and KEGG analysis with DAVID[@huang_david_2007] and still attracted by the visualization methods provided by [clusterProfiler](https://www.bioconductor.org/packages/clusterProfiler)[@paranjpe_genome_wid_2013]. To bridge the gap between DAVID and clusterProfiler, we implemented `enrichDAVID`. This function query enrichment analysis result from DAVID webserver via [RDAVIDWebService](https://www.bioconductor.org/packages/RDAVIDWebService)[@fresno_rdavidwebservice_2013] and stored the result as an `enrichResult` instance, so that we can use all the visualization functions in [clusterProfiler](https://www.bioconductor.org/packages/clusterProfiler) to visualize DAVID results. `enrichDAVID` is fully compatible with `compareCluster` function and comparing enrichment results from different gene clusters is now available with DAVID.


```r
library(clusterProfiler)
data(geneList, package="DOSE")
gene = names(geneList)[abs(geneList) > 2]
david <- enrichDAVID(gene = gene,
                     idType = "ENTREZ_GENE_ID",
                     listType = "Gene",
                     annotation = "KEGG_PATHWAY",
                     david.user = "clusterProfiler@hku.hk")
```

DAVID Web Service has the following limitations:

+ A job with more than 3000 genes to generate gene or term cluster report will not be handled by DAVID due to resource limit.
+ No more than 200 jobs in a day from one user or computer.
+ DAVID Team reserves right to suspend any improper uses of the web service without notice.

For more details, please refer to [http://david.abcc.ncifcrf.gov/content.jsp?file=WS.html](http://david.abcc.ncifcrf.gov/content.jsp?file=WS.html).

As user has limited usage, please [register](http://david.abcc.ncifcrf.gov/webservice/register.htm) and use your own user account to run `enrichDAVID`.


-->
</div>
<div id="no-gene-can-be-mapped" class="section level2" number="18.2">
<h2><span class="header-section-number">A.2</span> No gene can be mapped</h2>
<ul>
<li><a href="https://www.biostars.org/p/431270/" class="uri">https://www.biostars.org/p/431270/</a></li>
<li><a href="https://github.com/YuLab-SMU/clusterProfiler/issues/280" class="uri">https://github.com/YuLab-SMU/clusterProfiler/issues/280</a></li>
</ul>
</div>
<div id="showing-specific-pathways" class="section level2" number="18.3">
<h2><span class="header-section-number">A.3</span> Showing specific pathways</h2>
<p>By default, all the visualization methods provided by <a href="http://bioconductor.org/packages/enrichplot">enrichplot</a> display most significant pathways. If users are interested to show some specific pathways (e.g.¬†excluding some unimportant pathways among the top categories), users can pass a vector of selected pathways to the <code>showCategory</code> parameter in <code>dotplot()</code>, <code>barplot()</code>, <code>cnetplot()</code> and <code>emapplot()</code> etc.</p>


<div class="sourceCode" id="cb210"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb210-1"><a href="#cb210-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DOSE)</span>
<span id="cb210-2"><a href="#cb210-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(enrichplot)</span>
<span id="cb210-3"><a href="#cb210-3" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(geneList)</span>
<span id="cb210-4"><a href="#cb210-4" aria-hidden="true" tabindex="-1"></a>de <span class="ot">&lt;-</span> <span class="fu">names</span>(geneList)[<span class="dv">1</span><span class="sc">:</span><span class="dv">100</span>]</span>
<span id="cb210-5"><a href="#cb210-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb210-6"><a href="#cb210-6" aria-hidden="true" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="fu">enrichDO</span>(de)</span>
<span id="cb210-7"><a href="#cb210-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb210-8"><a href="#cb210-8" aria-hidden="true" tabindex="-1"></a><span class="do">## show top 10 most significant pathways and want to exclude the second one</span></span>
<span id="cb210-9"><a href="#cb210-9" aria-hidden="true" tabindex="-1"></a><span class="do">## dotplot(x, showCategory = x$Description[1:10][-2])</span></span>
<span id="cb210-10"><a href="#cb210-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb210-11"><a href="#cb210-11" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">2020-10-27</span>)</span>
<span id="cb210-12"><a href="#cb210-12" aria-hidden="true" tabindex="-1"></a>selected_pathways <span class="ot">&lt;-</span> <span class="fu">sample</span>(x<span class="sc">$</span>Description, <span class="dv">10</span>)</span>
<span id="cb210-13"><a href="#cb210-13" aria-hidden="true" tabindex="-1"></a>selected_pathways</span></code></pre></div>
<pre><code>##  [1] &quot;in situ carcinoma&quot;                              
##  [2] &quot;hypersensitivity reaction type IV disease&quot;      
##  [3] &quot;lymph node disease&quot;                             
##  [4] &quot;esophageal carcinoma&quot;                           
##  [5] &quot;atopic dermatitis&quot;                              
##  [6] &quot;pulmonary sarcoidosis&quot;                          
##  [7] &quot;esophageal cancer&quot;                              
##  [8] &quot;Human immunodeficiency virus infectious disease&quot;
##  [9] &quot;ovarian cancer&quot;                                 
## [10] &quot;breast carcinoma&quot;</code></pre>
<div class="sourceCode" id="cb212"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb212-1"><a href="#cb212-1" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">dotplot</span>(x, <span class="at">showCategory =</span> <span class="dv">10</span>, <span class="at">font.size=</span><span class="dv">14</span>)</span>
<span id="cb212-2"><a href="#cb212-2" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">dotplot</span>(x, <span class="at">showCategory =</span> selected_pathways, <span class="at">font.size=</span><span class="dv">14</span>)</span>
<span id="cb212-3"><a href="#cb212-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-4"><a href="#cb212-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-5"><a href="#cb212-5" aria-hidden="true" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(p1, p2, <span class="at">labels=</span>LETTERS[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>])</span></code></pre></div>
<div class="figure"><span id="fig:selectedPaths"></span>
<img src="biomedicalKnowledge_files/figure-html/selectedPaths-1.png" alt="Showing specific pathways. Top ten most significant pathways (A), selected ten pathways (B)." width="95%" />
<p class="caption">
Figure A.1: <strong>Showing specific pathways.</strong> Top ten most significant pathways (A), selected ten pathways (B).
</p>
</div>
<p>Note: Another solution is using the <code>filter</code> verb to extract a subset of the result as described in <a href="#clusterProfiler-dplyr">Chapter 16</a>.</p>
</div>
<div id="how-to-extract-genes-of-a-specific-termpathway" class="section level2" number="18.4">
<h2><span class="header-section-number">A.4</span> How to extract genes of a specific term/pathway</h2>
<div class="sourceCode" id="cb213"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb213-1"><a href="#cb213-1" aria-hidden="true" tabindex="-1"></a>id <span class="ot">&lt;-</span> x<span class="sc">$</span>ID[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>]</span>
<span id="cb213-2"><a href="#cb213-2" aria-hidden="true" tabindex="-1"></a>id</span></code></pre></div>
<pre><code>## [1] &quot;DOID:0060071&quot; &quot;DOID:5295&quot;    &quot;DOID:8719&quot;</code></pre>
<div class="sourceCode" id="cb215"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb215-1"><a href="#cb215-1" aria-hidden="true" tabindex="-1"></a>x[[id[<span class="dv">1</span>]]]</span></code></pre></div>
<pre><code>## [1] &quot;6280&quot;  &quot;6278&quot;  &quot;10232&quot; &quot;332&quot;   &quot;4321&quot;</code></pre>
<div class="sourceCode" id="cb217"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb217-1"><a href="#cb217-1" aria-hidden="true" tabindex="-1"></a><span class="fu">geneInCategory</span>(x)[id]</span></code></pre></div>
<pre><code>## $`DOID:0060071`
## [1] &quot;6280&quot;  &quot;6278&quot;  &quot;10232&quot; &quot;332&quot;   &quot;4321&quot; 
## 
## $`DOID:5295`
## [1] &quot;4312&quot;  &quot;6279&quot;  &quot;3627&quot;  &quot;10563&quot; &quot;4283&quot;  &quot;890&quot;   &quot;366&quot;   &quot;4902&quot;  &quot;3620&quot; 
## 
## $`DOID:8719`
## [1] &quot;6280&quot;  &quot;6278&quot;  &quot;10232&quot; &quot;332&quot;</code></pre>

</div>
</div>
<div id="references" class="section level1 unnumbered">
<h1>References</h1>

<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-omer_ncg" class="csl-entry">
A., Omer, Giovanni M. D., Thanos P. M., and Francesca D. C. 2016. <span>‚ÄúNCG 5.0: Updates of a Manually Curated Repository of Cancer Genes and Associated Properties from Cancer Mutational Screenings.‚Äù</span> <em>Nucleic Acids Research</em> 44 (D1): D992‚Äì99. <a href="https://doi.org/10.1093/nar/gkv1123">https://doi.org/10.1093/nar/gkv1123</a>.
</div>
<div id="ref-boyle2004" class="csl-entry">
Boyle, Elizabeth I, Shuai Weng, Jeremy Gollub, Heng Jin, David Botstein, J Michael Cherry, and Gavin Sherlock. 2004. <span>‚Äú<span class="nocase">GO::TermFinder‚Äìopen</span> Source Software for Accessing Gene Ontology Information and Finding Significantly Enriched Gene Ontology Terms Associated with a List of Genes.‚Äù</span> <em>Bioinformatics (Oxford, England)</em> 20 (18): 3710‚Äì15. <a href="https://doi.org/10.1093/bioinformatics/bth456">https://doi.org/10.1093/bioinformatics/bth456</a>.
</div>
<div id="ref-Du15062009" class="csl-entry">
Du, Pan, Gang Feng, Jared Flatow, Jie Song, Michelle Holko, Warren A. Kibbe, and Simon M. Lin. 2009. <span>‚ÄúFrom Disease Ontology to Disease-Ontology Lite: Statistical Methods to Adapt a General-Purpose Ontology for the Test of Gene-Ontology Associations.‚Äù</span> <em>Bioinformatics</em> 25 (12): i63‚Äì68. <a href="https://doi.org/10.1093/bioinformatics/btp193">https://doi.org/10.1093/bioinformatics/btp193</a>.
</div>
<div id="ref-janet_disgenet" class="csl-entry">
Janet, P., N√∫ria Q. R., √Älex B., Jordi D. P., Anna B. M., Martin B., Ferran S., and Laura I. F. 2015. <span>‚ÄúDisGeNET: A Discovery Platform for the Dynamical Exploration of Human Diseases and Their Genes.‚Äù</span> <em>Database</em> 2015 (March): bav028. <a href="https://doi.org/10.1093/database/bav028">https://doi.org/10.1093/database/bav028</a>.
</div>
<div id="ref-jiang_semantic_1997" class="csl-entry">
Jiang, Jay J., and David W. Conrath. 1997. <span>‚ÄúSemantic Similarity Based on Corpus Statistics and Lexical Taxonomy.‚Äù</span> <em>Proceedings of 10th International Conference on Research In Computational Linguistics</em>. <a href="http://www.citebase.org/abstract?id=oai:arXiv.org:cmp-lg/9709008">http://www.citebase.org/abstract?id=oai:arXiv.org:cmp-lg/9709008</a>.
</div>
<div id="ref-korotkevich_fast_2019" class="csl-entry">
Korotkevich, Gennady, Vladimir Sukhov, and Alexey Sergushichev. 2019. <span>‚ÄúFast Gene Set Enrichment Analysis.‚Äù</span> <em><span>bioRxiv</span></em>, October, 060012. <a href="https://doi.org/10.1101/060012">https://doi.org/10.1101/060012</a>.
</div>
<div id="ref-lin_information-theoretic_1998" class="csl-entry">
Lin, Dekang. 1998. <span>‚ÄúAn <span>Information-Theoretic</span> Definition of Similarity.‚Äù</span> <em>In Proceedings of the 15th International Conference on Machine Learning</em>, 296‚Äî304. <a href="https://doi.org/10.1.1.55.1832">https://doi.org/10.1.1.55.1832</a>.
</div>
<div id="ref-luo_pathview" class="csl-entry">
Luo, Weijun, and Cory Brouwer. 2013. <span>‚ÄúPathview: An <span>R/Bioconductor</span> Package for Pathway-Based Data Integration and Visualization.‚Äù</span> <em>Bioinformatics</em> 29 (July): 1830‚Äì31. <a href="https://doi.org/10.1093/bioinformatics/btt285">https://doi.org/10.1093/bioinformatics/btt285</a>.
</div>
<div id="ref-philip_semantic_1999" class="csl-entry">
Philip, Resnik. 1999. <span>‚ÄúSemantic Similarity in a Taxonomy: An <span>Information-Based</span> Measure and Its Application to Problems of Ambiguity in Natural Language.‚Äù</span> <em>Journal of Artificial Intelligence Research</em> 11: 95‚Äì130.
</div>
<div id="ref-schlicker_new_2006" class="csl-entry">
Schlicker, Andreas, Francisco S Domingues, J√∂rg Rahnenf√ºhrer, and Thomas Lengauer. 2006. <span>‚ÄúA New Measure for Functional Similarity of Gene Products Based on Gene Ontology.‚Äù</span> <em>BMC Bioinformatics</em> 7: 302. <a href="https://doi.org/1471-2105-7-302">https://doi.org/1471-2105-7-302</a>.
</div>
<div id="ref-schmidt2008" class="csl-entry">
Schmidt, Marcus, Daniel B?hm, Christian von T?rne, Eric Steiner, Alexander Puhl, Henryk Pilch, Hans-Anton Lehr, Jan G. Hengstler, Heinz K?lbl, and Mathias Gehrmann. 2008. <span>‚ÄúThe Humoral Immune System Has a Key Prognostic Impact in Node-Negative Breast Cancer.‚Äù</span> <em>Cancer Research</em> 68 (13): 5405‚Äì13. <a href="https://doi.org/10.1158/0008-5472.CAN-07-5206">https://doi.org/10.1158/0008-5472.CAN-07-5206</a>.
</div>
<div id="ref-schriml_disease_2011" class="csl-entry">
Schriml, L. M., C. Arze, S. Nadendla, Y.-W. W. Chang, M. Mazaitis, V. Felix, G. Feng, and W. A. Kibbe. 2011. <span>‚ÄúDisease Ontology: A Backbone for Disease Semantic Integration.‚Äù</span> <em>Nucleic Acids Research</em> 40 (D1): D940‚Äì46. <a href="https://doi.org/10.1093/nar/gkr972">https://doi.org/10.1093/nar/gkr972</a>.
</div>
<div id="ref-subramanian_gene_2005" class="csl-entry">
Subramanian, Aravind, Pablo Tamayo, Vamsi K. Mootha, Sayan Mukherjee, Benjamin L. Ebert, Michael A. Gillette, Amanda Paulovich, et al. 2005. <span>‚ÄúGene Set Enrichment Analysis: A Knowledge-Based Approach for Interpreting Genome-Wide Expression Profiles.‚Äù</span> <em>Proceedings of the National Academy of Sciences of the United States of America</em> 102 (43): 15545‚Äì50. <a href="https://doi.org/10.1073/pnas.0506580102">https://doi.org/10.1073/pnas.0506580102</a>.
</div>
<div id="ref-wang_new_2007" class="csl-entry">
Wang, James Z, Zhidian Du, Rapeeporn Payattakool, Philip S Yu, and Chin-Fu Chen. 2007. <span>‚ÄúA New Method to Measure the Semantic Similarity of GO Terms.‚Äù</span> <em>Bioinformatics (Oxford, England)</em> 23 (May): 1274‚Äì81. <a href="https://doi.org/btm087">https://doi.org/btm087</a>.
</div>
<div id="ref-wu_clusterprofiler_2021" class="csl-entry">
Wu, Tianzhi, Erqiang Hu, Shuangbin Xu, Meijun Chen, Pingfan Guo, Zehan Dai, Tingze Feng, et al. 2021. <span>‚Äú<span class="nocase">clusterProfiler</span> 4.0: A Universal Enrichment Tool for Interpreting Omics Data.‚Äù</span> <em>The Innovation</em> 2 (3): 100141. <a href="https://doi.org/10.1016/j.xinn.2021.100141">https://doi.org/10.1016/j.xinn.2021.100141</a>.
</div>
<div id="ref-yu_meshes_2018" class="csl-entry">
Yu, Guangchuang. 2018. <span>‚ÄúUsing Meshes for <span>MeSH</span> Term Enrichment and Semantic Analyses.‚Äù</span> <em>Bioinformatics</em> 34 (21): 3766‚Äì67. <a href="https://doi.org/10.1093/bioinformatics/bty410">https://doi.org/10.1093/bioinformatics/bty410</a>.
</div>
<div id="ref-yu_gosemsim_2020" class="csl-entry">
‚Äî‚Äî‚Äî. 2020. <span>‚ÄúGene Ontology Semantic Similarity Analysis Using <span>GOSemSim</span>.‚Äù</span> <em>Methods in Molecular Biology (Clifton, N.J.)</em> 2117: 207‚Äì15. <a href="https://doi.org/10.1007/978-1-0716-0301-7_11">https://doi.org/10.1007/978-1-0716-0301-7_11</a>.
</div>
<div id="ref-yu_reactomepa_2016" class="csl-entry">
Yu, Guangchuang, and Qing-Yu He. 2016. <span>‚Äú<span>ReactomePA</span>: An r/Bioconductor Package for Reactome Pathway Analysis and Visualization.‚Äù</span> <em>Molecular <span>BioSystems</span></em> 12 (2): 477‚Äì79. <a href="https://doi.org/10.1039/C5MB00663E">https://doi.org/10.1039/C5MB00663E</a>.
</div>
<div id="ref-yu2010" class="csl-entry">
Yu, Guangchuang, Fei Li, Yide Qin, Xiaochen Bo, Yibo Wu, and Shengqi Wang. 2010. <span>‚ÄúGOSemSim: An r Package for Measuring Semantic Similarity Among GO Terms and Gene Products.‚Äù</span> <em>Bioinformatics</em> 26 (7): 976‚Äì78. <a href="https://doi.org/10.1093/bioinformatics/btq064">https://doi.org/10.1093/bioinformatics/btq064</a>.
</div>
<div id="ref-yu2012" class="csl-entry">
Yu, Guangchuang, Le-Gen Wang, Yanyan Han, and Qing-Yu He. 2012. <span>‚ÄúclusterProfiler: An r Package for Comparing Biological Themes Among Gene Clusters.‚Äù</span> <em>OMICS: A Journal of Integrative Biology</em> 16 (5): 284‚Äì87. <a href="https://doi.org/10.1089/omi.2011.0118">https://doi.org/10.1089/omi.2011.0118</a>.
</div>
<div id="ref-yu_chipseeker_2015" class="csl-entry">
Yu, Guangchuang, Li-Gen Wang, and Qing-Yu He. 2015. <span>‚ÄúChIPseeker: An r/Bioconductor Package for ChIP Peak Annotation, Comparison and Visualization.‚Äù</span> <em>Bioinformatics</em> 31 (14): 2382‚Äì83. <a href="https://doi.org/10.1093/bioinformatics/btv145">https://doi.org/10.1093/bioinformatics/btv145</a>.
</div>
<div id="ref-yu_dose_2015" class="csl-entry">
Yu, Guangchuang, Li-Gen Wang, Guang-Rong Yan, and Qing-Yu He. 2015. <span>‚Äú<span>DOSE</span>: An r/Bioconductor Package for Disease Ontology Semantic and Enrichment Analysis.‚Äù</span> <em>Bioinformatics</em> 31 (4): 608‚Äì9. <a href="https://doi.org/10.1093/bioinformatics/btu684">https://doi.org/10.1093/bioinformatics/btu684</a>.
</div>
</div>
</div>
<div class="footnotes">
<hr />
<ol>
<li id="fn1"><p>example adopted from <a href="https://guangchuangyu.github.io/2012/04/enrichment-analysis/" class="uri">https://guangchuangyu.github.io/2012/04/enrichment-analysis/</a><a href="#fnref1" class="footnote-back">‚Ü©Ô∏é</a></p></li>
</ol>
</div>
            </section>

          </div>
        </div>
      </div>


    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
