<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 6 GO enrichment analysis | Biomedical Knowledge Mining using GOSemSim and clusterProfiler</title>
<meta name="author" content="Guangchuang Yu">
<meta name="description" content="GO comprises three orthogonal ontologies, i.e.Â molecular function (MF), biological process (BP), and cellular component (CC).  6.1 Supported organisms GO analyses (groupGO(), enrichGO() and...">
<meta name="generator" content="bookdown 0.26 with bs4_book()">
<meta property="og:title" content="Chapter 6 GO enrichment analysis | Biomedical Knowledge Mining using GOSemSim and clusterProfiler">
<meta property="og:type" content="book">
<meta property="og:description" content="GO comprises three orthogonal ontologies, i.e.Â molecular function (MF), biological process (BP), and cellular component (CC).  6.1 Supported organisms GO analyses (groupGO(), enrichGO() and...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Chapter 6 GO enrichment analysis | Biomedical Knowledge Mining using GOSemSim and clusterProfiler">
<meta name="twitter:description" content="GO comprises three orthogonal ontologies, i.e.Â molecular function (MF), biological process (BP), and cellular component (CC).  6.1 Supported organisms GO analyses (groupGO(), enrichGO() and...">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.3.1/transition.js"></script><script src="libs/bs3compat-0.3.1/tabs.js"></script><script src="libs/bs3compat-0.3.1/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<style type="text/css">
    /* Used with Pandoc 2.11+ new --citeproc when CSL is used */
    div.csl-bib-body { }
    div.csl-entry {
      clear: both;
        }
    .hanging div.csl-entry {
      margin-left:2em;
      text-indent:-2em;
    }
    div.csl-left-margin {
      min-width:2em;
      float:left;
    }
    div.csl-right-inline {
      margin-left:2em;
      padding-left:1em;
    }
    div.csl-indent {
      margin-left: 2em;
    }
  </style>
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">Biomedical Knowledge Mining using GOSemSim and clusterProfiler</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html">ðŸ“– Introduction</a></li>
<li class="book-part">Part I: Semantic Similarity Analysis</li>
<li><a class="" href="semantic-similarity-overview.html"><span class="header-section-number">1</span> Overview of semantic similarity analysis</a></li>
<li><a class="" href="GOSemSim.html"><span class="header-section-number">2</span> GO semantic similarity analysis</a></li>
<li><a class="" href="DOSE-semantic-similarity.html"><span class="header-section-number">3</span> DO semantic similarity analysis</a></li>
<li><a class="" href="meshes-semantic-similarity.html"><span class="header-section-number">4</span> MeSH semantic similarity analysis</a></li>
<li class="book-part">Part II: Enrichment analysis</li>
<li><a class="" href="enrichment-overview.html"><span class="header-section-number">5</span> Overview of enrichment analysis</a></li>
<li><a class="active" href="clusterprofiler-go.html"><span class="header-section-number">6</span> GO enrichment analysis</a></li>
<li><a class="" href="clusterprofiler-kegg.html"><span class="header-section-number">7</span> KEGG enrichment analysis</a></li>
<li><a class="" href="wikipathways-analysis.html"><span class="header-section-number">8</span> WikiPathways analysis</a></li>
<li><a class="" href="reactomepa.html"><span class="header-section-number">9</span> Reactome enrichment analysis</a></li>
<li><a class="" href="dose-enrichment.html"><span class="header-section-number">10</span> Disease enrichment analysis</a></li>
<li><a class="" href="meshes-enrichment.html"><span class="header-section-number">11</span> MeSH enrichment analysis</a></li>
<li><a class="" href="universal-api.html"><span class="header-section-number">12</span> Universal enrichment analysis</a></li>
<li><a class="" href="chipseeker-enrichment.html"><span class="header-section-number">13</span> Functional enrichment analysis of genomic coordinations</a></li>
<li><a class="" href="clusterprofiler-comparecluster.html"><span class="header-section-number">14</span> Biological theme comparison</a></li>
<li><a class="" href="enrichplot.html"><span class="header-section-number">15</span> Visualization of functional enrichment result</a></li>
<li class="book-part">Part III: Miscellaneous topics</li>
<li><a class="" href="clusterProfiler-dplyr.html"><span class="header-section-number">16</span> dplyr verbs for manipulating enrichment result</a></li>
<li><a class="" href="useful-utilities.html"><span class="header-section-number">17</span> Useful utilities</a></li>
<li class="book-part">Appendix</li>
<li><a class="" href="faq.html"><span class="header-section-number">A</span> Frequently asked questions</a></li>
<li><a class="" href="references.html">References</a></li>
</ul>

        <div class="book-extra">
          
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="clusterprofiler-go" class="section level1" number="6">
<h1>
<span class="header-section-number">6</span> GO enrichment analysis<a class="anchor" aria-label="anchor" href="#clusterprofiler-go"><i class="fas fa-link"></i></a>
</h1>
<p>GO comprises three orthogonal ontologies, i.e.Â molecular function (MF), biological
process (BP), and cellular component (CC).</p>
<div id="clusterProfiler-go-supported-organisms" class="section level2" number="6.1">
<h2>
<span class="header-section-number">6.1</span> Supported organisms<a class="anchor" aria-label="anchor" href="#clusterProfiler-go-supported-organisms"><i class="fas fa-link"></i></a>
</h2>
<p>GO analyses (<code><a href="https://rdrr.io/pkg/clusterProfiler/man/groupGO.html">groupGO()</a></code>, <code><a href="https://rdrr.io/pkg/clusterProfiler/man/enrichGO.html">enrichGO()</a></code> and <code><a href="https://rdrr.io/pkg/clusterProfiler/man/gseGO.html">gseGO()</a></code>) support organisms that have an <code>OrgDb</code> object available (see also <a href="GOSemSim.html#gosemsim-supported-organisms">session 2.2</a>).</p>
<p>If a user has GO annotation data (in a <code>data.frame</code> format with the first column as gene ID and the second column as GO ID), they can use the <code><a href="https://rdrr.io/pkg/clusterProfiler/man/enricher.html">enricher()</a></code> and <code><a href="https://rdrr.io/pkg/clusterProfiler/man/GSEA.html">GSEA()</a></code> functions to perform an over-representation test and gene set enrichment analysis.</p>
<p>If the genes are annotated by direction annotation, they should also be annotated by their ancestor GO nodes (indirect annotation). If a user only has direct annotation, they can pass their annotation to the <code>buildGOmap</code> function, which will infer indirect annotation and generate a <code>data.frame</code> that is suitable for both <code><a href="https://rdrr.io/pkg/clusterProfiler/man/enricher.html">enricher()</a></code> and <code><a href="https://rdrr.io/pkg/clusterProfiler/man/GSEA.html">GSEA()</a></code>.</p>
</div>
<div id="go-classification" class="section level2" number="6.2">
<h2>
<span class="header-section-number">6.2</span> GO classification<a class="anchor" aria-label="anchor" href="#go-classification"><i class="fas fa-link"></i></a>
</h2>
<p>In <a href="http://bioconductor.org/packages/clusterProfiler">clusterProfiler</a>, the <code><a href="https://rdrr.io/pkg/clusterProfiler/man/groupGO.html">groupGO()</a></code> function is designed for gene classification based on GO distribution at a specific level. Here we use the dataset <code>geneList</code> provided by <a href="http://bioconductor.org/packages/DOSE">DOSE</a>.</p>
<div class="sourceCode" id="cb60"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://yulab-smu.top/biomedical-knowledge-mining-book/">clusterProfiler</a></span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">geneList</span>, package<span class="op">=</span><span class="st">"DOSE"</span><span class="op">)</span>
<span class="va">gene</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">geneList</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">abs</a></span><span class="op">(</span><span class="va">geneList</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">2</span><span class="op">]</span>

<span class="co"># Entrez gene ID</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">gene</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] "4312"  "8318"  "10874" "55143" "55388" "991"</code></pre>
<div class="sourceCode" id="cb62"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ggo</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/groupGO.html">groupGO</a></span><span class="op">(</span>gene     <span class="op">=</span> <span class="va">gene</span>,
               OrgDb    <span class="op">=</span> <span class="va">org.Hs.eg.db</span>,
               ont      <span class="op">=</span> <span class="st">"CC"</span>,
               level    <span class="op">=</span> <span class="fl">3</span>,
               readable <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">ggo</span><span class="op">)</span></code></pre></div>
<pre><code>##                    ID                Description Count GeneRatio geneID
## GO:0000133 GO:0000133                 polarisome     0     0/207       
## GO:0000408 GO:0000408          EKC/KEOPS complex     0     0/207       
## GO:0000417 GO:0000417                HIR complex     0     0/207       
## GO:0000444 GO:0000444    MIS12/MIND type complex     0     0/207       
## GO:0000808 GO:0000808 origin recognition complex     0     0/207       
## GO:0000930 GO:0000930      gamma-tubulin complex     0     0/207</code></pre>
<p>The <code>gene</code> parameter is a vector of gene IDs (can be any ID type that is supported by the corresponding <code>OrgDb</code>, see also <a href="useful-utilities.html#id-convert">session 16.1</a>). If <code>readable</code> is set to <code>TRUE</code>, the input gene IDs will be converted to gene symbols.</p>
</div>
<div id="clusterprofiler-go-ora" class="section level2" number="6.3">
<h2>
<span class="header-section-number">6.3</span> GO over-representation analysis<a class="anchor" aria-label="anchor" href="#clusterprofiler-go-ora"><i class="fas fa-link"></i></a>
</h2>
<p>The <a href="http://bioconductor.org/packages/clusterProfiler">clusterProfiler</a> package implements <code><a href="https://rdrr.io/pkg/clusterProfiler/man/enrichGO.html">enrichGO()</a></code> for gene ontology over-representation test.</p>
<div class="sourceCode" id="cb64"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ego</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/enrichGO.html">enrichGO</a></span><span class="op">(</span>gene          <span class="op">=</span> <span class="va">gene</span>,
                universe      <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">geneList</span><span class="op">)</span>,
                OrgDb         <span class="op">=</span> <span class="va">org.Hs.eg.db</span>,
                ont           <span class="op">=</span> <span class="st">"CC"</span>,
                pAdjustMethod <span class="op">=</span> <span class="st">"BH"</span>,
                pvalueCutoff  <span class="op">=</span> <span class="fl">0.01</span>,
                qvalueCutoff  <span class="op">=</span> <span class="fl">0.05</span>,
        readable      <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">ego</span><span class="op">)</span></code></pre></div>
<pre><code>##                    ID                              Description GeneRatio
## GO:0005819 GO:0005819                                  spindle    26/201
## GO:0000779 GO:0000779 condensed chromosome, centromeric region    16/201
## GO:0072686 GO:0072686                          mitotic spindle    17/201
## GO:0000775 GO:0000775           chromosome, centromeric region    18/201
## GO:0098687 GO:0098687                       chromosomal region    23/201
## GO:0000776 GO:0000776                              kinetochore    15/201
##              BgRatio       pvalue     p.adjust       qvalue
## GO:0005819 306/11853 1.072029e-11 3.151766e-09 2.888837e-09
## GO:0000779 114/11853 7.709944e-11 8.659125e-09 7.936756e-09
## GO:0072686 133/11853 8.835841e-11 8.659125e-09 7.936756e-09
## GO:0000775 158/11853 1.684987e-10 1.179661e-08 1.081250e-08
## GO:0098687 272/11853 2.006225e-10 1.179661e-08 1.081250e-08
## GO:0000776 106/11853 2.733425e-10 1.339378e-08 1.227644e-08
##                                                                                                                                                               geneID
## GO:0005819 CDCA8/CDC20/KIF23/CENPE/ASPM/DLGAP5/SKA1/NUSAP1/TPX2/TACC3/NEK2/CDK1/MAD2L1/KIF18A/BIRC5/KIF11/TRAT1/TTK/AURKB/PRC1/KIFC1/KIF18B/KIF20A/AURKA/CCNB1/KIF4A
## GO:0000779                                                             CENPE/NDC80/HJURP/SKA1/NEK2/CENPM/CENPN/ERCC6L/MAD2L1/KIF18A/CDT1/BIRC5/TTK/NCAPG/AURKB/CCNB1
## GO:0072686                                                      KIF23/CENPE/ASPM/SKA1/NUSAP1/TPX2/TACC3/CDK1/MAD2L1/KIF18A/KIF11/TRAT1/AURKB/PRC1/KIFC1/KIF18B/AURKA
## GO:0000775                                                 CDCA8/CENPE/NDC80/TOP2A/HJURP/SKA1/NEK2/CENPM/CENPN/ERCC6L/MAD2L1/KIF18A/CDT1/BIRC5/TTK/NCAPG/AURKB/CCNB1
## GO:0098687                   CDCA8/CENPE/NDC80/TOP2A/HJURP/SKA1/NEK2/CENPM/RAD51AP1/CENPN/CDK1/ERCC6L/MAD2L1/KIF18A/CDT1/BIRC5/EZH2/TTK/NCAPG/AURKB/CHEK1/CCNB1/MCM5
## GO:0000776                                                                   CENPE/NDC80/HJURP/SKA1/NEK2/CENPM/CENPN/ERCC6L/MAD2L1/KIF18A/CDT1/BIRC5/TTK/AURKB/CCNB1
##            Count
## GO:0005819    26
## GO:0000779    16
## GO:0072686    17
## GO:0000775    18
## GO:0098687    23
## GO:0000776    15</code></pre>
<p>Any gene ID type that is supported in <code>OrgDb</code> can be directly used in GO analyses. Users need to specify the <code>keyType</code> parameter to specify the input gene ID type.</p>
<div class="sourceCode" id="cb66"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gene.df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/bitr.html">bitr</a></span><span class="op">(</span><span class="va">gene</span>, fromType <span class="op">=</span> <span class="st">"ENTREZID"</span>,
        toType <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"ENSEMBL"</span>, <span class="st">"SYMBOL"</span><span class="op">)</span>,
        OrgDb <span class="op">=</span> <span class="va">org.Hs.eg.db</span><span class="op">)</span>

<span class="va">ego2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/enrichGO.html">enrichGO</a></span><span class="op">(</span>gene         <span class="op">=</span> <span class="va">gene.df</span><span class="op">$</span><span class="va">ENSEMBL</span>,
                OrgDb         <span class="op">=</span> <span class="va">org.Hs.eg.db</span>,
                keyType       <span class="op">=</span> <span class="st">'ENSEMBL'</span>,
                ont           <span class="op">=</span> <span class="st">"CC"</span>,
                pAdjustMethod <span class="op">=</span> <span class="st">"BH"</span>,
                pvalueCutoff  <span class="op">=</span> <span class="fl">0.01</span>,
                qvalueCutoff  <span class="op">=</span> <span class="fl">0.05</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">ego2</span>, <span class="fl">3</span><span class="op">)</span>                </code></pre></div>
<pre><code>##                    ID                              Description GeneRatio
## GO:0005819 GO:0005819                                  spindle    30/233
## GO:0072686 GO:0072686                          mitotic spindle    21/233
## GO:0000779 GO:0000779 condensed chromosome, centromeric region    17/233
##              BgRatio       pvalue     p.adjust       qvalue
## GO:0005819 422/21916 1.954166e-16 5.413039e-14 4.175744e-14
## GO:0072686 179/21916 3.911063e-16 5.416822e-14 4.178662e-14
## GO:0000779 154/21916 7.525516e-13 6.948560e-11 5.360280e-11
##                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     geneID
## GO:0005819 ENSG00000134690/ENSG00000117399/ENSG00000137807/ENSG00000138778/ENSG00000066279/ENSG00000126787/ENSG00000154839/ENSG00000262634/ENSG00000137804/ENSG00000088325/ENSG00000013810/ENSG00000117650/ENSG00000170312/ENSG00000164109/ENSG00000121621/ENSG00000089685/ENSG00000138160/ENSG00000163519/ENSG00000112742/ENSG00000178999/ENSG00000198901/ENSG00000237649/ENSG00000233450/ENSG00000056678/ENSG00000204197/ENSG00000186185/ENSG00000112984/ENSG00000087586/ENSG00000134057/ENSG00000090889
## GO:0072686                                                                                                                                                 ENSG00000137807/ENSG00000138778/ENSG00000066279/ENSG00000154839/ENSG00000262634/ENSG00000137804/ENSG00000088325/ENSG00000013810/ENSG00000170312/ENSG00000164109/ENSG00000121621/ENSG00000138160/ENSG00000163519/ENSG00000178999/ENSG00000198901/ENSG00000237649/ENSG00000233450/ENSG00000056678/ENSG00000204197/ENSG00000186185/ENSG00000087586
## GO:0000779                                                                                                                                                                                                                 ENSG00000138778/ENSG00000080986/ENSG00000123485/ENSG00000154839/ENSG00000262634/ENSG00000117650/ENSG00000100162/ENSG00000166451/ENSG00000186871/ENSG00000164109/ENSG00000121621/ENSG00000167513/ENSG00000089685/ENSG00000112742/ENSG00000109805/ENSG00000178999/ENSG00000134057
##            Count
## GO:0005819    30
## GO:0072686    21
## GO:0000779    17</code></pre>
<p>Gene IDs can be mapped to gene Symbols by using the parameter <code>readable=TRUE</code> or <a href="useful-utilities.html#setReadable"><code>setReadable()</code> function</a>.</p>
</div>
<div id="go-gene-set-enrichment-analysis" class="section level2" number="6.4">
<h2>
<span class="header-section-number">6.4</span> GO Gene Set Enrichment Analysis<a class="anchor" aria-label="anchor" href="#go-gene-set-enrichment-analysis"><i class="fas fa-link"></i></a>
</h2>
<p>The <a href="http://bioconductor.org/packages/clusterProfiler">clusterProfiler</a> package provides the <code><a href="https://rdrr.io/pkg/clusterProfiler/man/gseGO.html">gseGO()</a></code> function for <a href="enrichment-overview.html#gsea-algorithm">gene set enrichment analysis</a> using gene ontology.</p>
<div class="sourceCode" id="cb68"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ego3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/gseGO.html">gseGO</a></span><span class="op">(</span>geneList     <span class="op">=</span> <span class="va">geneList</span>,
              OrgDb        <span class="op">=</span> <span class="va">org.Hs.eg.db</span>,
              ont          <span class="op">=</span> <span class="st">"CC"</span>,
              minGSSize    <span class="op">=</span> <span class="fl">100</span>,
              maxGSSize    <span class="op">=</span> <span class="fl">500</span>,
              pvalueCutoff <span class="op">=</span> <span class="fl">0.05</span>,
              verbose      <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<p>The format of input data, <code>geneList</code>, was documented in the <a href="faq.html#genelist">FAQ</a>. Beware that only gene Set size in <code>[minGSSize, maxGSSize]</code> will be tested.</p>
</div>
<div id="clusterprofiler-go-non-model" class="section level2" number="6.5">
<h2>
<span class="header-section-number">6.5</span> GO analysis for non-model organisms<a class="anchor" aria-label="anchor" href="#clusterprofiler-go-non-model"><i class="fas fa-link"></i></a>
</h2>
<p>Both the <code><a href="https://rdrr.io/pkg/clusterProfiler/man/enrichGO.html">enrichGO()</a></code> and <code><a href="https://rdrr.io/pkg/clusterProfiler/man/gseGO.html">gseGO()</a></code> functions require an <code>OrgDb</code> object as the background annotation. For organisms that donâ€™t have <code>OrgDb</code> provided by <code>Bioconductor</code>, users can query one (if available) online via <a href="http://bioconductor.org/packages/AnnotationHub">AnnotationHub</a>. If there is no <code>OrgDb</code> available, users can obtain GO annotation from other sources, e.g.Â from <a href="http://bioconductor.org/packages/biomaRt">biomaRt</a> or <a href="https://www.blast2go.com/">Blast2GO</a>. Then the <code><a href="https://rdrr.io/pkg/clusterProfiler/man/enricher.html">enricher()</a></code> or <code><a href="https://rdrr.io/pkg/clusterProfiler/man/GSEA.html">GSEA()</a></code> functions can be used to perform GO analysis for these organisms, similar to the examples using wikiPathways and MSigDB. Another solution is to create an <code>OrgDb</code> on your own using <a href="http://bioconductor.org/packages/AnnotationForge">AnnotationForge</a> package.</p>
</div>
<div id="visualize-enriched-go-terms-as-a-directed-acyclic-graph" class="section level2" number="6.6">
<h2>
<span class="header-section-number">6.6</span> Visualize enriched GO terms as a directed acyclic graph<a class="anchor" aria-label="anchor" href="#visualize-enriched-go-terms-as-a-directed-acyclic-graph"><i class="fas fa-link"></i></a>
</h2>
<p>The <code><a href="https://rdrr.io/pkg/enrichplot/man/goplot.html">goplot()</a></code> function can accept the output of <code>enrichGO</code> and visualize the enriched GO induced graph.</p>


<div class="sourceCode" id="cb69"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/goplot.html">goplot</a></span><span class="op">(</span><span class="va">ego</span><span class="op">)</span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:goplot"></span>
<img src="biomedicalKnowledge_files/figure-html/goplot-1.png" alt="Goplot of enrichment analysis." width="98%"><p class="caption">
Figure 6.1: <strong>Goplot of enrichment analysis.</strong>
</p>
</div>
</div>
<div id="clusterprofiler-go-summary" class="section level2" number="6.7">
<h2>
<span class="header-section-number">6.7</span> Summary<a class="anchor" aria-label="anchor" href="#clusterprofiler-go-summary"><i class="fas fa-link"></i></a>
</h2>
<p>GO semantic similarity can be calculated by <a href="http://bioconductor.org/packages/GOSemSim">GOSemSim</a> <span class="citation">(<a href="references.html#ref-yu2010" role="doc-biblioref">Yu et al. 2010</a>)</span>. We can use it to cluster genes/proteins into different clusters based on their functional similarity and can also use it to measure the similarities among GO terms to reduce the redundancy of GO enrichment results.</p>

</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="enrichment-overview.html"><span class="header-section-number">5</span> Overview of enrichment analysis</a></div>
<div class="next"><a href="clusterprofiler-kegg.html"><span class="header-section-number">7</span> KEGG enrichment analysis</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#clusterprofiler-go"><span class="header-section-number">6</span> GO enrichment analysis</a></li>
<li><a class="nav-link" href="#clusterProfiler-go-supported-organisms"><span class="header-section-number">6.1</span> Supported organisms</a></li>
<li><a class="nav-link" href="#go-classification"><span class="header-section-number">6.2</span> GO classification</a></li>
<li><a class="nav-link" href="#clusterprofiler-go-ora"><span class="header-section-number">6.3</span> GO over-representation analysis</a></li>
<li><a class="nav-link" href="#go-gene-set-enrichment-analysis"><span class="header-section-number">6.4</span> GO Gene Set Enrichment Analysis</a></li>
<li><a class="nav-link" href="#clusterprofiler-go-non-model"><span class="header-section-number">6.5</span> GO analysis for non-model organisms</a></li>
<li><a class="nav-link" href="#visualize-enriched-go-terms-as-a-directed-acyclic-graph"><span class="header-section-number">6.6</span> Visualize enriched GO terms as a directed acyclic graph</a></li>
<li><a class="nav-link" href="#clusterprofiler-go-summary"><span class="header-section-number">6.7</span> Summary</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
          
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Biomedical Knowledge Mining using GOSemSim and clusterProfiler</strong>" was written by Guangchuang Yu. It was last built on 2022-04-24.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer><!-- dynamically load mathjax for compatibility with self-contained --><script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script><script type="text/x-mathjax-config">const popovers = document.querySelectorAll('a.footnote-ref[data-toggle="popover"]');
for (let popover of popovers) {
  const div = document.createElement('div');
  div.setAttribute('style', 'position: absolute; top: 0, left:0; width:0, height:0, overflow: hidden; visibility: hidden;');
  div.innerHTML = popover.getAttribute('data-content');

  var has_math = div.querySelector("span.math");
  if (has_math) {
    document.body.appendChild(div);
    MathJax.Hub.Queue(["Typeset", MathJax.Hub, div]);
    MathJax.Hub.Queue(function() {
      popover.setAttribute('data-content', div.innerHTML);
      document.body.removeChild(div);
    })
  }
}
</script>
</body>
</html>
