<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 7 KEGG enrichment analysis | Biomedical Knowledge Mining using GOSemSim and clusterProfiler</title>
<meta name="author" content="Guangchuang Yu">
<meta name="description" content="The KEGG FTP service is not freely available for academic since 2012, and there are many software packages using out-dated KEGG annotation data. The clusterProfiler package supports downloading...">
<meta name="generator" content="bookdown 0.23 with bs4_book()">
<meta property="og:title" content="Chapter 7 KEGG enrichment analysis | Biomedical Knowledge Mining using GOSemSim and clusterProfiler">
<meta property="og:type" content="book">
<meta property="og:description" content="The KEGG FTP service is not freely available for academic since 2012, and there are many software packages using out-dated KEGG annotation data. The clusterProfiler package supports downloading...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Chapter 7 KEGG enrichment analysis | Biomedical Knowledge Mining using GOSemSim and clusterProfiler">
<meta name="twitter:description" content="The KEGG FTP service is not freely available for academic since 2012, and there are many software packages using out-dated KEGG annotation data. The clusterProfiler package supports downloading...">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/header-attrs-2.10/header-attrs.js"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.2.5.1/tabs.js"></script><script src="libs/bs3compat-0.2.5.1/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS -->
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">Biomedical Knowledge Mining using GOSemSim and clusterProfiler</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html">üìñ Introduction</a></li>
<li class="book-part">Part I: Semantic Similarity Analysis</li>
<li><a class="" href="semantic-similarity-overview.html"><span class="header-section-number">1</span> Overview of semantic similarity analysis</a></li>
<li><a class="" href="GOSemSim.html"><span class="header-section-number">2</span> GO semantic similarity analysis</a></li>
<li><a class="" href="DOSE-semantic-similarity.html"><span class="header-section-number">3</span> DO semantic similarity analysis</a></li>
<li><a class="" href="meshes-semantic-similarity.html"><span class="header-section-number">4</span> MeSH semantic similarity analysis</a></li>
<li class="book-part">Part II: Enrichment analysis</li>
<li><a class="" href="enrichment-overview.html"><span class="header-section-number">5</span> Overview of enrichment analysis</a></li>
<li><a class="" href="clusterprofiler-go.html"><span class="header-section-number">6</span> GO enrichment analysis</a></li>
<li><a class="active" href="clusterprofiler-kegg.html"><span class="header-section-number">7</span> KEGG enrichment analysis</a></li>
<li><a class="" href="wikipathways-analysis.html"><span class="header-section-number">8</span> WikiPathways analysis</a></li>
<li><a class="" href="reactomepa.html"><span class="header-section-number">9</span> Reactome enrichment analysis</a></li>
<li><a class="" href="dose-enrichment.html"><span class="header-section-number">10</span> Disease enrichment analysis</a></li>
<li><a class="" href="meshes-enrichment.html"><span class="header-section-number">11</span> MeSH enrichment analysis</a></li>
<li><a class="" href="universal-api.html"><span class="header-section-number">12</span> Universal enrichment analysis</a></li>
<li><a class="" href="chipseeker-enrichment.html"><span class="header-section-number">13</span> Functional enrichment analysis of genomic coordinations</a></li>
<li><a class="" href="clusterprofiler-comparecluster.html"><span class="header-section-number">14</span> Biological theme comparison</a></li>
<li><a class="" href="enrichplot.html"><span class="header-section-number">15</span> Visualization of functional enrichment result</a></li>
<li class="book-part">Part III: Miscellaneous topics</li>
<li><a class="" href="clusterProfiler-dplyr.html"><span class="header-section-number">16</span> dplyr verbs for manipulating enrichment result</a></li>
<li><a class="" href="useful-utilities.html"><span class="header-section-number">17</span> Useful utilities</a></li>
<li class="book-part">Appendix</li>
<li><a class="" href="faq.html"><span class="header-section-number">A</span> Frequently asked questions</a></li>
<li><a class="" href="references.html">References</a></li>
</ul>

        <div class="book-extra">
          
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="clusterprofiler-kegg" class="section level1" number="7">
<h1>
<span class="header-section-number">7</span> KEGG enrichment analysis<a class="anchor" aria-label="anchor" href="#clusterprofiler-kegg"><i class="fas fa-link"></i></a>
</h1>
<p>The KEGG FTP service is not freely available for academic since 2012, and there are many software packages using out-dated KEGG annotation data. The <a href="http://bioconductor.org/packages/clusterProfiler">clusterProfiler</a> package supports downloading latest online version of KEGG data using the <a href="https://www.kegg.jp">KEGG website</a>, which is freely available for academic users. Both KEGG pathway and module are supported in <a href="http://bioconductor.org/packages/clusterProfiler">clusterProfiler</a>.</p>
<div id="clusterProfiler-kegg-supported-organisms" class="section level2" number="7.1">
<h2>
<span class="header-section-number">7.1</span> Supported organisms<a class="anchor" aria-label="anchor" href="#clusterProfiler-kegg-supported-organisms"><i class="fas fa-link"></i></a>
</h2>
<p>The <a href="http://bioconductor.org/packages/clusterProfiler">clusterProfiler</a> package supports all organisms that have KEGG annotation data available in KEGG database. User should pass abbreviation of academic name to the <code>organism</code> parameter. The full list of KEGG supported organisms can be accessed via <a href="http://www.genome.jp/kegg/catalog/org_list.html" class="uri">http://www.genome.jp/kegg/catalog/org_list.html</a>. <a href="https://www.genome.jp/kegg/ko.html">KEGG Orthology</a> (KO) Database is also supported by specifying <code>organism = "ko"</code>.</p>
<p>The <a href="http://bioconductor.org/packages/clusterProfiler">clusterProfiler</a> package provides <code><a href="https://rdrr.io/pkg/clusterProfiler/man/search_kegg_organism.html">search_kegg_organism()</a></code> function to help searching supported organisms.</p>
<div class="sourceCode" id="cb70"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://yulab-smu.top/biomedical-knowledge-mining-book/">clusterProfiler</a></span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/search_kegg_organism.html">search_kegg_organism</a></span><span class="op">(</span><span class="st">'ece'</span>, by<span class="op">=</span><span class="st">'kegg_code'</span><span class="op">)</span></code></pre></div>
<pre><code>##     kegg_code                        scientific_name common_name
## 366       ece Escherichia coli O157:H7 EDL933 (EHEC)        &lt;NA&gt;</code></pre>
<div class="sourceCode" id="cb72"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ecoli</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/search_kegg_organism.html">search_kegg_organism</a></span><span class="op">(</span><span class="st">'Escherichia coli'</span>, by<span class="op">=</span><span class="st">'scientific_name'</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">ecoli</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 65  3</code></pre>
<div class="sourceCode" id="cb74"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">ecoli</span><span class="op">)</span></code></pre></div>
<pre><code>##     kegg_code                        scientific_name common_name
## 361       eco           Escherichia coli K-12 MG1655        &lt;NA&gt;
## 362       ecj            Escherichia coli K-12 W3110        &lt;NA&gt;
## 363       ecd            Escherichia coli K-12 DH10B        &lt;NA&gt;
## 364       ebw                Escherichia coli BW2952        &lt;NA&gt;
## 365      ecok            Escherichia coli K-12 MDS42        &lt;NA&gt;
## 366       ece Escherichia coli O157:H7 EDL933 (EHEC)        &lt;NA&gt;</code></pre>
</div>
<div id="clusterprofiler-kegg-pathway-ora" class="section level2" number="7.2">
<h2>
<span class="header-section-number">7.2</span> KEGG pathway over-representation analysis<a class="anchor" aria-label="anchor" href="#clusterprofiler-kegg-pathway-ora"><i class="fas fa-link"></i></a>
</h2>
<div class="sourceCode" id="cb76"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">geneList</span>, package<span class="op">=</span><span class="st">"DOSE"</span><span class="op">)</span>
<span class="va">gene</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">geneList</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">abs</a></span><span class="op">(</span><span class="va">geneList</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">2</span><span class="op">]</span>

<span class="va">kk</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/enrichKEGG.html">enrichKEGG</a></span><span class="op">(</span>gene         <span class="op">=</span> <span class="va">gene</span>,
                 organism     <span class="op">=</span> <span class="st">'hsa'</span>,
                 pvalueCutoff <span class="op">=</span> <span class="fl">0.05</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">kk</span><span class="op">)</span></code></pre></div>
<pre><code>##                ID                                                   Description
## hsa04110 hsa04110                                                    Cell cycle
## hsa04114 hsa04114                                                Oocyte meiosis
## hsa04218 hsa04218                                           Cellular senescence
## hsa04061 hsa04061 Viral protein interaction with cytokine and cytokine receptor
## hsa03320 hsa03320                                        PPAR signaling pathway
## hsa04914 hsa04914                       Progesterone-mediated oocyte maturation
##          GeneRatio  BgRatio       pvalue     p.adjust       qvalue
## hsa04110     11/94 124/8091 1.651714e-07 3.419048e-05 3.355587e-05
## hsa04114     10/94 129/8091 2.175890e-06 2.252047e-04 2.210247e-04
## hsa04218     10/94 156/8091 1.198811e-05 8.271799e-04 8.118267e-04
## hsa04061      8/94 100/8091 1.904909e-05 9.857904e-04 9.674932e-04
## hsa03320      7/94  76/8091 2.595975e-05 1.074734e-03 1.054786e-03
## hsa04914      7/94 100/8091 1.516194e-04 5.230871e-03 5.133781e-03
##                                                      geneID Count
## hsa04110 8318/991/9133/890/983/4085/7272/1111/891/4174/9232    11
## hsa04114    991/9133/983/4085/51806/6790/891/9232/3708/5241    10
## hsa04218     2305/4605/9133/890/983/51806/1111/891/776/3708    10
## hsa04061           3627/10563/6373/4283/6362/6355/9547/1524     8
## hsa03320                 4312/9415/9370/5105/2167/3158/5346     7
## hsa04914                    9133/890/983/4085/6790/891/5241     7</code></pre>
<p>Input ID type can be <code>kegg</code>, <code>ncbi-geneid</code>, <code>ncbi-proteinid</code> or <code>uniprot</code> (see also <a href="useful-utilities.html#bitr_kegg">session 16.1.2</a>). Unlike <code><a href="https://rdrr.io/pkg/clusterProfiler/man/enrichGO.html">enrichGO()</a></code>, there is no <code>readable</code> parameter for <code><a href="https://rdrr.io/pkg/clusterProfiler/man/enrichKEGG.html">enrichKEGG()</a></code>. However, users can use the <a href="useful-utilities.html#setReadable"><code>setReadable()</code> function</a> if there is an <code>OrgDb</code> available for the species.</p>
</div>
<div id="clusterprofiler-kegg-pathway-gsea" class="section level2" number="7.3">
<h2>
<span class="header-section-number">7.3</span> KEGG pathway gene set enrichment analysis<a class="anchor" aria-label="anchor" href="#clusterprofiler-kegg-pathway-gsea"><i class="fas fa-link"></i></a>
</h2>
<div class="sourceCode" id="cb78"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">kk2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/gseKEGG.html">gseKEGG</a></span><span class="op">(</span>geneList     <span class="op">=</span> <span class="va">geneList</span>,
               organism     <span class="op">=</span> <span class="st">'hsa'</span>,
               minGSSize    <span class="op">=</span> <span class="fl">120</span>,
               pvalueCutoff <span class="op">=</span> <span class="fl">0.05</span>,
               verbose      <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">kk2</span><span class="op">)</span></code></pre></div>
<pre><code>##                ID                              Description setSize
## hsa05169 hsa05169             Epstein-Barr virus infection     193
## hsa05166 hsa05166  Human T-cell leukemia virus 1 infection     200
## hsa04613 hsa04613  Neutrophil extracellular trap formation     130
## hsa04218 hsa04218                      Cellular senescence     141
## hsa04510 hsa04510                           Focal adhesion     190
## hsa05170 hsa05170 Human immunodeficiency virus 1 infection     189
##          enrichmentScore       NES       pvalue     p.adjust      qvalues rank
## hsa05169       0.4335010  1.925520 6.472840e-08 4.919358e-06 3.270488e-06 2820
## hsa05166       0.3926594  1.757442 3.012634e-06 8.065327e-05 5.361990e-05 1955
## hsa04613       0.4496569  1.890892 3.183682e-06 8.065327e-05 5.361990e-05 2575
## hsa04218       0.4115945  1.754241 3.701280e-05 6.591880e-04 4.382413e-04 1155
## hsa04510      -0.4169068 -1.703069 4.336763e-05 6.591880e-04 4.382413e-04 2183
## hsa05170       0.3711150  1.648181 1.414590e-04 1.610393e-03 1.070621e-03 3178
##                            leading_edge
## hsa05169 tags=39%, list=23%, signal=31%
## hsa05166 tags=26%, list=16%, signal=23%
## hsa04613 tags=37%, list=21%, signal=30%
## hsa04218  tags=17%, list=9%, signal=16%
## hsa04510 tags=27%, list=17%, signal=22%
## hsa05170 tags=38%, list=25%, signal=29%
##                                                                                                                                                                                                                                                                                                                                                                                core_enrichment
## hsa05169 3627/890/6890/9636/898/9134/6502/6772/3126/3112/4609/917/5709/1869/3654/919/915/4067/4938/864/4940/5713/5336/11047/3066/54205/1871/578/1019/637/916/3383/4939/10213/23586/4793/5603/7979/7128/6891/930/5714/3452/6850/5702/4794/7124/3569/7097/5708/2208/8772/3119/5704/7186/5971/3135/1380/958/5610/4792/10018/8819/3134/10379/9641/1147/5718/6300/3109/811/5606/2923/3108/5707/1432
## hsa05166                                                                                                                      991/9133/890/4085/7850/1111/9232/8061/701/9700/898/4316/9134/3932/3559/3126/3112/4609/3561/917/1869/1029/915/114/2005/5902/55697/1871/1031/2224/292/1019/3689/916/3383/11200/706/3600/6513/3601/468/5604/7124/1030/3569/4049/4055/10393/3119/5901/5971/1959/3135
## hsa04613                                                                                                                                      820/366/51311/64581/3015/85236/55506/8970/8357/1535/2359/5336/4688/92815/3066/8336/292/1991/3689/8345/5603/4689/5880/10105/1184/6404/3018/6850/5604/3014/7097/1378/8290/1536/834/5605/1183/728/2215/8335/5594/9734/3674/5578/5582/7417/8331/6300
## hsa04218                                                                                                                                                                                                                                                                 2305/4605/9133/890/983/51806/1111/891/993/3576/1978/898/9134/4609/1869/1029/22808/1871/5499/91860/292/1019/11200/1875
## hsa04510                                                                                                                   5595/5228/7424/1499/4636/83660/7059/5295/1288/23396/3910/3371/3082/1291/394/3791/7450/596/3685/1280/3675/595/2318/3912/1793/1278/1277/1293/10398/55742/2317/7058/25759/56034/3693/3480/5159/857/1292/3908/3909/63923/3913/1287/3679/7060/3479/10451/80310/1311/1101
## hsa05170                   9133/9582/983/51806/1111/891/6890/200315/917/3654/919/915/5336/54205/91860/578/995/25939/637/1234/916/5603/2792/5880/6891/6921/3452/5604/7124/9978/7097/7852/8772/1174/7186/3135/164/60489/2787/356/7133/5605/27350/6199/1642/5594/4792/3134/5578/4893/8454/5582/2786/1147/3984/6300/200316/811/5606/2923/4775/162/1432/2784/836/5747/5058/3106/2770/5534/5579/4615</code></pre>
</div>
<div id="clusterprofiler-kegg-module-ora" class="section level2" number="7.4">
<h2>
<span class="header-section-number">7.4</span> KEGG module over-representation analysis<a class="anchor" aria-label="anchor" href="#clusterprofiler-kegg-module-ora"><i class="fas fa-link"></i></a>
</h2>
<p><a href="http://www.genome.jp/kegg/module.html">KEGG Module</a> is a collection of manually defined function units. In some situation, KEGG Modules have a more straightforward interpretation.</p>
<div class="sourceCode" id="cb80"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">mkk</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/enrichMKEGG.html">enrichMKEGG</a></span><span class="op">(</span>gene <span class="op">=</span> <span class="va">gene</span>,
                   organism <span class="op">=</span> <span class="st">'hsa'</span>,
                   pvalueCutoff <span class="op">=</span> <span class="fl">1</span>,
                   qvalueCutoff <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">mkk</span><span class="op">)</span>                   </code></pre></div>
<pre><code>##            ID                                             Description GeneRatio
## M00912 M00912      NAD biosynthesis, tryptophan =&gt; quinolinate =&gt; NAD       2/9
## M00095 M00095          C5 isoprenoid biosynthesis, mevalonate pathway       1/9
## M00053 M00053 Pyrimidine deoxyribonuleotide biosynthesis, CDP =&gt; dCTP       1/9
## M00938 M00938 Pyrimidine deoxyribonuleotide biosynthesis, UDP =&gt; dTTP       1/9
## M00003 M00003            Gluconeogenesis, oxaloacetate =&gt; fructose-6P       1/9
## M00049 M00049     Adenine ribonucleotide biosynthesis, IMP =&gt; ADP,ATP       1/9
##        BgRatio     pvalue   p.adjust     qvalue     geneID Count
## M00912  12/830 0.00652644 0.03915864 0.03434969 23475/3620     2
## M00095  10/830 0.10382973 0.21390091 0.18763237       3158     1
## M00053  11/830 0.11366575 0.21390091 0.18763237       6241     1
## M00938  14/830 0.14260060 0.21390091 0.18763237       6241     1
## M00003  18/830 0.17987328 0.21569867 0.18920936       5105     1
## M00049  22/830 0.21569867 0.21569867 0.18920936      26289     1</code></pre>
</div>
<div id="clusterprofiler-kegg-module-gsea" class="section level2" number="7.5">
<h2>
<span class="header-section-number">7.5</span> KEGG module gene set enrichment analysis<a class="anchor" aria-label="anchor" href="#clusterprofiler-kegg-module-gsea"><i class="fas fa-link"></i></a>
</h2>
<div class="sourceCode" id="cb82"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">mkk2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/gseMKEGG.html">gseMKEGG</a></span><span class="op">(</span>geneList <span class="op">=</span> <span class="va">geneList</span>,
                 organism <span class="op">=</span> <span class="st">'hsa'</span>,
                 pvalueCutoff <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">mkk2</span><span class="op">)</span></code></pre></div>
<pre><code>##            ID                                                      Description
## M00001 M00001        Glycolysis (Embden-Meyerhof pathway), glucose =&gt; pyruvate
## M00035 M00035                                           Methionine degradation
## M00002 M00002         Glycolysis, core module involving three-carbon compounds
## M00938 M00938          Pyrimidine deoxyribonuleotide biosynthesis, UDP =&gt; dTTP
## M00009 M00009                           Citrate cycle (TCA cycle, Krebs cycle)
## M00104 M00104 Bile acid biosynthesis, cholesterol =&gt; cholate/chenodeoxycholate
##        setSize enrichmentScore       NES      pvalue  p.adjust   qvalues rank
## M00001      24       0.5739036  1.777650 0.005589333 0.1788587 0.1647382 2886
## M00035      10       0.6784636  1.646743 0.028210658 0.3072796 0.2830207 1555
## M00002      11       0.6421781  1.614229 0.037225486 0.3072796 0.2830207 1381
## M00938      10       0.6648004  1.613580 0.038409946 0.3072796 0.2830207  648
## M00009      22       0.4504911  1.374948 0.089974293 0.5758355 0.5303748 3514
## M00104      10      -0.5876900 -1.326415 0.150592217 0.7851100 0.7231276  961
##                          leading_edge
## M00001 tags=54%, list=23%, signal=42%
## M00035 tags=50%, list=12%, signal=44%
## M00002 tags=55%, list=11%, signal=49%
## M00938  tags=40%, list=5%, signal=38%
## M00009 tags=50%, list=28%, signal=36%
## M00104  tags=50%, list=8%, signal=46%
##                                                         core_enrichment
## M00001 5214/3101/2821/7167/2597/5230/2023/5223/5315/3099/5232/2027/5211
## M00035                                           875/1789/191/1788/1786
## M00002                                    7167/2597/5230/2023/5223/5315
## M00938                                              6241/7298/4830/1841
## M00009            3418/50/4190/3419/2271/3421/55753/3417/1431/6389/4191
## M00104                                        6342/10998/1581/3295/8309</code></pre>
</div>
<div id="visualize-enriched-kegg-pathways" class="section level2" number="7.6">
<h2>
<span class="header-section-number">7.6</span> Visualize enriched KEGG pathways<a class="anchor" aria-label="anchor" href="#visualize-enriched-kegg-pathways"><i class="fas fa-link"></i></a>
</h2>
<p>The <a href="http://bioconductor.org/packages/enrichplot">enrichplot</a> package implemented <a href="enrichplot.html#enrichplot">several methods</a> to visualize enriched terms. Most of them are general methods that can be used to GO, KEGG, MSigDb and other gene set annotations. Here, we introduce <code><a href="https://rdrr.io/pkg/clusterProfiler/man/browseKEGG.html">clusterProfiler::browseKEGG()</a></code> and <code><a href="https://rdrr.io/pkg/pathview/man/pathview.html">pathview::pathview()</a></code> functions to help users exploring enriched KEGG pathway with interested genes.</p>
<p>To view the KEGG pathway, user can use <code>browseKEGG</code> function, which will open web browser and highlight enriched genes.</p>
<div class="sourceCode" id="cb84"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/browseKEGG.html">browseKEGG</a></span><span class="op">(</span><span class="va">kk</span>, <span class="st">'hsa04110'</span><span class="op">)</span></code></pre></div>


<div class="figure">
<span style="display:block;" id="fig:browseKEGG"></span>
<img src="figures/browseKEGG.png" alt="Explore selected KEGG pathway. Differentially expressed genes that enriched in the selected pathway will be highlighted." width="100%"><p class="caption">
Figure 7.1: <strong>Explore selected KEGG pathway.</strong> Differentially expressed genes that enriched in the selected pathway will be highlighted.
</p>
</div>
<p>Users can also use the <code><a href="https://rdrr.io/pkg/pathview/man/pathview.html">pathview()</a></code> function from the <a href="http://bioconductor.org/packages/pathview">pathview</a> <span class="citation">(<a href="references.html#ref-luo_pathview" role="doc-biblioref">Luo and Brouwer 2013</a>)</span> to visualize enriched KEGG pathways identify by the <a href="http://bioconductor.org/packages/clusterProfiler">clusterProfiler</a> package <span class="citation">(<a href="references.html#ref-yu2012" role="doc-biblioref">Yu et al. 2012</a>)</span>.</p>
<p>The following example illustrate how to visualize ‚Äúhsa04110‚Äù pathway, which was enriched in our previous analysis.</p>
<div class="sourceCode" id="cb85"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/datapplab/pathview">"pathview"</a></span><span class="op">)</span>
<span class="va">hsa04110</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/pathview/man/pathview.html">pathview</a></span><span class="op">(</span>gene.data  <span class="op">=</span> <span class="va">geneList</span>,
                     pathway.id <span class="op">=</span> <span class="st">"hsa04110"</span>,
                     species    <span class="op">=</span> <span class="st">"hsa"</span>,
                     limit      <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>gene<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">abs</a></span><span class="op">(</span><span class="va">geneList</span><span class="op">)</span><span class="op">)</span>, cpd<span class="op">=</span><span class="fl">1</span><span class="op">)</span><span class="op">)</span></code></pre></div>


<div class="figure">
<span style="display:block;" id="fig:pathview"></span>
<img src="figures/hsa04110_pathview.png" alt="Visualize selected KEGG pathway by pathview(). Gene expression values can be mapped to gradient color scale." width="100%"><p class="caption">
Figure 7.2: <strong>Visualize selected KEGG pathway by <code><a href="https://rdrr.io/pkg/pathview/man/pathview.html">pathview()</a></code>.</strong> Gene expression values can be mapped to gradient color scale.
</p>
</div>

</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="clusterprofiler-go.html"><span class="header-section-number">6</span> GO enrichment analysis</a></div>
<div class="next"><a href="wikipathways-analysis.html"><span class="header-section-number">8</span> WikiPathways analysis</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#clusterprofiler-kegg"><span class="header-section-number">7</span> KEGG enrichment analysis</a></li>
<li><a class="nav-link" href="#clusterProfiler-kegg-supported-organisms"><span class="header-section-number">7.1</span> Supported organisms</a></li>
<li><a class="nav-link" href="#clusterprofiler-kegg-pathway-ora"><span class="header-section-number">7.2</span> KEGG pathway over-representation analysis</a></li>
<li><a class="nav-link" href="#clusterprofiler-kegg-pathway-gsea"><span class="header-section-number">7.3</span> KEGG pathway gene set enrichment analysis</a></li>
<li><a class="nav-link" href="#clusterprofiler-kegg-module-ora"><span class="header-section-number">7.4</span> KEGG module over-representation analysis</a></li>
<li><a class="nav-link" href="#clusterprofiler-kegg-module-gsea"><span class="header-section-number">7.5</span> KEGG module gene set enrichment analysis</a></li>
<li><a class="nav-link" href="#visualize-enriched-kegg-pathways"><span class="header-section-number">7.6</span> Visualize enriched KEGG pathways</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
          
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Biomedical Knowledge Mining using GOSemSim and clusterProfiler</strong>" was written by Guangchuang Yu. It was last built on 2021-08-27.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer><!-- dynamically load mathjax for compatibility with self-contained --><script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script><script type="text/x-mathjax-config">const popovers = document.querySelectorAll('a.footnote-ref[data-toggle="popover"]');
for (let popover of popovers) {
  const div = document.createElement('div');
  div.setAttribute('style', 'position: absolute; top: 0, left:0; width:0, height:0, overflow: hidden; visibility: hidden;');
  div.innerHTML = popover.getAttribute('data-content');

  var has_math = div.querySelector("span.math");
  if (has_math) {
    document.body.appendChild(div);
    MathJax.Hub.Queue(["Typeset", MathJax.Hub, div]);
    MathJax.Hub.Queue(function() {
      popover.setAttribute('data-content', div.innerHTML);
      document.body.removeChild(div);
    })
  }
}
</script>
</body>
</html>
