<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 14 Biological theme comparison | Biomedical Knowledge Mining using GOSemSim and clusterProfiler</title>
<meta name="author" content="Guangchuang Yu">
<meta name="description" content="The clusterProfiler package was developed for biological theme comparison (Yu et al. 2012; Wu et al. 2021), and it provides a function, compareCluster, to automatically calculate enriched...">
<meta name="generator" content="bookdown 0.23 with bs4_book()">
<meta property="og:title" content="Chapter 14 Biological theme comparison | Biomedical Knowledge Mining using GOSemSim and clusterProfiler">
<meta property="og:type" content="book">
<meta property="og:description" content="The clusterProfiler package was developed for biological theme comparison (Yu et al. 2012; Wu et al. 2021), and it provides a function, compareCluster, to automatically calculate enriched...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Chapter 14 Biological theme comparison | Biomedical Knowledge Mining using GOSemSim and clusterProfiler">
<meta name="twitter:description" content="The clusterProfiler package was developed for biological theme comparison (Yu et al. 2012; Wu et al. 2021), and it provides a function, compareCluster, to automatically calculate enriched...">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/header-attrs-2.10/header-attrs.js"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.2.5.1/tabs.js"></script><script src="libs/bs3compat-0.2.5.1/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS -->
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">Biomedical Knowledge Mining using GOSemSim and clusterProfiler</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html">ðŸ“– Introduction</a></li>
<li class="book-part">Part I: Semantic Similarity Analysis</li>
<li><a class="" href="semantic-similarity-overview.html"><span class="header-section-number">1</span> Overview of semantic similarity analysis</a></li>
<li><a class="" href="GOSemSim.html"><span class="header-section-number">2</span> GO semantic similarity analysis</a></li>
<li><a class="" href="DOSE-semantic-similarity.html"><span class="header-section-number">3</span> DO semantic similarity analysis</a></li>
<li><a class="" href="meshes-semantic-similarity.html"><span class="header-section-number">4</span> MeSH semantic similarity analysis</a></li>
<li class="book-part">Part II: Enrichment analysis</li>
<li><a class="" href="enrichment-overview.html"><span class="header-section-number">5</span> Overview of enrichment analysis</a></li>
<li><a class="" href="clusterprofiler-go.html"><span class="header-section-number">6</span> GO enrichment analysis</a></li>
<li><a class="" href="clusterprofiler-kegg.html"><span class="header-section-number">7</span> KEGG enrichment analysis</a></li>
<li><a class="" href="wikipathways-analysis.html"><span class="header-section-number">8</span> WikiPathways analysis</a></li>
<li><a class="" href="reactomepa.html"><span class="header-section-number">9</span> Reactome enrichment analysis</a></li>
<li><a class="" href="dose-enrichment.html"><span class="header-section-number">10</span> Disease enrichment analysis</a></li>
<li><a class="" href="meshes-enrichment.html"><span class="header-section-number">11</span> MeSH enrichment analysis</a></li>
<li><a class="" href="universal-api.html"><span class="header-section-number">12</span> Universal enrichment analysis</a></li>
<li><a class="" href="chipseeker-enrichment.html"><span class="header-section-number">13</span> Functional enrichment analysis of genomic coordinations</a></li>
<li><a class="active" href="clusterprofiler-comparecluster.html"><span class="header-section-number">14</span> Biological theme comparison</a></li>
<li><a class="" href="enrichplot.html"><span class="header-section-number">15</span> Visualization of functional enrichment result</a></li>
<li class="book-part">Part III: Miscellaneous topics</li>
<li><a class="" href="clusterProfiler-dplyr.html"><span class="header-section-number">16</span> dplyr verbs for manipulating enrichment result</a></li>
<li><a class="" href="useful-utilities.html"><span class="header-section-number">17</span> Useful utilities</a></li>
<li class="book-part">Appendix</li>
<li><a class="" href="faq.html"><span class="header-section-number">A</span> Frequently asked questions</a></li>
<li><a class="" href="references.html">References</a></li>
</ul>

        <div class="book-extra">
          
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="clusterprofiler-comparecluster" class="section level1" number="14">
<h1>
<span class="header-section-number">14</span> Biological theme comparison<a class="anchor" aria-label="anchor" href="#clusterprofiler-comparecluster"><i class="fas fa-link"></i></a>
</h1>
<p>The <a href="http://bioconductor.org/packages/clusterProfiler">clusterProfiler</a> package was developed for biological theme comparison <span class="citation">(<a href="references.html#ref-yu2012" role="doc-biblioref">Yu et al. 2012</a>; <a href="references.html#ref-wu_clusterprofiler_2021" role="doc-biblioref">Wu et al. 2021</a>)</span>, and it provides a function, <code>compareCluster</code>, to automatically calculate enriched functional profiles of each gene clusters and aggregate the results into a single object. Comparing functional profiles can reveal functional consensus and differences among different experiments and helps in identifying differential functional modules in omics datasets.</p>
<div id="comparing-multiple-gene-lists" class="section level2" number="14.1">
<h2>
<span class="header-section-number">14.1</span> Comparing multiple gene lists<a class="anchor" aria-label="anchor" href="#comparing-multiple-gene-lists"><i class="fas fa-link"></i></a>
</h2>
<p>The <code><a href="https://rdrr.io/pkg/clusterProfiler/man/compareCluster.html">compareCluster()</a></code> function applies selected function (via the <code>fun</code> parameter) to perform enrichment analysis for each gene list.</p>
<div class="sourceCode" id="cb143"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">gcSample</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/str.html">str</a></span><span class="op">(</span><span class="va">gcSample</span><span class="op">)</span> </code></pre></div>
<pre><code>## List of 8
##  $ X1: chr [1:216] "4597" "7111" "5266" "2175" ...
##  $ X2: chr [1:805] "23450" "5160" "7126" "26118" ...
##  $ X3: chr [1:392] "894" "7057" "22906" "3339" ...
##  $ X4: chr [1:838] "5573" "7453" "5245" "23450" ...
##  $ X5: chr [1:929] "5982" "7318" "6352" "2101" ...
##  $ X6: chr [1:585] "5337" "9295" "4035" "811" ...
##  $ X7: chr [1:582] "2621" "2665" "5690" "3608" ...
##  $ X8: chr [1:237] "2665" "4735" "1327" "3192" ...</code></pre>
<p>Users can use a named list of gene IDs as the input that passed to the <code>geneCluster</code> parameter.</p>
<div class="sourceCode" id="cb145"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ck</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/compareCluster.html">compareCluster</a></span><span class="op">(</span>geneCluster <span class="op">=</span> <span class="va">gcSample</span>, fun <span class="op">=</span> <span class="va">enrichKEGG</span><span class="op">)</span>
<span class="va">ck</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DOSE/man/setReadable.html">setReadable</a></span><span class="op">(</span><span class="va">ck</span>, OrgDb <span class="op">=</span> <span class="va">org.Hs.eg.db</span>, keyType<span class="op">=</span><span class="st">"ENTREZID"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">ck</span><span class="op">)</span> </code></pre></div>
<pre><code>##   Cluster       ID                            Description GeneRatio  BgRatio
## 1      X2 hsa04110                             Cell cycle    18/384 126/8093
## 2      X2 hsa05169           Epstein-Barr virus infection    23/384 202/8093
## 3      X2 hsa05340               Primary immunodeficiency     8/384  38/8093
## 4      X3 hsa04512               ECM-receptor interaction     9/187  88/8093
## 5      X3 hsa04060 Cytokine-cytokine receptor interaction    17/187 295/8093
## 6      X3 hsa04151             PI3K-Akt signaling pathway    19/187 354/8093
##         pvalue    p.adjust     qvalue
## 1 2.489396e-05 0.007592658 0.00710133
## 2 8.089523e-05 0.012336522 0.01153821
## 3 3.330480e-04 0.033859880 0.03166877
## 4 1.835745e-04 0.045747075 0.04274416
## 5 4.566657e-04 0.045747075 0.04274416
## 6 5.140121e-04 0.045747075 0.04274416
##                                                                                                                          geneID
## 1                          CDC20/E2F1/CCNA2/E2F3/BUB1B/CDC6/DBF4/PKMYT1/CDC7/ESPL1/CCNE2/CDKN2A/SFN/BUB1/CHEK2/ORC6/CDC14B/MCM4
## 2 LYN/ICAM1/TNFAIP3/E2F1/CCNA2/E2F3/BAK1/RUNX3/BID/IKBKE/TAP2/FAS/CCNE2/RELB/CD3E/ENTPD3/SYK/TRAF3/IKBKB/CD247/NEDD4/CD40/STAT1
## 3                                                                                     ADA/TAP2/LCK/CD79A/CD3E/CD8A/CD40/DCLRE1C
## 4                                                                         THBS1/HSPG2/COL9A3/ITGB7/CHAD/ITGA7/LAMA4/ITGB8/ITGB5
## 5                CXCL1/TNFRSF11B/LIFR/XCL1/GDF5/TNFRSF17/TNFRSF11A/IL5RA/EPOR/CSF2RA/TNFRSF25/BMP7/BMP4/GDF11/IL21R/IL23A/TGFB2
## 6                    CCND2/THBS1/STK11/FGF2/COL9A3/ITGB7/FGF7/ERBB4/CHAD/FGF18/CDK6/PCK1/ITGA7/HGF/EPOR/LAMA4/IKBKB/ITGB8/ITGB5
##   Count
## 1    18
## 2    23
## 3     8
## 4     9
## 5    17
## 6    19</code></pre>
</div>
<div id="formula-interface-of-comparecluster" class="section level2" number="14.2">
<h2>
<span class="header-section-number">14.2</span> Formula interface of compareCluster<a class="anchor" aria-label="anchor" href="#formula-interface-of-comparecluster"><i class="fas fa-link"></i></a>
</h2>
<p>As an alternaitve to using named list, the <code><a href="https://rdrr.io/pkg/clusterProfiler/man/compareCluster.html">compareCluster()</a></code> function also supports passing a formula to describe more complicated experimental designs (<em>e.g.</em>, <span class="math inline">\(Gene \sim time + treatment\)</span>).</p>
<div class="sourceCode" id="cb147"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">mydf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>Entrez<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">geneList</span><span class="op">)</span>, FC<span class="op">=</span><span class="va">geneList</span><span class="op">)</span>
<span class="va">mydf</span> <span class="op">&lt;-</span> <span class="va">mydf</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">abs</a></span><span class="op">(</span><span class="va">mydf</span><span class="op">$</span><span class="va">FC</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">1</span>,<span class="op">]</span>
<span class="va">mydf</span><span class="op">$</span><span class="va">group</span> <span class="op">&lt;-</span> <span class="st">"upregulated"</span>
<span class="va">mydf</span><span class="op">$</span><span class="va">group</span><span class="op">[</span><span class="va">mydf</span><span class="op">$</span><span class="va">FC</span> <span class="op">&lt;</span> <span class="fl">0</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"downregulated"</span>
<span class="va">mydf</span><span class="op">$</span><span class="va">othergroup</span> <span class="op">&lt;-</span> <span class="st">"A"</span>
<span class="va">mydf</span><span class="op">$</span><span class="va">othergroup</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">abs</a></span><span class="op">(</span><span class="va">mydf</span><span class="op">$</span><span class="va">FC</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">2</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"B"</span>

<span class="va">formula_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/compareCluster.html">compareCluster</a></span><span class="op">(</span><span class="va">Entrez</span><span class="op">~</span><span class="va">group</span><span class="op">+</span><span class="va">othergroup</span>, data<span class="op">=</span><span class="va">mydf</span>, fun<span class="op">=</span><span class="st">"enrichKEGG"</span><span class="op">)</span>
  
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">formula_res</span><span class="op">)</span></code></pre></div>
<pre><code>##           Cluster         group othergroup       ID
## 1 downregulated.A downregulated          A hsa04974
## 2 downregulated.A downregulated          A hsa04510
## 3 downregulated.A downregulated          A hsa04151
## 4 downregulated.A downregulated          A hsa04512
## 5 downregulated.B downregulated          B hsa03320
## 6   upregulated.A   upregulated          A hsa04110
##                        Description GeneRatio  BgRatio       pvalue     p.adjust
## 1 Protein digestion and absorption    16/319 103/8093 2.322875e-06 6.480820e-04
## 2                   Focal adhesion    20/319 201/8093 1.220834e-04 1.703063e-02
## 3       PI3K-Akt signaling pathway    28/319 354/8093 3.238142e-04 3.011472e-02
## 4         ECM-receptor interaction    11/319  88/8093 6.367805e-04 4.441544e-02
## 5           PPAR signaling pathway      5/43  76/8093 4.658778e-05 7.267694e-03
## 6                       Cell cycle    20/220 126/8093 1.370622e-10 3.426556e-08
##         qvalue
## 1 6.161731e-04
## 2 1.619211e-02
## 3 2.863199e-02
## 4 4.222860e-02
## 5 6.816528e-03
## 6 3.073080e-08
##                                                                                                                                             geneID
## 1                                                                 1281/50509/1290/477/1294/1360/1289/1292/1296/23428/1359/1300/1287/6505/2006/7373
## 2                                         55742/2317/7058/25759/56034/3693/3480/5159/857/1292/3908/3909/63923/3913/1287/3679/7060/3479/10451/80310
## 3 55970/5618/7058/10161/56034/3693/4254/3480/4908/5159/1292/3908/2690/3909/8817/9223/4915/3551/2791/63923/3913/9863/3667/1287/3679/7060/3479/80310
## 4                                                                                          7058/3693/3339/1292/3908/3909/63923/3913/1287/3679/7060
## 5                                                                                                                         9370/5105/2167/3158/5346
## 6                                                  4171/993/990/5347/701/9700/898/23594/4998/9134/4175/4173/10926/6502/994/699/4609/5111/1869/1029
##   Count
## 1    16
## 2    20
## 3    28
## 4    11
## 5     5
## 6    20</code></pre>
</div>
<div id="vis-compare-profile" class="section level2" number="14.3">
<h2>
<span class="header-section-number">14.3</span> Visualization of functional profile comparison<a class="anchor" aria-label="anchor" href="#vis-compare-profile"><i class="fas fa-link"></i></a>
</h2>
<div id="compare-dotplot" class="section level3" number="14.3.1">
<h3>
<span class="header-section-number">14.3.1</span> Dot plot<a class="anchor" aria-label="anchor" href="#compare-dotplot"><i class="fas fa-link"></i></a>
</h3>
<p>We can visualize the result using the <code><a href="https://rdrr.io/pkg/enrichplot/man/dotplot.html">dotplot()</a></code> method.</p>
<div class="sourceCode" id="cb149"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/dotplot.html">dotplot</a></span><span class="op">(</span><span class="va">ck</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/dotplot.html">dotplot</a></span><span class="op">(</span><span class="va">formula_res</span><span class="op">)</span></code></pre></div>


<div class="figure">
<span style="display:block;" id="fig:ccn"></span>
<img src="biomedicalKnowledge_files/figure-html/ccn-1.png" alt="Comparing enrichment results of multiple gene lists. (A) Using a named list of gene clusters, the results were displayed as multiple columns with each one represents an enrichment result of a gene cluster. (B) Using formula interface, the columns represent gene clusters defined by the formula." width="98%"><p class="caption">
Figure 14.1: <strong>Comparing enrichment results of multiple gene lists.</strong> (A) Using a named list of gene clusters, the results were displayed as multiple columns with each one represents an enrichment result of a gene cluster. (B) Using formula interface, the columns represent gene clusters defined by the formula.
</p>
</div>
<p>The fomula interface allows more complicated gene cluster definition. In Figure <a href="clusterprofiler-comparecluster.html#fig:ccn">14.1</a>B, the gene clusters were defined by two variables (i.e.Â <code>group</code> that divides genes into <code>upregulated</code> and <code>downregulated</code> and <code>othergroup</code> that divides the genes into two categories of <code>A</code> and <code>B</code>.). The <code><a href="https://rdrr.io/pkg/enrichplot/man/dotplot.html">dotplot()</a></code> function allows us to use one variable to divide the result into different facet and plot the result with other variables in each facet panel (Figure <a href="clusterprofiler-comparecluster.html#fig:ccf">14.2</a>).</p>


<div class="sourceCode" id="cb150"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/dotplot.html">dotplot</a></span><span class="op">(</span><span class="va">formula_res</span>, x<span class="op">=</span><span class="st">"group"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_grid.html">facet_grid</a></span><span class="op">(</span><span class="op">~</span><span class="va">othergroup</span><span class="op">)</span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:ccf"></span>
<img src="biomedicalKnowledge_files/figure-html/ccf-1.png" alt="Comparing enrichment results of multiple gene lists defined by multiple variable. The results were visualized as a dot plot with an x-axis representing one level of conditions (the group variable) and a facet panel indicating another level of conditions (the othergroup variable)." width="98%"><p class="caption">
Figure 14.2: <strong>Comparing enrichment results of multiple gene lists defined by multiple variable.</strong> The results were visualized as a dot plot with an x-axis representing one level of conditions (the <em>group</em> variable) and a facet panel indicating another level of conditions (the <em>othergroup</em> variable).
</p>
</div>
<p>By default, only top 5 (most significant) categories of each cluster
was plotted. User can changes the parameter <code>showCategory</code> to
specify how many categories of each cluster to be plotted, and if
<code>showCategory</code> was set to <code>NULL</code>, the whole result will
be plotted. The <code>showCategory</code> parameter also allows passing a vector of selected categories to <a href="faq.html#showing-specific-pathways">plot pathway of interests</a></p>
<p>The <code><a href="https://rdrr.io/pkg/enrichplot/man/dotplot.html">dotplot()</a></code> function tries to make the comparison among different clusters more informative and reasonable. After extracting <em>e.g.</em> 10 categories for each clusters, <a href="http://bioconductor.org/packages/clusterProfiler">clusterProfiler</a> tries to collect overlap of these categories among clusters. For example, <code>term A</code> is enriched in all the gene clusters (<em>e.g.</em>, <code>g1</code> and <code>g2</code>) and is in the 10 most significant categories of <code>g1</code> but not <code>g2</code>. <a href="http://bioconductor.org/packages/clusterProfiler">clusterProfiler</a> will capture this information and include <code>term A</code> in <code>g2</code> cluster to make the comparison in <code>dotplot</code> more reasonable. If users want to ignore these information, they can set <code>includeAll = FALSE</code> in <code><a href="https://rdrr.io/pkg/enrichplot/man/dotplot.html">dotplot()</a></code>, which is not recommended.</p>
<p>The <code><a href="https://rdrr.io/pkg/enrichplot/man/dotplot.html">dotplot()</a></code> function accepts a parameter <code>size</code> for setting the scale of dot sizes. The default parameter <code>size</code> is setting to <code>geneRatio</code>, which corresponding to the <code>GeneRatio</code> column of the output. If it was setting to <code>count</code>, the comparison will be based on gene counts, while if setting to <code>rowPercentage</code>, the dot sizes will be normalized by <code>count/(sum of each row)</code>. Users can also map the dot size to other variables or derived variables (see <a href="clusterProfiler-dplyr.html#clusterProfiler-dplyr">Chapter 16</a>).</p>
<p>To provide the full information, we also provide number of identified genes in each category (numbers in parentheses) when <code>by</code> is setting to <code>rowPercentage</code> and number of gene clusters in each cluster label (numbers in parentheses) when <code>by</code> is setting to <code>geneRatio</code>, as shown in Figure <a href="clusterprofiler-comparecluster.html#fig:ccn">14.1</a>.</p>
<p>The p-values indicate that which categories are more likely to have biological meanings. The dots in the plot are color-coded based on their corresponding adjusted p-values. Color gradient ranging from red to blue correspond to the order of increasing adjusted p-values. That is, red indicate low p-values (high enrichment), and blue indicate high p-values (low enrichment). Adjusted p-values were filtered out by the threshold giving by the parameter <code>pvalueCutoff</code>, and FDR can be estimated by <code>qvalue</code>.</p>
</div>
<div id="compare-cnetplot" class="section level3" number="14.3.2">
<h3>
<span class="header-section-number">14.3.2</span> Gene-Concept Network<a class="anchor" aria-label="anchor" href="#compare-cnetplot"><i class="fas fa-link"></i></a>
</h3>
<p>The <a href="enrichplot.html#cnetplot">cnetplot</a> also works with <code><a href="https://rdrr.io/pkg/clusterProfiler/man/compareCluster.html">compareCluster()</a></code> result.</p>


<div class="sourceCode" id="cb151"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/cnetplot.html">cnetplot</a></span><span class="op">(</span><span class="va">ck</span><span class="op">)</span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:mcnetplot"></span>
<img src="biomedicalKnowledge_files/figure-html/mcnetplot-1.png" alt="cnetplot() for comparing functional profiles of multiple gene clusters. Genes and functional categories (i.e., pathways) are encoded as pies to distinguish different gene clusters." width="98%"><p class="caption">
Figure 14.3: <strong><code><a href="https://rdrr.io/pkg/enrichplot/man/cnetplot.html">cnetplot()</a></code> for comparing functional profiles of multiple gene clusters.</strong> Genes and functional categories (<em>i.e.</em>, pathways) are encoded as pies to distinguish different gene clusters.
</p>
</div>
</div>
</div>
<div id="summary-1" class="section level2" number="14.4">
<h2>
<span class="header-section-number">14.4</span> Summary<a class="anchor" aria-label="anchor" href="#summary-1"><i class="fas fa-link"></i></a>
</h2>
<p>The comparison function was designed as a framework for comparing gene clusters of any kind of ontology associations, not only <a href="clusterprofiler-go.html#go-classification">groupGO</a>, <a href="clusterprofiler-go.html#clusterprofiler-go-ora">enrichGO</a>, <a href="clusterprofiler-kegg.html#clusterprofiler-kegg-pathway-ora">enrichKEGG</a>, <a href="clusterprofiler-kegg.html#clusterprofiler-kegg-module-ora">enrichMKEGG</a>, <a href="wikipathways-analysis.html#clusterprofiler-wikipathway-ora">enrichWP</a> and <a href="universal-api.html#universal-api">enricher</a> that were provided in this package, but also other biological and biomedical ontologies, including but not limited to <a href="reactomepa.html#reactomepa-ora">enrichPathway</a>, <a href="dose-enrichment.html#dose-do-ora">enrichDO</a>, <a href="dose-enrichment.html#dose-ncg-ora">enrichNCG</a>, <a href="dose-enrichment.html#dose-dgn-ora">enrichDGN</a> and <a href="meshes-enrichment.html#meshes-ora">enrichMeSH</a>.</p>
<p>In <span class="citation">(<a href="references.html#ref-yu2012" role="doc-biblioref">Yu et al. 2012</a>)</span>, we analyzed the publicly available expression dataset of breast tumour tissues from 200 patients (GSE11121, Gene Expression Omnibus) <span class="citation">(<a href="references.html#ref-schmidt2008" role="doc-biblioref">Schmidt et al. 2008</a>)</span>. We identified 8 gene clusters from differentially expressed genes, and using the <code><a href="https://rdrr.io/pkg/clusterProfiler/man/compareCluster.html">compareCluster()</a></code> function to compare these gene clusters by their enriched biological process. In <span class="citation">(<a href="references.html#ref-wu_clusterprofiler_2021" role="doc-biblioref">Wu et al. 2021</a>)</span>, we analyzed the GSE8057 dataset which contains expression data from ovarian cancer cells at multiple time points and under two treatment conditions. Eight groups of DEG lists were analyzed simultaneously using <code><a href="https://rdrr.io/pkg/clusterProfiler/man/compareCluster.html">compareCluster()</a></code> with WikiPathways. The result indicates that the two drgus have distinct effects at the begining but consistent effects in the later stages (Fig. 4 of <span class="citation">(<a href="references.html#ref-wu_clusterprofiler_2021" role="doc-biblioref">Wu et al. 2021</a>)</span>).</p>

</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="chipseeker-enrichment.html"><span class="header-section-number">13</span> Functional enrichment analysis of genomic coordinations</a></div>
<div class="next"><a href="enrichplot.html"><span class="header-section-number">15</span> Visualization of functional enrichment result</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#clusterprofiler-comparecluster"><span class="header-section-number">14</span> Biological theme comparison</a></li>
<li><a class="nav-link" href="#comparing-multiple-gene-lists"><span class="header-section-number">14.1</span> Comparing multiple gene lists</a></li>
<li><a class="nav-link" href="#formula-interface-of-comparecluster"><span class="header-section-number">14.2</span> Formula interface of compareCluster</a></li>
<li>
<a class="nav-link" href="#vis-compare-profile"><span class="header-section-number">14.3</span> Visualization of functional profile comparison</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#compare-dotplot"><span class="header-section-number">14.3.1</span> Dot plot</a></li>
<li><a class="nav-link" href="#compare-cnetplot"><span class="header-section-number">14.3.2</span> Gene-Concept Network</a></li>
</ul>
</li>
<li><a class="nav-link" href="#summary-1"><span class="header-section-number">14.4</span> Summary</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
          
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Biomedical Knowledge Mining using GOSemSim and clusterProfiler</strong>" was written by Guangchuang Yu. It was last built on 2021-09-02.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer><!-- dynamically load mathjax for compatibility with self-contained --><script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script><script type="text/x-mathjax-config">const popovers = document.querySelectorAll('a.footnote-ref[data-toggle="popover"]');
for (let popover of popovers) {
  const div = document.createElement('div');
  div.setAttribute('style', 'position: absolute; top: 0, left:0; width:0, height:0, overflow: hidden; visibility: hidden;');
  div.innerHTML = popover.getAttribute('data-content');

  var has_math = div.querySelector("span.math");
  if (has_math) {
    document.body.appendChild(div);
    MathJax.Hub.Queue(["Typeset", MathJax.Hub, div]);
    MathJax.Hub.Queue(function() {
      popover.setAttribute('data-content', div.innerHTML);
      document.body.removeChild(div);
    })
  }
}
</script>
</body>
</html>
