<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 3 DO semantic similarity analysis | Biomedical Knowledge Mining using GOSemSim and clusterProfiler</title>
<meta name="author" content="Guangchuang Yu">
<meta name="description" content="Public health is an important driving force behind biological and medical research. A major challenge of the post-genomic era is bridging the gap between fundamental biological research and its...">
<meta name="generator" content="bookdown 0.26 with bs4_book()">
<meta property="og:title" content="Chapter 3 DO semantic similarity analysis | Biomedical Knowledge Mining using GOSemSim and clusterProfiler">
<meta property="og:type" content="book">
<meta property="og:description" content="Public health is an important driving force behind biological and medical research. A major challenge of the post-genomic era is bridging the gap between fundamental biological research and its...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Chapter 3 DO semantic similarity analysis | Biomedical Knowledge Mining using GOSemSim and clusterProfiler">
<meta name="twitter:description" content="Public health is an important driving force behind biological and medical research. A major challenge of the post-genomic era is bridging the gap between fundamental biological research and its...">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.3.1/transition.js"></script><script src="libs/bs3compat-0.3.1/tabs.js"></script><script src="libs/bs3compat-0.3.1/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<style type="text/css">
    /* Used with Pandoc 2.11+ new --citeproc when CSL is used */
    div.csl-bib-body { }
    div.csl-entry {
      clear: both;
        }
    .hanging div.csl-entry {
      margin-left:2em;
      text-indent:-2em;
    }
    div.csl-left-margin {
      min-width:2em;
      float:left;
    }
    div.csl-right-inline {
      margin-left:2em;
      padding-left:1em;
    }
    div.csl-indent {
      margin-left: 2em;
    }
  </style>
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">Biomedical Knowledge Mining using GOSemSim and clusterProfiler</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html">ðŸ“– Introduction</a></li>
<li class="book-part">Part I: Semantic Similarity Analysis</li>
<li><a class="" href="semantic-similarity-overview.html"><span class="header-section-number">1</span> Overview of semantic similarity analysis</a></li>
<li><a class="" href="GOSemSim.html"><span class="header-section-number">2</span> GO semantic similarity analysis</a></li>
<li><a class="active" href="DOSE-semantic-similarity.html"><span class="header-section-number">3</span> DO semantic similarity analysis</a></li>
<li><a class="" href="meshes-semantic-similarity.html"><span class="header-section-number">4</span> MeSH semantic similarity analysis</a></li>
<li class="book-part">Part II: Enrichment analysis</li>
<li><a class="" href="enrichment-overview.html"><span class="header-section-number">5</span> Overview of enrichment analysis</a></li>
<li><a class="" href="clusterprofiler-go.html"><span class="header-section-number">6</span> GO enrichment analysis</a></li>
<li><a class="" href="clusterprofiler-kegg.html"><span class="header-section-number">7</span> KEGG enrichment analysis</a></li>
<li><a class="" href="wikipathways-analysis.html"><span class="header-section-number">8</span> WikiPathways analysis</a></li>
<li><a class="" href="reactomepa.html"><span class="header-section-number">9</span> Reactome enrichment analysis</a></li>
<li><a class="" href="dose-enrichment.html"><span class="header-section-number">10</span> Disease enrichment analysis</a></li>
<li><a class="" href="meshes-enrichment.html"><span class="header-section-number">11</span> MeSH enrichment analysis</a></li>
<li><a class="" href="universal-api.html"><span class="header-section-number">12</span> Universal enrichment analysis</a></li>
<li><a class="" href="chipseeker-enrichment.html"><span class="header-section-number">13</span> Functional enrichment analysis of genomic coordinations</a></li>
<li><a class="" href="clusterprofiler-comparecluster.html"><span class="header-section-number">14</span> Biological theme comparison</a></li>
<li><a class="" href="enrichplot.html"><span class="header-section-number">15</span> Visualization of functional enrichment result</a></li>
<li class="book-part">Part III: Miscellaneous topics</li>
<li><a class="" href="clusterProfiler-dplyr.html"><span class="header-section-number">16</span> dplyr verbs for manipulating enrichment result</a></li>
<li><a class="" href="useful-utilities.html"><span class="header-section-number">17</span> Useful utilities</a></li>
<li class="book-part">Appendix</li>
<li><a class="" href="faq.html"><span class="header-section-number">A</span> Frequently asked questions</a></li>
<li><a class="" href="references.html">References</a></li>
</ul>

        <div class="book-extra">
          
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="DOSE-semantic-similarity" class="section level1" number="3">
<h1>
<span class="header-section-number">3</span> DO semantic similarity analysis<a class="anchor" aria-label="anchor" href="#DOSE-semantic-similarity"><i class="fas fa-link"></i></a>
</h1>
<p>Public health is an important driving force behind biological and medical research.
A major challenge of the post-genomic era is bridging the gap between fundamental biological research and its clinical applications.
Recent research has increasingly demonstrated that many seemingly dissimilar diseases have common molecular mechanisms.
Understanding similarities among disease aids in early diagnosis and new drug development.</p>
<p>Formal knowledge representation of gene-disease association is demanded for this purpose.
Ontologies, such as Gene Ontology (GO), have been successfully applied to represent biological knowledge, and many related techniques have been adopted to extract information.
Disease Ontology (DO) <span class="citation">(<a href="references.html#ref-schriml_disease_2011" role="doc-biblioref">Schriml et al. 2011</a>)</span> was developed to create a consistent description of gene products with disease perspectives, and is essential for supporting functional genomics in disease context.
Accurate disease descriptions can discover new relationships between genes and disease, and new functions for previous uncharacteried genes and alleles.</p>
<p>Disease Ontology (DO) <span class="citation">(<a href="references.html#ref-schriml_disease_2011" role="doc-biblioref">Schriml et al. 2011</a>)</span> aims to provide an open source ontology for the integration of biomedical data that is associated with human disease. Unlike other clinical vocabularies that defined disease related concepts disparately, DO is organized as a directed acyclic graph, laying the foundation for quantitative computation of disease knowledge.</p>
<p>We developed an <code>R</code> package <a href="http://bioconductor.org/packages/DOSE">DOSE</a> <span class="citation">(<a href="references.html#ref-yu_dose_2015" role="doc-biblioref">Yu et al. 2015</a>)</span> for analyzing semantic similarities among DO terms and gene products annotated with DO terms.</p>
<div id="do-semantic-similarity" class="section level2" number="3.1">
<h2>
<span class="header-section-number">3.1</span> DO term semantic similarity measurement<a class="anchor" aria-label="anchor" href="#do-semantic-similarity"><i class="fas fa-link"></i></a>
</h2>
<p>Four methods determine the semantic similarity of two terms based on the Information Content of their common ancestor term were proposed by Resnik <span class="citation">(<a href="references.html#ref-philip_semantic_1999" role="doc-biblioref">Philip 1999</a>)</span>, Jiang <span class="citation">(<a href="references.html#ref-jiang_semantic_1997" role="doc-biblioref">Jiang and Conrath 1997</a>)</span>, Lin <span class="citation">(<a href="references.html#ref-lin_information-theoretic_1998" role="doc-biblioref">Lin 1998</a>)</span> and Schlicker <span class="citation">(<a href="references.html#ref-schlicker_new_2006" role="doc-biblioref">Schlicker et al. 2006</a>)</span>. Wang <span class="citation">(<a href="references.html#ref-wang_new_2007" role="doc-biblioref">Wang et al. 2007</a>)</span> presented a method to measure the similarity based on the graph structure. Each of these methods has its own advantage and weakness. <a href="http://bioconductor.org/packages/DOSE">DOSE</a> supports all these methods to compute semantic similarity among DO terms and gene products. For algorithm details, please refer to <a href="semantic-similarity-overview.html#semantic-similarity-overview">chapter 1</a>.</p>
<p>In <a href="http://bioconductor.org/packages/DOSE">DOSE</a>, we implemented <code>doSim</code> for calculating semantic similarity between two DO terms and two set of DO terms.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://yulab-smu.top/biomedical-knowledge-mining-book/">DOSE</a></span><span class="op">)</span>

<span class="va">a</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"DOID:14095"</span>, <span class="st">"DOID:5844"</span>, <span class="st">"DOID:2044"</span>, <span class="st">"DOID:8432"</span>, <span class="st">"DOID:9146"</span>,
       <span class="st">"DOID:10588"</span>, <span class="st">"DOID:3209"</span>, <span class="st">"DOID:848"</span>, <span class="st">"DOID:3341"</span>, <span class="st">"DOID:252"</span><span class="op">)</span>
<span class="va">b</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"DOID:9409"</span>, <span class="st">"DOID:2491"</span>, <span class="st">"DOID:4467"</span>, <span class="st">"DOID:3498"</span>, <span class="st">"DOID:11256"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/DOSE/man/doSim.html">doSim</a></span><span class="op">(</span><span class="va">a</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>, <span class="va">b</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>, measure<span class="op">=</span><span class="st">"Wang"</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 0.07142995</code></pre>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/DOSE/man/doSim.html">doSim</a></span><span class="op">(</span><span class="va">a</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>, <span class="va">b</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>, measure<span class="op">=</span><span class="st">"Resnik"</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 0</code></pre>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/DOSE/man/doSim.html">doSim</a></span><span class="op">(</span><span class="va">a</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>, <span class="va">b</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>, measure<span class="op">=</span><span class="st">"Lin"</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 0</code></pre>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">s</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DOSE/man/doSim.html">doSim</a></span><span class="op">(</span><span class="va">a</span>, <span class="va">b</span>, measure<span class="op">=</span><span class="st">"Wang"</span><span class="op">)</span>
<span class="va">s</span></code></pre></div>
<pre><code>##             DOID:9409  DOID:2491  DOID:4467  DOID:3498 DOID:11256
## DOID:14095 0.07142995 0.05714393 0.03676194 0.03676194 0.52749870
## DOID:5844  0.14897652 0.11564838 0.02801328 0.02801328 0.06134327
## DOID:2044  0.14897652 0.11564838 0.02801328 0.02801328 0.06134327
## DOID:8432  0.17347273 0.13877811 0.03676194 0.03676194 0.07142995
## DOID:9146  0.07142995 0.05714393 0.03676194 0.03676194 0.17347273
## DOID:10588 0.13240905 0.18401515 0.02208240 0.02208240 0.05452137
## DOID:3209  0.14897652 0.11564838 0.02801328 0.02801328 0.06134327
## DOID:848   0.14897652 0.11564838 0.02801328 0.02801328 0.06134327
## DOID:3341  0.13240905 0.09998997 0.02208240 0.02208240 0.05452137
## DOID:252   0.06134327 0.04761992 0.02801328 0.02801328 0.06134327</code></pre>
<p>The <code>doSim</code> function requires three parameter <code>DOID1</code>, <code>DOID2</code> and <code>measure</code>. <code>DOID1</code> and <code>DOID2</code> should be a vector of DO terms, while <code>measure</code> should be one of <em>Resnik</em>, <em>Jiang</em>, <em>Lin</em>, <em>Rel</em>, and <em>Wang</em>.</p>
<p>We also implement a plot function <code><a href="https://rdrr.io/pkg/DOSE/man/simplot.html">simplot()</a></code> to visualize the similarity result as demonstrated in Figure <a href="DOSE-semantic-similarity.html#fig:dose-simplot">3.1</a>.</p>


<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/DOSE/man/simplot.html">simplot</a></span><span class="op">(</span><span class="va">s</span>,
        color.low<span class="op">=</span><span class="st">"white"</span>, color.high<span class="op">=</span><span class="st">"red"</span>,
        labs<span class="op">=</span><span class="cn">TRUE</span>, digits<span class="op">=</span><span class="fl">2</span>, labs.size<span class="op">=</span><span class="fl">5</span>,
        font.size<span class="op">=</span><span class="fl">14</span>, xlab<span class="op">=</span><span class="st">""</span>, ylab<span class="op">=</span><span class="st">""</span><span class="op">)</span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:dose-simplot"></span>
<img src="biomedicalKnowledge_files/figure-html/dose-simplot-1.png" alt="Similarity matrix visualization. The simplot() function visualize similarity matrix as a heatmap." width="98%"><p class="caption">
Figure 3.1: <strong>Similarity matrix visualization</strong>. The <code><a href="https://rdrr.io/pkg/DOSE/man/simplot.html">simplot()</a></code> function visualize similarity matrix as a heatmap.
</p>
</div>
<p>Parameter <code>color.low</code> and <code>colow.high</code> are used to setting the color gradient; <code>labs</code> is a logical parameter indicating whether to show the similarity values or not, <code>digits</code> to indicate the number of decimal places to be used and <code>labs.size</code> control the font size of similarity values; <code>font.size</code> setting the font size of axis and label of the coordinate system.</p>
</div>
<div id="gene-do-semantic-similarity" class="section level2" number="3.2">
<h2>
<span class="header-section-number">3.2</span> Gene semantic similarity measurement<a class="anchor" aria-label="anchor" href="#gene-do-semantic-similarity"><i class="fas fa-link"></i></a>
</h2>
<p>On the basis of semantic similarity between DO terms, <a href="http://bioconductor.org/packages/DOSE">DOSE</a> can also compute semantic similarity among gene products. <a href="http://bioconductor.org/packages/DOSE">DOSE</a> provides four methods which called <code>max</code>, <code>avg</code>, <code>rcmax</code> and <code>BMA</code> to combine semantic similarity scores of multiple DO terms (as described in <a href="combine-methods">session 1.3</a>). The similarities among genes and gene clusters which annotated by multiple DO terms were also calculated by these combine methods.</p>
<p><a href="http://bioconductor.org/packages/DOSE">DOSE</a> implemented <code>geneSim</code> to measure semantic similarities among genes.</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">g1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"84842"</span>, <span class="st">"2524"</span>, <span class="st">"10590"</span>, <span class="st">"3070"</span>, <span class="st">"91746"</span><span class="op">)</span>
<span class="va">g2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"84289"</span>, <span class="st">"6045"</span>, <span class="st">"56999"</span>, <span class="st">"9869"</span><span class="op">)</span>

<span class="fu">DOSE</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/DOSE/man/geneSim.html">geneSim</a></span><span class="op">(</span><span class="va">g1</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>, <span class="va">g2</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>, measure<span class="op">=</span><span class="st">"Wang"</span>, combine<span class="op">=</span><span class="st">"BMA"</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 0.051</code></pre>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gs</span> <span class="op">&lt;-</span> <span class="fu">DOSE</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/DOSE/man/geneSim.html">geneSim</a></span><span class="op">(</span><span class="va">g1</span>, <span class="va">g2</span>, measure<span class="op">=</span><span class="st">"Wang"</span>, combine<span class="op">=</span><span class="st">"BMA"</span><span class="op">)</span>
<span class="va">gs</span></code></pre></div>
<pre><code>##       84289  6045 56999  9869
## 84842 0.051 0.135 0.355 0.103
## 2524  0.284 0.172 0.517 0.517
## 10590 0.150 0.173 0.242 0.262
## 3070  0.573 0.517 1.000 1.000
## 91746 0.351 0.308 0.527 0.496</code></pre>
<p>The <code>geneSim</code> requires four parameter <code>geneID1</code>, <code>geneID2</code>, <code>measure</code> and <code>combine</code>. <code>geneID1</code> and <code>geneID2</code> should be a vector of entrez gene IDs; <code>measure</code> should be one of <em>Resnik</em>, <em>Jiang</em>, <em>Lin</em>, <em>Rel</em>, and <em>Wang</em>, while <code>combine</code> should be one of <em>max</em>, <em>avg</em>, <em>rcmax</em> and <em>BMA</em>.</p>
</div>
<div id="gene-cluster-do-semantic-similarity" class="section level2" number="3.3">
<h2>
<span class="header-section-number">3.3</span> Gene cluster semantic similarity measurement<a class="anchor" aria-label="anchor" href="#gene-cluster-do-semantic-similarity"><i class="fas fa-link"></i></a>
</h2>
<p><a href="http://bioconductor.org/packages/DOSE">DOSE</a> also implemented <code>clusterSim</code> for calculating semantic similarity between two gene clusters and <code>mclusterSim</code> for calculating semantic similarities among multiple gene clusters.</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">DOSE</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/DOSE/man/clusterSim.html">clusterSim</a></span><span class="op">(</span><span class="va">g1</span>, <span class="va">g2</span>, measure<span class="op">=</span><span class="st">"Wang"</span>, combine<span class="op">=</span><span class="st">"BMA"</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 0.549</code></pre>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">g3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"57491"</span>, <span class="st">"6296"</span>, <span class="st">"51438"</span>, <span class="st">"5504"</span>, <span class="st">"27319"</span>, <span class="st">"1643"</span><span class="op">)</span>
<span class="va">clusters</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>a<span class="op">=</span><span class="va">g1</span>, b<span class="op">=</span><span class="va">g2</span>, c<span class="op">=</span><span class="va">g3</span><span class="op">)</span>
<span class="fu">DOSE</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/DOSE/man/mclusterSim.html">mclusterSim</a></span><span class="op">(</span><span class="va">clusters</span>, measure<span class="op">=</span><span class="st">"Wang"</span>, combine<span class="op">=</span><span class="st">"BMA"</span><span class="op">)</span></code></pre></div>
<pre><code>##       a     b     c
## a 1.000 0.549 0.425
## b 0.549 1.000 0.645
## c 0.425 0.645 1.000</code></pre>
<!--

# GO semantic similarity calculation

GO Semantic similarity can be calculated by [GOSemSim](http://bioconductor.org/packages/GOSemSim)[@yu2010].

We have developed another package [GOSemSim](http://bioconductor.org/packages/GOSemSim)[@yu2010] to explore the functional similarity at GO perspective, including molecular function (MF), biological process (BP) and cellular component (CC).


# MeSH semantic analysis

MeSH (Medical Subject Headings) is the NLM controlled vocabulary used
to manually index articles for MEDLINE/PubMed. [meshes](http://bioconductor.org/packages/meshes)
supports enrichment (hypergeometric test and GSEA) and semantic
similarity analyses for more than 70 species.

-->

</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="GOSemSim.html"><span class="header-section-number">2</span> GO semantic similarity analysis</a></div>
<div class="next"><a href="meshes-semantic-similarity.html"><span class="header-section-number">4</span> MeSH semantic similarity analysis</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#DOSE-semantic-similarity"><span class="header-section-number">3</span> DO semantic similarity analysis</a></li>
<li><a class="nav-link" href="#do-semantic-similarity"><span class="header-section-number">3.1</span> DO term semantic similarity measurement</a></li>
<li><a class="nav-link" href="#gene-do-semantic-similarity"><span class="header-section-number">3.2</span> Gene semantic similarity measurement</a></li>
<li><a class="nav-link" href="#gene-cluster-do-semantic-similarity"><span class="header-section-number">3.3</span> Gene cluster semantic similarity measurement</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
          
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Biomedical Knowledge Mining using GOSemSim and clusterProfiler</strong>" was written by Guangchuang Yu. It was last built on 2022-04-24.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer><!-- dynamically load mathjax for compatibility with self-contained --><script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script><script type="text/x-mathjax-config">const popovers = document.querySelectorAll('a.footnote-ref[data-toggle="popover"]');
for (let popover of popovers) {
  const div = document.createElement('div');
  div.setAttribute('style', 'position: absolute; top: 0, left:0; width:0, height:0, overflow: hidden; visibility: hidden;');
  div.innerHTML = popover.getAttribute('data-content');

  var has_math = div.querySelector("span.math");
  if (has_math) {
    document.body.appendChild(div);
    MathJax.Hub.Queue(["Typeset", MathJax.Hub, div]);
    MathJax.Hub.Queue(function() {
      popover.setAttribute('data-content', div.innerHTML);
      document.body.removeChild(div);
    })
  }
}
</script>
</body>
</html>
