<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 5 Overview of enrichment analysis | Biomedical Knowledge Mining using GOSemSim and clusterProfiler</title>
<meta name="author" content="Guangchuang Yu">
<!-- CSS --><!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.2"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/header-attrs-2.6/header-attrs.js"></script><script src="libs/jquery-3.5.1/jquery-3.5.1.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.5.2/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.5.2/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.2.1.9000/tabs.js"></script><script src="libs/bs3compat-0.2.1.9000/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="https://cdn.jsdelivr.net/autocomplete.js/0/autocomplete.jquery.min.js"></script><script src="https://cdn.jsdelivr.net/npm/mark.js@8.11.1/dist/mark.min.js"></script>
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">Biomedical Knowledge Mining using GOSemSim and clusterProfiler</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html">ðŸ“– Introduction</a></li>
<li class="book-part">Part I: Semantic Similarity Analysis</li>
<li><a class="" href="semantic-similarity-overview.html"><span class="header-section-number">1</span> Overview of semantic similarity analysis</a></li>
<li><a class="" href="GOSemSim.html"><span class="header-section-number">2</span> GO semantic similarity analysis</a></li>
<li><a class="" href="DOSE-semantic-similarity.html"><span class="header-section-number">3</span> DO semantic similarity analysis</a></li>
<li><a class="" href="meshes-semantic-similarity.html"><span class="header-section-number">4</span> MeSH semantic similarity analysis</a></li>
<li class="book-part">Part II: Enrichment analysis</li>
<li><a class="active" href="enrichment-overview.html"><span class="header-section-number">5</span> Overview of enrichment analysis</a></li>
<li><a class="" href="clusterprofiler-go.html"><span class="header-section-number">6</span> GO enrichment analysis</a></li>
<li><a class="" href="clusterprofiler-kegg.html"><span class="header-section-number">7</span> KEGG enrichment analysis</a></li>
<li><a class="" href="wikipathways-analysis.html"><span class="header-section-number">8</span> WikiPathways analysis</a></li>
<li><a class="" href="reactomepa.html"><span class="header-section-number">9</span> Reactome enrichment analysis</a></li>
<li><a class="" href="dose-enrichment.html"><span class="header-section-number">10</span> Disease enrichment analysis</a></li>
<li><a class="" href="meshes-enrichment.html"><span class="header-section-number">11</span> MeSH enrichment analysis</a></li>
<li><a class="" href="universal-api.html"><span class="header-section-number">12</span> Universal enrichment analysis</a></li>
<li><a class="" href="chipseeker-enrichment.html"><span class="header-section-number">13</span> Functional enrichment analysis of genomic coordinations</a></li>
<li><a class="" href="clusterprofiler-comparecluster.html"><span class="header-section-number">14</span> Biological theme comparison</a></li>
<li><a class="" href="enrichplot.html"><span class="header-section-number">15</span> Visualization of functional enrichment result</a></li>
<li class="book-part">Part III: Miscellaneous topics</li>
<li><a class="" href="clusterProfiler-dplyr.html"><span class="header-section-number">16</span> dplyr verbs for manipulating enrichment result</a></li>
<li><a class="" href="useful-utilities.html"><span class="header-section-number">17</span> Useful utilities</a></li>
<li class="book-part">Appendix</li>
<li><a class="" href="faq.html"><span class="header-section-number">A</span> Frequently asked questions</a></li>
<li><a class="" href="references.html">References</a></li>
</ul>

        <div class="book-extra">
          
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="enrichment-overview" class="section level1" number="5">
<h1>
<span class="header-section-number">5</span> Overview of enrichment analysis<a class="anchor" aria-label="anchor" href="#enrichment-overview"><i class="fas fa-link"></i></a>
</h1>
<div id="terminology" class="section level2" number="5.1">
<h2>
<span class="header-section-number">5.1</span> Terminology<a class="anchor" aria-label="anchor" href="#terminology"><i class="fas fa-link"></i></a>
</h2>
<div id="gene-sets-and-pathway" class="section level3" number="5.1.1">
<h3>
<span class="header-section-number">5.1.1</span> Gene sets and pathway<a class="anchor" aria-label="anchor" href="#gene-sets-and-pathway"><i class="fas fa-link"></i></a>
</h3>
<p>A gene set is an unordered collection of genes that are functional related. A pathway can be interpreted as a gene set by ignoring functional relationships among genes.</p>
</div>
<div id="gene-ontology-go" class="section level3" number="5.1.2">
<h3>
<span class="header-section-number">5.1.2</span> Gene Ontology (GO)<a class="anchor" aria-label="anchor" href="#gene-ontology-go"><i class="fas fa-link"></i></a>
</h3>
<p><a href="http://www.geneontology.org/">Gene Ontology</a> defines concepts/classes used to describe gene function, and relationships between these concepts. It classifies functions along three aspects:</p>
<ul>
<li>MF: Molecular Function
<ul>
<li>molecular activities of gene products</li>
</ul>
</li>
<li>CC: Cellular Component
<ul>
<li>where gene products are active</li>
</ul>
</li>
<li>BP: Biological Process
<ul>
<li>pathways and larger processes made up of the activities of multiple gene products</li>
</ul>
</li>
</ul>
<p>GO terms are organized in a directed acyclic graph, where edge between the terms represent parent-child relationship.</p>
</div>
<div id="kyoto-encyclopedia-of-genes-and-genomes-kegg" class="section level3" number="5.1.3">
<h3>
<span class="header-section-number">5.1.3</span> Kyoto Encyclopedia of Genes and Genomes (KEGG)<a class="anchor" aria-label="anchor" href="#kyoto-encyclopedia-of-genes-and-genomes-kegg"><i class="fas fa-link"></i></a>
</h3>
<p><a href="https://www.genome.jp/kegg/">KEGG</a> is a collection of manually drawn pathway maps representing molecular interaction and reaction networks. These pathways cover a wide range of biochemical processes that can be divided in <a href="https://www.genome.jp/kegg/pathway.html">7 broad categories</a>:</p>
<ol style="list-style-type: decimal">
<li>Metabolism</li>
<li>Genetic information processing</li>
<li>Environmental information processing</li>
<li>Cellular processes</li>
<li>Organismal systems</li>
<li>Human diseases</li>
<li>Drug development.</li>
</ol>
<!-- https://pathview.uncc.edu/data/khier.tsv -->
</div>
<div id="other-gene-sets" class="section level3" number="5.1.4">
<h3>
<span class="header-section-number">5.1.4</span> Other gene sets<a class="anchor" aria-label="anchor" href="#other-gene-sets"><i class="fas fa-link"></i></a>
</h3>
<p>GO and KEGG are most frequently used for the functional analysis. They are typically the first choice because their long-standing curation and availability for a wide range of species.</p>
<p>Other gene sets including but not limited to Disease Ontology (<a href="http://disease-ontology.org/">DO</a>), Disease Gene Network (<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4397996/">DisGeNET</a>), <a href="https://www.wikipathways.org">wikiPathways</a>, Molecular Signatures Database (<a href="http://software.broadinstitute.org/gsea/msigdb">MSigDb</a>).</p>
</div>
</div>
<div id="ora-algorithm" class="section level2" number="5.2">
<h2>
<span class="header-section-number">5.2</span> Over Representation Analysis<a class="anchor" aria-label="anchor" href="#ora-algorithm"><i class="fas fa-link"></i></a>
</h2>
<p>Over Representation Analysis (ORA) <span class="citation">(Boyle et al. <a href="references.html#ref-boyle2004" role="doc-biblioref">2004</a>)</span> is a widely used approach to determine whether known biological functions or processes are over-represented (= enriched) in an experimentally-derived gene list, <em>e.g.</em> a list of differentially expressed genes (DEGs).</p>
<p>The <em>p</em>-value can be calculated by hypergeometric distribution.</p>
<p><span class="math inline">\(p = 1 - \displaystyle\sum_{i = 0}^{k-1}\frac{{M \choose i}{{N-M} \choose {n-i}}} {{N \choose n}}\)</span></p>
<p>In this equation, <code>N</code> is the total number of genes in the background distribution,
<code>M</code> is the number of genes within that distribution that are annotated (either directly or indirectly) to the gene set of interest,
<code>n</code> is the size of the list of genes of interest and <code>k</code> is the number of genes within that list which
are annotated to the gene set. The background distribution by default is
all the genes that have annotation. <em>P</em>-values should be adjusted for <a href="https://en.wikipedia.org/wiki/Multiple_comparisons_problem">multiple comparison</a>.</p>
<p><strong>Example:</strong> Suppose we have 17,980 genes detected in a Microarray study and 57 genes were differentially expressed. Among the differential expressed genes, 28 are annotated to a gene set<a class="footnote-ref" tabindex="0" data-toggle="popover" data-content='&lt;p&gt;example adopted from &lt;a href="https://guangchuangyu.github.io/2012/04/enrichment-analysis/" class="uri"&gt;https://guangchuangyu.github.io/2012/04/enrichment-analysis/&lt;/a&gt;&lt;/p&gt;'><sup>1</sup></a>.</p>
<div class="sourceCode" id="cb55"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">d</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>gene.not.interest<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">2613</span>, <span class="fl">15310</span><span class="op">)</span>, gene.in.interest<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">28</span>, <span class="fl">29</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/row.names.html">row.names</a></span><span class="op">(</span><span class="va">d</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"In_category"</span>, <span class="st">"not_in_category"</span><span class="op">)</span>
<span class="va">d</span></code></pre></div>
<pre><code>##                 gene.not.interest gene.in.interest
## In_category                  2613               28
## not_in_category             15310               29</code></pre>
<p>whether the overlap of 25 genes are significantly over represented in the gene set can be assessed using hypergeometric distribution. This corresponding to a one-sided version of Fisherâ€™s exact test.</p>
<div class="sourceCode" id="cb57"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/stats/fisher.test.html">fisher.test</a></span><span class="op">(</span><span class="va">d</span>, alternative <span class="op">=</span> <span class="st">"greater"</span><span class="op">)</span></code></pre></div>
<pre><code>## 
##  Fisher's Exact Test for Count Data
## 
## data:  d
## p-value = 1
## alternative hypothesis: true odds ratio is greater than 1
## 95 percent confidence interval:
##  0.110242      Inf
## sample estimates:
## odds ratio 
##  0.1767937</code></pre>
</div>
<div id="gsea-algorithm" class="section level2" number="5.3">
<h2>
<span class="header-section-number">5.3</span> Gene Set Enrichment Analysis<a class="anchor" aria-label="anchor" href="#gsea-algorithm"><i class="fas fa-link"></i></a>
</h2>
<p>A common approach in analyzing gene expression profiles was identifying differential expressed genes that are deemed interesting. The <a href="enrichment-overview.html#ora-algorithm">ORA enrichment analysis</a> is based on these differential expressed genes. This approach will find genes where the difference is large, but it will not detect a situation where the difference is small, but evidenced in coordinated way in a set of related genes. Gene Set Enrichment Analysis (GSEA)<span class="citation">(Subramanian et al. <a href="references.html#ref-subramanian_gene_2005" role="doc-biblioref">2005</a>)</span> directly addresses this limitation.
All genes can be used in GSEA; GSEA aggregates the per gene statistics across genes within a gene set, therefore making it possible to detect situations where all genes in a predefined set change in a small but coordinated way. Since it is likely that many relevant phenotypic differences are manifested by small but consistent changes in a set of genes.</p>
<p>Genes are ranked based on their phenotypes. Given a priori defined set of gene <em>S</em> (e.g., genes shareing the same <em>DO</em> category), the goal of GSEA is to determine whether the members of <em>S</em> are randomly distributed throughout the ranked gene list (<em>L</em>) or primarily found at the top or bottom.</p>
<p>There are three key elements of the GSEA method:</p>
<ul>
<li>Calculation of an Enrichment Score.
<ul>
<li>The enrichment score (<em>ES</em>) represent the degree to which a set <em>S</em> is over-represented at the top or bottom of the ranked list <em>L</em>. The score is calculated by walking down the list <em>L</em>, increasing a running-sum statistic when we encounter a gene in <em>S</em> and decreasing when it is not. The magnitude of the increment depends on the gene statistics (e.g., correlation of the gene with phenotype). The <em>ES</em> is the maximum deviation from zero encountered in the random walk; it corresponds to a weighted Kolmogorov-Smirnov-like statistic <span class="citation">(Subramanian et al. <a href="references.html#ref-subramanian_gene_2005" role="doc-biblioref">2005</a>)</span>.</li>
</ul>
</li>
<li>Esimation of Significance Level of <em>ES</em>.
<ul>
<li>The <em>p</em>-value of the <em>ES</em> is calculated using permutation test. Specifically, we permute the gene labels of the gene list <em>L</em> and recompute the <em>ES</em> of the gene set for the permutated data, which generate a null distribution for the <em>ES</em>. The <em>p</em>-value of the observed <em>ES</em> is then calculated relative to this null distribution.</li>
</ul>
</li>
<li>Adjustment for Multiple Hypothesis Testing.
<ul>
<li>When the entire gene sets were evaluated, the estimated significance level was adjusted to account for multiple hypothesis testing and also <em>q</em>-values were calculated for FDR control.</li>
</ul>
</li>
</ul>
<p>We implemented GSEA algorithm proposed by Subramanian <span class="citation">(Subramanian et al. <a href="references.html#ref-subramanian_gene_2005" role="doc-biblioref">2005</a>)</span>. Alexey Sergushichev implemented an algorithm for fast GSEA calculation in the <a href="http://bioconductor.org/packages/fgsea">fgsea</a> <span class="citation">(Korotkevich, Sukhov, and Sergushichev <a href="references.html#ref-korotkevich_fast_2019" role="doc-biblioref">2019</a>)</span> package. In our packages (<a href="http://bioconductor.org/packages/clusterProfiler">clusterProfiler</a>, <a href="http://bioconductor.org/packages/DOSE">DOSE</a>, <a href="http://bioconductor.org/packages/meshes">meshes</a> and <a href="http://bioconductor.org/packages/ReactomePA">ReactomePA</a>), users can use GSEA algorithm implemented in <code>DOSE</code> or <code>fgsea</code> by specifying the parameter <code>by="DOSE"</code> or <code>by="fgsea"</code>. By default, the <code>fgsea</code> method will be used since it is much more faster.</p>
</div>
<div id="leading-edge-analysis-and-core-enriched-genes" class="section level2" number="5.4">
<h2>
<span class="header-section-number">5.4</span> Leading edge analysis and core enriched genes<a class="anchor" aria-label="anchor" href="#leading-edge-analysis-and-core-enriched-genes"><i class="fas fa-link"></i></a>
</h2>
<p>Leading edge analysis reports <code>Tags</code> to indicate the percentage of genes contributing to the enrichment score, <code>List</code> to indicate where in the list the enrichment score is attained and <code>Signal</code> for enrichment signal strength.</p>
<p>It would also be very interesting to get the core enriched genes that contribute to the enrichment. Our packages (<a href="http://bioconductor.org/packages/clusterProfiler">clusterProfiler</a>, <a href="http://bioconductor.org/packages/DOSE">DOSE</a>, <a href="http://bioconductor.org/packages/meshes">meshes</a> and <a href="http://bioconductor.org/packages/ReactomePA">ReactomePA</a>) support leading edge analysis and report core enriched genes in GSEA analysis.</p>

</div>
</div>

  <div class="chapter-nav">
<div class="prev"><a href="meshes-semantic-similarity.html"><span class="header-section-number">4</span> MeSH semantic similarity analysis</a></div>
<div class="next"><a href="clusterprofiler-go.html"><span class="header-section-number">6</span> GO enrichment analysis</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#enrichment-overview"><span class="header-section-number">5</span> Overview of enrichment analysis</a></li>
<li>
<a class="nav-link" href="#terminology"><span class="header-section-number">5.1</span> Terminology</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#gene-sets-and-pathway"><span class="header-section-number">5.1.1</span> Gene sets and pathway</a></li>
<li><a class="nav-link" href="#gene-ontology-go"><span class="header-section-number">5.1.2</span> Gene Ontology (GO)</a></li>
<li><a class="nav-link" href="#kyoto-encyclopedia-of-genes-and-genomes-kegg"><span class="header-section-number">5.1.3</span> Kyoto Encyclopedia of Genes and Genomes (KEGG)</a></li>
<li><a class="nav-link" href="#other-gene-sets"><span class="header-section-number">5.1.4</span> Other gene sets</a></li>
</ul>
</li>
<li><a class="nav-link" href="#ora-algorithm"><span class="header-section-number">5.2</span> Over Representation Analysis</a></li>
<li><a class="nav-link" href="#gsea-algorithm"><span class="header-section-number">5.3</span> Gene Set Enrichment Analysis</a></li>
<li><a class="nav-link" href="#leading-edge-analysis-and-core-enriched-genes"><span class="header-section-number">5.4</span> Leading edge analysis and core enriched genes</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
          
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Biomedical Knowledge Mining using GOSemSim and clusterProfiler</strong>" was written by Guangchuang Yu. It was last built on 2021-02-06.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer><!-- dynamically load mathjax for compatibility with self-contained --><script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>
</html>
