<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-1.8.25">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<meta name="author" content="Guangchuang Yu">
<title>13&nbsp; Visualization of functional enrichment result – Biomedical Knowledge Mining using GOSemSim and clusterProfiler</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>

<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./misc.html" rel="next">
<link href="./029-compareCluster.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js" type="module"></script>
<script src="site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="site_libs/quarto-html/axe/axe-check.js" type="module"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-7b89279ff1a6dce999919e0e67d4d9ec.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-70a47bd5681a7291082a5b9f83d58762.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light"><script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
</head>
<body class="nav-sidebar floating quarto-light">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top"><nav class="quarto-secondary-nav"><div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./02-Enrichment.html">Enrichment Analysis</a></li><li class="breadcrumb-item"><a href="./enrichplot.html"><span class="chapter-number">13</span>&nbsp; <span class="chapter-title">Visualization of functional enrichment result</span></a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav></header><!-- content --><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto"><div class="pt-lg-2 mt-2 text-left sidebar-header sidebar-header-stacked">
      <a href="./index.html" class="sidebar-logo-link">
      </a>
    <div class="sidebar-title mb-0 py-0">
      <a href="./">Biomedical Knowledge Mining using GOSemSim and clusterProfiler</a> 
        <div class="sidebar-tools-main">
    <a href="https://github.com/YuLab-SMU/biomedical-knowledge-mining-book/" title="Source Code" class="quarto-navigation-tool px-1" aria-label="Source Code"><i class="bi bi-github"></i></a>
  <a href="" class="quarto-reader-toggle quarto-navigation-tool px-1" onclick="window.quartoToggleReader(); return false;" title="Toggle reader mode">
  <div class="quarto-reader-toggle-btn">
  <i class="bi"></i>
  </div>
</a>
</div>
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Welcome</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="./01-semantic-similarity.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Semantic Similarity Analysis</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./011-GOSemSim.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">GO semantic similarity analysis</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./012-DOSE-semantic-similarity.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">DO semantic similarity analysis</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./013-meshes-semantic-similarity.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">MeSH semantic similarity analysis</span></span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="./02-Enrichment.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Enrichment Analysis</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./021-go.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">GO enrichment analysis</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./022-kegg.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">KEGG enrichment analysis</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./023-wikipathways.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">WikiPathways analysis</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./024-reactome.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Reactome enrichment analysis</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./025-do-enrichment.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Disease enrichment analysis</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./026-meshes-enrichment.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">9</span>&nbsp; <span class="chapter-title">MeSH enrichment analysis</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./027-universal-enrichment.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">10</span>&nbsp; <span class="chapter-title">Universal enrichment analysis</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./028-chipseeker.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">11</span>&nbsp; <span class="chapter-title">Functional enrichment analysis of genomic coordinations</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./029-compareCluster.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">12</span>&nbsp; <span class="chapter-title">Biological theme comparison</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./enrichplot.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">13</span>&nbsp; <span class="chapter-title">Visualization of functional enrichment result</span></span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="./misc.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Miscellaneous topics</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./dplyr.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">14</span>&nbsp; <span class="chapter-title">dplyr verbs for manipulating enrichment result</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./utilities.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">15</span>&nbsp; <span class="chapter-title">Useful utilities</span></span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="./appendix.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Appendix</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./FAQ.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">16</span>&nbsp; <span class="chapter-title">Frequently asked questions</span></span></a>
  </div>
</li>
      </ul>
</li>
    </ul>
</div>
</nav><div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active"><h2 id="toc-title">Table of contents</h2>
   
  <ul>
<li><a href="#bar-plot" id="toc-bar-plot" class="nav-link active" data-scroll-target="#bar-plot"><span class="header-section-number">13.1</span> Bar Plot</a></li>
  <li><a href="#dot-plot" id="toc-dot-plot" class="nav-link" data-scroll-target="#dot-plot"><span class="header-section-number">13.2</span> Dot plot</a></li>
  <li><a href="#cnetplot" id="toc-cnetplot" class="nav-link" data-scroll-target="#cnetplot"><span class="header-section-number">13.3</span> Gene-Concept Network</a></li>
  <li><a href="#heatmap-like-functional-classification" id="toc-heatmap-like-functional-classification" class="nav-link" data-scroll-target="#heatmap-like-functional-classification"><span class="header-section-number">13.4</span> Heatmap-like functional classification</a></li>
  <li><a href="#tree-plot" id="toc-tree-plot" class="nav-link" data-scroll-target="#tree-plot"><span class="header-section-number">13.5</span> Tree plot</a></li>
  <li><a href="#enrichment-map" id="toc-enrichment-map" class="nav-link" data-scroll-target="#enrichment-map"><span class="header-section-number">13.6</span> Enrichment Map</a></li>
  <li><a href="#biological-theme-comparison" id="toc-biological-theme-comparison" class="nav-link" data-scroll-target="#biological-theme-comparison"><span class="header-section-number">13.7</span> Biological theme comparison</a></li>
  <li><a href="#upset-plot" id="toc-upset-plot" class="nav-link" data-scroll-target="#upset-plot"><span class="header-section-number">13.8</span> UpSet Plot</a></li>
  <li><a href="#ridgeline-plot-for-expression-distribution-of-gsea-result" id="toc-ridgeline-plot-for-expression-distribution-of-gsea-result" class="nav-link" data-scroll-target="#ridgeline-plot-for-expression-distribution-of-gsea-result"><span class="header-section-number">13.9</span> ridgeline plot for expression distribution of GSEA result</a></li>
  <li><a href="#running-score-and-preranked-list-of-gsea-result" id="toc-running-score-and-preranked-list-of-gsea-result" class="nav-link" data-scroll-target="#running-score-and-preranked-list-of-gsea-result"><span class="header-section-number">13.10</span> running score and preranked list of GSEA result</a></li>
  <li><a href="#pubmed-trend-of-enriched-terms" id="toc-pubmed-trend-of-enriched-terms" class="nav-link" data-scroll-target="#pubmed-trend-of-enriched-terms"><span class="header-section-number">13.11</span> pubmed trend of enriched terms</a></li>
  <li><a href="#references" id="toc-references" class="nav-link" data-scroll-target="#references">References</a></li>
  </ul><div class="toc-actions"><ul><li><a href="https://github.com/YuLab-SMU/biomedical-knowledge-mining-book/edit/main/enrichplot.qmd" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/YuLab-SMU/biomedical-knowledge-mining-book/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content"><header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./02-Enrichment.html">Enrichment Analysis</a></li><li class="breadcrumb-item"><a href="./enrichplot.html"><span class="chapter-number">13</span>&nbsp; <span class="chapter-title">Visualization of functional enrichment result</span></a></li></ol></nav><div class="quarto-title">
<h1 class="title"><span id="enrichplot" class="quarto-section-identifier"><span class="chapter-number">13</span>&nbsp; <span class="chapter-title">Visualization of functional enrichment result</span></span></h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header><p>The <a href="http://bioconductor.org/packages/enrichplot">enrichplot</a> package implements several visualization methods to help interpreting enrichment results. It supports visualizing enrichment results obtained from <a href="http://bioconductor.org/packages/DOSE">DOSE</a> <span class="citation" data-cites="yu_dose_2015">(<a href="#ref-yu_dose_2015" role="doc-biblioref">Yu et al. 2015</a>)</span>, <a href="http://bioconductor.org/packages/clusterProfiler">clusterProfiler</a> <span class="citation" data-cites="yu2012 wu_clusterprofiler_2021">(<a href="#ref-yu2012" role="doc-biblioref">Yu et al. 2012</a>; <a href="#ref-wu_clusterprofiler_2021" role="doc-biblioref">Wu et al. 2021</a>)</span>, <a href="http://bioconductor.org/packages/ReactomePA">ReactomePA</a> <span class="citation" data-cites="yu_reactomepa_2016">(<a href="#ref-yu_reactomepa_2016" role="doc-biblioref">Yu and He 2016</a>)</span> and <a href="http://bioconductor.org/packages/meshes">meshes</a> <span class="citation" data-cites="yu_meshes_2018">(<a href="#ref-yu_meshes_2018" role="doc-biblioref">Yu 2018</a>)</span>. Both over representation analysis (ORA) and gene set enrichment analysis (GSEA) are supported.</p>
<p>Note: Several visualization methods were first implemented in <a href="http://bioconductor.org/packages/DOSE">DOSE</a> and rewrote from scratch using <a href="https://CRAN.R-project.org/package=ggplot2">ggplot2</a>. If you want to use the <a href="https://www.biostars.org/p/375555">old methods</a>, you can use the <a href="https://github.com/GuangchuangYu/doseplot">doseplot</a> package.</p>
<section id="bar-plot" class="level2" data-number="13.1"><h2 data-number="13.1" class="anchored" data-anchor-id="bar-plot">
<span class="header-section-number">13.1</span> Bar Plot</h2>
<p>Bar plot is the most widely used method to visualize enriched terms. It depicts the enrichment scores (<em>e.g.</em> p values) and gene count or ratio as bar height and color <a href="#fig-Barplot" class="quarto-xref">Figure&nbsp;<span>13.1</span></a>. Users can specify the number of terms (most significant) or selected terms (see also the <a href="FAQ.html#showing-specific-pathways">FAQ</a>) to display via the <code>showCategory</code> parameter.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb1"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://yulab-smu.top/contribution-knowledge-mining/">DOSE</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">geneList</span><span class="op">)</span></span>
<span><span class="va">de</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">geneList</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">abs</a></span><span class="op">(</span><span class="va">geneList</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">2</span><span class="op">]</span></span>
<span></span>
<span><span class="va">edo</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DOSE/man/enrichDGN.html">enrichDGN</a></span><span class="op">(</span><span class="va">de</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb2"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://yulab-smu.top/contribution-knowledge-mining/">enrichplot</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/barplot.html">barplot</a></span><span class="op">(</span><span class="va">edo</span>, showCategory<span class="op">=</span><span class="fl">20</span><span class="op">)</span> </span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>Other variables that derived using <code>mutate</code> can also be used as bar height or color as demonstrated in <a href="dplyr.html#sec-mutate" class="quarto-xref"><span>Section 14.4</span></a> and <a href="#fig-Barplot" class="quarto-xref">Figure&nbsp;<span>13.1</span></a>.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb3"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://yulab-smu.top/contribution-knowledge-mining/">clusterProfiler</a></span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span><span class="va">edo</span>, qscore <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html">log</a></span><span class="op">(</span><span class="va">p.adjust</span>, base<span class="op">=</span><span class="fl">10</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>    <span class="fu"><a href="https://rdrr.io/r/graphics/barplot.html">barplot</a></span><span class="op">(</span>x<span class="op">=</span><span class="st">"qscore"</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="cell-output-display">
<div id="fig-Barplot" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-scap="Bar plot of enriched terms. ">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-Barplot-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="enrichplot_files/figure-html/fig-Barplot-1.svg" class="img-fluid figure-img" style="width:98.0%" data-fig-scap="Bar plot of enriched terms. ">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-Barplot-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;13.1: <strong>Bar plot of enriched terms.</strong>
</figcaption></figure>
</div>
</div>
</div>
</section><section id="dot-plot" class="level2" data-number="13.2"><h2 data-number="13.2" class="anchored" data-anchor-id="dot-plot">
<span class="header-section-number">13.2</span> Dot plot</h2>
<p>Dot plot is similar to bar plot with the capability to encode another score as dot size.</p>
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb4"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span></span>
<span><span class="va">edo2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DOSE/man/gseDO.html">gseDO</a></span><span class="op">(</span><span class="va">geneList</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/dotplot.html">dotplot</a></span><span class="op">(</span><span class="va">edo</span>, showCategory<span class="op">=</span><span class="fl">30</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="st">"dotplot for ORA"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/dotplot.html">dotplot</a></span><span class="op">(</span><span class="va">edo2</span>, showCategory<span class="op">=</span><span class="fl">30</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="st">"dotplot for GSEA"</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell">
<div class="cell-output-display">
<div id="fig-Dotplotcap" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-scap="Dot plot of enriched terms. ">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-Dotplotcap-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="enrichplot_files/figure-html/fig-Dotplotcap-1.svg" class="img-fluid figure-img" style="width:98.0%" data-fig-scap="Dot plot of enriched terms. ">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-Dotplotcap-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;13.2: <strong>Dot plot of enriched terms.</strong>
</figcaption></figure>
</div>
</div>
</div>
<p>Note: The <code><a href="https://rdrr.io/pkg/enrichplot/man/dotplot.html">dotplot()</a></code> function also works with <a href="##compare-dotplot"><code>compareCluster()</code> output</a>.</p>
</section><section id="cnetplot" class="level2" data-number="13.3"><h2 data-number="13.3" class="anchored" data-anchor-id="cnetplot">
<span class="header-section-number">13.3</span> Gene-Concept Network</h2>
<p>Both the <code><a href="https://rdrr.io/r/graphics/barplot.html">barplot()</a></code> and <code><a href="https://rdrr.io/pkg/enrichplot/man/dotplot.html">dotplot()</a></code> only displayed most significant or selected enriched terms, while users may want to know which genes are involved in these significant terms. In order to consider the potentially biological complexities in which a gene may belong to multiple annotation categories and provide information of numeric changes if available, we developed the <code><a href="https://rdrr.io/pkg/ggtangle/man/cnetplot.html">cnetplot()</a></code> function to extract the complex association. The <code><a href="https://rdrr.io/pkg/ggtangle/man/cnetplot.html">cnetplot()</a></code> depicts the linkages of genes and biological concepts (<em>e.g.</em> GO terms or KEGG pathways) as a network. GSEA result is also supported with only core enriched genes displayed.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb5"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co">## convert gene ID to Symbol</span></span>
<span><span class="va">edox</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DOSE/man/setReadable.html">setReadable</a></span><span class="op">(</span><span class="va">edo</span>, <span class="st">'org.Hs.eg.db'</span>, <span class="st">'ENTREZID'</span><span class="op">)</span></span>
<span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtangle/man/cnetplot.html">cnetplot</a></span><span class="op">(</span><span class="va">edox</span>, foldChange<span class="op">=</span><span class="va">geneList</span><span class="op">)</span></span>
<span><span class="co">## categorySize can be scaled by 'pvalue' or 'geneNum'</span></span>
<span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtangle/man/cnetplot.html">cnetplot</a></span><span class="op">(</span><span class="va">edox</span>, categorySize<span class="op">=</span><span class="st">"pvalue"</span>, foldChange<span class="op">=</span><span class="va">geneList</span><span class="op">)</span></span>
<span><span class="va">p3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtangle/man/cnetplot.html">cnetplot</a></span><span class="op">(</span><span class="va">edox</span>, foldChange<span class="op">=</span><span class="va">geneList</span>, circular <span class="op">=</span> <span class="cn">TRUE</span>, colorEdge <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> </span>
<span><span class="fu"><a href="https://rdrr.io/pkg/aplot/man/plot_list.html">plot_list</a></span><span class="op">(</span><span class="va">p1</span>, <span class="va">p2</span>, <span class="va">p3</span>, ncol<span class="op">=</span><span class="fl">3</span>, tag_levels <span class="op">=</span> <span class="st">'A'</span>, widths <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">.8</span>, <span class="fl">.8</span>, <span class="fl">1.2</span><span class="op">)</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output-display">
<div id="fig-Networkplot" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-scap="Network plot of enriched terms. ">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-Networkplot-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="enrichplot_files/figure-html/fig-Networkplot-1.svg" class="img-fluid figure-img" style="width:98.0%" data-fig-scap="Network plot of enriched terms. ">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-Networkplot-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;13.3: <strong>Network plot of enriched terms.</strong>
</figcaption></figure>
</div>
</div>
</div>
<p>If you would like label subset of the nodes, you can use the <code>node_label</code> parameter, which supports 4 possible selections (i.e.&nbsp;“category”, “gene”, “all” and “none”), as demonstrated in <a href="#fig-cnetNodeLabel" class="quarto-xref">Figure&nbsp;<span>13.4</span></a>.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb6"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtangle/man/cnetplot.html">cnetplot</a></span><span class="op">(</span><span class="va">edox</span>, node_label<span class="op">=</span><span class="st">"category"</span><span class="op">)</span> </span>
<span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtangle/man/cnetplot.html">cnetplot</a></span><span class="op">(</span><span class="va">edox</span>, node_label<span class="op">=</span><span class="st">"gene"</span><span class="op">)</span> </span>
<span><span class="va">p3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtangle/man/cnetplot.html">cnetplot</a></span><span class="op">(</span><span class="va">edox</span>, node_label<span class="op">=</span><span class="st">"all"</span><span class="op">)</span> </span>
<span><span class="va">p4</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtangle/man/cnetplot.html">cnetplot</a></span><span class="op">(</span><span class="va">edox</span>, node_label<span class="op">=</span><span class="st">"none"</span>, </span>
<span>        color_category<span class="op">=</span><span class="st">'firebrick'</span>, </span>
<span>        color_gene<span class="op">=</span><span class="st">'steelblue'</span><span class="op">)</span> </span>
<span><span class="fu"><a href="https://rdrr.io/pkg/aplot/man/plot_list.html">plot_list</a></span><span class="op">(</span><span class="va">p1</span>, <span class="va">p2</span>, <span class="va">p3</span>, <span class="va">p4</span>, ncol<span class="op">=</span><span class="fl">2</span>, tag_levels <span class="op">=</span> <span class="st">'A'</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output-display">
<div id="fig-cnetNodeLabel" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-scap="Labelling nodes by selected subset. ">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-cnetNodeLabel-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="enrichplot_files/figure-html/fig-cnetNodeLabel-1.svg" class="img-fluid figure-img" style="width:98.0%" data-fig-scap="Labelling nodes by selected subset. ">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-cnetNodeLabel-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;13.4: <strong>Labelling nodes by selected subset.</strong> gene category (A), gene name (B), both gene category and gene name (C, default) and not to label at all (D).
</figcaption></figure>
</div>
</div>
</div>
<p>The <code>cnetplot</code> function can be used as a general method to visualize data relationships in a network diagram. Users can use a named list as input as demonstrated in <a href="#fig-cnetplotgeneral" class="quarto-xref">Figure&nbsp;<span>13.5</span></a>.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb7"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>A <span class="op">=</span> <span class="va">letters</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span>, B<span class="op">=</span><span class="va">letters</span><span class="op">[</span><span class="fl">5</span><span class="op">:</span><span class="fl">12</span><span class="op">]</span>, C<span class="op">=</span><span class="va">letters</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/sample.html">sample</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">26</span>, <span class="fl">15</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtangle/man/cnetplot.html">cnetplot</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span></span>
<span><span class="va">d</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/setNames.html">setNames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/Normal.html">rnorm</a></span><span class="op">(</span><span class="fl">26</span><span class="op">)</span>, <span class="va">letters</span><span class="op">)</span></span>
<span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtangle/man/cnetplot.html">cnetplot</a></span><span class="op">(</span><span class="va">x</span>, foldChange<span class="op">=</span><span class="va">d</span><span class="op">)</span> <span class="op">+</span> </span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_gradient.html">scale_color_gradient2</a></span><span class="op">(</span>name<span class="op">=</span><span class="st">'associated data'</span>, low<span class="op">=</span><span class="st">'darkgreen'</span>, high<span class="op">=</span><span class="st">'firebrick'</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/aplot/man/plot_list.html">plot_list</a></span><span class="op">(</span><span class="va">p1</span>, <span class="va">p2</span>, ncol<span class="op">=</span><span class="fl">2</span>, tag_levels <span class="op">=</span> <span class="st">'A'</span><span class="op">)</span>    </span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output-display">
<div id="fig-cnetplotgeneral" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-scap="Using `cnetplot` to visualize data relationships. ">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-cnetplotgeneral-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="enrichplot_files/figure-html/fig-cnetplotgeneral-1.svg" class="img-fluid figure-img" style="width:98.0%" data-fig-scap="Using `cnetplot` to visualize data relationships. ">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-cnetplotgeneral-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;13.5: <strong>Using <code>cnetplot</code> to visualize data relationships.</strong> relationships as a network diagram (A) and with associated data to color nodes (B).
</figcaption></figure>
</div>
</div>
</div>
<p>Note: The <code><a href="https://rdrr.io/pkg/ggtangle/man/cnetplot.html">cnetplot()</a></code> function also works with <a href="##compare-cnetplot"><code>compareCluster()</code> output</a>.</p>
</section><section id="heatmap-like-functional-classification" class="level2" data-number="13.4"><h2 data-number="13.4" class="anchored" data-anchor-id="heatmap-like-functional-classification">
<span class="header-section-number">13.4</span> Heatmap-like functional classification</h2>
<p>The <code>heatplot</code> is similar to <code>cnetplot</code>, while displaying the relationships as a heatmap. The gene-concept network may become too complicated if user want to show a large number significant terms. The <code>heatplot</code> can simplify the result and more easy to identify expression patterns.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb8"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/heatplot.html">heatplot</a></span><span class="op">(</span><span class="va">edox</span>, showCategory<span class="op">=</span><span class="fl">5</span><span class="op">)</span></span>
<span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/heatplot.html">heatplot</a></span><span class="op">(</span><span class="va">edox</span>, foldChange<span class="op">=</span><span class="va">geneList</span>, showCategory<span class="op">=</span><span class="fl">5</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/aplot/man/plot_list.html">plot_list</a></span><span class="op">(</span><span class="va">p1</span>, <span class="va">p2</span>, ncol<span class="op">=</span><span class="fl">1</span>, tag_levels <span class="op">=</span> <span class="st">'A'</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output-display">
<div id="fig-Heatplot" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-scap="Heatmap plot of enriched terms. ">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-Heatplot-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="enrichplot_files/figure-html/fig-Heatplot-1.svg" class="img-fluid figure-img" style="width:98.0%" data-fig-scap="Heatmap plot of enriched terms. ">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-Heatplot-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;13.6: <strong>Heatmap plot of enriched terms.</strong> default (A), <code>foldChange=geneList</code> (B)
</figcaption></figure>
</div>
</div>
</div>
</section><section id="tree-plot" class="level2" data-number="13.5"><h2 data-number="13.5" class="anchored" data-anchor-id="tree-plot">
<span class="header-section-number">13.5</span> Tree plot</h2>
<p>The <code><a href="https://rdrr.io/pkg/enrichplot/man/treeplot.html">treeplot()</a></code> function performs hierarchical clustering of enriched terms. It relies on the pairwise similarities of the enriched terms calculated by the <code><a href="https://rdrr.io/pkg/enrichplot/man/pairwise_termsim.html">pairwise_termsim()</a></code> function, which by default using Jaccard’s similarity index (JC). Users can also use semantic similarity values if it is supported (<em>e.g.</em>, <a href="011-GOSemSim.html">GO</a>, <a href="012-DOSE-semantic-similarity.html">DO</a> and <a href="013-meshes-semantic-similarity.html">MeSH</a>).</p>
<p>The default agglomeration method in <code><a href="https://rdrr.io/pkg/enrichplot/man/treeplot.html">treeplot()</a></code> is <code>ward.D</code> and users can specify other methods via the <code>cluster_method</code> parameter (<em>e.g.</em>, ‘average’, ‘complete’, ‘median’, ‘centroid’, <em>etc.</em>, see also the document of the <code><a href="https://rdrr.io/r/stats/hclust.html">hclust()</a></code> function). The <code><a href="https://rdrr.io/pkg/enrichplot/man/treeplot.html">treeplot()</a></code> function will cut the tree into several subtrees (specify by the <code>nCluster</code> parameter (default is 5)) and labels subtrees using high-frequency words. This will reduce the complexity of the enriched result and improve user interpretation ability.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb9"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">edox2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/pairwise_termsim.html">pairwise_termsim</a></span><span class="op">(</span><span class="va">edox</span><span class="op">)</span></span>
<span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/treeplot.html">treeplot</a></span><span class="op">(</span><span class="va">edox2</span><span class="op">)</span></span>
<span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/treeplot.html">treeplot</a></span><span class="op">(</span><span class="va">edox2</span>, cluster_method <span class="op">=</span> <span class="st">"average"</span><span class="op">)</span></span>
<span><span class="fu">aplot</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/aplot/man/plot_list.html">plot_list</a></span><span class="op">(</span><span class="va">p1</span>, <span class="va">p2</span>, tag_levels<span class="op">=</span><span class="st">'A'</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output-display">
<div id="fig-treeplot" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-scap="Tree plot of enriched terms. ">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-treeplot-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="enrichplot_files/figure-html/fig-treeplot-1.svg" class="img-fluid figure-img" style="width:98.0%" data-fig-scap="Tree plot of enriched terms. ">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-treeplot-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;13.7: <strong>Tree plot of enriched terms.</strong> default (A), <code>hclust_method = "average"</code> (B)
</figcaption></figure>
</div>
</div>
</div>
</section><section id="enrichment-map" class="level2" data-number="13.6"><h2 data-number="13.6" class="anchored" data-anchor-id="enrichment-map">
<span class="header-section-number">13.6</span> Enrichment Map</h2>
<p>Enrichment map organizes enriched terms into a network with edges connecting overlapping gene sets. In this way, mutually overlapping gene sets are tend to cluster together, making it easy to identify functional module.</p>
<p>The <code>emapplot</code> function supports results obtained from hypergeometric test and gene set enrichment analysis. The <code>size_category</code> parameter can be used to resize nodes and the <code>layout</code> parameter can adjust the layout, as demonstrated in <a href="#fig-Enrichment" class="quarto-xref">Figure&nbsp;<span>13.8</span></a>.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb10"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">edo</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/pairwise_termsim.html">pairwise_termsim</a></span><span class="op">(</span><span class="va">edo</span><span class="op">)</span></span>
<span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/emapplot.html">emapplot</a></span><span class="op">(</span><span class="va">edo</span><span class="op">)</span></span>
<span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/emapplot.html">emapplot</a></span><span class="op">(</span><span class="va">edo</span>, size_category<span class="op">=</span><span class="fl">1.5</span><span class="op">)</span></span>
<span><span class="va">p3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/emapplot.html">emapplot</a></span><span class="op">(</span><span class="va">edo</span>, layout<span class="op">=</span><span class="st">"kk"</span><span class="op">)</span></span>
<span><span class="va">p4</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/emapplot.html">emapplot</a></span><span class="op">(</span><span class="va">edo</span>, size_category<span class="op">=</span><span class="fl">1.5</span>,layout<span class="op">=</span><span class="st">"kk"</span><span class="op">)</span> </span>
<span><span class="fu"><a href="https://rdrr.io/pkg/aplot/man/plot_list.html">plot_list</a></span><span class="op">(</span><span class="va">p1</span>, <span class="va">p2</span>, <span class="va">p3</span>, <span class="va">p4</span>, ncol<span class="op">=</span><span class="fl">2</span>, tag_levels <span class="op">=</span> <span class="st">'A'</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output-display">
<div id="fig-Enrichment" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-scap="Plot for results obtained from hypergeometric test and gene set enrichment analysis. ">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-Enrichment-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="enrichplot_files/figure-html/fig-Enrichment-1.svg" class="img-fluid figure-img" style="width:98.0%" data-fig-scap="Plot for results obtained from hypergeometric test and gene set enrichment analysis. ">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-Enrichment-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;13.8: <strong>Plot for results obtained from hypergeometric test and gene set enrichment analysis.</strong> default (A), <code>size_category=1.5</code> (B), <code>layout="kk"</code> (C) and <code>size_category=1.5,layout="kk"</code> (D).
</figcaption></figure>
</div>
</div>
</div>
</section><section id="biological-theme-comparison" class="level2" data-number="13.7"><h2 data-number="13.7" class="anchored" data-anchor-id="biological-theme-comparison">
<span class="header-section-number">13.7</span> Biological theme comparison</h2>
<p>The <code>emapplot</code> function also supports results obtained from <code>compareCluster</code> function of <code>clusterProfiler</code> package. In addition to <code>size_category</code> and <code>layout</code> parameters, the number of circles in the bottom left corner can be adjusted using the <code>legend_n</code> parameteras, and proportion of clusters in the pie chart can be adjusted using the <code>pie</code> parameter, when <code>pie="count"</code>, the proportion of clusters in the pie chart is determined by the number of genes, as demonstrated in <a href="#fig-Enrichment2" class="quarto-xref">Figure&nbsp;<span>13.9</span></a>.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb11"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://yulab-smu.top/contribution-knowledge-mining/">clusterProfiler</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">gcSample</span><span class="op">)</span></span>
<span><span class="va">xx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/compareCluster.html">compareCluster</a></span><span class="op">(</span><span class="va">gcSample</span>, fun<span class="op">=</span><span class="st">"enrichKEGG"</span>,</span>
<span>                     organism<span class="op">=</span><span class="st">"hsa"</span>, pvalueCutoff<span class="op">=</span><span class="fl">0.05</span><span class="op">)</span></span>
<span><span class="va">xx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/pairwise_termsim.html">pairwise_termsim</a></span><span class="op">(</span><span class="va">xx</span><span class="op">)</span>                     </span>
<span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/emapplot.html">emapplot</a></span><span class="op">(</span><span class="va">xx</span><span class="op">)</span></span>
<span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/emapplot.html">emapplot</a></span><span class="op">(</span><span class="va">xx</span><span class="op">)</span> </span>
<span><span class="va">p3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/emapplot.html">emapplot</a></span><span class="op">(</span><span class="va">xx</span>, pie<span class="op">=</span><span class="st">"count"</span><span class="op">)</span></span>
<span><span class="va">p4</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/emapplot.html">emapplot</a></span><span class="op">(</span><span class="va">xx</span>, pie<span class="op">=</span><span class="st">"count"</span>, size_category<span class="op">=</span><span class="fl">1.5</span>, layout<span class="op">=</span><span class="st">"kk"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/aplot/man/plot_list.html">plot_list</a></span><span class="op">(</span><span class="va">p1</span>, <span class="va">p2</span>, <span class="va">p3</span>, <span class="va">p4</span>, ncol<span class="op">=</span><span class="fl">2</span>, tag_levels <span class="op">=</span> <span class="st">'A'</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output-display">
<div id="fig-Enrichment2" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-scap="Plot for results obtained from `compareCluster` function of `clusterProfiler` package. ">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-Enrichment2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="enrichplot_files/figure-html/fig-Enrichment2-1.svg" class="img-fluid figure-img" style="width:98.0%" data-fig-scap="Plot for results obtained from `compareCluster` function of `clusterProfiler` package. ">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-Enrichment2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;13.9: <strong>Plot for results obtained from <code>compareCluster</code> function of <code>clusterProfiler</code> package.</strong> default (A), <code>legend_n=2</code> (B), <code>pie="count"</code> (C) and <code>pie="count", size_category=1.5, layout="kk"</code> (D).
</figcaption></figure>
</div>
</div>
</div>
</section><section id="upset-plot" class="level2" data-number="13.8"><h2 data-number="13.8" class="anchored" data-anchor-id="upset-plot">
<span class="header-section-number">13.8</span> UpSet Plot</h2>
<p>The <code>upsetplot</code> is an alternative to <code>cnetplot</code> for visualizing the complex association between genes and gene sets. It emphasizes the gene overlapping among different gene sets.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb12"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/upsetplot-methods.html">upsetplot</a></span><span class="op">(</span><span class="va">edo</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output-display">
<div id="fig-upsetORA" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-scap="Upsetplot for over-representation analysis. ">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-upsetORA-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="enrichplot_files/figure-html/fig-upsetORA-1.svg" class="img-fluid figure-img" style="width:98.0%" data-fig-scap="Upsetplot for over-representation analysis. ">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-upsetORA-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;13.10: <strong>Upsetplot for over-representation analysis.</strong>
</figcaption></figure>
</div>
</div>
</div>
<p>For over-representation analysis, <code>upsetplot</code> will calculate the overlaps among different gene sets as demonstrated in <a href="#fig-upsetORA" class="quarto-xref">Figure&nbsp;<span>13.10</span></a>. For GSEA result, it will plot the fold change distributions of different categories (e.g.&nbsp;unique to pathway, overlaps among different pathways).</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb13"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">kk2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/gseKEGG.html">gseKEGG</a></span><span class="op">(</span>geneList     <span class="op">=</span> <span class="va">geneList</span>,</span>
<span>               organism     <span class="op">=</span> <span class="st">'hsa'</span>,</span>
<span>               minGSSize    <span class="op">=</span> <span class="fl">120</span>,</span>
<span>               pvalueCutoff <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>               verbose      <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/upsetplot-methods.html">upsetplot</a></span><span class="op">(</span><span class="va">kk2</span><span class="op">)</span> </span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output-display">
<div id="fig-upsetGSEA" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-scap="Upsetplot for gene set enrichment analysis. ">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-upsetGSEA-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="enrichplot_files/figure-html/fig-upsetGSEA-1.svg" class="img-fluid figure-img" style="width:98.0%" data-fig-scap="Upsetplot for gene set enrichment analysis. ">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-upsetGSEA-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;13.11: <strong>Upsetplot for gene set enrichment analysis.</strong>
</figcaption></figure>
</div>
</div>
</div>
</section><section id="ridgeline-plot-for-expression-distribution-of-gsea-result" class="level2" data-number="13.9"><h2 data-number="13.9" class="anchored" data-anchor-id="ridgeline-plot-for-expression-distribution-of-gsea-result">
<span class="header-section-number">13.9</span> ridgeline plot for expression distribution of GSEA result</h2>
<p>The <code>ridgeplot</code> will visualize expression distributions of core enriched genes for GSEA enriched categories. It helps users to interpret up/down-regulated pathways.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb14"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/ridgeplot.html">ridgeplot</a></span><span class="op">(</span><span class="va">edo2</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output-display">
<div id="fig-ridgeplot" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-scap="Ridgeplot for gene set enrichment analysis. ">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-ridgeplot-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="enrichplot_files/figure-html/fig-ridgeplot-1.svg" class="img-fluid figure-img" style="width:98.0%" data-fig-scap="Ridgeplot for gene set enrichment analysis. ">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-ridgeplot-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;13.12: <strong>Ridgeplot for gene set enrichment analysis.</strong>
</figcaption></figure>
</div>
</div>
</div>
</section><section id="running-score-and-preranked-list-of-gsea-result" class="level2" data-number="13.10"><h2 data-number="13.10" class="anchored" data-anchor-id="running-score-and-preranked-list-of-gsea-result">
<span class="header-section-number">13.10</span> running score and preranked list of GSEA result</h2>
<p>Running score and preranked list are traditional methods for visualizing GSEA result. The <a href="http://bioconductor.org/packages/enrichplot">enrichplot</a> package supports both of them to visualize the distribution of the gene set and the enrichment score.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb15"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/gseaplot.html">gseaplot</a></span><span class="op">(</span><span class="va">edo2</span>, geneSetID <span class="op">=</span> <span class="fl">1</span>, by <span class="op">=</span> <span class="st">"runningScore"</span>, title <span class="op">=</span> <span class="va">edo2</span><span class="op">$</span><span class="va">Description</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/gseaplot.html">gseaplot</a></span><span class="op">(</span><span class="va">edo2</span>, geneSetID <span class="op">=</span> <span class="fl">1</span>, by <span class="op">=</span> <span class="st">"preranked"</span>, title <span class="op">=</span> <span class="va">edo2</span><span class="op">$</span><span class="va">Description</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">p3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/gseaplot.html">gseaplot</a></span><span class="op">(</span><span class="va">edo2</span>, geneSetID <span class="op">=</span> <span class="fl">1</span>, title <span class="op">=</span> <span class="va">edo2</span><span class="op">$</span><span class="va">Description</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/aplot/man/plot_list.html">plot_list</a></span><span class="op">(</span><span class="va">p1</span>, <span class="va">p2</span>, <span class="va">p3</span>, ncol<span class="op">=</span><span class="fl">1</span>, tag_levels<span class="op">=</span><span class="st">'A'</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output-display">
<div id="fig-gseaplot" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-scap="gseaplot for GSEA result(`by = &quot;runningScore&quot;`). ">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-gseaplot-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="enrichplot_files/figure-html/fig-gseaplot-1.svg" class="img-fluid figure-img" style="width:98.0%" data-fig-scap="gseaplot for GSEA result(`by = &quot;runningScore&quot;`). ">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-gseaplot-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;13.13: <strong>gseaplot for GSEA result(<code>by = "runningScore"</code>).</strong> <code>by = "runningScore"</code> (A), <code>by = "preranked"</code> (B), default (C)
</figcaption></figure>
</div>
</div>
</div>
<p>Another method to plot GSEA result is the <code>gseaplot2</code> function:</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb16"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/gseaplot2.html">gseaplot2</a></span><span class="op">(</span><span class="va">edo2</span>, geneSetID <span class="op">=</span> <span class="fl">1</span>, title <span class="op">=</span> <span class="va">edo2</span><span class="op">$</span><span class="va">Description</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output-display">
<div id="fig-gseaplot2" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-scap="Gseaplot2 for GSEA result. ">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-gseaplot2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="enrichplot_files/figure-html/fig-gseaplot2-1.svg" class="img-fluid figure-img" style="width:98.0%" data-fig-scap="Gseaplot2 for GSEA result. ">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-gseaplot2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;13.14: <strong>Gseaplot2 for GSEA result.</strong>
</figcaption></figure>
</div>
</div>
</div>
<p>The <code>gseaplot2</code> also supports multile gene sets to be displayed on the same figure:</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb17"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/gseaplot2.html">gseaplot2</a></span><span class="op">(</span><span class="va">edo2</span>, geneSetID <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output-display">
<div id="fig-gseaplot22" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-scap="Gseaplot2 for GSEA result of multile gene sets. ">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-gseaplot22-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="enrichplot_files/figure-html/fig-gseaplot22-1.svg" class="img-fluid figure-img" style="width:98.0%" data-fig-scap="Gseaplot2 for GSEA result of multile gene sets. ">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-gseaplot22-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;13.15: <strong>Gseaplot2 for GSEA result of multile gene sets.</strong>
</figcaption></figure>
</div>
</div>
</div>
<p>User can also displaying the pvalue table on the plot via <code>pvalue_table</code> parameter:</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb18"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/gseaplot2.html">gseaplot2</a></span><span class="op">(</span><span class="va">edo2</span>, geneSetID <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">3</span>, pvalue_table <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>          color <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"#E495A5"</span>, <span class="st">"#86B875"</span>, <span class="st">"#7DB0DD"</span><span class="op">)</span>, ES_geom <span class="op">=</span> <span class="st">"dot"</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output-display">
<div id="fig-gseaplot23" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-scap="Gseaplot2 for GSEA result of multile gene sets(add pvalue_table). ">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-gseaplot23-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="enrichplot_files/figure-html/fig-gseaplot23-1.svg" class="img-fluid figure-img" style="width:98.0%" data-fig-scap="Gseaplot2 for GSEA result of multile gene sets(add pvalue_table). ">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-gseaplot23-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;13.16: <strong>Gseaplot2 for GSEA result of multile gene sets(add pvalue_table).</strong>
</figcaption></figure>
</div>
</div>
</div>
<p>The <code>pvalue_table</code> can be customized using the following parameters:</p>
<ul>
<li>
<code>pvalue_table_rownames</code> to specify row names of the table (if NULL, no row names will be displayed)</li>
<li>
<code>pvalue_table_colnames</code> to specify column names of the table</li>
</ul>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb19"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/gseaplot2.html">gseaplot2</a></span><span class="op">(</span><span class="va">edo2</span>, geneSetID <span class="op">=</span> <span class="fl">1</span>, pvalue_table <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>          pvalue_table_rownames <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>          pvalue_table_columns <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"ID"</span>, <span class="st">"NES"</span>, <span class="st">"p.adjust"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output-display">
<div id="fig-gseaplot-pvalue" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-scap="Gseaplot2 with customized pvalue table. ">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-gseaplot-pvalue-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="enrichplot_files/figure-html/fig-gseaplot-pvalue-1.svg" class="img-fluid figure-img" style="width:98.0%" data-fig-scap="Gseaplot2 with customized pvalue table. ">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-gseaplot-pvalue-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;13.17: <strong>Gseaplot2 with customized pvalue table.</strong>
</figcaption></figure>
</div>
</div>
</div>
<p>User can specify <code>subplots</code> to only display a subset of plots:</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb20"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/gseaplot2.html">gseaplot2</a></span><span class="op">(</span><span class="va">edo2</span>, geneSetID <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">3</span>, subplots <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/gseaplot2.html">gseaplot2</a></span><span class="op">(</span><span class="va">edo2</span>, geneSetID <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">3</span>, subplots <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/aplot/man/plot_list.html">plot_list</a></span><span class="op">(</span><span class="va">p1</span>, <span class="va">p2</span>, ncol<span class="op">=</span><span class="fl">1</span>, tag_levels <span class="op">=</span> <span class="st">'A'</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output-display">
<div id="fig-gseaplot24" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-scap="Gseaplot2 for GSEA result of multile gene sets(add subplots). ">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-gseaplot24-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="enrichplot_files/figure-html/fig-gseaplot24-1.svg" class="img-fluid figure-img" style="width:98.0%" data-fig-scap="Gseaplot2 for GSEA result of multile gene sets(add subplots). ">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-gseaplot24-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;13.18: <strong>Gseaplot2 for GSEA result of multile gene sets(add subplots).</strong> <code>subplots = 1</code> (A),<code>subplots = 1:2</code> (B)
</figcaption></figure>
</div>
</div>
</div>
<p>The <code>gsearank</code> function plot the ranked list of genes belong to the specific gene set.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb21"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/gsearank.html">gsearank</a></span><span class="op">(</span><span class="va">edo2</span>, <span class="fl">1</span>, title <span class="op">=</span> <span class="va">edo2</span><span class="op">[</span><span class="fl">1</span>, <span class="st">"Description"</span><span class="op">]</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output-display">
<div id="fig-gsearank" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-scap="Ranked list of genes belong to the specific gene set. ">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-gsearank-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="enrichplot_files/figure-html/fig-gsearank-1.svg" class="img-fluid figure-img" style="width:98.0%" data-fig-scap="Ranked list of genes belong to the specific gene set. ">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-gsearank-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;13.19: <strong>Ranked list of genes belong to the specific gene set.</strong>
</figcaption></figure>
</div>
</div>
</div>
<p>Multiple gene sets can be aligned using <code>cowplot</code>:</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb22"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">pp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span>, <span class="kw">function</span><span class="op">(</span><span class="va">i</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">anno</span> <span class="op">&lt;-</span> <span class="va">edo2</span><span class="op">[</span><span class="va">i</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"NES"</span>, <span class="st">"pvalue"</span>, <span class="st">"p.adjust"</span><span class="op">)</span><span class="op">]</span></span>
<span>    <span class="va">lab</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">anno</span><span class="op">)</span>, <span class="st">"="</span>,  <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="va">anno</span>, <span class="fl">3</span><span class="op">)</span>, collapse<span class="op">=</span><span class="st">"\n"</span><span class="op">)</span></span>
<span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/gsearank.html">gsearank</a></span><span class="op">(</span><span class="va">edo2</span>, <span class="va">i</span>, <span class="va">edo2</span><span class="op">[</span><span class="va">i</span>, <span class="fl">2</span><span class="op">]</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">xlab</a></span><span class="op">(</span><span class="cn">NULL</span><span class="op">)</span> <span class="op">+</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ylab</a></span><span class="op">(</span><span class="cn">NULL</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/annotate.html">annotate</a></span><span class="op">(</span><span class="st">"text"</span>, <span class="fl">10000</span>, <span class="va">edo2</span><span class="op">[</span><span class="va">i</span>, <span class="st">"enrichmentScore"</span><span class="op">]</span> <span class="op">*</span> <span class="fl">.75</span>, label <span class="op">=</span> <span class="va">lab</span>, hjust<span class="op">=</span><span class="fl">0</span>, vjust<span class="op">=</span><span class="fl">0</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/aplot/man/plot_list.html">plot_list</a></span><span class="op">(</span>gglist<span class="op">=</span><span class="va">pp</span>, ncol<span class="op">=</span><span class="fl">1</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output-display">
<div id="fig-gsearank2" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-scap="Gsearank for multiple gene sets. ">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-gsearank2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="enrichplot_files/figure-html/fig-gsearank2-1.svg" class="img-fluid figure-img" style="width:98.0%" data-fig-scap="Gsearank for multiple gene sets. ">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-gsearank2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;13.20: <strong>Gsearank for multiple gene sets.</strong>
</figcaption></figure>
</div>
</div>
</div>
</section><section id="pubmed-trend-of-enriched-terms" class="level2" data-number="13.11"><h2 data-number="13.11" class="anchored" data-anchor-id="pubmed-trend-of-enriched-terms">
<span class="header-section-number">13.11</span> pubmed trend of enriched terms</h2>
<p>One of the problem of enrichment analysis is to find pathways for further investigation. Here, we provide <code>pmcplot</code> function to plot the number/proportion of publications trend based on the query result from PubMed Central. Of course, users can use <code>pmcplot</code> in other scenarios. All text that can be queried on PMC is valid as input of <code>pmcplot</code>.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb23"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">terms</span> <span class="op">&lt;-</span> <span class="va">edo</span><span class="op">$</span><span class="va">Description</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span>
<span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/pmcplot.html">pmcplot</a></span><span class="op">(</span><span class="va">terms</span>, <span class="fl">2010</span><span class="op">:</span><span class="fl">2020</span><span class="op">)</span></span>
<span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/pmcplot.html">pmcplot</a></span><span class="op">(</span><span class="va">terms</span>, <span class="fl">2010</span><span class="op">:</span><span class="fl">2020</span>, proportion<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/aplot/man/plot_list.html">plot_list</a></span><span class="op">(</span><span class="va">p</span>, <span class="va">p2</span>, ncol<span class="op">=</span><span class="fl">2</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output-display">
<div id="fig-pmcplot" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-scap="Pmcplot of enrichment analysis. ">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-pmcplot-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="enrichplot_files/figure-html/fig-pmcplot-1.svg" class="img-fluid figure-img" style="width:98.0%" data-fig-scap="Pmcplot of enrichment analysis. ">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-pmcplot-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;13.21: <strong>Pmcplot of enrichment analysis.</strong>
</figcaption></figure>
</div>
</div>
</div>
<!--

using ggplot2 api

`[` accessor


ggplot(egobp[1:20], aes(x=reorder(Description, -pvalue), y=Count, fill=-p.adjust)) +
    geom_bar(stat = "identity") +
    coord_flip() +
    scale_fill_continuous(low="blue", high="red") +
    labs(x = "", y = "", fill = "p.adjust") +
    theme(axis.text=element_text(size=11))

-->
</section><section id="references" class="level2 unnumbered"><h2 class="unnumbered anchored" data-anchor-id="references">References</h2>


<div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list">
<div id="ref-wu_clusterprofiler_2021" class="csl-entry" role="listitem">
Wu, Tianzhi, Erqiang Hu, Shuangbin Xu, Meijun Chen, Pingfan Guo, Zehan Dai, Tingze Feng, et al. 2021. <span>“<span class="nocase">clusterProfiler</span> 4.0: A Universal Enrichment Tool for Interpreting Omics Data.”</span> <em>The Innovation</em> 2 (3): 100141. <a href="https://doi.org/10.1016/j.xinn.2021.100141">https://doi.org/10.1016/j.xinn.2021.100141</a>.
</div>
<div id="ref-yu_meshes_2018" class="csl-entry" role="listitem">
Yu, Guangchuang. 2018. <span>“Using Meshes for <span>MeSH</span> Term Enrichment and Semantic Analyses.”</span> <em>Bioinformatics</em> 34 (21): 3766–67. <a href="https://doi.org/10.1093/bioinformatics/bty410">https://doi.org/10.1093/bioinformatics/bty410</a>.
</div>
<div id="ref-yu_reactomepa_2016" class="csl-entry" role="listitem">
Yu, Guangchuang, and Qing-Yu He. 2016. <span>“<span>ReactomePA</span>: An r/Bioconductor Package for Reactome Pathway Analysis and Visualization.”</span> <em>Molecular <span>BioSystems</span></em> 12 (2): 477–79. <a href="https://doi.org/10.1039/C5MB00663E">https://doi.org/10.1039/C5MB00663E</a>.
</div>
<div id="ref-yu2012" class="csl-entry" role="listitem">
Yu, Guangchuang, Le-Gen Wang, Yanyan Han, and Qing-Yu He. 2012. <span>“clusterProfiler: An r Package for Comparing Biological Themes Among Gene Clusters.”</span> <em>OMICS: A Journal of Integrative Biology</em> 16 (5): 284–87. <a href="https://doi.org/10.1089/omi.2011.0118">https://doi.org/10.1089/omi.2011.0118</a>.
</div>
<div id="ref-yu_dose_2015" class="csl-entry" role="listitem">
Yu, Guangchuang, Li-Gen Wang, Guang-Rong Yan, and Qing-Yu He. 2015. <span>“<span>DOSE</span>: An r/Bioconductor Package for Disease Ontology Semantic and Enrichment Analysis.”</span> <em>Bioinformatics</em> 31 (4): 608–9. <a href="https://doi.org/10.1093/bioinformatics/btu684">https://doi.org/10.1093/bioinformatics/btu684</a>.
</div>
</div>
</section></main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
      const outerScaffold = trigger.parentElement.cloneNode(true);
      const codeEl = outerScaffold.querySelector('code');
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp("https:\/\/yulab-smu\.top\/biomedical-knowledge-mining-book\/");
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script><nav class="page-navigation"><div class="nav-page nav-page-previous">
      <a href="./029-compareCluster.html" class="pagination-link" aria-label="Biological theme comparison">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">12</span>&nbsp; <span class="chapter-title">Biological theme comparison</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./misc.html" class="pagination-link" aria-label="Miscellaneous topics">
        <span class="nav-page-text">Miscellaneous topics</span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->
<footer class="footer"><div class="nav-footer">
    <div class="nav-footer-left">
<p>Biomedical Knowledge Mining using GOSemSim and clusterProfiler was written by Guangchuang Yu.</p>
</div>   
    <div class="nav-footer-center">
      &nbsp;
    <div class="toc-actions d-sm-block d-md-none"><ul><li><a href="https://github.com/YuLab-SMU/biomedical-knowledge-mining-book/edit/main/enrichplot.qmd" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/YuLab-SMU/biomedical-knowledge-mining-book/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div></div>
    <div class="nav-footer-right">
<p>This book was built with <a href="https://quarto.org/">Quarto</a>.</p>
</div>
  </div>
</footer>


</body></html>