<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 15 Visualization of functional enrichment result | Biomedical Knowledge Mining using GOSemSim and clusterProfiler</title>
<meta name="author" content="Guangchuang Yu">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.2"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/header-attrs-2.9/header-attrs.js"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.2.5.1/tabs.js"></script><script src="libs/bs3compat-0.2.5.1/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="https://cdn.jsdelivr.net/autocomplete.js/0/autocomplete.jquery.min.js"></script><script src="https://cdn.jsdelivr.net/npm/mark.js@8.11.1/dist/mark.min.js"></script><!-- CSS -->
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">Biomedical Knowledge Mining using GOSemSim and clusterProfiler</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html">üìñ Introduction</a></li>
<li class="book-part">Part I: Semantic Similarity Analysis</li>
<li><a class="" href="semantic-similarity-overview.html"><span class="header-section-number">1</span> Overview of semantic similarity analysis</a></li>
<li><a class="" href="GOSemSim.html"><span class="header-section-number">2</span> GO semantic similarity analysis</a></li>
<li><a class="" href="DOSE-semantic-similarity.html"><span class="header-section-number">3</span> DO semantic similarity analysis</a></li>
<li><a class="" href="meshes-semantic-similarity.html"><span class="header-section-number">4</span> MeSH semantic similarity analysis</a></li>
<li class="book-part">Part II: Enrichment analysis</li>
<li><a class="" href="enrichment-overview.html"><span class="header-section-number">5</span> Overview of enrichment analysis</a></li>
<li><a class="" href="clusterprofiler-go.html"><span class="header-section-number">6</span> GO enrichment analysis</a></li>
<li><a class="" href="clusterprofiler-kegg.html"><span class="header-section-number">7</span> KEGG enrichment analysis</a></li>
<li><a class="" href="wikipathways-analysis.html"><span class="header-section-number">8</span> WikiPathways analysis</a></li>
<li><a class="" href="reactomepa.html"><span class="header-section-number">9</span> Reactome enrichment analysis</a></li>
<li><a class="" href="dose-enrichment.html"><span class="header-section-number">10</span> Disease enrichment analysis</a></li>
<li><a class="" href="meshes-enrichment.html"><span class="header-section-number">11</span> MeSH enrichment analysis</a></li>
<li><a class="" href="universal-api.html"><span class="header-section-number">12</span> Universal enrichment analysis</a></li>
<li><a class="" href="chipseeker-enrichment.html"><span class="header-section-number">13</span> Functional enrichment analysis of genomic coordinations</a></li>
<li><a class="" href="clusterprofiler-comparecluster.html"><span class="header-section-number">14</span> Biological theme comparison</a></li>
<li><a class="active" href="enrichplot.html"><span class="header-section-number">15</span> Visualization of functional enrichment result</a></li>
<li class="book-part">Part III: Miscellaneous topics</li>
<li><a class="" href="clusterProfiler-dplyr.html"><span class="header-section-number">16</span> dplyr verbs for manipulating enrichment result</a></li>
<li><a class="" href="useful-utilities.html"><span class="header-section-number">17</span> Useful utilities</a></li>
<li class="book-part">Appendix</li>
<li><a class="" href="faq.html"><span class="header-section-number">A</span> Frequently asked questions</a></li>
<li><a class="" href="references.html">References</a></li>
</ul>

        <div class="book-extra">
          
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="enrichplot" class="section level1" number="15">
<h1>
<span class="header-section-number">15</span> Visualization of functional enrichment result<a class="anchor" aria-label="anchor" href="#enrichplot"><i class="fas fa-link"></i></a>
</h1>
<p>The <a href="http://bioconductor.org/packages/enrichplot">enrichplot</a> package implements several visualization methods to help interpreting enrichment results. It supports visualizing enrichment results obtained from <a href="http://bioconductor.org/packages/DOSE">DOSE</a> <span class="citation">(<a href="references.html#ref-yu_dose_2015" role="doc-biblioref">Yu et al. 2015</a>)</span>, <a href="http://bioconductor.org/packages/clusterProfiler">clusterProfiler</a> <span class="citation">(<a href="references.html#ref-yu2012" role="doc-biblioref">Yu et al. 2012</a>)</span>,
<a href="http://bioconductor.org/packages/ReactomePA">ReactomePA</a> <span class="citation">(<a href="references.html#ref-yu_reactomepa_2016" role="doc-biblioref">Yu and He 2016</a>)</span> and <a href="http://bioconductor.org/packages/meshes">meshes</a> <span class="citation">(<a href="references.html#ref-yu_meshes_2018" role="doc-biblioref">Yu 2018</a>)</span>. Both over representation analysis (ORA) and gene set enrichment analysis (GSEA) are supported.</p>
<p>Several visualization methods were first implemented in <a href="http://bioconductor.org/packages/DOSE">DOSE</a> and rewrote from scratch using <a href="https://CRAN.R-project.org/package=ggplot2">ggplot2</a>. If you want to use the <a href="https://www.biostars.org/p/375555">old methods</a>, you can use the <a href="https://github.com/GuangchuangYu/doseplot">doseplot</a> package.</p>
<div id="bar-plot" class="section level2" number="15.1">
<h2>
<span class="header-section-number">15.1</span> Bar Plot<a class="anchor" aria-label="anchor" href="#bar-plot"><i class="fas fa-link"></i></a>
</h2>
<p>Bar plot is the most widely used method to visualize enriched terms. It depicts the enrichment scores (<em>e.g.</em> p values) and gene count or ratio as bar height
and color.</p>
<div class="sourceCode" id="cb147"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://yulab-smu.top/biomedical-knowledge-mining-book/">DOSE</a></span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">geneList</span><span class="op">)</span>
<span class="va">de</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">geneList</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">abs</a></span><span class="op">(</span><span class="va">geneList</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">2</span><span class="op">]</span>

<span class="va">edo</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DOSE/man/enrichDGN.html">enrichDGN</a></span><span class="op">(</span><span class="va">de</span><span class="op">)</span></code></pre></div>


<div class="sourceCode" id="cb148"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://yulab-smu.top/biomedical-knowledge-mining-book/">enrichplot</a></span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/barplot.html">barplot</a></span><span class="op">(</span><span class="va">edo</span>, showCategory<span class="op">=</span><span class="fl">20</span><span class="op">)</span> </code></pre></div>
<div class="figure">
<span id="fig:Barplot"></span>
<img src="biomedicalKnowledge_files/figure-html/Barplot-1.png" alt="Bar plot of enriched terms." width="95%"><p class="caption">
Figure 15.1: <strong>Bar plot of enriched terms.</strong>
</p>
</div>
</div>
<div id="dot-plot" class="section level2" number="15.2">
<h2>
<span class="header-section-number">15.2</span> Dot plot<a class="anchor" aria-label="anchor" href="#dot-plot"><i class="fas fa-link"></i></a>
</h2>
<p>Dot plot is similar to bar plot with the capability to encode another score as dot size.</p>
<div class="sourceCode" id="cb149"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">edo2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DOSE/man/gseDO.html">gseDO</a></span><span class="op">(</span><span class="va">geneList</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/dotplot.html">dotplot</a></span><span class="op">(</span><span class="va">edo</span>, showCategory<span class="op">=</span><span class="fl">30</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="st">"dotplot for ORA"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/dotplot.html">dotplot</a></span><span class="op">(</span><span class="va">edo2</span>, showCategory<span class="op">=</span><span class="fl">30</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="st">"dotplot for GSEA"</span><span class="op">)</span></code></pre></div>


<div class="figure">
<span id="fig:Dotplotcap"></span>
<img src="biomedicalKnowledge_files/figure-html/Dotplotcap-1.png" alt="Dot plot of enriched terms." width="95%"><p class="caption">
Figure 15.2: <strong>Dot plot of enriched terms.</strong>
</p>
</div>
</div>
<div id="gene-concept-network" class="section level2" number="15.3">
<h2>
<span class="header-section-number">15.3</span> Gene-Concept Network<a class="anchor" aria-label="anchor" href="#gene-concept-network"><i class="fas fa-link"></i></a>
</h2>
<p>Both the <code><a href="https://rdrr.io/r/graphics/barplot.html">barplot()</a></code> and <code><a href="https://rdrr.io/pkg/enrichplot/man/dotplot.html">dotplot()</a></code> only displayed most significant enriched terms,
while users may want to know which genes are involved in these significant
terms.
In order to consider the potentially biological complexities in which a gene may belong to multiple annotation categories and provide information of numeric changes if available, we developed <code><a href="https://rdrr.io/pkg/enrichplot/man/cnetplot.html">cnetplot()</a></code> function to extract the complex association.
The <code><a href="https://rdrr.io/pkg/enrichplot/man/cnetplot.html">cnetplot()</a></code> depicts the linkages of genes and biological concepts (<em>e.g.</em> GO terms or KEGG pathways) as a network. GSEA result is also supported
with only core enriched genes displayed.</p>


<div class="sourceCode" id="cb150"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## convert gene ID to Symbol</span>
<span class="va">edox</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DOSE/man/setReadable.html">setReadable</a></span><span class="op">(</span><span class="va">edo</span>, <span class="st">'org.Hs.eg.db'</span>, <span class="st">'ENTREZID'</span><span class="op">)</span>
<span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/cnetplot.html">cnetplot</a></span><span class="op">(</span><span class="va">edox</span>, foldChange<span class="op">=</span><span class="va">geneList</span><span class="op">)</span>
<span class="co">## categorySize can be scaled by 'pvalue' or 'geneNum'</span>
<span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/cnetplot.html">cnetplot</a></span><span class="op">(</span><span class="va">edox</span>, categorySize<span class="op">=</span><span class="st">"pvalue"</span>, foldChange<span class="op">=</span><span class="va">geneList</span><span class="op">)</span>
<span class="va">p3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/cnetplot.html">cnetplot</a></span><span class="op">(</span><span class="va">edox</span>, foldChange<span class="op">=</span><span class="va">geneList</span>, circular <span class="op">=</span> <span class="cn">TRUE</span>, colorEdge <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="fu">cowplot</span><span class="fu">::</span><span class="fu"><a href="https://wilkelab.org/cowplot/reference/plot_grid.html">plot_grid</a></span><span class="op">(</span><span class="va">p1</span>, <span class="va">p2</span>, <span class="va">p3</span>, ncol<span class="op">=</span><span class="fl">3</span>, labels<span class="op">=</span><span class="va">LETTERS</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span>, rel_widths<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">.8</span>, <span class="fl">.8</span>, <span class="fl">1.2</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<div class="figure">
<span id="fig:Networkplot"></span>
<img src="biomedicalKnowledge_files/figure-html/Networkplot-1.png" alt="Network plot of enriched terms." width="95%"><p class="caption">
Figure 15.3: <strong>Network plot of enriched terms.</strong>
</p>
</div>
<p>If you would like label subset of the nodes, you can use the <code>node_label</code> parameter, which supports 4 possible selections (i.e.¬†‚Äúcategory,‚Äù ‚Äúgene,‚Äù ‚Äúall‚Äù and ‚Äúnone‚Äù), as demonstrated in Figure <a href="enrichplot.html#fig:cnetNodeLabel">15.4</a>.</p>


<div class="sourceCode" id="cb151"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/cnetplot.html">cnetplot</a></span><span class="op">(</span><span class="va">edox</span>, node_label<span class="op">=</span><span class="st">"category"</span><span class="op">)</span> 
<span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/cnetplot.html">cnetplot</a></span><span class="op">(</span><span class="va">edox</span>, node_label<span class="op">=</span><span class="st">"gene"</span><span class="op">)</span> 
<span class="va">p3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/cnetplot.html">cnetplot</a></span><span class="op">(</span><span class="va">edox</span>, node_label<span class="op">=</span><span class="st">"all"</span><span class="op">)</span> 
<span class="va">p4</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/cnetplot.html">cnetplot</a></span><span class="op">(</span><span class="va">edox</span>, node_label<span class="op">=</span><span class="st">"none"</span><span class="op">)</span> 
<span class="fu">cowplot</span><span class="fu">::</span><span class="fu"><a href="https://wilkelab.org/cowplot/reference/plot_grid.html">plot_grid</a></span><span class="op">(</span><span class="va">p1</span>, <span class="va">p2</span>, <span class="va">p3</span>, <span class="va">p4</span>, ncol<span class="op">=</span><span class="fl">2</span>, labels<span class="op">=</span><span class="va">LETTERS</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<div class="figure">
<span id="fig:cnetNodeLabel"></span>
<img src="biomedicalKnowledge_files/figure-html/cnetNodeLabel-1.png" alt="Labelling nodes by selected subset. gene category (A), gene name (B), both gene category and gene name (C, default) and not to label at all (D)." width="95%"><p class="caption">
Figure 15.4: <strong>Labelling nodes by selected subset.</strong> gene category (A), gene name (B), both gene category and gene name (C, default) and not to label at all (D).
</p>
</div>
</div>
<div id="heatmap-like-functional-classification" class="section level2" number="15.4">
<h2>
<span class="header-section-number">15.4</span> Heatmap-like functional classification<a class="anchor" aria-label="anchor" href="#heatmap-like-functional-classification"><i class="fas fa-link"></i></a>
</h2>
<p>The <code>heatplot</code> is similar to <code>cnetplot</code>, while displaying the relationships as a
heatmap. The gene-concept network may become too complicated if user want to
show a large number significant terms. The <code>heatplot</code> can simplify the result
and more easy to identify expression patterns.</p>


<div class="sourceCode" id="cb152"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/heatplot.html">heatplot</a></span><span class="op">(</span><span class="va">edox</span><span class="op">)</span>
<span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/heatplot.html">heatplot</a></span><span class="op">(</span><span class="va">edox</span>, foldChange<span class="op">=</span><span class="va">geneList</span><span class="op">)</span>
<span class="fu">cowplot</span><span class="fu">::</span><span class="fu"><a href="https://wilkelab.org/cowplot/reference/plot_grid.html">plot_grid</a></span><span class="op">(</span><span class="va">p1</span>, <span class="va">p2</span>, ncol<span class="op">=</span><span class="fl">1</span>, labels<span class="op">=</span><span class="va">LETTERS</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<div class="figure">
<span id="fig:Heatplot"></span>
<img src="biomedicalKnowledge_files/figure-html/Heatplot-1.png" alt="Heatmap plot of enriched terms. default (A), foldChange=geneList (B)" width="95%"><p class="caption">
Figure 15.5: <strong>Heatmap plot of enriched terms.</strong> default (A), <code>foldChange=geneList</code> (B)
</p>
</div>
</div>
<div id="enrichment-map" class="section level2" number="15.5">
<h2>
<span class="header-section-number">15.5</span> Enrichment Map<a class="anchor" aria-label="anchor" href="#enrichment-map"><i class="fas fa-link"></i></a>
</h2>
<p>Enrichment map organizes enriched terms into a network with edges connecting
overlapping gene sets. In this way, mutually overlapping gene sets are tend to
cluster together, making it easy to identify functional module.</p>
<p>The <code>emapplot</code> function supports results obtained from hypergeometric test and gene set enrichment analysis. The <code>cex_category</code> parameter can be used to resize nodes, as demonstrated in Figure <a href="enrichplot.html#fig:Enrichment">15.6</a> B, and the <code>layout</code> parameter can adjust the layout, as demonstrated in Figure <a href="enrichplot.html#fig:Enrichment">15.6</a> C and D.</p>


<div class="sourceCode" id="cb153"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">edo</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/pairwise_termsim.html">pairwise_termsim</a></span><span class="op">(</span><span class="va">edo</span><span class="op">)</span>
<span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/emapplot.html">emapplot</a></span><span class="op">(</span><span class="va">edo</span><span class="op">)</span>
<span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/emapplot.html">emapplot</a></span><span class="op">(</span><span class="va">edo</span>, cex_category<span class="op">=</span><span class="fl">1.5</span><span class="op">)</span>
<span class="va">p3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/emapplot.html">emapplot</a></span><span class="op">(</span><span class="va">edo</span>, layout<span class="op">=</span><span class="st">"kk"</span><span class="op">)</span>
<span class="va">p4</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/emapplot.html">emapplot</a></span><span class="op">(</span><span class="va">edo</span>, cex_category<span class="op">=</span><span class="fl">1.5</span>,layout<span class="op">=</span><span class="st">"kk"</span><span class="op">)</span> 
<span class="fu">cowplot</span><span class="fu">::</span><span class="fu"><a href="https://wilkelab.org/cowplot/reference/plot_grid.html">plot_grid</a></span><span class="op">(</span><span class="va">p1</span>, <span class="va">p2</span>, <span class="va">p3</span>, <span class="va">p4</span>, ncol<span class="op">=</span><span class="fl">2</span>, labels<span class="op">=</span><span class="va">LETTERS</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<div class="figure">
<span id="fig:Enrichment"></span>
<img src="biomedicalKnowledge_files/figure-html/Enrichment-1.png" alt="Plot for results obtained from hypergeometric test and gene set enrichment analysis. default (A), cex_category=1.5 (B), layout=" kk and cex_category='1.5,layout="kk"' width="95%"><p class="caption">
Figure 15.6: <strong>Plot for results obtained from hypergeometric test and gene set enrichment analysis.</strong> default (A), <code>cex_category=1.5</code> (B), <code>layout="kk"</code> (C) and <code>cex_category=1.5,layout="kk"</code> (D).
</p>
</div>
</div>
<div id="biological-theme-comparison" class="section level2" number="15.6">
<h2>
<span class="header-section-number">15.6</span> Biological theme comparison<a class="anchor" aria-label="anchor" href="#biological-theme-comparison"><i class="fas fa-link"></i></a>
</h2>
<p>The <code>emapplot</code> function also supports results obtained from <code>compareCluster</code> function of <code>clusterProfiler</code> package. In addition to <code>cex_category</code> and <code>layout</code> parameters, the number of circles in the bottom left corner can be adjusted using the <code>legend_n</code> parameteras, as demonstrated in Figure <a href="enrichplot.html#fig:Enrichment2">15.7</a> B. And proportion of clusters in the pie chart can be adjusted using the <code>pie</code> parameter, when <code>pie="count"</code>, the proportion of clusters in the pie chart is determined by the number of genes, as demonstrated in Figure <a href="enrichplot.html#fig:Enrichment2">15.7</a> C and D.</p>


<div class="sourceCode" id="cb154"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://yulab-smu.top/biomedical-knowledge-mining-book/">clusterProfiler</a></span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">gcSample</span><span class="op">)</span>
<span class="va">xx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/compareCluster.html">compareCluster</a></span><span class="op">(</span><span class="va">gcSample</span>, fun<span class="op">=</span><span class="st">"enrichKEGG"</span>,
                     organism<span class="op">=</span><span class="st">"hsa"</span>, pvalueCutoff<span class="op">=</span><span class="fl">0.05</span><span class="op">)</span>
<span class="va">xx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/pairwise_termsim.html">pairwise_termsim</a></span><span class="op">(</span><span class="va">xx</span><span class="op">)</span>                     
<span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/emapplot.html">emapplot</a></span><span class="op">(</span><span class="va">xx</span><span class="op">)</span>
<span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/emapplot.html">emapplot</a></span><span class="op">(</span><span class="va">xx</span>, legend_n<span class="op">=</span><span class="fl">2</span><span class="op">)</span> 
<span class="va">p3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/emapplot.html">emapplot</a></span><span class="op">(</span><span class="va">xx</span>, pie<span class="op">=</span><span class="st">"count"</span><span class="op">)</span>
<span class="va">p4</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/emapplot.html">emapplot</a></span><span class="op">(</span><span class="va">xx</span>, pie<span class="op">=</span><span class="st">"count"</span>, cex_category<span class="op">=</span><span class="fl">1.5</span>, layout<span class="op">=</span><span class="st">"kk"</span><span class="op">)</span>
<span class="fu">cowplot</span><span class="fu">::</span><span class="fu"><a href="https://wilkelab.org/cowplot/reference/plot_grid.html">plot_grid</a></span><span class="op">(</span><span class="va">p1</span>, <span class="va">p2</span>, <span class="va">p3</span>, <span class="va">p4</span>, ncol<span class="op">=</span><span class="fl">2</span>, labels<span class="op">=</span><span class="va">LETTERS</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<div class="figure">
<span id="fig:Enrichment2"></span>
<img src="biomedicalKnowledge_files/figure-html/Enrichment2-1.png" alt="Plot for results obtained from compareCluster function of clusterProfiler package. default (A), legend_n=2 (B), pie=" count and pie="count" cex_category="1.5," layout="kk" width="95%"><p class="caption">
Figure 15.7: <strong>Plot for results obtained from <code>compareCluster</code> function of <code>clusterProfiler</code> package.</strong> default (A), <code>legend_n=2</code> (B), <code>pie="count"</code> (C) and <code>pie="count", cex_category=1.5, layout="kk"</code> (D).
</p>
</div>
</div>
<div id="upset-plot" class="section level2" number="15.7">
<h2>
<span class="header-section-number">15.7</span> UpSet Plot<a class="anchor" aria-label="anchor" href="#upset-plot"><i class="fas fa-link"></i></a>
</h2>
<p>The <code>upsetplot</code> is an alternative to <code>cnetplot</code> for visualizing the complex
association between genes and gene sets. It emphasizes the gene overlapping
among different gene sets.</p>


<div class="sourceCode" id="cb155"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/upsetplot-methods.html">upsetplot</a></span><span class="op">(</span><span class="va">edo</span><span class="op">)</span></code></pre></div>
<div class="figure">
<span id="fig:upsetORA"></span>
<img src="biomedicalKnowledge_files/figure-html/upsetORA-1.png" alt="Upsetplot for over-representation analysis." width="95%"><p class="caption">
Figure 15.8: <strong>Upsetplot for over-representation analysis.</strong>
</p>
</div>
<p>For over-representation analysis, <code>upsetplot</code> will calculate the overlaps among different gene sets as demonstrated in Figure <a href="enrichplot.html#fig:upsetORA">15.8</a>. For GSEA result, it will plot the fold change distributions of different categories (e.g.¬†unique to pathway, overlaps among different pathways).</p>


<div class="sourceCode" id="cb156"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/upsetplot-methods.html">upsetplot</a></span><span class="op">(</span><span class="va">kk2</span><span class="op">)</span> </code></pre></div>
<div class="figure">
<span id="fig:upsetGSEA"></span>
<img src="biomedicalKnowledge_files/figure-html/upsetGSEA-1.png" alt="Upsetplot for gene set enrichment analysis." width="95%"><p class="caption">
Figure 15.9: <strong>Upsetplot for gene set enrichment analysis.</strong>
</p>
</div>
</div>
<div id="ridgeline-plot-for-expression-distribution-of-gsea-result" class="section level2" number="15.8">
<h2>
<span class="header-section-number">15.8</span> ridgeline plot for expression distribution of GSEA result<a class="anchor" aria-label="anchor" href="#ridgeline-plot-for-expression-distribution-of-gsea-result"><i class="fas fa-link"></i></a>
</h2>
<p>The <code>ridgeplot</code> will visualize expression distributions of core enriched genes
for GSEA enriched categories. It helps users to interpret up/down-regulated pathways.</p>


<div class="sourceCode" id="cb157"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/ridgeplot.html">ridgeplot</a></span><span class="op">(</span><span class="va">edo2</span><span class="op">)</span></code></pre></div>
<div class="figure">
<span id="fig:ridgeplot"></span>
<img src="biomedicalKnowledge_files/figure-html/ridgeplot-1.png" alt="Ridgeplot for gene set enrichment analysis." width="95%"><p class="caption">
Figure 15.10: <strong>Ridgeplot for gene set enrichment analysis.</strong>
</p>
</div>
</div>
<div id="running-score-and-preranked-list-of-gsea-result" class="section level2" number="15.9">
<h2>
<span class="header-section-number">15.9</span> running score and preranked list of GSEA result<a class="anchor" aria-label="anchor" href="#running-score-and-preranked-list-of-gsea-result"><i class="fas fa-link"></i></a>
</h2>
<p>Running score and preranked list are traditional methods for visualizing GSEA
result. The <a href="http://bioconductor.org/packages/enrichplot">enrichplot</a> package supports both of them to visualize
the distribution of the gene set and the enrichment score.</p>


<div class="sourceCode" id="cb158"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/gseaplot.html">gseaplot</a></span><span class="op">(</span><span class="va">edo2</span>, geneSetID <span class="op">=</span> <span class="fl">1</span>, by <span class="op">=</span> <span class="st">"runningScore"</span>, title <span class="op">=</span> <span class="va">edo2</span><span class="op">$</span><span class="va">Description</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span>
<span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/gseaplot.html">gseaplot</a></span><span class="op">(</span><span class="va">edo2</span>, geneSetID <span class="op">=</span> <span class="fl">1</span>, by <span class="op">=</span> <span class="st">"preranked"</span>, title <span class="op">=</span> <span class="va">edo2</span><span class="op">$</span><span class="va">Description</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span>
<span class="va">p3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/gseaplot.html">gseaplot</a></span><span class="op">(</span><span class="va">edo2</span>, geneSetID <span class="op">=</span> <span class="fl">1</span>, title <span class="op">=</span> <span class="va">edo2</span><span class="op">$</span><span class="va">Description</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span>
<span class="fu">cowplot</span><span class="fu">::</span><span class="fu"><a href="https://wilkelab.org/cowplot/reference/plot_grid.html">plot_grid</a></span><span class="op">(</span><span class="va">p1</span>, <span class="va">p2</span>, <span class="va">p3</span>, ncol<span class="op">=</span><span class="fl">1</span>, labels<span class="op">=</span><span class="va">LETTERS</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<div class="figure">
<span id="fig:gseaplot"></span>
<img src="biomedicalKnowledge_files/figure-html/gseaplot-1.png" alt="gseaplot for GSEA result(by = " runningscore by="runningScore" default width="95%"><p class="caption">
Figure 15.11: <strong>gseaplot for GSEA result(<code>by = "runningScore"</code>).</strong> <code>by = "runningScore"</code> (A), <code>by = "preranked"</code> (B), default (C)
</p>
</div>
<p>Another method to plot GSEA result is the <code>gseaplot2</code> function:</p>


<div class="sourceCode" id="cb159"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/gseaplot2.html">gseaplot2</a></span><span class="op">(</span><span class="va">edo2</span>, geneSetID <span class="op">=</span> <span class="fl">1</span>, title <span class="op">=</span> <span class="va">edo2</span><span class="op">$</span><span class="va">Description</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<div class="figure">
<span id="fig:gseaplot2"></span>
<img src="biomedicalKnowledge_files/figure-html/gseaplot2-1.png" alt="Gseaplot2 for GSEA result." width="95%"><p class="caption">
Figure 15.12: <strong>Gseaplot2 for GSEA result.</strong>
</p>
</div>
<p>The <code>gseaplot2</code> also supports multile gene sets to be displayed on the same figure:</p>


<div class="sourceCode" id="cb160"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/gseaplot2.html">gseaplot2</a></span><span class="op">(</span><span class="va">edo2</span>, geneSetID <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">)</span></code></pre></div>
<div class="figure">
<span id="fig:gseaplot22"></span>
<img src="biomedicalKnowledge_files/figure-html/gseaplot22-1.png" alt="Gseaplot2 for GSEA result of multile gene sets." width="95%"><p class="caption">
Figure 15.13: <strong>Gseaplot2 for GSEA result of multile gene sets.</strong>
</p>
</div>
<p>User can also displaying the pvalue table on the plot via <code>pvalue_table</code>
parameter:</p>


<div class="sourceCode" id="cb161"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/gseaplot2.html">gseaplot2</a></span><span class="op">(</span><span class="va">edo2</span>, geneSetID <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">3</span>, pvalue_table <span class="op">=</span> <span class="cn">TRUE</span>,
          color <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"#E495A5"</span>, <span class="st">"#86B875"</span>, <span class="st">"#7DB0DD"</span><span class="op">)</span>, ES_geom <span class="op">=</span> <span class="st">"dot"</span><span class="op">)</span></code></pre></div>
<div class="figure">
<span id="fig:gseaplot23"></span>
<img src="biomedicalKnowledge_files/figure-html/gseaplot23-1.png" alt="Gseaplot2 for GSEA result of multile gene sets(add pvalue_table)." width="95%"><p class="caption">
Figure 15.14: <strong>Gseaplot2 for GSEA result of multile gene sets(add pvalue_table).</strong>
</p>
</div>
<p>User can specify <code>subplots</code> to only display a subset of plots:</p>


<div class="sourceCode" id="cb162"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/gseaplot2.html">gseaplot2</a></span><span class="op">(</span><span class="va">edo2</span>, geneSetID <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">3</span>, subplots <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>
<span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/gseaplot2.html">gseaplot2</a></span><span class="op">(</span><span class="va">edo2</span>, geneSetID <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">3</span>, subplots <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">)</span>
<span class="fu">cowplot</span><span class="fu">::</span><span class="fu"><a href="https://wilkelab.org/cowplot/reference/plot_grid.html">plot_grid</a></span><span class="op">(</span><span class="va">p1</span>, <span class="va">p2</span>, ncol<span class="op">=</span><span class="fl">1</span>, labels<span class="op">=</span><span class="va">LETTERS</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<div class="figure">
<span id="fig:gseaplot24"></span>
<img src="biomedicalKnowledge_files/figure-html/gseaplot24-1.png" alt="Gseaplot2 for GSEA result of multile gene sets(add subplots). subplots = 1 (A),subplots = 1:2 (B)" width="95%"><p class="caption">
Figure 15.15: <strong>Gseaplot2 for GSEA result of multile gene sets(add subplots).</strong> <code>subplots = 1</code> (A),<code>subplots = 1:2</code> (B)
</p>
</div>
<p>The <code>gsearank</code> function plot the ranked list of genes belong to the specific
gene set.</p>


<div class="sourceCode" id="cb163"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/gsearank.html">gsearank</a></span><span class="op">(</span><span class="va">edo2</span>, <span class="fl">1</span>, title <span class="op">=</span> <span class="va">edo2</span><span class="op">[</span><span class="fl">1</span>, <span class="st">"Description"</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<div class="figure">
<span id="fig:gsearank"></span>
<img src="biomedicalKnowledge_files/figure-html/gsearank-1.png" alt="Ranked list of genes belong to the specific gene set." width="95%"><p class="caption">
Figure 15.16: <strong>Ranked list of genes belong to the specific gene set.</strong>
</p>
</div>
<p>Multiple gene sets can be aligned using <code>cowplot</code>:</p>


<div class="sourceCode" id="cb164"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://wilkelab.org/cowplot/">cowplot</a></span><span class="op">)</span>

<span class="va">pp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span>, <span class="kw">function</span><span class="op">(</span><span class="va">i</span><span class="op">)</span> <span class="op">{</span>
    <span class="va">anno</span> <span class="op">&lt;-</span> <span class="va">edo2</span><span class="op">[</span><span class="va">i</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"NES"</span>, <span class="st">"pvalue"</span>, <span class="st">"p.adjust"</span><span class="op">)</span><span class="op">]</span>
    <span class="va">lab</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">anno</span><span class="op">)</span>, <span class="st">"="</span>,  <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="va">anno</span>, <span class="fl">3</span><span class="op">)</span>, collapse<span class="op">=</span><span class="st">"\n"</span><span class="op">)</span>

    <span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/gsearank.html">gsearank</a></span><span class="op">(</span><span class="va">edo2</span>, <span class="va">i</span>, <span class="va">edo2</span><span class="op">[</span><span class="va">i</span>, <span class="fl">2</span><span class="op">]</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">xlab</a></span><span class="op">(</span><span class="cn">NULL</span><span class="op">)</span> <span class="op">+</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ylab</a></span><span class="op">(</span><span class="cn">NULL</span><span class="op">)</span> <span class="op">+</span>
        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/annotate.html">annotate</a></span><span class="op">(</span><span class="st">"text"</span>, <span class="fl">10000</span>, <span class="va">edo2</span><span class="op">[</span><span class="va">i</span>, <span class="st">"enrichmentScore"</span><span class="op">]</span> <span class="op">*</span> <span class="fl">.75</span>, label <span class="op">=</span> <span class="va">lab</span>, hjust<span class="op">=</span><span class="fl">0</span>, vjust<span class="op">=</span><span class="fl">0</span><span class="op">)</span>
<span class="op">}</span><span class="op">)</span>
<span class="fu"><a href="https://wilkelab.org/cowplot/reference/plot_grid.html">plot_grid</a></span><span class="op">(</span>plotlist<span class="op">=</span><span class="va">pp</span>, ncol<span class="op">=</span><span class="fl">1</span><span class="op">)</span></code></pre></div>
<div class="figure">
<span id="fig:gsearank2"></span>
<img src="biomedicalKnowledge_files/figure-html/gsearank2-1.png" alt="Gsearank for multiple gene sets." width="95%"><p class="caption">
Figure 15.17: <strong>Gsearank for multiple gene sets.</strong>
</p>
</div>
</div>
<div id="pubmed-trend-of-enriched-terms" class="section level2" number="15.10">
<h2>
<span class="header-section-number">15.10</span> pubmed trend of enriched terms<a class="anchor" aria-label="anchor" href="#pubmed-trend-of-enriched-terms"><i class="fas fa-link"></i></a>
</h2>
<p>One of the problem of enrichment analysis is to find pathways for further
investigation. Here, we provide <code>pmcplot</code> function to plot the number/proportion
of publications trend based on the query result from PubMed Central. Of course,
users can use <code>pmcplot</code> in other scenarios. All text that can be queried on PMC
is valid as input of <code>pmcplot</code>.</p>


<div class="sourceCode" id="cb165"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">terms</span> <span class="op">&lt;-</span> <span class="va">edo</span><span class="op">$</span><span class="va">Description</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span>
<span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/pmcplot.html">pmcplot</a></span><span class="op">(</span><span class="va">terms</span>, <span class="fl">2010</span><span class="op">:</span><span class="fl">2020</span><span class="op">)</span>
<span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/pmcplot.html">pmcplot</a></span><span class="op">(</span><span class="va">terms</span>, <span class="fl">2010</span><span class="op">:</span><span class="fl">2020</span>, proportion<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span>
<span class="fu"><a href="https://wilkelab.org/cowplot/reference/plot_grid.html">plot_grid</a></span><span class="op">(</span><span class="va">p</span>, <span class="va">p2</span>, ncol<span class="op">=</span><span class="fl">2</span><span class="op">)</span></code></pre></div>
<div class="figure">
<span id="fig:pmcplot"></span>
<img src="biomedicalKnowledge_files/figure-html/pmcplot-1.png" alt="Pmcplot of enrichment analysis." width="95%"><p class="caption">
Figure 15.18: <strong>Pmcplot of enrichment analysis.</strong>
</p>
</div>
<!--

using ggplot2 api

`[` accessor


ggplot(egobp[1:20], aes(x=reorder(Description, -pvalue), y=Count, fill=-p.adjust)) +
    geom_bar(stat = "identity") +
    coord_flip() +
    scale_fill_continuous(low="blue", high="red") +
    labs(x = "", y = "", fill = "p.adjust") +
    theme(axis.text=element_text(size=11))

-->

</div>
</div>



  <div class="chapter-nav">
<div class="prev"><a href="clusterprofiler-comparecluster.html"><span class="header-section-number">14</span> Biological theme comparison</a></div>
<div class="next"><a href="clusterProfiler-dplyr.html"><span class="header-section-number">16</span> dplyr verbs for manipulating enrichment result</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#enrichplot"><span class="header-section-number">15</span> Visualization of functional enrichment result</a></li>
<li><a class="nav-link" href="#bar-plot"><span class="header-section-number">15.1</span> Bar Plot</a></li>
<li><a class="nav-link" href="#dot-plot"><span class="header-section-number">15.2</span> Dot plot</a></li>
<li><a class="nav-link" href="#gene-concept-network"><span class="header-section-number">15.3</span> Gene-Concept Network</a></li>
<li><a class="nav-link" href="#heatmap-like-functional-classification"><span class="header-section-number">15.4</span> Heatmap-like functional classification</a></li>
<li><a class="nav-link" href="#enrichment-map"><span class="header-section-number">15.5</span> Enrichment Map</a></li>
<li><a class="nav-link" href="#biological-theme-comparison"><span class="header-section-number">15.6</span> Biological theme comparison</a></li>
<li><a class="nav-link" href="#upset-plot"><span class="header-section-number">15.7</span> UpSet Plot</a></li>
<li><a class="nav-link" href="#ridgeline-plot-for-expression-distribution-of-gsea-result"><span class="header-section-number">15.8</span> ridgeline plot for expression distribution of GSEA result</a></li>
<li><a class="nav-link" href="#running-score-and-preranked-list-of-gsea-result"><span class="header-section-number">15.9</span> running score and preranked list of GSEA result</a></li>
<li><a class="nav-link" href="#pubmed-trend-of-enriched-terms"><span class="header-section-number">15.10</span> pubmed trend of enriched terms</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
          
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Biomedical Knowledge Mining using GOSemSim and clusterProfiler</strong>" was written by Guangchuang Yu. It was last built on 2021-08-06.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer><!-- dynamically load mathjax for compatibility with self-contained --><script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>
</html>
