<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>üìñ Introduction | Biomedical Knowledge Mining using GOSemSim and clusterProfiler</title>
<meta name="author" content="Guangchuang Yu">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.2"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/header-attrs-2.8/header-attrs.js"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.2.5.1/tabs.js"></script><script src="libs/bs3compat-0.2.5.1/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="https://cdn.jsdelivr.net/autocomplete.js/0/autocomplete.jquery.min.js"></script><script src="https://cdn.jsdelivr.net/npm/mark.js@8.11.1/dist/mark.min.js"></script><!-- CSS -->
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">Biomedical Knowledge Mining using GOSemSim and clusterProfiler</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="active" href="index.html">üìñ Introduction</a></li>
<li class="book-part">Part I: Semantic Similarity Analysis</li>
<li><a class="" href="semantic-similarity-overview.html"><span class="header-section-number">1</span> Overview of semantic similarity analysis</a></li>
<li><a class="" href="GOSemSim.html"><span class="header-section-number">2</span> GO semantic similarity analysis</a></li>
<li><a class="" href="DOSE-semantic-similarity.html"><span class="header-section-number">3</span> DO semantic similarity analysis</a></li>
<li><a class="" href="meshes-semantic-similarity.html"><span class="header-section-number">4</span> MeSH semantic similarity analysis</a></li>
<li class="book-part">Part II: Enrichment analysis</li>
<li><a class="" href="enrichment-overview.html"><span class="header-section-number">5</span> Overview of enrichment analysis</a></li>
<li><a class="" href="clusterprofiler-go.html"><span class="header-section-number">6</span> GO enrichment analysis</a></li>
<li><a class="" href="clusterprofiler-kegg.html"><span class="header-section-number">7</span> KEGG enrichment analysis</a></li>
<li><a class="" href="wikipathways-analysis.html"><span class="header-section-number">8</span> WikiPathways analysis</a></li>
<li><a class="" href="reactomepa.html"><span class="header-section-number">9</span> Reactome enrichment analysis</a></li>
<li><a class="" href="dose-enrichment.html"><span class="header-section-number">10</span> Disease enrichment analysis</a></li>
<li><a class="" href="meshes-enrichment.html"><span class="header-section-number">11</span> MeSH enrichment analysis</a></li>
<li><a class="" href="universal-api.html"><span class="header-section-number">12</span> Universal enrichment analysis</a></li>
<li><a class="" href="chipseeker-enrichment.html"><span class="header-section-number">13</span> Functional enrichment analysis of genomic coordinations</a></li>
<li><a class="" href="clusterprofiler-comparecluster.html"><span class="header-section-number">14</span> Biological theme comparison</a></li>
<li><a class="" href="enrichplot.html"><span class="header-section-number">15</span> Visualization of functional enrichment result</a></li>
<li class="book-part">Part III: Miscellaneous topics</li>
<li><a class="" href="clusterProfiler-dplyr.html"><span class="header-section-number">16</span> dplyr verbs for manipulating enrichment result</a></li>
<li><a class="" href="useful-utilities.html"><span class="header-section-number">17</span> Useful utilities</a></li>
<li class="book-part">Appendix</li>
<li><a class="" href="faq.html"><span class="header-section-number">A</span> Frequently asked questions</a></li>
<li><a class="" href="references.html">References</a></li>
</ul>

        <div class="book-extra">
          
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><!--bookdown:title:end--><!--bookdown:title:start--><div id="introduction" class="section level1 unnumbered">
<h1>üìñ Introduction<a class="anchor" aria-label="anchor" href="#introduction"><i class="fas fa-link"></i></a>
</h1>
<p><a href="https://github.com/YuLab-SMU/ChIPseeker"><img src="https://img.shields.io/badge/ChIPseeker-1.28.3-green.svg"></a>
<a href="https://github.com/YuLab-SMU/clusterProfiler"><img src="https://img.shields.io/badge/clusterProfiler-4.1.1-green.svg"></a>
<a href="https://github.com/YuLab-SMU/DOSE"><img src="https://img.shields.io/badge/DOSE-3.19.1-green.svg"></a>
<a href="https://github.com/YuLab-SMU/enrichplot"><img src="https://img.shields.io/badge/enrichplot-1.13.1-green.svg"></a>
<a href="https://github.com/YuLab-SMU/GOSemSim"><img src="https://img.shields.io/badge/GOSemSim-2.18.0-green.svg"></a>
<a href="https://github.com/YuLab-SMU/meshes"><img src="https://img.shields.io/badge/meshes-1.18.0-green.svg"></a>
<a href="https://github.com/YuLab-SMU/ReactomePA"><img src="https://img.shields.io/badge/ReactomePA-1.36.0-green.svg"></a></p>
<p>hello</p>
<div class="inline-figure"><img src="figures/clusterProfiler-diagram.png" width="95%"></div>
<!--
In recently years, high-throughput experimental techniques such as
microarray, RNA-Seq and mass spectrometry can detect cellular
molecules at systems-level. These kinds of analyses generate huge
quantitaties of data, which need to be given a biological
interpretation. A commonly used approach is via clustering in the gene
dimension for grouping different genes based on their similarities[@yu2010].

To search for shared functions among genes, a common way is to
incorporate the biological knowledge, such as Gene Ontology (GO) and
Kyoto Encyclopedia of Genes and Genomes (KEGG), for identifying
predominant biological themes of a collection of genes.

After clustering analysis, researchers not only want to determine
whether there is a common theme of a particular gene cluster, but also
to compare the biological themes among gene clusters. The manual step
to choose interesting clusters followed by enrichment analysis on each
selected cluster is slow and tedious. To bridge this gap, we designed
[clusterProfiler](https://www.bioconductor.org/packages/clusterProfiler)[@yu2012], for comparing and visualizing functional
profiles among gene clusters.

-->
<div id="motivation" class="section level2 unnumbered">
<h2>üéØ Motivation<a class="anchor" aria-label="anchor" href="#motivation"><i class="fas fa-link"></i></a>
</h2>
<p>The book is meant as a guide for mining biological knowledge to elucidate or interpret molecular mechanisms using a suite of R packages, including <a href="http://bioconductor.org/packages/ChIPseeker">ChIPseeker</a>, <a href="http://bioconductor.org/packages/clusterProfiler">clusterProfiler</a>, <a href="http://bioconductor.org/packages/DOSE">DOSE</a>, <a href="http://bioconductor.org/packages/enrichplot">enrichplot</a>, <a href="http://bioconductor.org/packages/GOSemSim">GOSemSim</a>, <a href="http://bioconductor.org/packages/meshes">meshes</a> and <a href="http://bioconductor.org/packages/ReactomePA">ReactomePA</a>. Hence, if you are starting to read this book, we assume you have a working knowledge of how to use R.</p>
</div>
<div id="citation" class="section level2 unnumbered">
<h2>üìù Citation<a class="anchor" aria-label="anchor" href="#citation"><i class="fas fa-link"></i></a>
</h2>
<p>If you use the software suite in published research, please cite the most appropriate paper(s) from this list:</p>
<ol style="list-style-type: decimal">
<li>T Wu<sup>#</sup>, E Hu<sup>#</sup>, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan, X Fu, S Liu, X Bo<sup>*</sup>, and <strong>G Yu</strong><sup>*</sup>. clusterProfiler 4.0: A universal enrichment tool for interpreting omics data. <strong><em>The Innovation</em></strong>. 2021, accepted.
doi: <a href="https://doi.org/10.1016/j.xinn.2021.100141">10.1016/j.xinn.2021.100141</a>
</li>
<li>
<strong>G Yu</strong><sup>*</sup>. Gene Ontology Semantic Similarity Analysis Using GOSemSim. In: Kidder B. (eds) Stem Cell Transcriptional Networks. <strong><em>Methods in Molecular Biology</em></strong>. 2020, 2117:207-215. Humana, New York, NY.
doi: <a href="https://doi.org/10.1007/978-1-0716-0301-7_11">10.1007/978-1-0716-0301-7_11</a>
</li>
<li>
<strong>G Yu</strong><sup>*</sup>. Using meshes for MeSH term enrichment and semantic analyses. <strong><em>Bioinformatics</em></strong>. 2018, 34(21):3766‚Äì3767.
doi: <a href="https://doi.org/10.1093/bioinformatics/bty410">10.1093/bioinformatics/bty410</a>
</li>
<li>
<strong>G Yu</strong>, QY He<sup>*</sup>. ReactomePA: an R/Bioconductor package for reactome pathway analysis and visualization. <strong><em>Molecular BioSystems</em></strong>. 2016, 12(2):477-479.
doi: <a href="https://doi.org/10.1039/C5MB00663E">10.1039/C5MB00663E</a>
</li>
<li>
<strong>G Yu</strong><sup>*</sup>, LG Wang, and QY He<sup>*</sup>. ChIPseeker: an R/Bioconductor package for ChIP peak annotation, comparison and visualization. <strong><em>Bioinformatics</em></strong>. 2015, 31(14):2382-2383.
doi: <a href="https://doi.org/10.1093/bioinformatics/btv145">10.1093/bioinformatics/btv145</a>
</li>
<li>
<strong>G Yu</strong><sup>*</sup>, LG Wang, GR Yan, QY He<sup>*</sup>. DOSE: an R/Bioconductor package for Disease Ontology Semantic and Enrichment analysis. <strong><em>Bioinformatics</em></strong>. 2015, 31(4):608-609.
doi: <a href="https://doi.org/10.1093/bioinformatics/btu684">10.1093/bioinformatics/btu684</a>
</li>
<li>
<strong>G Yu</strong>, LG Wang, Y Han and QY He<sup>*</sup>. clusterProfiler: an R package for comparing biological themes among gene clusters. <strong><em>OMICS: A Journal of Integrative Biology</em></strong>. 2012, 16(5):284-287.
doi: <a href="https://doi.org/10.1089/omi.2011.0118">10.1089/omi.2011.0118</a>
</li>
<li>
<strong>G Yu</strong>, F Li, Y Qin, X Bo<sup>*</sup>, Y Wu, S Wang<sup>*</sup>. GOSemSim: an R package for measuring semantic similarity among GO terms and gene products. <strong><em>Bioinformatics</em></strong>. 2010, 26(7):976-978.
doi: <a href="https://doi.org/10.1093/bioinformatics/btq064">10.1093/bioinformatics/btq064</a>
</li>
</ol>
</div>
<div id="book-structure" class="section level2 unnumbered">
<h2>üìö Book structure<a class="anchor" aria-label="anchor" href="#book-structure"><i class="fas fa-link"></i></a>
</h2>
<ul>
<li>Part 1 (Semantic similarity analysis) describes <a href="http://bioconductor.org/packages/GOSemSim">GOSemSim</a>, <a href="http://bioconductor.org/packages/DOSE">DOSE</a> and <a href="http://bioconductor.org/packages/meshes">meshes</a> packages for measuring semantic similarity of genes or gene products based on Gene Ontology, Disease Ontology and Medical Subject Headings.</li>
<li>Part 2 (Enrichment analysis) introduces over-reprensentation analysis and gene set enrichment analysis using <a href="http://bioconductor.org/packages/clusterProfiler">clusterProfiler</a> (supports GO, KEGG, MSigDb, WikiPathway, and many others via universal interface), <a href="http://bioconductor.org/packages/DOSE">DOSE</a> (DO, Disease-Gene Network, Network of Cancer Genes), <a href="http://bioconductor.org/packages/meshes">meshes</a> (MeSH), and <a href="http://bioconductor.org/packages/ReactomePA">ReactomePA</a> (Reactome pathway). Functional enrichment analysis of Genomic coordination is supported via <a href="http://bioconductor.org/packages/ChIPseeker">ChIPseeker</a> and comparison among multiple conditions is also supported by <a href="http://bioconductor.org/packages/clusterProfiler">clusterProfiler</a>. We implemented a number of visualization methods in the <a href="http://bioconductor.org/packages/enrichplot">enrichplot</a> package to help users to interpret their results.</li>
<li>Part 3 (Miscellaneous topics) describes useful utilities including translating gene IDs and manipulating enrichment results.</li>
</ul>
</div>
<div id="want-to-help" class="section level2 unnumbered">
<h2>üíñ Want to help?<a class="anchor" aria-label="anchor" href="#want-to-help"><i class="fas fa-link"></i></a>
</h2>
<p>The book‚Äôs source code is hosted on GitHub, at <a href="https://github.com/YuLab-SMU/biomedical-knowledge-mining-book" class="uri">https://github.com/YuLab-SMU/biomedical-knowledge-mining-book</a>. Any feedback on the book is very welcome. Feel free to <a href="https://github.com/YuLab-SMU/biomedical-knowledge-mining-book/issues/new">open an issue</a> on GitHub or send me a pull request if you notice typos or other issues (I‚Äôm not a native English speaker ;) ).</p>

</div>
</div>



  <div class="chapter-nav">
<div class="empty"></div>
<div class="next"><a href="semantic-similarity-overview.html"><span class="header-section-number">1</span> Overview of semantic similarity analysis</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#introduction">üìñ Introduction</a></li>
<li><a class="nav-link" href="#motivation">üéØ Motivation</a></li>
<li><a class="nav-link" href="#citation">üìù Citation</a></li>
<li><a class="nav-link" href="#book-structure">üìö Book structure</a></li>
<li><a class="nav-link" href="#want-to-help">üíñ Want to help?</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
          
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Biomedical Knowledge Mining using GOSemSim and clusterProfiler</strong>" was written by Guangchuang Yu. It was last built on 2021-07-05.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer><!-- dynamically load mathjax for compatibility with self-contained --><script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>
</html>
