<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 9 Reactome enrichment analysis | Biomedical Knowledge Mining using GOSemSim and clusterProfiler</title>
<meta name="author" content="Guangchuang Yu">
<meta name="description" content="ReactomePA is designed for reactome pathway based analysis (Yu and He 2016). Reactome is an open-source, open access, manually curated and peer-reviewed pathway database.  9.1 Supported organisms...">
<meta name="generator" content="bookdown 0.23 with bs4_book()">
<meta property="og:title" content="Chapter 9 Reactome enrichment analysis | Biomedical Knowledge Mining using GOSemSim and clusterProfiler">
<meta property="og:type" content="book">
<meta property="og:description" content="ReactomePA is designed for reactome pathway based analysis (Yu and He 2016). Reactome is an open-source, open access, manually curated and peer-reviewed pathway database.  9.1 Supported organisms...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Chapter 9 Reactome enrichment analysis | Biomedical Knowledge Mining using GOSemSim and clusterProfiler">
<meta name="twitter:description" content="ReactomePA is designed for reactome pathway based analysis (Yu and He 2016). Reactome is an open-source, open access, manually curated and peer-reviewed pathway database.  9.1 Supported organisms...">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/header-attrs-2.10/header-attrs.js"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.2.5.1/tabs.js"></script><script src="libs/bs3compat-0.2.5.1/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS -->
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">Biomedical Knowledge Mining using GOSemSim and clusterProfiler</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html">ðŸ“– Introduction</a></li>
<li class="book-part">Part I: Semantic Similarity Analysis</li>
<li><a class="" href="semantic-similarity-overview.html"><span class="header-section-number">1</span> Overview of semantic similarity analysis</a></li>
<li><a class="" href="GOSemSim.html"><span class="header-section-number">2</span> GO semantic similarity analysis</a></li>
<li><a class="" href="DOSE-semantic-similarity.html"><span class="header-section-number">3</span> DO semantic similarity analysis</a></li>
<li><a class="" href="meshes-semantic-similarity.html"><span class="header-section-number">4</span> MeSH semantic similarity analysis</a></li>
<li class="book-part">Part II: Enrichment analysis</li>
<li><a class="" href="enrichment-overview.html"><span class="header-section-number">5</span> Overview of enrichment analysis</a></li>
<li><a class="" href="clusterprofiler-go.html"><span class="header-section-number">6</span> GO enrichment analysis</a></li>
<li><a class="" href="clusterprofiler-kegg.html"><span class="header-section-number">7</span> KEGG enrichment analysis</a></li>
<li><a class="" href="wikipathways-analysis.html"><span class="header-section-number">8</span> WikiPathways analysis</a></li>
<li><a class="active" href="reactomepa.html"><span class="header-section-number">9</span> Reactome enrichment analysis</a></li>
<li><a class="" href="dose-enrichment.html"><span class="header-section-number">10</span> Disease enrichment analysis</a></li>
<li><a class="" href="meshes-enrichment.html"><span class="header-section-number">11</span> MeSH enrichment analysis</a></li>
<li><a class="" href="universal-api.html"><span class="header-section-number">12</span> Universal enrichment analysis</a></li>
<li><a class="" href="chipseeker-enrichment.html"><span class="header-section-number">13</span> Functional enrichment analysis of genomic coordinations</a></li>
<li><a class="" href="clusterprofiler-comparecluster.html"><span class="header-section-number">14</span> Biological theme comparison</a></li>
<li><a class="" href="enrichplot.html"><span class="header-section-number">15</span> Visualization of functional enrichment result</a></li>
<li class="book-part">Part III: Miscellaneous topics</li>
<li><a class="" href="clusterProfiler-dplyr.html"><span class="header-section-number">16</span> dplyr verbs for manipulating enrichment result</a></li>
<li><a class="" href="useful-utilities.html"><span class="header-section-number">17</span> Useful utilities</a></li>
<li class="book-part">Appendix</li>
<li><a class="" href="faq.html"><span class="header-section-number">A</span> Frequently asked questions</a></li>
<li><a class="" href="references.html">References</a></li>
</ul>

        <div class="book-extra">
          
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="reactomepa" class="section level1" number="9">
<h1>
<span class="header-section-number">9</span> Reactome enrichment analysis<a class="anchor" aria-label="anchor" href="#reactomepa"><i class="fas fa-link"></i></a>
</h1>
<p><a href="http://bioconductor.org/packages/ReactomePA">ReactomePA</a> is designed for reactome pathway based analysis <span class="citation">(<a href="references.html#ref-yu_reactomepa_2016" role="doc-biblioref">Yu and He 2016</a>)</span>. Reactome is an open-source, open access, manually curated and peer-reviewed pathway database.</p>
<div id="reactomepa-supported-organisms" class="section level2" number="9.1">
<h2>
<span class="header-section-number">9.1</span> Supported organisms<a class="anchor" aria-label="anchor" href="#reactomepa-supported-organisms"><i class="fas fa-link"></i></a>
</h2>
<p>Currently <a href="http://bioconductor.org/packages/ReactomePA">ReactomePA</a> supports several model organisms, including â€˜celegans,â€™ â€˜fly,â€™ â€˜human,â€™ â€˜mouse,â€™ â€˜rat,â€™ â€˜yeastâ€™ and â€˜zebrafish.â€™ The input gene ID should be Entrez gene ID. We recommend using <a href="useful-utilities.html#bitr"><code>clusterProfiler::bitr()</code></a> to convert biological IDs.</p>
</div>
<div id="reactomepa-ora" class="section level2" number="9.2">
<h2>
<span class="header-section-number">9.2</span> Reactome pathway over-representation analysis<a class="anchor" aria-label="anchor" href="#reactomepa-ora"><i class="fas fa-link"></i></a>
</h2>
<p>Enrichment analysis is a widely used approach to identify biological
themes. <a href="http://bioconductor.org/packages/ReactomePA">ReactomePA</a> implemented <code><a href="https://rdrr.io/pkg/ReactomePA/man/enrichPathway.html">enrichPathway()</a></code> that uses <a href="enrichment-overview.html#ora-algorithm">hypergeometric model</a> to assess whether the number of selected genes associated with a reactome pathway is larger than expected.</p>
<div class="sourceCode" id="cb92"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://yulab-smu.top/biomedical-knowledge-mining-book/">ReactomePA</a></span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">geneList</span><span class="op">)</span>
<span class="va">de</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">geneList</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">abs</a></span><span class="op">(</span><span class="va">geneList</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">1.5</span><span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">de</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] "4312"  "8318"  "10874" "55143" "55388" "991"</code></pre>
<div class="sourceCode" id="cb94"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ReactomePA/man/enrichPathway.html">enrichPathway</a></span><span class="op">(</span>gene<span class="op">=</span><span class="va">de</span>, pvalueCutoff <span class="op">=</span> <span class="fl">0.05</span>, readable<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></code></pre></div>
<pre><code>##                          ID
## R-HSA-69620     R-HSA-69620
## R-HSA-69618     R-HSA-69618
## R-HSA-2500257 R-HSA-2500257
## R-HSA-141424   R-HSA-141424
## R-HSA-141444   R-HSA-141444
## R-HSA-9648025 R-HSA-9648025
##                                                                                        Description
## R-HSA-69620                                                                 Cell Cycle Checkpoints
## R-HSA-69618                                                             Mitotic Spindle Checkpoint
## R-HSA-2500257                                              Resolution of Sister Chromatid Cohesion
## R-HSA-141424                                         Amplification of signal from the kinetochores
## R-HSA-141444  Amplification  of signal from unattached  kinetochores via a MAD2  inhibitory signal
## R-HSA-9648025                                           EML4 and NUDC in mitotic spindle formation
##               GeneRatio   BgRatio       pvalue     p.adjust       qvalue
## R-HSA-69620      38/328 294/10856 2.142796e-14 1.647810e-11 1.470635e-11
## R-HSA-69618      22/328 113/10856 2.206937e-12 7.273823e-10 6.491729e-10
## R-HSA-2500257    23/328 126/10856 2.837642e-12 7.273823e-10 6.491729e-10
## R-HSA-141424     20/328  96/10856 6.071311e-12 9.337676e-10 8.333673e-10
## R-HSA-141444     20/328  96/10856 6.071311e-12 9.337676e-10 8.333673e-10
## R-HSA-9648025    20/328 117/10856 2.674965e-10 3.428414e-08 3.059785e-08
##                                                                                                                                                                                                                                       geneID
## R-HSA-69620   CDC45/CDCA8/MCM10/CDC20/CENPE/CCNB2/NDC80/UBE2C/SKA1/CENPM/CENPN/UBE2S/CCNA2/CDK1/ERCC6L/MAD2L1/KIF18A/BIRC5/AURKB/CHEK1/CCNB1/MCM5/MCM2/KIF2C/CDC25A/CDC6/PLK1/BUB1B/GTSE1/EXO1/ZWINT/CENPU/SPC25/CENPI/CCNE1/ORC6/ORC1/TAOK1
## R-HSA-69618                                                                                             CDCA8/CDC20/CENPE/NDC80/UBE2C/SKA1/CENPM/CENPN/UBE2S/ERCC6L/MAD2L1/KIF18A/BIRC5/AURKB/KIF2C/PLK1/BUB1B/ZWINT/CENPU/SPC25/CENPI/TAOK1
## R-HSA-2500257                                                                                      CDCA8/CDC20/CENPE/CCNB2/NDC80/SKA1/CENPM/CENPN/CDK1/ERCC6L/MAD2L1/KIF18A/BIRC5/AURKB/CCNB1/KIF2C/PLK1/BUB1B/ZWINT/CENPU/SPC25/CENPI/TAOK1
## R-HSA-141424                                                                                                        CDCA8/CDC20/CENPE/NDC80/SKA1/CENPM/CENPN/ERCC6L/MAD2L1/KIF18A/BIRC5/AURKB/KIF2C/PLK1/BUB1B/ZWINT/CENPU/SPC25/CENPI/TAOK1
## R-HSA-141444                                                                                                        CDCA8/CDC20/CENPE/NDC80/SKA1/CENPM/CENPN/ERCC6L/MAD2L1/KIF18A/BIRC5/AURKB/KIF2C/PLK1/BUB1B/ZWINT/CENPU/SPC25/CENPI/TAOK1
## R-HSA-9648025                                                                                                       CDCA8/CDC20/CENPE/NDC80/SKA1/CENPM/CENPN/ERCC6L/MAD2L1/KIF18A/BIRC5/AURKB/KIF2C/PLK1/BUB1B/ZWINT/CENPU/SPC25/CENPI/TAOK1
##               Count
## R-HSA-69620      38
## R-HSA-69618      22
## R-HSA-2500257    23
## R-HSA-141424     20
## R-HSA-141444     20
## R-HSA-9648025    20</code></pre>
</div>
<div id="reactomepa-gsea" class="section level2" number="9.3">
<h2>
<span class="header-section-number">9.3</span> Reactome pathway gene set enrichment analysis<a class="anchor" aria-label="anchor" href="#reactomepa-gsea"><i class="fas fa-link"></i></a>
</h2>
<div class="sourceCode" id="cb96"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">y</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ReactomePA/man/gsePathway.html">gsePathway</a></span><span class="op">(</span><span class="va">geneList</span>, 
                pvalueCutoff <span class="op">=</span> <span class="fl">0.2</span>,
                pAdjustMethod <span class="op">=</span> <span class="st">"BH"</span>, 
                verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">y</span><span class="op">)</span></code></pre></div>
<pre><code>##                        ID
## R-HSA-141424 R-HSA-141424
## R-HSA-141444 R-HSA-141444
## R-HSA-162909 R-HSA-162909
## R-HSA-174143 R-HSA-174143
## R-HSA-174178 R-HSA-174178
## R-HSA-174184 R-HSA-174184
##                                                                                                           Description
## R-HSA-141424                                                            Amplification of signal from the kinetochores
## R-HSA-141444                     Amplification  of signal from unattached  kinetochores via a MAD2  inhibitory signal
## R-HSA-162909                                                                         Host Interactions of HIV factors
## R-HSA-174143                                                        APC/C-mediated degradation of cell cycle proteins
## R-HSA-174178 APC/C:Cdh1 mediated degradation of Cdc20 and other APC/C:Cdh1 targeted proteins in late mitosis/early G1
## R-HSA-174184                                                     Cdc20:Phospho-APC/C mediated degradation of Cyclin A
##              setSize enrichmentScore      NES pvalue     p.adjust      qvalues
## R-HSA-141424      81       0.7111667 2.781869  1e-10 4.034984e-09 2.992922e-09
## R-HSA-141444      81       0.7111667 2.781869  1e-10 4.034984e-09 2.992922e-09
## R-HSA-162909     122       0.5860187 2.450643  1e-10 4.034984e-09 2.992922e-09
## R-HSA-174143      79       0.6940241 2.701439  1e-10 4.034984e-09 2.992922e-09
## R-HSA-174178      65       0.6868145 2.578477  1e-10 4.034984e-09 2.992922e-09
## R-HSA-174184      64       0.6941964 2.606047  1e-10 4.034984e-09 2.992922e-09
##              rank                   leading_edge
## R-HSA-141424  759  tags=31%, list=6%, signal=29%
## R-HSA-141444  759  tags=31%, list=6%, signal=29%
## R-HSA-162909 3321 tags=57%, list=27%, signal=43%
## R-HSA-174143 1851 tags=47%, list=15%, signal=40%
## R-HSA-174178 1851 tags=46%, list=15%, signal=40%
## R-HSA-174184 1851 tags=45%, list=15%, signal=39%
##                                                                                                                                                                                                                                                                                                                                                                 core_enrichment
## R-HSA-141424                                                                                                                                                                                                                             55143/991/1062/10403/220134/79019/55839/54821/4085/81930/332/9212/11004/5347/701/11130/79682/57405/2491/1058/699/1063/55055/79980/9735
## R-HSA-141444                                                                                                                                                                                                                             55143/991/1062/10403/220134/79019/55839/54821/4085/81930/332/9212/11004/5347/701/11130/79682/57405/2491/1058/699/1063/55055/79980/9735
## R-HSA-162909 3159/3932/5688/9688/3055/5709/5698/919/5693/5902/5713/5721/5691/292/1104/5685/1794/9631/5690/5684/5686/5695/11168/10213/940/23198/7979/6396/23165/5699/5714/6921/55706/8815/5702/23636/9978/5905/5708/1174/5692/5704/5901/164/60489/4927/79902/5683/10762/1175/5694/8480/11097/5718/5682/5716/9972/904/81929/162/5707/57122/4869/926/5696/7514/79023/3837/920/9818
## R-HSA-174143                                                                                                                                                                          991/11065/4751/27338/890/983/4085/9212/6790/891/9232/5347/701/6502/5688/5709/26271/5698/5693/5713/5721/5691/5685/5690/5684/5686/5695/10213/23198/7979/5699/5714/5702/5708/5692/10393/5704
## R-HSA-174178                                                                                                                                                                                                          991/11065/27338/9212/6790/9232/5347/6502/5688/5709/5698/5693/5713/5721/5691/5685/5690/5684/5686/5695/10213/23198/7979/5699/5714/5702/5708/5692/10393/5704
## R-HSA-174184                                                                                                                                                                                                                  991/11065/27338/890/983/4085/701/5688/5709/5698/5693/5713/5721/5691/5685/5690/5684/5686/5695/10213/23198/7979/5699/5714/5702/5708/5692/10393/5704</code></pre>
</div>
<div id="pathway-visualization" class="section level2" number="9.4">
<h2>
<span class="header-section-number">9.4</span> Pathway Visualization<a class="anchor" aria-label="anchor" href="#pathway-visualization"><i class="fas fa-link"></i></a>
</h2>
<p><a href="http://bioconductor.org/packages/ReactomePA">ReactomePA</a> implemented the <code><a href="https://rdrr.io/pkg/ReactomePA/man/viewPathway.html">viewPathway()</a></code> to visualized selected reactome pathway. More general purpose of visualization methods for ORA and GSEA results are provided in the <a href="http://bioconductor.org/packages/enrichplot">enrichplot</a> package and are documented on <a href="enrichplot.html#enrichplot">Chapter 14</a>.</p>


<div class="sourceCode" id="cb98"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/ReactomePA/man/viewPathway.html">viewPathway</a></span><span class="op">(</span><span class="st">"E2F mediated regulation of DNA replication"</span>, 
            readable <span class="op">=</span> <span class="cn">TRUE</span>, 
            foldChange <span class="op">=</span> <span class="va">geneList</span><span class="op">)</span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:viewpathway"></span>
<img src="biomedicalKnowledge_files/figure-html/viewpathway-1.png" alt="Visualize reactome pathway." width="95%"><p class="caption">
Figure 9.1: <strong>Visualize reactome pathway.</strong>
</p>
</div>
<!--

## Pathway analysis of NGS data

Pathway analysis using NGS data (eg, RNA-Seq and ChIP-Seq) can be performed by linking coding and non-coding regions to coding genes via [ChIPseeker](http://bioconductor.org/packages/ChIPseeker) package, which can annotates genomic regions to their nearest genes, host genes, and flanking genes respectivly. In addtion, it provides a function, __*seq2gene*__, that simultaneously considering host genes, promoter region and flanking gene from intergenic region that may under control via cis-regulation. This function maps genomic regions to genes in a many-to-many manner and facilitate functional analysis. For more details, please refer to [ChIPseeker](http://bioconductor.org/packages/ChIPseeker)[@yu_chipseeker_2015].


## Visualize enrichment result

We implement barplot, dotplot enrichment map and category-gene-network for visualization. It is very common to visualize the enrichment result in bar or pie chart. We believe the pie chart is misleading and only provide bar chart.
```r fig.height=6, fig.width=12}
barplot(x, showCategory=8)
```
```r fig.height=6, fig.width=12}
dotplot(x, showCategory=15)
```

Enrichment map can be viusalized by __*enrichMap*__:
```r fig.height=10, fig.width=10}
emapplot(x)
```



In order to consider the potentially biological complexities in which a gene may belong to multiple annotation categories, we developed __*cnetplot*__ function to extract the complex association between genes and diseases.
```r fig.height=8, fig.width=8}
cnetplot(x, categorySize="pvalue", foldChange=geneList)
```


## Visualize GSEA result

```r fig.height=8, fig.width=8}
emapplot(y, color="pvalue")
```


```r fig.height=7, fig.width=10}
gseaplot(y, geneSetID = "R-HSA-69242")
```


## Comparing enriched reactome pathways among gene clusters with clusterProfiler

We have developed an `R` package [clusterProfiler](http://bioconductor.org/packages/clusterProfiler)[@yu_clusterprofiler:_2012] for comparing biological themes among gene clusters. [ReactomePA](http://bioconductor.org/packages/ReactomePA) works fine with [clusterProfiler](http://bioconductor.org/packages/clusterProfiler) and can compare biological themes at reactome pathway perspective.

```r fig.height=8, fig.width=13, eval=FALSE}
require(clusterProfiler)
data(gcSample)
res <- compareCluster(gcSample, fun="enrichPathway")
dotplot(res)
```


-->

</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="wikipathways-analysis.html"><span class="header-section-number">8</span> WikiPathways analysis</a></div>
<div class="next"><a href="dose-enrichment.html"><span class="header-section-number">10</span> Disease enrichment analysis</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#reactomepa"><span class="header-section-number">9</span> Reactome enrichment analysis</a></li>
<li><a class="nav-link" href="#reactomepa-supported-organisms"><span class="header-section-number">9.1</span> Supported organisms</a></li>
<li><a class="nav-link" href="#reactomepa-ora"><span class="header-section-number">9.2</span> Reactome pathway over-representation analysis</a></li>
<li><a class="nav-link" href="#reactomepa-gsea"><span class="header-section-number">9.3</span> Reactome pathway gene set enrichment analysis</a></li>
<li><a class="nav-link" href="#pathway-visualization"><span class="header-section-number">9.4</span> Pathway Visualization</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
          
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Biomedical Knowledge Mining using GOSemSim and clusterProfiler</strong>" was written by Guangchuang Yu. It was last built on 2021-08-27.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer><!-- dynamically load mathjax for compatibility with self-contained --><script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script><script type="text/x-mathjax-config">const popovers = document.querySelectorAll('a.footnote-ref[data-toggle="popover"]');
for (let popover of popovers) {
  const div = document.createElement('div');
  div.setAttribute('style', 'position: absolute; top: 0, left:0; width:0, height:0, overflow: hidden; visibility: hidden;');
  div.innerHTML = popover.getAttribute('data-content');

  var has_math = div.querySelector("span.math");
  if (has_math) {
    document.body.appendChild(div);
    MathJax.Hub.Queue(["Typeset", MathJax.Hub, div]);
    MathJax.Hub.Queue(function() {
      popover.setAttribute('data-content', div.innerHTML);
      document.body.removeChild(div);
    })
  }
}
</script>
</body>
</html>
