<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 1 Overview of semantic similarity analysis | Biomedical Knowledge Mining using GOSemSim and clusterProfiler</title>
<meta name="author" content="Guangchuang Yu">
<meta name="description" content="Functional similarity of gene products can be estimated by controlled biological vocabularies, such as Gene Ontology (GO), Disease Ontology (DO) and Medical Subject Headings (MeSH). Four methods...">
<meta name="generator" content="bookdown 0.23 with bs4_book()">
<meta property="og:title" content="Chapter 1 Overview of semantic similarity analysis | Biomedical Knowledge Mining using GOSemSim and clusterProfiler">
<meta property="og:type" content="book">
<meta property="og:description" content="Functional similarity of gene products can be estimated by controlled biological vocabularies, such as Gene Ontology (GO), Disease Ontology (DO) and Medical Subject Headings (MeSH). Four methods...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Chapter 1 Overview of semantic similarity analysis | Biomedical Knowledge Mining using GOSemSim and clusterProfiler">
<meta name="twitter:description" content="Functional similarity of gene products can be estimated by controlled biological vocabularies, such as Gene Ontology (GO), Disease Ontology (DO) and Medical Subject Headings (MeSH). Four methods...">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/header-attrs-2.10/header-attrs.js"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.2.5.1/tabs.js"></script><script src="libs/bs3compat-0.2.5.1/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS -->
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">Biomedical Knowledge Mining using GOSemSim and clusterProfiler</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html">ðŸ“– Introduction</a></li>
<li class="book-part">Part I: Semantic Similarity Analysis</li>
<li><a class="active" href="semantic-similarity-overview.html"><span class="header-section-number">1</span> Overview of semantic similarity analysis</a></li>
<li><a class="" href="GOSemSim.html"><span class="header-section-number">2</span> GO semantic similarity analysis</a></li>
<li><a class="" href="DOSE-semantic-similarity.html"><span class="header-section-number">3</span> DO semantic similarity analysis</a></li>
<li><a class="" href="meshes-semantic-similarity.html"><span class="header-section-number">4</span> MeSH semantic similarity analysis</a></li>
<li class="book-part">Part II: Enrichment analysis</li>
<li><a class="" href="enrichment-overview.html"><span class="header-section-number">5</span> Overview of enrichment analysis</a></li>
<li><a class="" href="clusterprofiler-go.html"><span class="header-section-number">6</span> GO enrichment analysis</a></li>
<li><a class="" href="clusterprofiler-kegg.html"><span class="header-section-number">7</span> KEGG enrichment analysis</a></li>
<li><a class="" href="wikipathways-analysis.html"><span class="header-section-number">8</span> WikiPathways analysis</a></li>
<li><a class="" href="reactomepa.html"><span class="header-section-number">9</span> Reactome enrichment analysis</a></li>
<li><a class="" href="dose-enrichment.html"><span class="header-section-number">10</span> Disease enrichment analysis</a></li>
<li><a class="" href="meshes-enrichment.html"><span class="header-section-number">11</span> MeSH enrichment analysis</a></li>
<li><a class="" href="universal-api.html"><span class="header-section-number">12</span> Universal enrichment analysis</a></li>
<li><a class="" href="chipseeker-enrichment.html"><span class="header-section-number">13</span> Functional enrichment analysis of genomic coordinations</a></li>
<li><a class="" href="clusterprofiler-comparecluster.html"><span class="header-section-number">14</span> Biological theme comparison</a></li>
<li><a class="" href="enrichplot.html"><span class="header-section-number">15</span> Visualization of functional enrichment result</a></li>
<li class="book-part">Part III: Miscellaneous topics</li>
<li><a class="" href="clusterProfiler-dplyr.html"><span class="header-section-number">16</span> dplyr verbs for manipulating enrichment result</a></li>
<li><a class="" href="useful-utilities.html"><span class="header-section-number">17</span> Useful utilities</a></li>
<li class="book-part">Appendix</li>
<li><a class="" href="faq.html"><span class="header-section-number">A</span> Frequently asked questions</a></li>
<li><a class="" href="references.html">References</a></li>
</ul>

        <div class="book-extra">
          
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="semantic-similarity-overview" class="section level1" number="1">
<h1>
<span class="header-section-number">1</span> Overview of semantic similarity analysis<a class="anchor" aria-label="anchor" href="#semantic-similarity-overview"><i class="fas fa-link"></i></a>
</h1>
<p>Functional similarity of gene products can be estimated by controlled
biological vocabularies, such as Gene Ontology (GO), Disease Ontology (DO) and Medical Subject Headings (MeSH).</p>
<p>Four methods including Resnik <span class="citation">(<a href="references.html#ref-philip_semantic_1999" role="doc-biblioref">Philip 1999</a>)</span>, Jiang <span class="citation">(<a href="references.html#ref-jiang_semantic_1997" role="doc-biblioref">Jiang and Conrath 1997</a>)</span>, Lin <span class="citation">(<a href="references.html#ref-lin_information-theoretic_1998" role="doc-biblioref">Lin 1998</a>)</span> and Schlicker <span class="citation">(<a href="references.html#ref-schlicker_new_2006" role="doc-biblioref">Schlicker et al. 2006</a>)</span> have been presented to determine the semantic similarity of two GO terms based on the annotation statistics of their common ancestor terms. Wang <span class="citation">(<a href="references.html#ref-wang_new_2007" role="doc-biblioref">Wang et al. 2007</a>)</span>
proposed a method to measure the similarity based on the graph structure of GO. Each of these methods has its own advantages and
weaknesses and can be applied to other ontologies that have similar structure (i.e.Â directed acyclic graph).</p>
<div id="information-content-based-methods" class="section level2" number="1.1">
<h2>
<span class="header-section-number">1.1</span> Information content-based methods<a class="anchor" aria-label="anchor" href="#information-content-based-methods"><i class="fas fa-link"></i></a>
</h2>
<p>Four methods proposed by Resnik <span class="citation">(<a href="references.html#ref-philip_semantic_1999" role="doc-biblioref">Philip 1999</a>)</span>,
Jiang <span class="citation">(<a href="references.html#ref-jiang_semantic_1997" role="doc-biblioref">Jiang and Conrath 1997</a>)</span>, Lin <span class="citation">(<a href="references.html#ref-lin_information-theoretic_1998" role="doc-biblioref">Lin 1998</a>)</span>
and Schlicker <span class="citation">(<a href="references.html#ref-schlicker_new_2006" role="doc-biblioref">Schlicker et al. 2006</a>)</span> are information content (IC) based, which depend on the frequencies of two GO terms involved and that of their closest common ancestor term in a specific corpus of GO
annotations. The information content of a GO term is computed by the
negative log probability of the term occurring in GO corpus. A rarely used term contains a greater amount of information.</p>
<p>The frequency of a term t is defined as:</p>
<p><span class="math display">\[p(t) = \frac{n_{t'}}{N} | t' \in \left\{t, \; children\: of\: t \right\}\]</span></p>
<p>where <span class="math inline">\(n_{t'}\)</span> is the number of term <span class="math inline">\(t'\)</span>, and <span class="math inline">\(N\)</span> is the total number of terms in GO corpus.</p>
<p>Thus the information content is defined as:</p>
<p><span class="math display">\[IC(t) = -\log(p(t))\]</span></p>
<p>As GO allow multiple parents for each concept, two terms can share
parents by multiple paths. IC-based methods calculate similarity of two GO terms based on the information content of their closest common ancestor term, which was also called most informative common ancestor (MICA).</p>
<div id="resnik-method" class="section level3" number="1.1.1">
<h3>
<span class="header-section-number">1.1.1</span> Resnik method<a class="anchor" aria-label="anchor" href="#resnik-method"><i class="fas fa-link"></i></a>
</h3>
<p>The Resnik method is defined as:</p>
<p><span class="math display">\[sim_{Resnik}(t_1,t_2) = IC(MICA)\]</span></p>
</div>
<div id="lin-method" class="section level3" number="1.1.2">
<h3>
<span class="header-section-number">1.1.2</span> Lin method<a class="anchor" aria-label="anchor" href="#lin-method"><i class="fas fa-link"></i></a>
</h3>
<p>The Lin method is defined as:</p>
<p><span class="math display">\[sim_{Lin}(t_1,t_2) = \frac{2IC(MICA)}{IC(t_1)+IC(t_2)}\]</span></p>
</div>
<div id="rel-method" class="section level3" number="1.1.3">
<h3>
<span class="header-section-number">1.1.3</span> Rel method<a class="anchor" aria-label="anchor" href="#rel-method"><i class="fas fa-link"></i></a>
</h3>
<p>The Relevance method, which was proposed by Schlicker, combine Resnikâ€™s and Linâ€™s method and is defined as:</p>
<p><span class="math display">\[sim_{Rel}(t_1,t_2) = \frac{2IC(MICA)(1-p(MICA))}{IC(t_1)+IC(t_2)}\]</span></p>
</div>
<div id="jiang-method" class="section level3" number="1.1.4">
<h3>
<span class="header-section-number">1.1.4</span> Jiang method<a class="anchor" aria-label="anchor" href="#jiang-method"><i class="fas fa-link"></i></a>
</h3>
<p>The Jiang and Conrathâ€™s method is defined as:</p>
<p><span class="math display">\[sim_{Jiang}(t_1,t_2) = 1-\min(1, IC(t_1) + IC(t_2) - 2IC(MICA))\]</span></p>
</div>
</div>
<div id="graph-based-method" class="section level2" number="1.2">
<h2>
<span class="header-section-number">1.2</span> Graph-based method<a class="anchor" aria-label="anchor" href="#graph-based-method"><i class="fas fa-link"></i></a>
</h2>
<p>Graph-based methods using the topology of GO graph structure to
compute semantic similarity. Formally, a GO term A can be represented
as <span class="math inline">\(DAG_{A}=(A,T_{A},E_{A})\)</span> where <span class="math inline">\(T_{A}\)</span> is the set of GO terms in
<span class="math inline">\(DAG_{A}\)</span>, including term A and all of its ancestor terms in the GO
graph, and <span class="math inline">\(E_{A}\)</span> is the set of edges connecting the GO terms in
<span class="math inline">\(DAG_{A}\)</span>.</p>
<div id="wang-method" class="section level3" number="1.2.1">
<h3>
<span class="header-section-number">1.2.1</span> Wang method<a class="anchor" aria-label="anchor" href="#wang-method"><i class="fas fa-link"></i></a>
</h3>
<p>To encode the semantic of a GO term in a measurable format to enable a quantitative comparison, Wang<span class="citation">(<a href="references.html#ref-wang_new_2007" role="doc-biblioref">Wang et al. 2007</a>)</span> firstly defined the semantic value of term A as the aggregate contribution of all terms in <span class="math inline">\(DAG_{A}\)</span> to the semantics of term A, terms closer to term A in <span class="math inline">\(DAG_{A}\)</span> contribute more to its semantics. Thus, defined the contribution of a GO term <span class="math inline">\(t\)</span> to the semantic of GO term <span class="math inline">\(A\)</span> as the S-value of GO term <span class="math inline">\(t\)</span> related to term <span class="math inline">\(A\)</span>.</p>
<p>For any of term <span class="math inline">\(t\)</span> in <span class="math inline">\(DAG_{A}\)</span>, its S-value related to term <span class="math inline">\(A\)</span>, <span class="math inline">\(S_{A}(\textit{t})\)</span> is defined as:</p>
<p><span class="math display">\[\left\{\begin{array}{l} S_{A}(A)=1 \\ S_{A}(\textit{t})=\max\{w_{e} \times S_{A}(\textit{t}') | \textit{t}' \in children \: of(\textit{t}) \} \; if \: \textit{t} \ne A \end{array} \right.\]</span></p>
<p>where <span class="math inline">\(w_{e}\)</span> is the semantic contribution factor for edge <span class="math inline">\(e \in E_{A}\)</span> linking term <span class="math inline">\(t\)</span> with its child term <span class="math inline">\(t'\)</span>.
Term <span class="math inline">\(A\)</span> contributes to its own is defined as 1. After obtaining the S-values for all terms in <span class="math inline">\(DAG_{A}\)</span>,
the semantic value of DO term A, <span class="math inline">\(SV(A)\)</span>, is calculated as:</p>
<p><span class="math display">\[SV(A)=\displaystyle\sum_{t \in T_{A}} S_{A}(t)\]</span></p>
<p>Thus given two GO terms A and B, the semantic similarity between these two terms is defined as:</p>
<p><span class="math display">\[sim_{Wang}(A, B) = \frac{\displaystyle\sum_{t \in T_{A} \cap T_{B}}{S_{A}(t) + S_{B}(t)}}{SV(A) + SV(B)}\]</span></p>
<p>where <span class="math inline">\(S_{A}(\textit{t})\)</span> is the S-value of GO term <span class="math inline">\(t\)</span> related to term <span class="math inline">\(A\)</span>
and <span class="math inline">\(S_{B}(\textit{t})\)</span> is the S-value of GO term <span class="math inline">\(t\)</span> related to term <span class="math inline">\(B\)</span>.</p>
<p>This method proposed by Wang <span class="citation">(<a href="references.html#ref-wang_new_2007" role="doc-biblioref">Wang et al. 2007</a>)</span> determines the semantic
similarity of two GO terms based on both the locations of these terms
in the GO graph and their relations with their ancestor terms.</p>
</div>
</div>
<div id="combine-methods" class="section level2" number="1.3">
<h2>
<span class="header-section-number">1.3</span> Combine methods<a class="anchor" aria-label="anchor" href="#combine-methods"><i class="fas fa-link"></i></a>
</h2>
<p>Since a gene product can be annotated by multiple GO terms, semantic similarity among gene products needs to be aggregated from different semantic similarity scores of multiple GO terms associated with genes, including <code>max</code>, <code>avg</code>, <code>rcmax</code> and <code>BMA</code>.</p>
<div id="max" class="section level3" number="1.3.1">
<h3>
<span class="header-section-number">1.3.1</span> max<a class="anchor" aria-label="anchor" href="#max"><i class="fas fa-link"></i></a>
</h3>
<p>The <code>max</code> method calculates the maximum semantic similarity score over all pairs of GO terms between these two GO term sets.</p>
<p><span class="math display">\[sim_{max}(g_1, g_2) = \displaystyle\max_{1 \le i \le m, 1 \le j \le n} sim(go_{1i}, go_{2j})\]</span></p>
</div>
<div id="avg" class="section level3" number="1.3.2">
<h3>
<span class="header-section-number">1.3.2</span> avg<a class="anchor" aria-label="anchor" href="#avg"><i class="fas fa-link"></i></a>
</h3>
<p>The <code>avg</code> calculates the average semantic similarity score over all pairs of GO terms.</p>
<p><span class="math display">\[sim_{avg}(g_1, g_2) = \frac{\displaystyle\sum_{i=1}^m\sum_{j=1}^nsim(go_{1i}, go_{2j})}{m \times n}\]</span></p>
</div>
<div id="rcmax" class="section level3" number="1.3.3">
<h3>
<span class="header-section-number">1.3.3</span> rcmax<a class="anchor" aria-label="anchor" href="#rcmax"><i class="fas fa-link"></i></a>
</h3>
<p>Similarities among two sets of GO terms form a matrix, the <code>rcmax</code> method uses the maximum of <code>RowScore</code> and <code>ColumnScore</code>, where <code>RowScore</code> (or <code>ColumnScore</code>) is the average of maximum similarity on each row (or column).</p>
<p><span class="math display">\[sim_{rcmax}(g_1, g_2) = \max(\frac{\displaystyle\sum_{i=1}^m \max_{1 \le j \le n} sim(go_{1i}, go_{2j})}{m},\frac{\displaystyle\sum_{j=1}^n \max_{1 \le i \le m} sim(go_{1i},go_{2j})}{n})\]</span></p>
</div>
<div id="bma" class="section level3" number="1.3.4">
<h3>
<span class="header-section-number">1.3.4</span> BMA<a class="anchor" aria-label="anchor" href="#bma"><i class="fas fa-link"></i></a>
</h3>
<p>The <code>BMA</code> method, used the <strong>B</strong>est-<strong>M</strong>atch <strong>A</strong>verage strategy, calculates the average of all maximum similarities on each row and column, and is defined as:</p>
<p><span class="math display">\[sim_{BMA}(g_1, g_2) = \frac{\displaystyle\sum_{1=i}^m \max_{1 \le j \le n}sim(go_{1i}, go_{2j}) + \displaystyle\sum_{1=j}^n \max_{1 \le i \le m}sim(go_{1i}, go_{2j})} {m+n}\]</span></p>
</div>
</div>
<div id="summary" class="section level2" number="1.4">
<h2>
<span class="header-section-number">1.4</span> Summary<a class="anchor" aria-label="anchor" href="#summary"><i class="fas fa-link"></i></a>
</h2>
<p>The idea behind semantic similarity measurement is the notion that genes with similar function should have similar annotation vocabulary and have a close relationship in the ontology strucutre. Measuring similarity is critical for expanding knownledge, since similar objects tend to behave similarly, which supports many bioinformatics applications to infer gene/protein functions, miRNA function, genetic interaction, protein-protein interaction, miRNA-mRNA interaction and celluar localization.</p>
<p>We developed several Bioconductor packages, including <a href="http://bioconductor.org/packages/GOSemSim">GOSemSim</a> <span class="citation">(<a href="references.html#ref-yu2010" role="doc-biblioref">Yu et al. 2010</a>; <a href="references.html#ref-yu_gosemsim_2020" role="doc-biblioref">Yu 2020</a>)</span> for computing semantic similarity among GO terms, sets of GO terms, gene products and gene clusters (see also <a href="GOSemSim.html#GOSemSim">Chapter 2</a>), <a href="http://bioconductor.org/packages/DOSE">DOSE</a> <span class="citation">(<a href="references.html#ref-yu_dose_2015" role="doc-biblioref">Yu et al. 2015</a>)</span> for Disease Ontology (DO) (see also <a href="DOSE-semantic-similarity.html#DOSE-semantic-similarity">Chapter 3</a>) and <a href="http://bioconductor.org/packages/meshes">meshes</a> <span class="citation">(<a href="references.html#ref-yu_meshes_2018" role="doc-biblioref">Yu 2018</a>)</span> that based on Medical Subject Headings (MeSH) (see also <a href="meshes-semantic-similarity.html#meshes-semantic-similarity">Chapter 4</a>).</p>

</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="index.html">ðŸ“– Introduction</a></div>
<div class="next"><a href="GOSemSim.html"><span class="header-section-number">2</span> GO semantic similarity analysis</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#semantic-similarity-overview"><span class="header-section-number">1</span> Overview of semantic similarity analysis</a></li>
<li>
<a class="nav-link" href="#information-content-based-methods"><span class="header-section-number">1.1</span> Information content-based methods</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#resnik-method"><span class="header-section-number">1.1.1</span> Resnik method</a></li>
<li><a class="nav-link" href="#lin-method"><span class="header-section-number">1.1.2</span> Lin method</a></li>
<li><a class="nav-link" href="#rel-method"><span class="header-section-number">1.1.3</span> Rel method</a></li>
<li><a class="nav-link" href="#jiang-method"><span class="header-section-number">1.1.4</span> Jiang method</a></li>
</ul>
</li>
<li>
<a class="nav-link" href="#graph-based-method"><span class="header-section-number">1.2</span> Graph-based method</a><ul class="nav navbar-nav"><li><a class="nav-link" href="#wang-method"><span class="header-section-number">1.2.1</span> Wang method</a></li></ul>
</li>
<li>
<a class="nav-link" href="#combine-methods"><span class="header-section-number">1.3</span> Combine methods</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#max"><span class="header-section-number">1.3.1</span> max</a></li>
<li><a class="nav-link" href="#avg"><span class="header-section-number">1.3.2</span> avg</a></li>
<li><a class="nav-link" href="#rcmax"><span class="header-section-number">1.3.3</span> rcmax</a></li>
<li><a class="nav-link" href="#bma"><span class="header-section-number">1.3.4</span> BMA</a></li>
</ul>
</li>
<li><a class="nav-link" href="#summary"><span class="header-section-number">1.4</span> Summary</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
          
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Biomedical Knowledge Mining using GOSemSim and clusterProfiler</strong>" was written by Guangchuang Yu. It was last built on 2021-08-23.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer><!-- dynamically load mathjax for compatibility with self-contained --><script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script><script type="text/x-mathjax-config">const popovers = document.querySelectorAll('a.footnote-ref[data-toggle="popover"]');
for (let popover of popovers) {
  const div = document.createElement('div');
  div.setAttribute('style', 'position: absolute; top: 0, left:0; width:0, height:0, overflow: hidden; visibility: hidden;');
  div.innerHTML = popover.getAttribute('data-content');

  var has_math = div.querySelector("span.math");
  if (has_math) {
    document.body.appendChild(div);
    MathJax.Hub.Queue(["Typeset", MathJax.Hub, div]);
    MathJax.Hub.Queue(function() {
      popover.setAttribute('data-content', div.innerHTML);
      document.body.removeChild(div);
    })
  }
}
</script>
</body>
</html>
