<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 12 Universal enrichment analysis | Biomedical Knowledge Mining using GOSemSim and clusterProfiler</title>
<meta name="author" content="Guangchuang Yu">
<meta name="description" content="The clusterProfiler package (Yu et al. 2012) supports both hypergeometric test and gene set enrichment analyses of many ontology/pathway, but itâ€™s still not enough for users may want to analyze...">
<meta name="generator" content="bookdown 0.23 with bs4_book()">
<meta property="og:title" content="Chapter 12 Universal enrichment analysis | Biomedical Knowledge Mining using GOSemSim and clusterProfiler">
<meta property="og:type" content="book">
<meta property="og:description" content="The clusterProfiler package (Yu et al. 2012) supports both hypergeometric test and gene set enrichment analyses of many ontology/pathway, but itâ€™s still not enough for users may want to analyze...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Chapter 12 Universal enrichment analysis | Biomedical Knowledge Mining using GOSemSim and clusterProfiler">
<meta name="twitter:description" content="The clusterProfiler package (Yu et al. 2012) supports both hypergeometric test and gene set enrichment analyses of many ontology/pathway, but itâ€™s still not enough for users may want to analyze...">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/header-attrs-2.10/header-attrs.js"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.2.5.1/tabs.js"></script><script src="libs/bs3compat-0.2.5.1/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS -->
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">Biomedical Knowledge Mining using GOSemSim and clusterProfiler</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html">ðŸ“– Introduction</a></li>
<li class="book-part">Part I: Semantic Similarity Analysis</li>
<li><a class="" href="semantic-similarity-overview.html"><span class="header-section-number">1</span> Overview of semantic similarity analysis</a></li>
<li><a class="" href="GOSemSim.html"><span class="header-section-number">2</span> GO semantic similarity analysis</a></li>
<li><a class="" href="DOSE-semantic-similarity.html"><span class="header-section-number">3</span> DO semantic similarity analysis</a></li>
<li><a class="" href="meshes-semantic-similarity.html"><span class="header-section-number">4</span> MeSH semantic similarity analysis</a></li>
<li class="book-part">Part II: Enrichment analysis</li>
<li><a class="" href="enrichment-overview.html"><span class="header-section-number">5</span> Overview of enrichment analysis</a></li>
<li><a class="" href="clusterprofiler-go.html"><span class="header-section-number">6</span> GO enrichment analysis</a></li>
<li><a class="" href="clusterprofiler-kegg.html"><span class="header-section-number">7</span> KEGG enrichment analysis</a></li>
<li><a class="" href="wikipathways-analysis.html"><span class="header-section-number">8</span> WikiPathways analysis</a></li>
<li><a class="" href="reactomepa.html"><span class="header-section-number">9</span> Reactome enrichment analysis</a></li>
<li><a class="" href="dose-enrichment.html"><span class="header-section-number">10</span> Disease enrichment analysis</a></li>
<li><a class="" href="meshes-enrichment.html"><span class="header-section-number">11</span> MeSH enrichment analysis</a></li>
<li><a class="active" href="universal-api.html"><span class="header-section-number">12</span> Universal enrichment analysis</a></li>
<li><a class="" href="chipseeker-enrichment.html"><span class="header-section-number">13</span> Functional enrichment analysis of genomic coordinations</a></li>
<li><a class="" href="clusterprofiler-comparecluster.html"><span class="header-section-number">14</span> Biological theme comparison</a></li>
<li><a class="" href="enrichplot.html"><span class="header-section-number">15</span> Visualization of functional enrichment result</a></li>
<li class="book-part">Part III: Miscellaneous topics</li>
<li><a class="" href="clusterProfiler-dplyr.html"><span class="header-section-number">16</span> dplyr verbs for manipulating enrichment result</a></li>
<li><a class="" href="useful-utilities.html"><span class="header-section-number">17</span> Useful utilities</a></li>
<li class="book-part">Appendix</li>
<li><a class="" href="faq.html"><span class="header-section-number">A</span> Frequently asked questions</a></li>
<li><a class="" href="references.html">References</a></li>
</ul>

        <div class="book-extra">
          
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="universal-api" class="section level1" number="12">
<h1>
<span class="header-section-number">12</span> Universal enrichment analysis<a class="anchor" aria-label="anchor" href="#universal-api"><i class="fas fa-link"></i></a>
</h1>
<p>The <a href="http://bioconductor.org/packages/clusterProfiler">clusterProfiler</a> package <span class="citation">(<a href="references.html#ref-yu2012" role="doc-biblioref">Yu et al. 2012</a>)</span> supports both hypergeometric test and gene set enrichment analyses of many ontology/pathway, but itâ€™s still not enough for users may want to analyze their data with unsupported organisms, slim version of GO, novel functional annotation (e.g.Â GO via BlastGO or KEGG via KAAS), unsupported ontologies/pathways or customized annotations.</p>
<p>The <a href="http://bioconductor.org/packages/clusterProfiler">clusterProfiler</a> package provides <code><a href="https://rdrr.io/pkg/clusterProfiler/man/enricher.html">enricher()</a></code> function for hypergeometric test and <code><a href="https://rdrr.io/pkg/clusterProfiler/man/GSEA.html">GSEA()</a></code> function for gene set enrichment analysis that are designed to accept user defined annotation. They accept two additional parameters <code>TERM2GENE</code> and <code>TERM2NAME</code>. As indicated in the parameter names, <code>TERM2GENE</code> is a data.frame with first column of term ID and second column of corresponding mapped gene and <code>TERM2NAME</code> is a <code>data.frame</code> with first column of term ID and second column of corresponding term name. <code>TERM2NAME</code> is optional.</p>
<div id="input-data" class="section level2" number="12.1">
<h2>
<span class="header-section-number">12.1</span> Input data<a class="anchor" aria-label="anchor" href="#input-data"><i class="fas fa-link"></i></a>
</h2>
<p>For over representation analysis, all we need is a gene vector, that is a vector of gene IDs. These gene IDs can be obtained by differential expression analysis (<em>e.g.</em> with the <a href="http://bioconductor.org/packages/DESeq2">DESeq2</a> package).</p>
<p>For gene set enrichment analysis, we need a ranked list of genes. <a href="http://bioconductor.org/packages/DOSE">DOSE</a> provides an example dataset <code>geneList</code> which was derived from <code>R</code> package <a href="http://bioconductor.org/packages/breastCancerMAINZ">breastCancerMAINZ</a> that contained 200 samples, including 29 samples in grade I, 136 samples in grade II and 35 samples in grade III. We computed the ratios of geometric means of grade III samples versus geometric means of grade I samples. Logarithm of these ratios (base 2) were stored in <code>geneList</code> dataset. If you want to prepare your own <code>geneList</code>, please refer to the <a href="faq.html#genelist">FAQ</a>.</p>
<p>We can load the sample data into R via:</p>
<div class="sourceCode" id="cb120"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">geneList</span>, package<span class="op">=</span><span class="st">"DOSE"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">geneList</span><span class="op">)</span></code></pre></div>
<pre><code>##     4312     8318    10874    55143    55388      991 
## 4.572613 4.514594 4.418218 4.144075 3.876258 3.677857</code></pre>
<p>Suppose we define fold change greater than 2 as DEGs:</p>
<div class="sourceCode" id="cb122"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gene</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">geneList</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">abs</a></span><span class="op">(</span><span class="va">geneList</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">2</span><span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">gene</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] "4312"  "8318"  "10874" "55143" "55388" "991"</code></pre>
</div>
<div id="cell-marker" class="section level2" number="12.2">
<h2>
<span class="header-section-number">12.2</span> Cell Marker<a class="anchor" aria-label="anchor" href="#cell-marker"><i class="fas fa-link"></i></a>
</h2>
<!--     # tidyr::unite("cellMarker", tissueType, cancerType, cellName, sep=", ")  -->
<div class="sourceCode" id="cb124"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cell_marker_data</span> <span class="op">&lt;-</span> <span class="fu">vroom</span><span class="fu">::</span><span class="fu"><a href="https://vroom.r-lib.org/reference/vroom.html">vroom</a></span><span class="op">(</span><span class="st">'http://bio-bigdata.hrbmu.edu.cn/CellMarker/download/Human_cell_markers.txt'</span><span class="op">)</span>

<span class="co">## instead of `cellName`, users can use other features (e.g. `cancerType`)</span>
<span class="va">cells</span> <span class="op">&lt;-</span> <span class="va">cell_marker_data</span> <span class="op">%&gt;%</span>
    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">cellName</span>, <span class="va">geneID</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>geneID <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/strsplit.html">strsplit</a></span><span class="op">(</span><span class="va">geneID</span>, <span class="st">', '</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu">tidyr</span><span class="fu">::</span><span class="fu"><a href="https://tidyr.tidyverse.org/reference/nest.html">unnest</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<div id="cell-marker-ora" class="section level3" number="12.2.1">
<h3>
<span class="header-section-number">12.2.1</span> Cell Marker over-presentaton analysis<a class="anchor" aria-label="anchor" href="#cell-marker-ora"><i class="fas fa-link"></i></a>
</h3>
<div class="sourceCode" id="cb125"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/enricher.html">enricher</a></span><span class="op">(</span><span class="va">gene</span>, TERM2GENE <span class="op">=</span> <span class="va">cells</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></code></pre></div>
<pre><code>##                                            ID            Description GeneRatio
## Neural progenitor cell Neural progenitor cell Neural progenitor cell    33/153
## MKI67+ progenitor cell MKI67+ progenitor cell MKI67+ progenitor cell    20/153
## DCLK1+ progenitor cell DCLK1+ progenitor cell DCLK1+ progenitor cell     7/153
##                          BgRatio       pvalue     p.adjust       qvalue
## Neural progenitor cell 176/11582 3.395466e-29 3.259648e-27 2.716373e-27
## MKI67+ progenitor cell 111/11582 1.426062e-17 6.845100e-16 5.704250e-16
## DCLK1+ progenitor cell  99/11582 3.298755e-04 1.055601e-02 8.796679e-03
##                                                                                                                                                                                              geneID
## Neural progenitor cell 991/9493/9833/9133/10403/7153/259266/6241/9787/11065/55355/55872/51203/83461/22974/10460/79019/55839/890/983/4085/5080/332/3832/9212/51659/9055/891/4174/9232/1602/2018/4137
## MKI67+ progenitor cell                                                                       991/9133/10403/7153/259266/6241/9787/11065/51203/22974/10460/890/983/332/9212/9055/891/9232/2167/51705
## DCLK1+ progenitor cell                                                                                                                                        1307/730/54829/10234/79901/57758/4969
##                        Count
## Neural progenitor cell    33
## MKI67+ progenitor cell    20
## DCLK1+ progenitor cell     7</code></pre>
</div>
<div id="cell-marker-gsea" class="section level3" number="12.2.2">
<h3>
<span class="header-section-number">12.2.2</span> Cell Marker gene set enrichment analysis<a class="anchor" aria-label="anchor" href="#cell-marker-gsea"><i class="fas fa-link"></i></a>
</h3>
<div class="sourceCode" id="cb127"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">y</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/GSEA.html">GSEA</a></span><span class="op">(</span><span class="va">geneList</span>, TERM2GENE <span class="op">=</span> <span class="va">cells</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">y</span><span class="op">)</span></code></pre></div>
<pre><code>##                                                            ID
## FGFR1HighNME5- epithelial cell FGFR1HighNME5- epithelial cell
## FOXN4+ cell                                       FOXN4+ cell
## Leydig precursor cell                   Leydig precursor cell
## Neural progenitor cell                 Neural progenitor cell
## Paneth cell                                       Paneth cell
## Ciliated epithelial cell             Ciliated epithelial cell
##                                                   Description setSize
## FGFR1HighNME5- epithelial cell FGFR1HighNME5- epithelial cell     131
## FOXN4+ cell                                       FOXN4+ cell      63
## Leydig precursor cell                   Leydig precursor cell     184
## Neural progenitor cell                 Neural progenitor cell     149
## Paneth cell                                       Paneth cell     261
## Ciliated epithelial cell             Ciliated epithelial cell     217
##                                enrichmentScore       NES       pvalue
## FGFR1HighNME5- epithelial cell      -0.5739472 -2.249412 1.000000e-10
## FOXN4+ cell                          0.7136090  2.703078 1.000000e-10
## Leydig precursor cell               -0.5710583 -2.324742 1.000000e-10
## Neural progenitor cell               0.7160571  3.127855 1.000000e-10
## Paneth cell                          0.5381280  2.509634 1.000000e-10
## Ciliated epithelial cell            -0.5012031 -2.084643 2.644335e-10
##                                   p.adjust      qvalues rank
## FGFR1HighNME5- epithelial cell 3.30000e-09 1.915789e-09 1404
## FOXN4+ cell                    3.30000e-09 1.915789e-09 1419
## Leydig precursor cell          3.30000e-09 1.915789e-09 2609
## Neural progenitor cell         3.30000e-09 1.915789e-09  684
## Paneth cell                    3.30000e-09 1.915789e-09 2187
## Ciliated epithelial cell       7.27192e-09 4.221657e-09 1440
##                                                  leading_edge
## FGFR1HighNME5- epithelial cell tags=39%, list=11%, signal=35%
## FOXN4+ cell                    tags=51%, list=11%, signal=45%
## Leydig precursor cell          tags=45%, list=21%, signal=36%
## Neural progenitor cell          tags=47%, list=5%, signal=45%
## Paneth cell                    tags=45%, list=18%, signal=38%
## Ciliated epithelial cell       tags=29%, list=12%, signal=27%
##                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         core_enrichment
## FGFR1HighNME5- epithelial cell                                                                                                                                                                                                                                                                                                                                            3931/6586/23284/7106/5919/316/8829/4060/10559/8642/2846/58494/79987/8835/8076/64221/5118/83700/79776/50509/1513/7058/4313/55314/91851/10536/2674/3305/5159/443/3908/5101/2690/55733/2550/6857/10631/81035/4330/83468/576/2200/5157/7049/51380/1307/196740/2532/1308/8483/4239
## FOXN4+ cell                                                                                                                                                                                                                                                                                                                                                                                                                                                                   79733/4751/79019/983/81620/4069/8836/3070/10733/81831/3932/10926/56938/7298/6491/262/1029/1719/8894/5902/10950/5496/2023/124222/10263/51142/11168/3638/51491/8751/6637/86
## Leydig precursor cell                                                                                                                                                                                   7169/120/2949/22871/79822/7070/3624/10761/3671/80114/6326/25959/2013/9902/8654/4017/4811/26010/1291/92689/126393/7474/7168/27303/51454/1306/678/8425/7837/590/5919/23635/8829/1831/30008/1277/7078/4747/6591/56944/8406/1281/11167/1290/9037/2946/283298/1009/4313/6320/3075/6387/5159/1289/1292/116039/3908/5101/1909/83716/5376/633/6857/10631/54674/54587/30846/83468/3487/8404/2200/1634/1287/3479/5348/2006/7373/730/10234/4239/79901/4969
## Neural progenitor cell                                                                                                                                                                                                                                            991/9493/9833/9133/10403/7153/259266/6241/9787/11065/55355/55872/51203/83461/22974/10460/79019/55839/890/983/4085/5080/332/3832/9212/51659/9055/891/4174/9232/4171/1033/1164/11004/51512/11130/79682/26227/57405/10615/3149/23594/56992/9768/11339/3070/4288/1163/4175/4173/2237/29028/3838/54478/3015/699/81611/1063/5111/26271/51053/10535/1719/8543/9289/4176/1841/7112/58516/3148
## Paneth cell                    597/3627/8140/1844/4283/7850/2921/7941/4318/4069/3576/1075/10797/56833/713/1230/200315/712/51338/64581/57823/10437/8767/5341/91543/6351/3055/6614/10859/330/9332/313/51765/1051/22797/4067/80896/864/5226/9046/5336/8942/3162/4688/29887/5641/11314/79888/942/6916/1439/1116/10288/7127/4064/1794/483/963/3689/5788/3383/962/23250/3937/2207/3394/5552/54504/1514/79879/64092/409/7805/4360/3936/4689/5880/7128/10105/9308/51296/10148/10287/8477/6503/929/945/10261/4794/9775/8013/5912/7124/5329/10875/7097/7852/1378/5743/6398/7454/3119/951/6039/1240/5142/1536/3071/834/8676/7305/7133/3310/958/5724/728/11309/3059
## Ciliated epithelial cell                                                                                                                                                                                                                                                          9666/1490/9284/7840/51281/126070/5021/55779/80823/79781/11001/81890/79645/79411/8777/4753/2701/2259/9854/8100/323/55152/51364/9940/23345/216/146542/22832/582/57728/4246/7220/55064/79819/54681/5002/79884/10610/26018/79659/6542/53832/7033/8938/23245/26960/55638/9576/114327/54585/1287/9358/55259/643314/64799/6038/80129/4857/80736/2066/79083/79846/11122/79838</code></pre>
</div>
</div>
<div id="msigdb-analysis" class="section level2" number="12.3">
<h2>
<span class="header-section-number">12.3</span> MSigDb analysis<a class="anchor" aria-label="anchor" href="#msigdb-analysis"><i class="fas fa-link"></i></a>
</h2>
<p><a href="http://software.broadinstitute.org/gsea/msigdb">Molecular Signatures Database</a> is a collection of annotated gene sets. It contains 8 major collections:</p>
<ul>
<li>H: hallmark gene sets</li>
<li>C1: positional gene sets</li>
<li>C2: curated gene sets</li>
<li>C3: motif gene sets</li>
<li>C4: computational gene sets</li>
<li>C5: GO gene sets</li>
<li>C6: oncogenic signatures</li>
<li>C7: immunologic signatures</li>
</ul>
<p>Users can download <a href="www.broadinstitute.org/cancer/software/gsea/wiki/index.php/Data_formats#GMT:_Gene_Matrix_Transposed_file_format_.28.2A.gmt.29">GMT files</a> from <a href="http://software.broadinstitute.org/gsea/msigdb">Broad Institute</a> and use the <code><a href="https://rdrr.io/pkg/clusterProfiler/man/read-gmt.html">read.gmt()</a></code> function to parse the file to be used in <code><a href="https://rdrr.io/pkg/clusterProfiler/man/enricher.html">enricher()</a></code> and <code><a href="https://rdrr.io/pkg/clusterProfiler/man/GSEA.html">GSEA()</a></code>.</p>
<p>There is an R package, <a href="https://cran.r-project.org/package=msigdbr">msigdbr</a>, that already packed the MSigDB gene sets in tidy data format that can be used directly with <a href="http://bioconductor.org/packages/clusterProfiler">clusterProfiler</a> <span class="citation">(<a href="references.html#ref-yu2012" role="doc-biblioref">Yu et al. 2012</a>)</span>.</p>
<p>It supports several specices:</p>
<div class="sourceCode" id="cb129"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://igordot.github.io/msigdbr/">msigdbr</a></span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/msigdbr/man/msigdbr_show_species.html">msigdbr_show_species</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>##  [1] "Anolis carolinensis"             "Bos taurus"                     
##  [3] "Caenorhabditis elegans"          "Canis lupus familiaris"         
##  [5] "Danio rerio"                     "Drosophila melanogaster"        
##  [7] "Equus caballus"                  "Felis catus"                    
##  [9] "Gallus gallus"                   "Homo sapiens"                   
## [11] "Macaca mulatta"                  "Monodelphis domestica"          
## [13] "Mus musculus"                    "Ornithorhynchus anatinus"       
## [15] "Pan troglodytes"                 "Rattus norvegicus"              
## [17] "Saccharomyces cerevisiae"        "Schizosaccharomyces pombe 972h-"
## [19] "Sus scrofa"                      "Xenopus tropicalis"</code></pre>
<p>We can retrieve all human gene sets:</p>
<div class="sourceCode" id="cb131"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">m_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/msigdbr/man/msigdbr.html">msigdbr</a></span><span class="op">(</span>species <span class="op">=</span> <span class="st">"Homo sapiens"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">m_df</span>, <span class="fl">2</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">as.data.frame</span></code></pre></div>
<pre><code>##   gs_cat      gs_subcat        gs_name gene_symbol entrez_gene    ensembl_gene
## 1     C3 MIR:MIR_Legacy AAACCAC_MIR140       ABCC4       10257 ENSG00000125257
## 2     C3 MIR:MIR_Legacy AAACCAC_MIR140    ABRAXAS2       23172 ENSG00000165660
##   human_gene_symbol human_entrez_gene human_ensembl_gene  gs_id gs_pmid
## 1             ABCC4             10257    ENSG00000125257 M12609        
## 2          ABRAXAS2             23172    ENSG00000165660 M12609        
##   gs_geoid gs_exact_source gs_url
## 1                                
## 2                                
##                                                                                                                                                                                           gs_description
## 1 Genes having at least one occurence of the motif AAACCAC in their 3' untranslated region. The motif represents putative target (that is, seed match) of human mature miRNA hsa-miR-140 (v7.1 miRBase).
## 2 Genes having at least one occurence of the motif AAACCAC in their 3' untranslated region. The motif represents putative target (that is, seed match) of human mature miRNA hsa-miR-140 (v7.1 miRBase).</code></pre>
<p>Or specific collection. Here we use C6, oncogenic gene sets as an example:</p>
<div class="sourceCode" id="cb133"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">m_t2g</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/msigdbr/man/msigdbr.html">msigdbr</a></span><span class="op">(</span>species <span class="op">=</span> <span class="st">"Homo sapiens"</span>, category <span class="op">=</span> <span class="st">"C6"</span><span class="op">)</span> <span class="op">%&gt;%</span> 
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">gs_name</span>, <span class="va">entrez_gene</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">m_t2g</span><span class="op">)</span></code></pre></div>
<pre><code>## # A tibble: 6 Ã— 2
##   gs_name              entrez_gene
##   &lt;chr&gt;                      &lt;int&gt;
## 1 AKT_UP_MTOR_DN.V1_DN       25864
## 2 AKT_UP_MTOR_DN.V1_DN          95
## 3 AKT_UP_MTOR_DN.V1_DN      137872
## 4 AKT_UP_MTOR_DN.V1_DN         134
## 5 AKT_UP_MTOR_DN.V1_DN       55326
## 6 AKT_UP_MTOR_DN.V1_DN       55326</code></pre>
<div id="msigdb-ora" class="section level3" number="12.3.1">
<h3>
<span class="header-section-number">12.3.1</span> MSigDb over-presentaton analysis<a class="anchor" aria-label="anchor" href="#msigdb-ora"><i class="fas fa-link"></i></a>
</h3>
<div class="sourceCode" id="cb135"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">em</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/enricher.html">enricher</a></span><span class="op">(</span><span class="va">gene</span>, TERM2GENE<span class="op">=</span><span class="va">m_t2g</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">em</span><span class="op">)</span></code></pre></div>
<pre><code>##                                            ID            Description GeneRatio
## RPS14_DN.V1_DN                 RPS14_DN.V1_DN         RPS14_DN.V1_DN    22/183
## GCNP_SHH_UP_LATE.V1_UP GCNP_SHH_UP_LATE.V1_UP GCNP_SHH_UP_LATE.V1_UP    16/183
## PRC2_EZH2_UP.V1_DN         PRC2_EZH2_UP.V1_DN     PRC2_EZH2_UP.V1_DN    15/183
## VEGF_A_UP.V1_DN               VEGF_A_UP.V1_DN        VEGF_A_UP.V1_DN    15/183
## RB_P107_DN.V1_UP             RB_P107_DN.V1_UP       RB_P107_DN.V1_UP    10/183
## E2F1_UP.V1_UP                   E2F1_UP.V1_UP          E2F1_UP.V1_UP    12/183
##                          BgRatio       pvalue     p.adjust       qvalue
## RPS14_DN.V1_DN         186/10915 4.716365e-13 7.970657e-11 6.503619e-11
## GCNP_SHH_UP_LATE.V1_UP 181/10915 5.815455e-08 4.914059e-06 4.009603e-06
## PRC2_EZH2_UP.V1_DN     192/10915 7.635020e-07 3.446929e-05 2.812505e-05
## VEGF_A_UP.V1_DN        193/10915 8.158412e-07 3.446929e-05 2.812505e-05
## RB_P107_DN.V1_UP       130/10915 6.428560e-05 2.024672e-03 1.652021e-03
## E2F1_UP.V1_UP          188/10915 7.522623e-05 2.024672e-03 1.652021e-03
##                                                                                                                                        geneID
## RPS14_DN.V1_DN         10874/55388/991/9493/1062/4605/9133/23397/79733/9787/55872/83461/54821/51659/9319/9055/10112/4174/5105/2532/7021/79901
## GCNP_SHH_UP_LATE.V1_UP                                      55388/7153/79733/6241/9787/51203/983/9212/1111/9319/9055/3833/6790/4174/3169/1580
## PRC2_EZH2_UP.V1_DN                                          8318/55388/4605/23397/9787/55355/10460/6362/81620/2146/7272/9212/11182/3887/24137
## VEGF_A_UP.V1_DN                                                   8318/9493/1062/9133/10403/6241/9787/4085/332/3832/7272/891/23362/2167/10234
## RB_P107_DN.V1_UP                                                                         8318/23397/79733/6241/4085/8208/9055/24137/4174/1307
## E2F1_UP.V1_UP                                                                  55388/7153/23397/79733/9787/2146/2842/9212/8208/1111/9055/3833
##                        Count
## RPS14_DN.V1_DN            22
## GCNP_SHH_UP_LATE.V1_UP    16
## PRC2_EZH2_UP.V1_DN        15
## VEGF_A_UP.V1_DN           15
## RB_P107_DN.V1_UP          10
## E2F1_UP.V1_UP             12</code></pre>
</div>
<div id="msigdb-gsea" class="section level3" number="12.3.2">
<h3>
<span class="header-section-number">12.3.2</span> MSigDb gene set enrichment analysis<a class="anchor" aria-label="anchor" href="#msigdb-gsea"><i class="fas fa-link"></i></a>
</h3>
<p>In over-presentaton analysis, we use oncogenic gene sets (i.e.Â C6) to test whether the DE genes are involved in the process that leads to cancer. In this example, we will use the C3 category to test whether genes are up/down-regulated by sharing specific motif using GSEA approach.</p>
<div class="sourceCode" id="cb137"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">C3_t2g</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/msigdbr/man/msigdbr.html">msigdbr</a></span><span class="op">(</span>species <span class="op">=</span> <span class="st">"Homo sapiens"</span>, category <span class="op">=</span> <span class="st">"C3"</span><span class="op">)</span> <span class="op">%&gt;%</span> 
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">gs_name</span>, <span class="va">entrez_gene</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">C3_t2g</span><span class="op">)</span></code></pre></div>
<pre><code>## # A tibble: 6 Ã— 2
##   gs_name        entrez_gene
##   &lt;chr&gt;                &lt;int&gt;
## 1 AAACCAC_MIR140       10257
## 2 AAACCAC_MIR140       23172
## 3 AAACCAC_MIR140          81
## 4 AAACCAC_MIR140          81
## 5 AAACCAC_MIR140          90
## 6 AAACCAC_MIR140        8754</code></pre>
<div class="sourceCode" id="cb139"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">em2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/GSEA.html">GSEA</a></span><span class="op">(</span><span class="va">geneList</span>, TERM2GENE <span class="op">=</span> <span class="va">C3_t2g</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">em2</span><span class="op">)</span></code></pre></div>
<pre><code>##                                        ID          Description setSize
## E2F1_Q3                           E2F1_Q3              E2F1_Q3     187
## HSD17B8_TARGET_GENES HSD17B8_TARGET_GENES HSD17B8_TARGET_GENES     401
## E2F1_Q6                           E2F1_Q6              E2F1_Q6     185
## SGCGSSAAA_E2F1DP2_01 SGCGSSAAA_E2F1DP2_01 SGCGSSAAA_E2F1DP2_01     129
## E2F_Q4                             E2F_Q4               E2F_Q4     193
## E2F_Q6                             E2F_Q6               E2F_Q6     190
##                      enrichmentScore      NES       pvalue     p.adjust
## E2F1_Q3                    0.4983913 2.203638 1.000000e-10 1.655000e-07
## HSD17B8_TARGET_GENES       0.6368099 3.075013 1.000000e-10 1.655000e-07
## E2F1_Q6                    0.4906657 2.172309 1.585570e-10 1.749413e-07
## SGCGSSAAA_E2F1DP2_01       0.5408182 2.285356 1.207636e-09 9.993184e-07
## E2F_Q4                     0.4685394 2.087470 1.704036e-09 1.128072e-06
## E2F_Q6                     0.4672345 2.077292 2.664743e-09 1.302665e-06
##                           qvalues rank                   leading_edge
## E2F1_Q3              1.312105e-07 2045 tags=36%, list=16%, signal=30%
## HSD17B8_TARGET_GENES 1.312105e-07 1042  tags=36%, list=8%, signal=34%
## E2F1_Q6              1.386957e-07 1819 tags=34%, list=15%, signal=30%
## SGCGSSAAA_E2F1DP2_01 7.922725e-07 1797 tags=38%, list=14%, signal=33%
## E2F_Q4               8.943498e-07 1663 tags=29%, list=13%, signal=26%
## E2F_Q6               1.032770e-06 1663 tags=29%, list=13%, signal=25%
##                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             core_enrichment
## E2F1_Q3                                                                                                                                                                                                                                                                                                                                                                                                                                                 79733/6241/983/5080/2146/8715/4171/993/990/56992/9768/4998/10733/7037/4175/54962/54954/29028/3015/2643/85236/5111/64785/26271/51053/1869/3925/114/5427/4176/7112/4436/1871/79915/9088/6632/284403/7290/2597/7398/4172/5424/124222/1633/10849/9824/4678/6646/29902/63967/56886/6059/23234/1786/204/51087/23636/2048/83463/10393/79173/688/3007/79677/9462/56946/9221
## HSD17B8_TARGET_GENES 55143/55388/991/2305/9493/1062/4605/9833/9133/10403/7153/23397/79733/259266/6241/55165/9787/11065/55355/9582/220134/55872/83461/10460/4751/10635/55839/890/9415/983/54821/4085/9837/5080/332/7272/64151/2842/9212/51659/9319/9055/3833/146909/10112/51514/4174/9232/1033/9928/3161/11004/993/4603/79801/990/5347/55215/701/55723/51512/55635/9156/11130/10024/57405/10615/1894/2491/8438/9700/5888/7083/898/3149/56992/9768/4288/10733/1163/4175/5307/2237/29899/55010/59336/56942/5984/29028/3838/1058/54478/3015/699/6491/81611/1063/27346/64785/9401/26271/51026/641/1869/10535/1029/28998/1763/8970/54892/55159/864/93594/1789/4176/7111/3148/116832/9585/9735/55732/81624/23310/1871/1031/79915/995/10051/30012/1104/80178/78995/7468/284403/5230/8549/7884/5558/4172/5424/79866/3182/56902/83990
## E2F1_Q6                                                                                                                                                                                                                                                                                                                                                                                                                                                                          79733/6241/983/5080/81620/2146/4171/993/990/3159/9768/4998/4175/4173/29028/3015/4609/85236/5111/64785/26271/51053/1869/3925/5427/4176/7112/5902/4436/1871/79915/9088/6632/284403/7290/2597/4172/5424/124222/1633/10849/7374/9824/2189/4678/2289/6646/29902/50/63967/56886/23234/6839/1786/204/51087/23636/11335/1174/83463/10492/79173/688
## SGCGSSAAA_E2F1DP2_01                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     79733/6241/4171/993/990/3159/9768/4175/4173/29028/3015/4609/85236/5111/64785/26271/51053/1869/5427/4176/5902/4436/1871/79915/9088/6632/284403/7290/4172/5424/124222/1633/7374/2189/4678/2289/50/63967/56886/23234/6839/1786/204/51087/23636/1174/83463/10492/79173
## E2F_Q4                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 8318/79733/6241/983/5080/81620/2146/4171/993/990/5888/9768/4175/4173/29028/3015/85236/5111/64785/26271/51053/1869/3925/5427/23649/4176/7112/5902/4436/1871/79915/9088/6632/284403/7290/2597/4172/5424/124222/1633/10849/7473/7374/9824/7353/4678/6646/50/63967/56886/23234/1786/204/6427/51087/23636
## E2F_Q6                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      8318/79733/6241/983/5080/81620/2146/4171/993/990/5888/9768/4175/4173/29028/3015/85236/5111/64785/26271/51053/1869/3925/5427/23649/4176/7112/5902/4436/1871/79915/9088/6632/284403/7290/2597/4172/5424/124222/1633/10849/7374/9824/7353/4678/6646/50/63967/56886/23234/1786/204/6427/51087/23636</code></pre>

</div>
</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="meshes-enrichment.html"><span class="header-section-number">11</span> MeSH enrichment analysis</a></div>
<div class="next"><a href="chipseeker-enrichment.html"><span class="header-section-number">13</span> Functional enrichment analysis of genomic coordinations</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#universal-api"><span class="header-section-number">12</span> Universal enrichment analysis</a></li>
<li><a class="nav-link" href="#input-data"><span class="header-section-number">12.1</span> Input data</a></li>
<li>
<a class="nav-link" href="#cell-marker"><span class="header-section-number">12.2</span> Cell Marker</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#cell-marker-ora"><span class="header-section-number">12.2.1</span> Cell Marker over-presentaton analysis</a></li>
<li><a class="nav-link" href="#cell-marker-gsea"><span class="header-section-number">12.2.2</span> Cell Marker gene set enrichment analysis</a></li>
</ul>
</li>
<li>
<a class="nav-link" href="#msigdb-analysis"><span class="header-section-number">12.3</span> MSigDb analysis</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#msigdb-ora"><span class="header-section-number">12.3.1</span> MSigDb over-presentaton analysis</a></li>
<li><a class="nav-link" href="#msigdb-gsea"><span class="header-section-number">12.3.2</span> MSigDb gene set enrichment analysis</a></li>
</ul>
</li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
          
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Biomedical Knowledge Mining using GOSemSim and clusterProfiler</strong>" was written by Guangchuang Yu. It was last built on 2021-08-23.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer><!-- dynamically load mathjax for compatibility with self-contained --><script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script><script type="text/x-mathjax-config">const popovers = document.querySelectorAll('a.footnote-ref[data-toggle="popover"]');
for (let popover of popovers) {
  const div = document.createElement('div');
  div.setAttribute('style', 'position: absolute; top: 0, left:0; width:0, height:0, overflow: hidden; visibility: hidden;');
  div.innerHTML = popover.getAttribute('data-content');

  var has_math = div.querySelector("span.math");
  if (has_math) {
    document.body.appendChild(div);
    MathJax.Hub.Queue(["Typeset", MathJax.Hub, div]);
    MathJax.Hub.Queue(function() {
      popover.setAttribute('data-content', div.innerHTML);
      document.body.removeChild(div);
    })
  }
}
</script>
</body>
</html>
